---
title: "Hamster Hybrid Testes RNASeq"
author: "Kelsie E. Hunnicutt"
output:
  html_document: default
  pdf_document: default
---

<h3>Loading libraries and importing data</h3>
```{r setup}
#load packages; make sure these are installed on your machine
suppressMessages(library("edgeR")) #for differential expression analysis
suppressMessages(library("limma")) #for differential expression analysis
suppressMessages(library("ComplexHeatmap")) #for heatmap generation
suppressMessages(library("reshape2")) #for creating the conditions dataframe 
suppressMessages(library("data.table")) #for creating and manipulating data frames 
suppressMessages(library("ggplot2")) #for data visualizing
suppressMessages(library("vegan")) # for decostand count normalization method
suppressMessages(library("matrixStats")) #for calculating median/max values across rows of data frames
suppressMessages(library("ggbeeswarm")) #for offsetting overlapping points in boxplots and violins
suppressMessages(library("tidyr")) #for transforming data frames from wide to long format or long to wide format
suppressMessages(library("FSA")) #for dunn's test
suppressMessages(library("magick")) #for better Heatmap rasterization
suppressMessages(library("ggsignif")) #add significance annotation data to ggplot graphs
suppressMessages(library("dplyr")) #for recoding when doing orthology assignments
suppressMessages(library("knitr")) #to display tables with kable
suppressMessages(library("ggalluvial")) #for alluvial diagrams
suppressMessages(library("dendextend")) #for dendogram plotting
suppressMessages(library("cowplot")) # for multipanel plot organization
suppressMessages(library("gprofiler2")) # for multipanel plot organization
suppressMessages(library("ggnewscale")) # for making additional fill layers for data subsets in ggplot
suppressMessages(library("mygene")) #for annotation of overlap genes
suppressMessages(library("GOSim")) #retrieve go terms associated with a given gene

#set working directory and get today's date; set to the directory corresponding to where you downloaded this repository
knitr::opts_knit$set(root.dir = "your_working_directory")
all_labels()
unlink(".RData")
options(scipen = 999)

todays_date <- gsub("^[A-z][A-z][A-z] *([A-z][A-z][A-z]) *([0-9]+).*20([0-9][0-9])", "\\2\\1\\3", date())
todays_date

sessionInfo()
```

```{r import-data}
#select a counts file then read in raw counts; no multimapping was allowed for the main text figures but the multi-mapped files are provided for both dwarf hamster and mouse if desired; general patterns are the same
count_file.path <- "input_data/Hamsters_psun_14Mar23_singleMapping_featureCounts.csv"
count_data <- read.table(count_file.path, header = TRUE, row.names=NULL, sep = "\t")

#select a metadata file then read in info
metadata_file.path <- "input_data/Hamsters_psun_14Mar23_singleMapping_featureCounts.csv.summary"
metadata <- read.table(metadata_file.path, header = TRUE, row.names=1, sep = "\t")

#read in annotation and tissue specific data from Moore et al. 2022
annotations_raw <- read.csv("input_data/psun_annotations.csv", header = TRUE, na.strings = "#N/A")
annotations <- annotations_raw[c("Length", "Chr", "Start", "End", "Geneid", "Mouse_ID", "Gene_name")]
rownames(annotations) <- annotations$Geneid
colnames(annotations) <- c("Length", "scaffold", "start", "stop", "gene", "mouse_ensemble_ID", "mouse_gene_symbol")

#renames the main scaffolds to chromosome name for scaffolds that were successfully assigned to linkage groups; for chromosomes with more than one long scaffold, give scaffolds unique chromosome names
chrom_key <- read.csv("input_data/psun_chromosome_scaffold_key.csv")[,c(2,3,5)]
colnames(chrom_key) <- c("position_cM", "scaffold", "chr")
chrom_key$scaffold <- gsub(";HRSC", "_HRSC", chrom_key$scaffold)
chrom_sizes <- read.csv("input_data/psun_chrom_sizes.csv", header = TRUE)
colnames(chrom_sizes) <- c("scaffold", "size_bp")
chrom_key <- merge(chrom_key, chrom_sizes, by = "scaffold", all = TRUE)
chrom_key[is.na(chrom_key$chr),]$chr <- chrom_key[is.na(chrom_key$chr),]$scaffold

#add additional chromosome information to annotations
annotations <- merge(annotations, chrom_key, by = c("scaffold"), all.x = TRUE)

#limit to chromosomes to those over 1Mb
chrom_key_large <- chrom_key[chrom_key$size_bp > 1000000,]$chr
paste(dim(annotations[!annotations$chr %in% chrom_key_large,])[1], 
      "/", 
      dim(annotations)[1], 
      " genes were elimated because they are not located on the main chromosomes", 
      sep = "")

annotations <- annotations[annotations$chr %in% chrom_key_large,]
chrom_key <- chrom_key[chrom_key$chr %in% chrom_key_large,]

annotations$X_category <- "Auto"
annotations[annotations$chr == "X",]$X_category <- "X"

#reorder chr_key and annotations dataframes by chromosome name
chrom_key$chr <- factor(chrom_key$chr, levels = c("1.1", "1.2", "2.1", "3.1", "3.2", "4.1", "5.1", "5.2", "5.3", "5.4", "5.5", "6.1", "7.1", "8.1", "9.1", "10.1", "10.2", "11.1", "12.1", "13.1", "13.2", "X"))
chrom_key <- chrom_key[order(chrom_key$chr),]

annotations$chr <- factor(annotations$chr, levels = c("1.1", "1.2", "2.1", "3.1", "3.2", "4.1", "5.1", "5.2", "5.3", "5.4", "5.5", "6.1", "7.1", "8.1", "9.1", "10.1", "10.2", "11.1", "12.1", "13.1", "13.2", "X"))
annotations$X_category <- factor(annotations$X_category, levels = c("Auto", "X"))
annotations <- annotations[order(annotations$gene),]

chrom_sizes <- chrom_key[,3:4]
chrom_sizes$start <- 1
colnames(chrom_sizes) <- c("chromosome", "end", "start")

#write information about chromosome sizes (both all scaffolds and just the long scaffolds) to files for circos diagrams of hybrid and parent DE genes (Supplemental Figures S7 and S8)
write.csv(chrom_sizes, file = "circos_processing_scripts/hamster_circos_hybrids/data/psun_chromosomes.csv", row.names = FALSE)
```

<h3>Reformatting condition names</h3>
```{r renaming-treatments-across-dfs}
#edit metadata to include condition type (condition_name) for each sample
#transpose, rename columns, delete redundant row
metadata <- t(metadata)
head(rownames(metadata))
rownames(metadata) <- gsub(".*ers.sorted\\.", "", rownames(metadata))
rownames(metadata) <- gsub('\\.', '-', rownames(metadata))
rownames(metadata) <- gsub("\\.([D,R,L,W,S])", "-\\1",rownames(metadata))
rownames(metadata) <- gsub('M-', '-', rownames(metadata))
rownames(metadata) <- gsub("^([A-Z]+)_([0-9A-Z]*-[0-9A-Z]*)_([A-Z]+).*", "\\1_\\2_\\3", rownames(metadata))
head(rownames(metadata))

#split status into meaningful info
conditions_df <- colsplit(row.names(metadata), pattern = "_", names = c("cross","male_number","stage"))
conditions_df$sample_name <- paste(conditions_df$cross, conditions_df$male_number, conditions_df$stage, sep = ".")
#group names
conditions_df$condition_name <- paste(conditions_df$cross, conditions_df$stage, sep = "-") 
#combined original metadata with conditions matrix
metadata <- cbind(metadata, conditions_df)

#in preparation for DGE creation, make a dataframe of just gene name and counts
just_counts <- count_data[,grep("bam", colnames(count_data))]
row.names(just_counts) <- count_data[,1]
#match order of count dataframe to annotations
just_counts <- just_counts[annotations$gene, ]
head(colnames(just_counts))

#match just_counts IDs with metadata IDs (i.e. remove extraneous info from featureCount filename)
colnames(just_counts) <- gsub(".*ers.sorted\\.", "", colnames(just_counts))
colnames(just_counts) <- gsub('\\.', '-', colnames(just_counts))
colnames(just_counts) <- gsub("\\.([D,R,L,W,S])", "-\\1",colnames(just_counts))
colnames(just_counts) <- gsub('M-', '-', colnames(just_counts))
colnames(just_counts) <- gsub("^([A-Z]+)_([0-9A-Z]*-[0-9A-Z]*)_([A-Z]+).*", "\\1_\\2_\\3", colnames(just_counts))

head(colnames(just_counts))

dim(metadata)
dim(just_counts)
dim(annotations)
```

<h3>Color Set-Up</h3>
```{r color-setup}
#coloring indicates stage and colors used here are used throughout
#Red = spermatogonia, Yellow = leptotene-zygotene, Green = diplotene, and Blue = round spermatids
sp_color <- "#E07A5F"
sp_color_dark <- "#D85531"
sp_color_darkest <- "#AB3F21"
lz_color <- "#EEBC6D"
lz_color_dark <- "#E7A336"
lz_color_darkest <- "#DB911A"
dip_color <- "#81B29A"
dip_color_dark <- "#5D987B"
dip_color_darkest <- "#46725D"
rs_color <- "#6C719D"
rs_color_dark <- "#3D405B"
rs_color_darkest <- "#292B3D"

gray_1 <- "#EBEBEB"
gray_2 <- "#D6D6D6"
gray_3 <- "#C2C2C2"
gray_4 <- "#999999"
gray_5 <- "#646464"
gray_6 <- "#525252"
gray_7 <- "#474747"

#combining colors into vectors based on the color saturation
color_list <- c(SP = sp_color, 
                LZ = lz_color, 
                DIP = dip_color, 
                RS = rs_color)
color_list_dark <- c(SP = sp_color_dark, 
                     LZ = lz_color_dark, 
                     DIP = dip_color_dark, 
                     RS = rs_color_dark)
color_list_darkest <- c(SP = sp_color_darkest, 
                        LZ = lz_color_darkest, 
                        DIP = dip_color_darkest, 
                        RS = rs_color_darkest)

#a function to convert a vector from raw p-values to significance symbols
sig_conversion <- function(misc_dataframe) {
  triple <- misc_dataframe <= 0.001
  double <- misc_dataframe >= 0.001 & misc_dataframe<0.01 
  single <- misc_dataframe >= 0.01 & misc_dataframe<0.05 
  nonsig <- misc_dataframe >= 0.05 
  misc_dataframe[single] <- "*"
  misc_dataframe[double] <- "**"
  misc_dataframe[triple] <- "***"
  misc_dataframe[nonsig] <- "n.s."
  return(misc_dataframe)
}

#a function for calculating standard error of the mean from an input vector of data
calculate_sem <- function(vector_of_interest) {
  sd(vector_of_interest)/sqrt(length((vector_of_interest)))
}

#graphing parameters
point_size <- 0.5
text_size <- 6
boxplot_width <- 0.1
vjust_value <- 0.8
boxplot_lwd <- 0.2
axis_size <- 0.2
sigText_size <- 2
sigLine_size <- 0.25

#set figure export dimensions
half_plot_width <- 3.25
full_plot_width <- 6.75
half_plot_height <- 3
full_plot_height <- 6

#should figures should be written to files or plotted in R?
exportFigs <- TRUE
```

<h3>Hamster fertility data</h3>
```{r fertility-input-and-processing}
#read in fertility data, select columns of interest, then rename column names
fertility <- read.csv("input_data/Hamster_fertility_phenotype_data_7Mar22.csv", header = TRUE)
fertility <- fertility[c("Age", "Count.Rows", "Count.Total", "Dilution", "EarLength", "FootLength", "HamsterLength", "Motility.Mot", "Motility.Non.Mot", "RelativeSVWeight..mg.g.", "RelativeTestesWeight..mg.g.", "SampleID", "Strain", "SVWeight.g.", "TailLength", "TestesOnly.G.", "Weight..g.")]
colnames(fertility) <- c("Age", "SpermRows", "SpermCount", "SpermDilution", "Ear", "Foot", "BodyLength", "MotileSperm", "NonMotileSperm", "RelativeSVWeight", "RelativeTestesWeight", "ID", "Strain", "SVWeight", "Tail", "TestesWeight", "BodyWeight")

#reformat IDs and strain to match metadata
fertility$ID <- gsub("^([B|S][B|S]).([B|S][B|S])", "\\1\\2_", fertility$ID)
fertility$ID <- gsub("\\.", "-", fertility$ID)
fertility$Strain <- gsub("\\.", "", fertility$Strain)
fertility$Strain <- factor(fertility$Strain, levels = c("BBBB", "BBSS", "SSSS"))

#hamster age; set NA and recast as numeric
fertility$Age[fertility$Age == "#VALUE!"] <- NA
fertility$Age <- as.numeric(fertility$Age)

#calculate sperm motility (number motile sperm/ number motile + nonmotile sperm)
fertility$MotileSperm <- gsub("TNTC", NA, fertility$MotileSperm)
fertility$MotileSperm <- as.numeric(fertility$MotileSperm)
fertility$NonMotileSperm <- gsub("TNTC", NA, fertility$NonMotileSperm)
fertility$NonMotileSperm <- as.numeric(fertility$NonMotileSperm)
fertility$Motility <- fertility$MotileSperm/(fertility$MotileSperm + fertility$NonMotileSperm)
fertility[is.nan(fertility$Motility),]$Motility <- NA

#Calculate normalized sperm count (number sperm counted/number rows counted * sperm dilution )
fertility$SpermCount <- gsub("TNTC", NA, fertility$SpermCount)
fertility$SpermRows <- gsub("TNTC", NA, fertility$SpermRows)
fertility$NormSpermCount <- (5*as.numeric(fertility$SpermCount)) / (as.numeric(fertility$SpermRows)*as.numeric(fertility$SpermDilution))

#fixing record errors in original phenotype datasheet
#fixing misrecorded body measurements
fertility[fertility$ID == "BBBB_197-3M",]$Tail <- 8
fertility[fertility$ID == "SSSS_188-5M",]$Tail <- 7
fertility[fertility$ID == "BBSS_90-2M", ]$BodyLength <- 85

#set all ruptured SVs to NA
fertility[fertility$ID == "BBBB_195-4M",]$RelativeSVWeight <- NA
fertility[fertility$ID == "BBBB_295A-3M",]$RelativeSVWeight <- NA
fertility[fertility$ID == "SSSS_155-6M",]$RelativeSVWeight <- NA
fertility[fertility$ID == "SSSS_182-5M",]$RelativeSVWeight <- NA

#incorrect body weight for BBSS_100-2M because measured without organs; corrected with the addition of testes weight and SV weight (missing liver weight)
fertility[fertility$ID == "BBSS_100-2M",]$BodyWeight <- fertility[fertility$ID == "BBSS_100-2M",]$BodyWeight + fertility[fertility$ID == "BBSS_100-2M",]$TestesWeight + fertility[fertility$ID == "BBSS_100-2M",]$SVWeight
fertility[fertility$ID == "BBSS_100-2M",]$RelativeTestesWeight <- 100*fertility[fertility$ID == "BBSS_100-2M",]$TestesWeight/fertility[fertility$ID == "BBSS_100-2M",]$BodyWeight
fertility[fertility$ID == "BBSS_100-2M",]$RelativeSVWeight <- 100*fertility[fertility$ID == "BBSS_100-2M",]$SVWeight/fertility[fertility$ID == "BBSS_100-2M",]$BodyWeight

#hamster not weighed
fertility[fertility$ID == "SSSS_188-4M",]$BodyWeight <- NA
fertility[fertility$ID == "SSSS_188-4M",]$RelativeTestesWeight <- NA
fertility[fertility$ID == "SSSS_188-4M",]$RelativeSVWeight <- NA

#ear/foot swapped on datasheet
fertility[fertility$ID == "BBSS_7-1M",]$Foot <- 9.86
fertility[fertility$ID == "BBSS_7-1M",]$Ear <- 12.98

#incorrect motility
fertility[fertility$ID == "SSSS_152-4M",]$Motility <- NA

#toggle to exclude WT from analysis (collected at a different time than cell sorts); manuscript includes WT individuals
wt_individuals <- gsub("_WT", "", grep("WT", colnames(just_counts), value = TRUE))
#fertility <- fertility[!(fertility$ID %in% wt_individuals),]
```

```{r fertility-stats-and-plots}
#tests if testes weight and SV weight are correlated with body weight to determine if relative testes weight and SV weight should be used instead; they are both correlated so relative weight used in manuscript
#first investigate testes weight correlation
TW_corr <- cor.test(fertility$BodyWeight, fertility$TestesWeight, method = c("pearson"))
TW_rho <- TW_corr$estimate
TW_pValue <- TW_corr$p.value

#correlation plot between body weight and testes weight
bw_v_tw_plot <- ggplot(fertility[!is.na(fertility$BodyWeight),], 
                       aes(x=BodyWeight, y = TestesWeight, shape = Strain, col = Strain, fill = Strain)) +
  geom_point(size=2, stroke = 1) +  
  scale_color_manual(values=c(gray_5, gray_4, gray_2)) + 
  scale_fill_manual(values=c(gray_5, gray_4, gray_2)) + 
  scale_shape_manual(values=c(24, 4, 25)) +
  theme_linedraw() +
  xlab("Body weight (g)") +
  ylab("Testes weight (mg)") + 
  annotate(geom="text", x=20, y = 1.7, label=paste("rho = ", round(TW_rho, 3), sep = "")) + 
  annotate(geom="text", x=20, y = 1.5, label=paste("p = ", round(TW_pValue, 3), sep = ""))

#save or print correlation graph
if(exportFigs == TRUE) {
  ggsave(file = "exported_figures/hamster_bodyWeight_by_testesWeight.pdf", bw_v_tw_plot)
} else {
  if(TW_pValue < 0.05) {
    print(paste("Testes weight is correlated with body weight (Pearson’s r(", TW_corr$parameter, ") = ", round(TW_rho, 3), ", p = ", round(TW_pValue, 3), sep = ""))
  } else {
    print(paste("Testes weight is NOT correlated with body weight (Pearson’s r(", TW_corr$parameter, ") = ", round(TW_rho, 3), ", p = ", round(TW_pValue, 3), sep = ""))
  }
  bw_v_tw_plot
}

#second, investigate SV weight correlation
SVW_corr <- cor.test(fertility$BodyWeight, fertility$SVWeight, method = c("pearson"))
SVW_rho <- SVW_corr$estimate
SVW_pValue <- SVW_corr$p.value

#SV weight correlation plot
bw_v_svw_plot <- ggplot(fertility[!is.na(fertility$BodyWeight) & !is.na(fertility$SVWeight),], aes(x=BodyWeight, y = SVWeight, shape = Strain, col = Strain, fill = Strain)) +
  geom_point(size=2, stroke = 1) +  
  scale_color_manual(values=c(gray_5, gray_4, gray_2)) + 
  scale_fill_manual(values=c(gray_5, gray_4, gray_2)) + 
  scale_shape_manual(values=c(24, 4, 25)) +
  theme_linedraw() +
  xlab("Body weight (g)") +
  ylab("Seminal vesicle weight (mg)") + 
  annotate(geom="text", x=25, y = 0.8, label=paste("rho = ", round(SVW_rho, 3), sep = "")) + 
  annotate(geom="text", x=25, y = 0.7, label=paste("p = ", round(SVW_pValue, 3), sep = ""))

#either save or print SV correlation graph
if(exportFigs == TRUE) {
  ggsave(file = "exported_figures/hamster_bodyWeight_by_SVWeight.pdf", bw_v_svw_plot)
} else {
  if(SVW_pValue < 0.05) {
    print(paste("SV weight is correlated with body weight (Pearson’s r(", SVW_corr$parameter, ") = ", round(SVW_rho, 3), ", p = ", round(SVW_pValue, 3), sep = ""))
  } else {
    print(paste("SV weight is NOT correlated with body weight (Pearson’s r(", SVW_corr$parameter, ") = ", round(SVW_rho, 3), ", p = ", round(SVW_pValue, 3), sep = ""))
  }
  bw_v_svw_plot
}


#plot focal fertility phenotypes by strain type and test for statistical differences between parents and hybrids
calculate_fertility_differences <- function(trait, y_label) {
  
  #remove individuals missing the relevant trait data
  fertility_subset <- fertility[!is.na(fertility[,grep(paste("^", trait, sep = ""), colnames(fertility))]),]
  fertility_subset <- fertility_subset[,c("Strain", trait)]
  
  colnames(fertility_subset) <- c("Strain", "Trait")
  
  #kruskal wallis test followed by dunn's posthoc test
  example_trait_kw <- kruskal.test(Trait ~ Strain, data = fertility_subset)
  example_trait_dt <- dunnTest(Trait ~ Strain, data = fertility_subset, method = "bonferroni")
  #assign Dunn's test object for use in stats printing later
  assign(paste(trait, "_dt", sep = ""), example_trait_dt, envir = .GlobalEnv)
  
  #adjust p-values for multi-species comparison
  example_trait_padj <- example_trait_dt$res$P.adj
  names(example_trait_padj) <- gsub(" ","", example_trait_dt$res$Comparison)
  
  #convert p-values to symbols for graphing
  example_trait_letters <- data.frame(multcompView::multcompLetters(example_trait_padj)$Letters)
  example_trait_letters$Strain <- rownames(example_trait_letters)
  colnames(example_trait_letters) <- c("Letters", "Strain")
  example_trait_padj <- sig_conversion(example_trait_padj)
  
  #plot fertility metric for each species/hybrid
  example_trait_plot <- ggplot(fertility_subset, aes(x = Strain, y = Trait)) + 
    geom_quasirandom(aes(fill = Strain, color = Strain, shape = Strain), alpha = 1, size = point_size) +
    geom_boxplot(width=boxplot_width, outlier.shape = NA, lwd = boxplot_lwd, alpha = 0.2) +
    scale_fill_manual(values=c(gray_5, gray_4, gray_2)) + 
    scale_color_manual(values=c(gray_5, gray_4, gray_2)) + 
    scale_shape_manual(values=c(24,4,25)) + 
    theme_bw() + 
    theme(legend.position = "none", 
          text = element_text(size = text_size),
          axis.title.x=element_blank(), 
          axis.title.y = element_text(size = text_size), 
          axis.text.x = element_text(angle = 45, vjust = vjust_value, size = text_size), 
          plot.margin = unit(c(0.4, 0.07, 0, 0.07), "cm"), 
          panel.grid.major.x = element_blank(), 
          panel.grid.minor = element_blank(), 
          axis.line.y = element_line(colour = "black", linewidth = axis_size), 
          panel.border = element_blank(), 
          axis.ticks.length.x = unit(0, "cm")) +
    scale_x_discrete(labels = c("P. campbelli", "Hybrid", "P. sungorus")) + ylab(y_label) + 
    geom_signif(
      y_position = c(summary(fertility_subset[fertility_subset$Strain == "BBBB",]$Trait)[6]*1.18,
                     summary(fertility_subset[fertility_subset$Strain == "BBBB",]$Trait)[6]*1.10,
                     summary(fertility_subset[fertility_subset$Strain == "BBBB",]$Trait)[6]*1.02), 
      xmin = c(1,1,2), 
      xmax = c(3,2,3), 
      annotation = c(example_trait_padj[2],example_trait_padj[1],example_trait_padj[3]), 
      tip_length = 0, textsize = sigText_size, size = sigLine_size)
  #+  ylim(0, summary(fertility_subset[fertility_subset$Strain == "BBBB",]$Trait)[6]*1.25)
  
  #save plot object
  assign(paste(trait, "plot", sep = "_"), example_trait_plot, envir = .GlobalEnv)
  
  #calculate mean and SEM for each cross
  BBBB_example_trait_mean <- mean(fertility_subset[fertility_subset$Strain == "BBBB",]$Trait)
  BBBB_example_trait_sem <- calculate_sem(fertility_subset[fertility_subset$Strain == "BBBB",]$Trait)
  BBSS_example_trait_mean <- mean(fertility_subset[fertility_subset$Strain == "BBSS",]$Trait)
  BBSS_example_trait_sem <- calculate_sem(fertility_subset[fertility_subset$Strain == "BBSS",]$Trait)
  SSSS_example_trait_mean <- mean(fertility_subset[fertility_subset$Strain == "SSSS",]$Trait)
  SSSS_example_trait_sem <- calculate_sem(fertility_subset[fertility_subset$Strain == "SSSS",]$Trait)
  
  #assign means to objects
  assign(paste("BBBB", trait, "mean", sep = "_"), BBBB_example_trait_mean, envir = .GlobalEnv)
  assign(paste("BBSS", trait, "mean", sep = "_"), BBBB_example_trait_mean, envir = .GlobalEnv)
  assign(paste("SSSS", trait, "mean", sep = "_"), BBBB_example_trait_mean, envir = .GlobalEnv)
  
  #assign SEMs to objects
  assign(paste("BBBB", trait, "sem", sep = "_"), BBBB_example_trait_sem, envir = .GlobalEnv)
  assign(paste("BBSS", trait, "sem", sep = "_"), BBBB_example_trait_sem, envir = .GlobalEnv)
  assign(paste("SSSS", trait, "sem", sep = "_"), BBBB_example_trait_sem, envir = .GlobalEnv)
}

traits_to_plot <- c("BodyWeight", "RelativeTestesWeight", "TestesWeight", "RelativeSVWeight", "NormSpermCount", "Motility", "SVWeight")
corresponding_y_labs <- c("Body weight (g)", "Relative testis weight (mg/g)", "Absolute testis weight (mg)", "Relative SV weight (mg/g)", "Sperm count (1 x 10^6)", "Proportion motile sperm", "Absolute testis weight (mg)")

invisible(mapply(calculate_fertility_differences,
                 traits_to_plot,
                 corresponding_y_labs))

#print summary stats for inclusion in results
paste("P. sungorus males had smaller testes and seminal vesicles (testes = ",
      round(SSSS_TestesWeight_mean, 3), "g +/- ", round(SSSS_TestesWeight_sem,3),
      "; seminal vesicles = ",
      round(SSSS_SVWeight_mean, 3), "g +/- ", round(SSSS_SVWeight_sem,3),
      "; mean +/- se) than P. campbelli males (testes = ",
      round(BBBB_TestesWeight_mean, 3), "g +/- ", round(BBBB_TestesWeight_sem,3),
      "; seminal vesicles = ",
      round(BBBB_SVWeight_mean, 3), "g +/- ", round(BBBB_SVWeight_sem,3),
      "; mean +/- se)(Dunn's Test relative testes weight p = ",
      round(RelativeTestesWeight_dt$res[2,4], 4),
      " and relative seminal vesicle weight p = ",
      round(RelativeSVWeight_dt$res[2,4], 4),
      "; figure X).",
      sep = "")

paste("Parental males did not significantly differ in normalized sperm counts (P. campbelli = ",
      round(BBBB_NormSpermCount_mean,3), " +/- ", round(BBBB_NormSpermCount_sem, 3), " x 10^6",
      "; P. sungorus = ",
      round(SSSS_NormSpermCount_mean,3), " +/- ", round(SSSS_NormSpermCount_sem,3), " x 10^6",
      "); p = ",
      round(NormSpermCount_dt$res[2,4], 4),
      ") or in sperm motility (P. campbelli = ",
      round(BBBB_Motility_mean,3)*100, " +/- ", round(BBBB_Motility_sem,3)*100, "%",
      "; P. sungorus = ",
      round(SSSS_Motility_mean,3)*100, " +/- ", round(SSSS_Motility_sem,3)*100, "%",
      "); p = ",
      round(Motility_dt$res[4], 4),
      ").",
      sep = "")

paste("Hybrid males had smaller testes (",
      round(BBSS_TestesWeight_mean,3), "g +/- ", round(BBSS_TestesWeight_sem,3), "; p = ",
      round(RelativeTestesWeight_dt$res[1,4], 6),
      ") and seminal vesicles (",
      round(BBSS_SVWeight_mean,3), "g +/- ", round(BBSS_SVWeight_sem,3), "; p = ",
      round(RelativeTestesWeight_dt$res[1,4], 4),
      ") than male P. campbelli hamsters and produced almost no mature spermatozoa.",
      sep = "")

#saving fertility plots
#either save or print figure 1
if(exportFigs == TRUE) {
  ggsave(file = "exported_figures/sterility_phenotypes.pdf", plot_grid(
    RelativeTestesWeight_plot,
    RelativeSVWeight_plot,
    NormSpermCount_plot,
    Motility_plot, 
    nrow = 1), 
    width = full_plot_width, height = 2)
} else {
  plot_grid(
    RelativeTestesWeight_plot,
    RelativeSVWeight_plot,
    NormSpermCount_plot,
    Motility_plot, 
    nrow = 1)
}

if(exportFigs == TRUE) {
  ggsave(file = "exported_figures/hamster_absolute_TW_by_cross.pdf", TestesWeight_plot)
  ggsave(file = "exported_figures/hamster_bodyWeight_by_cross.pdf", BodyWeight_plot)
  ggsave(file = "exported_figures/hamster_relativeTestesWeight_by_cross.pdf", RelativeTestesWeight_plot)
  ggsave(file = "exported_figures/hamster_relativeSVWeight_by_cross.pdf", RelativeSVWeight_plot)
  ggsave(file = "exported_figures/hamster_motility_by_cross.pdf", Motility_plot)
  ggsave(file = "exported_figures/hamster_spermCount_by_cross.pdf", NormSpermCount_plot)
}

#exporting cleaned data for table S1
cleaned_fertility_data <- data.frame(Strain = gsub("_.*", "", fertility$ID),
                                     Family = gsub(".*_(.*)-.*", "\\1", fertility$ID),
                                     Individual = gsub(".*-", "", fertility$ID),
                                     Age = fertility$Age,
                                     BodyWeight = round(fertility$BodyWeight, 1),
                                     TestesWeight = round(fertility$TestesWeight, 2),
                                     RelativeTestesWeight = round(fertility$RelativeTestesWeight, 2),
                                     SVWeight = round(fertility$SVWeight, 2),
                                     RelativeSVWeight = round(fertility$RelativeSVWeight, 2),
                                     Motility = round(fertility$Motility, 2),
                                     NormalizedSpermCount = round(fertility$NormSpermCount, 0))

write.csv(cleaned_fertility_data, file = "exported_files/cleaned_hamster_ferility_data.csv")

#cleanup; remove unused objects and plots
rm(list = ls(pattern = "_mean"))
ls(pattern = "^[B,S].*_sem$")

rm(list = ls(pattern = "_rho"))
rm(list = ls(pattern = "_corr"))
rm(list = ls(pattern = "_dt"))
```

<h3>Gene expression settings</h3>
```{r settings-and-base-expression-stats}
#set thresholds for determining if a gene is "expressed"; needs to have an rpkm of at least 1 in at least 3 replicates
rpkm.cutoff <- 1
min.reps <- 3

#methods used for normalizing data; RLE used in manuscript
norm.method <- "RLE"
#norm.method <- "TMM"

#define orders of datasets, stages, and genotypes for factor purposes; WT not used in this manuscript but is present in countfiles
#BBBB = parental P. campbelli, BBSS = F1 campbelli x sungorus males, SSSS = parental P. sungorus
stages <- c("SP", "LZ", "DIP", "RS")
genotypes <- c("BBBB", "SSSS","BBSS")

#printing for record keeping purposes then cleanup unused values
paste("countfile used:", count_file.path, sep = " ")
paste("metadata file used:", metadata_file.path, sep = " ")
rm(count_file.path, metadata_file.path)

#printing for record keeping purposes
print("Parameters Used:")
paste("All expressed genes have a minimum rpkm of:", rpkm.cutoff, sep = " ")
paste("An expressed gene must be expressed in at least", min.reps, "samples", sep = " ")
paste("Counts are normalized with:", norm.method, sep = " ")

#make DGEs for sorted cell data (i.e. remove whole testes samples which are not used in hybrid analyses)
just_counts <- just_counts[,grep("WT", colnames(just_counts), invert = TRUE)]
metadata <- metadata[grep("WT", rownames(metadata), invert = TRUE),]

dgeHamster <- DGEList(just_counts, genes=annotations, group=metadata$condition_name)
paste("Expressed genes in sorted cell dataset (no cutoffs):", nrow(dgeHamster), sep = " ")

#create function normalize_dge which reads in a dge, applies the rpkm and minimum replicate cutoffs for determining if a gene is expressed, then it re-normalizes the data using the normalization method specified above
normalize_dge <- function(dge_name) {
  keep <- rowSums(rpkm(dge_name) >= rpkm.cutoff) >= min.reps 
  dge_name <- dge_name[keep, keep.lib.sizes=FALSE]
  dge_name <- calcNormFactors(dge_name, method = norm.method)
}

#filter DGEs with genes meeting minimum requirements using normalize_dge function defined above; Dev throughout refers to dataset consisting of sorted developmental stages (i.e., not including whole testes samples)
dgeHamster <- normalize_dge(dgeHamster)

#print expressed genes for each DGE- post filter
paste("Expressed genes in sorted cell dataset (with filter):", nrow(dgeHamster), sep = " ")
```

<h3>Categorize expressed genes by chromosome</h3>
```{r categorize-genes-by-chromosome}
#the X chromosome is ScmyWZ3_7747_HRSCAF_7900; this is defining the recombining and non-recombining arms of the dwarf hamster X chromosome
Auto_genes <- annotations[annotations$X_category == "Auto",]$gene
X_genes <- annotations[annotations$X_category == "X",]$gene

filtered_allAuto_genes <- Auto_genes[Auto_genes %in% dgeHamster$genes$gene]
filtered_justX_genes <- X_genes[X_genes %in% dgeHamster$genes$gene]

#subset count data into counts for genes on all Autosomes, on the X, and the subsets of the X (counts are not normalized)
just_Auto_counts <- just_counts[rownames(just_counts) %in% Auto_genes,]
just_X_counts <- just_counts[rownames(just_counts) %in%   X_genes,]

#summary line
paste("Annotation stats:", length(rownames(annotations)), "Auto_genes:", length(Auto_genes), "X_genes:", length(X_genes), sep = " ")

#number of expressed genes in combined data
hamster_numbers <- paste("total_genes:", length(dgeHamster$genes$gene), 
                         "; Auto_genes:", sum(dgeHamster$genes$gene %in% Auto_genes), 
                         "; X_genes:", sum(dgeHamster$genes$gene %in% X_genes), 
                         "; x/a=", round(sum(dgeHamster$genes$gene %in% X_genes)/
                                           length(dgeHamster$genes$gene)*100, 2),  
                         sep = " ")

kable(rbind(c(length(rownames(annotations)), 
              length(Auto_genes), 
              length(X_genes), 
              round(100*length(X_genes)/
                      length(rownames(annotations)), 2)),
            c(length(dgeHamster$genes$gene), 
              sum(dgeHamster$genes$gene %in% Auto_genes), 
              sum(dgeHamster$genes$gene %in% X_genes), 
              round(sum(dgeHamster$genes$gene %in% X_genes)/
                      length(dgeHamster$genes$gene)*100,2))), 
      "simple") 

paste("Total expressed genes:", hamster_numbers)
#cleanup
rm(hamster_numbers)
```

<h3>Make all subsetted DGEs for MDSplots using all genes</h3>
```{r dge-subsets}
#make all subsetted DGEs for MDSplots by specifying which countfiles, annotations, and grouping info to use for each dge then normalize that dge with the above normalize_dge function

#subsetted by spermatogenic stage
dgeSP <- DGEList(just_counts[,grep("SP",colnames(just_counts))], genes=annotations, group=grep("SP",metadata$condition_name, value = TRUE))
dgeSP <- normalize_dge(dgeSP)

dgeLZ <- DGEList(just_counts[,grep("LZ",colnames(just_counts))], genes=annotations, group=grep("LZ",metadata$condition_name, value = TRUE))
dgeLZ <- normalize_dge(dgeLZ)

dgeDIP <- DGEList(just_counts[,grep("DIP",colnames(just_counts))], genes=annotations, group=grep("DIP",metadata$condition_name, value = TRUE))
dgeDIP <- normalize_dge(dgeDIP)

dgeRS <- DGEList(just_counts[,grep("RS",colnames(just_counts))], genes=annotations, group=grep("RS",metadata$condition_name, value = TRUE))
dgeRS <- normalize_dge(dgeRS)

#subsetted by genotype
dgeSSSS <- DGEList(just_counts[,grep("SSSS",colnames(just_counts))], genes=annotations, group=grep("SSSS-[D,L,S,R]",metadata$condition_name, value = TRUE))
dgeSSSS <- normalize_dge(dgeSSSS)

dgeBBSS <- DGEList(just_counts[,grep("BBSS",colnames(just_counts))], genes=annotations, group=grep("BBSS-[D,L,S,R]",metadata$condition_name, value = TRUE))
dgeBBSS <- normalize_dge(dgeBBSS)

dgeBBBB <- DGEList(just_counts[,grep("BBBB",colnames(just_counts))], genes=annotations, group=grep("BBBB-[D,L,S,R]",metadata$condition_name, value = TRUE))
dgeBBBB <- normalize_dge(dgeBBBB)

#sex chromosome/autosome DGEs
dgeX <- DGEList(just_X_counts, 
                genes=annotations[annotations$gene %in% X_genes,], 
                group=metadata$condition_name)
dgeX <- normalize_dge(dgeX)

dgeAuto <- DGEList(just_Auto_counts, 
                   genes=annotations[annotations$gene %in% Auto_genes,], 
                   group=metadata$condition_name)
dgeAuto <- normalize_dge(dgeAuto)
```

<h3>Define the base order of the heatmap columns</h3>
```{r heatmap-settings}
#formatting metadata
metadata$cross <- factor(metadata$cross, levels = genotypes)
genotype_order <- seq(1,3)[metadata$cross]
metadata$stage <- factor(metadata$stage, levels = stages)
stage_order <- seq(1,4)[metadata$stage]

#heatmap order for all samples from Hamster dataset and just sorted cells
Full_heatmap_order <- rownames(metadata[order(stage_order, genotype_order),])
rm(genotype_order, stage_order)

#heatmap order by genotype
SSSS_heatmap_order <- grep("SSSS", Full_heatmap_order, value = TRUE)
BBBB_heatmap_order <- grep("BBBB", Full_heatmap_order, value = TRUE)
BBSS_heatmap_order <- grep("BBSS", Full_heatmap_order, value = TRUE)

#heatmap order by stage
SP_heatmap_order <- grep("SP", Full_heatmap_order, value = TRUE)
LZ_heatmap_order <- grep("LZ", Full_heatmap_order, value = TRUE)
DIP_heatmap_order <- grep("DIP", Full_heatmap_order, value = TRUE)
RS_heatmap_order <- grep("RS", Full_heatmap_order, value = TRUE)

#define color palette for all heatmaps
colfunc <- colorRampPalette(c("#F6F3F3", "#231A1A"))
browns <- colfunc(255)
rm(colfunc)
```

<h3>Set up CPM, rpkm, and Normalized FPKM data frames - not subsetted</h3>
```{r create-count-dfs-not-subsetted}
#counts per million dataframes for the whole dataset and by sex chromosome
cpm.log <- cpm(dgeHamster, log = TRUE)
cpm_justX.log <- subset(cpm.log, rownames(cpm.log) %in% X_genes)
cpm_allAuto.log <- subset(cpm.log, rownames(cpm.log) %in% Auto_genes)

#reads of kilobase of transcript, per million mapped reads (rpkm) dataframes for each of the three datasets (Ham, dev, wt) and for each by sex chromosome
rpkm.nonLog <- as.data.frame(rpkm(dgeHamster, normalized.lib.sizes = TRUE, log = FALSE))
rpkm_justX.nonLog <- subset(rpkm.nonLog, rownames(rpkm.nonLog) %in% X_genes)
rpkm_allAuto.nonLog <- subset(rpkm.nonLog, rownames(rpkm.nonLog) %in% Auto_genes)

#log2 reads of kilobase of transcript, per million mapped reads (log2(rpkm)) dataframes for each of the three datasets (Ham, dev, wt) and for eac dataset by sex chromosome
rpkm.log  <- as.data.frame(rpkm(dgeHamster, normalized.lib.sizes = TRUE, log = TRUE))
rpkm_justX.log <- subset(rpkm.log, rownames(rpkm.log) %in% X_genes)
rpkm_allAuto.log <- subset(rpkm.log, rownames(rpkm.log) %in% Auto_genes)

#normalized reads of kilobase of transcript, per million mapped reads so that sum of squares is 1 (normalized rpkm) dataframes for each of the three datasets (Ham, dev, wt) and for each dataset by sex chromosome
rpkm.norm  <- decostand(as.data.frame(rpkm(dgeHamster,  normalized.lib.sizes = TRUE)), "normalize")
rpkm_justX.norm <- subset(rpkm.norm, rownames(rpkm.norm) %in% X_genes)
rpkm_allAuto.norm <- subset(rpkm.norm, rownames(rpkm.norm) %in% Auto_genes)
```

<h3>Set up CPM, rpkm, and Normalized FPKM data frames - subsetted by stage</h3>
```{r create-count-dfs-subsetted-by-stage}
#create a function to subet cpm, rpkm, and rpkm.norm dataframes for each stage with both genotypes
subset_counts_by_stage <- function(stage) {
  #subset cpm/rpkm/rpkm.norm dataframes by stage of interest
  cpm_example <- cpm.log[,grep(stage, colnames(cpm.log))]
  rpkm_example <- rpkm.nonLog[,grep(stage, colnames(rpkm.nonLog))]
  rpkm.norm_example <- rpkm.norm[,grep(stage, colnames(rpkm.norm))]
  
  #assign subsets to new objects then print out the list of objects created
  assign(paste("cpm", stage, sep = "_"), cpm_example, envir = .GlobalEnv)
  assign(paste("rpkm", stage, sep = "_"), rpkm_example, envir = .GlobalEnv)
  assign(paste(paste("rpkm", stage, sep = "_"),"norm", sep = "."), rpkm.norm_example, envir = .GlobalEnv)
  
  return(paste("Created the following objects:", 
               paste("cpm", stage, sep = "_"),
               paste("rpkm", stage, sep = "_"),
               paste(paste("rpkm", stage, sep = "_"),"norm", sep = "."),
               sep = " "))
}

#apply function to all stages
lapply(stages, subset_counts_by_stage)
```

<h3>Set up CPM, rpkm, and Normalized FPKM data frames - subsetted by cross</h3>
```{r create-count-dfs-subsetted-by-cross}
#create a function to subet cpm, rpkm, and rpkm.norm dataframes for each genotype with all crosses
subset_counts_by_genotype <- function(genotype) {
  #subset cpm/rpkm/rpkm.norm dataframes by genotype of interest
  cpm_example <- cpm.log[,grep(genotype, colnames(cpm.log))]
  rpkm_example <- rpkm.nonLog[,grep(genotype, colnames(rpkm.nonLog))]
  rpkm.norm_example <- rpkm.norm[,grep(genotype, colnames(rpkm.norm))]
  
  #assign subsets to new objects then print out the list of objects created
  assign(paste("cpm", genotype, sep = "_"), cpm_example, envir = .GlobalEnv)
  assign(paste("rpkm", genotype, sep = "_"), rpkm_example, envir = .GlobalEnv)
  assign(paste(paste("rpkm", genotype, sep = "_"),"norm", sep = "."), rpkm.norm_example, envir = .GlobalEnv)
  
  return(paste("Created the following objects:", 
               paste("cpm", genotype, sep = "_"),
               paste("rpkm", genotype, sep = "_"),
               paste(paste("rpkm", genotype, sep = "_"),"norm", sep = "."),
               sep = " "))
}

#apply function to all genotypes
lapply(genotypes, subset_counts_by_genotype)
```

<h3>Set up CPM, rpkm, and Normalized FPKM data frames - subsetted by genotype and stage but not by chromosome</h3>
```{r create-count-dfs-subsetted-by-cross-and-stage}
#create a function to subet cpm, rpkm, and rpkm.norm dataframes for each genotype and each cross
subset_counts_by_genotype_and_stage <- function(genotype, stage) {
  
  #subset cpm/rpkm/rpkm.norm dataframes by genotype and stage of interest
  cpm_example <- cpm.log[,grep(paste(genotype, stage, sep = ".*"), colnames(cpm.log))]
  rpkm_example <- rpkm.nonLog[,grep(paste(genotype, stage, sep = ".*"), colnames(rpkm.nonLog))]
  rpkm.norm_example <- rpkm.norm[,grep(paste(genotype, stage, sep = ".*"), colnames(rpkm.norm))]
  
  #assign subsets to new objects then print out the list of objects created
  assign(paste("cpm", genotype, stage, sep = "_"), cpm_example, envir = .GlobalEnv)
  assign(paste("rpkm", genotype, stage, sep = "_"), rpkm_example, envir = .GlobalEnv)
  assign(paste(paste("rpkm", genotype, stage, sep = "_"),"norm", sep = "."), rpkm.norm_example, envir = .GlobalEnv)
  
  return(paste("Created the following objects:", 
               paste("cpm", genotype, stage, sep = "_"),
               paste("rpkm", genotype, stage, sep = "_"),
               paste(paste("rpkm", genotype, stage, sep = "_"),"norm", sep = "."),
               sep = " "))
}

#apply function to all genotypes and stages
mapply(subset_counts_by_genotype_and_stage, expand.grid(genotypes, stages)$Var1, expand.grid(genotypes, stages)$Var2)
```

<h3>Create sex chromosomes subsets by genotype and genotype x stage</h3>
```{r create-count-dfs-justX-by-cross-and-stage}
#create a function to subet cpm, rpkm, and rpkm.norm dataframes for each genotype with all crosses - justX
subset_counts_by_genotype_justX <- function(genotype) {
  #subset cpm/rpkm/rpkm.norm dataframes by genotype of interest
  cpm_example <- cpm_justX.log[,grep(genotype, colnames(cpm_justX.log))]
  rpkm_example <- rpkm_justX.nonLog[,grep(genotype, colnames(rpkm_justX.nonLog))]
  rpkm.norm_example <- rpkm_justX.norm[,grep(genotype, colnames(rpkm_justX.norm))]
  
  #assign subsets to new objects then print out the list of objects created
  assign(paste("cpm", genotype, "justX", sep = "_"), cpm_example, envir = .GlobalEnv)
  assign(paste("rpkm", genotype, "justX", sep = "_"), rpkm_example, envir = .GlobalEnv)
  assign(paste(paste("rpkm", genotype, "justX", sep = "_"),"norm", sep = "."), rpkm.norm_example, envir = .GlobalEnv)
  
  return(paste("Created the following objects:", 
               paste("cpm", genotype, "justX", sep = "_"),
               paste("rpkm", genotype, "justX", sep = "_"),
               paste(paste("rpkm", genotype, "justX", sep = "_"),"norm", sep = "."),
               sep = " "))
}

#apply function to all genotypes
lapply(genotypes, subset_counts_by_genotype_justX)


#create a function to subet cpm, rpkm, and rpkm.norm dataframes for each genotype and each cross
subset_counts_by_genotype_and_stage_justX <- function(genotype, stage) {
  
  #subset cpm/rpkm/rpkm.norm dataframes by genotype and stage of interest
  cpm_example <- cpm_justX.log[,grep(paste(genotype, stage, sep = ".*"), colnames(cpm_justX.log))]
  rpkm_example <- rpkm_justX.nonLog[,grep(paste(genotype, stage, sep = ".*"), colnames(rpkm_justX.nonLog))]
  rpkm.norm_example <- rpkm_justX.norm[,grep(paste(genotype, stage, sep = ".*"), colnames(rpkm_justX.norm))]
  
  #assign subsets to new objects then print out the list of objects created
  assign(paste("cpm", genotype, stage, "justX", sep = "_"), cpm_example, envir = .GlobalEnv)
  assign(paste("rpkm", genotype, stage, "justX", sep = "_"), rpkm_example, envir = .GlobalEnv)
  assign(paste(paste("rpkm", genotype, stage, "justX", sep = "_"),"norm", sep = "."), rpkm.norm_example, envir = .GlobalEnv)
  
  return(paste("Created the following objects:", 
               paste("cpm", genotype, stage, "justX", sep = "_"),
               paste("rpkm", genotype, stage, "justX", sep = "_"),
               paste(paste("rpkm", genotype, stage, "justX", sep = "_"),"norm", sep = "."),
               sep = " "))
}

#apply function to all genotypes and stages
mapply(subset_counts_by_genotype_and_stage_justX, expand.grid(genotypes, stages)$Var1, expand.grid(genotypes, stages)$Var2)
```

```{r create-count-dfs-allAuto-by-cross-and-stage}
#create a function to subet cpm, rpkm, and rpkm.norm dataframes for each genotype with all crosses - allAuto
subset_counts_by_genotype_allAuto <- function(genotype) {
  #subset cpm/rpkm/rpkm.norm dataframes by genotype of interest
  cpm_example <- cpm_allAuto.log[,grep(genotype, colnames(cpm_allAuto.log))]
  rpkm_example <- rpkm_allAuto.nonLog[,grep(genotype, colnames(rpkm_allAuto.nonLog))]
  rpkm.norm_example <- rpkm_allAuto.norm[,grep(genotype, colnames(rpkm_allAuto.norm))]
  
  #assign subsets to new objects then print out the list of objects created
  assign(paste("cpm", genotype, "allAuto", sep = "_"), cpm_example, envir = .GlobalEnv)
  assign(paste("rpkm", genotype, "allAuto", sep = "_"), rpkm_example, envir = .GlobalEnv)
  assign(paste(paste("rpkm", genotype, "allAuto", sep = "_"),"norm", sep = "."), rpkm.norm_example, envir = .GlobalEnv)
  
  return(paste("Created the following objects:", 
               paste("cpm", genotype, "allAuto", sep = "_"),
               paste("rpkm", genotype, "allAuto", sep = "_"),
               paste(paste("rpkm", genotype, "allAuto", sep = "_"),"norm", sep = "."),
               sep = " "))
}

#apply function to all genotypes
lapply(genotypes, subset_counts_by_genotype_allAuto)


#create a function to subet cpm, rpkm, and rpkm.norm dataframes for each genotype and each cross
subset_counts_by_genotype_and_stage_allAuto <- function(genotype, stage) {
  
  #subset cpm/rpkm/rpkm.norm dataframes by genotype and stage of interest
  cpm_example <- cpm_allAuto.log[,grep(paste(genotype, stage, sep = ".*"), colnames(cpm_allAuto.log))]
  rpkm_example <- rpkm_allAuto.nonLog[,grep(paste(genotype, stage, sep = ".*"), colnames(rpkm_allAuto.nonLog))]
  rpkm.norm_example <- rpkm_allAuto.norm[,grep(paste(genotype, stage, sep = ".*"), colnames(rpkm_allAuto.norm))]
  
  #assign subsets to new objects then print out the list of objects created
  assign(paste("cpm", genotype, stage, "allAuto", sep = "_"), cpm_example, envir = .GlobalEnv)
  assign(paste("rpkm", genotype, stage, "allAuto", sep = "_"), rpkm_example, envir = .GlobalEnv)
  assign(paste(paste("rpkm", genotype, stage, "allAuto", sep = "_"),"norm", sep = "."), rpkm.norm_example, envir = .GlobalEnv)
  
  return(paste("Created the following objects:", 
               paste("cpm", genotype, stage, "allAuto", sep = "_"),
               paste("rpkm", genotype, stage, "allAuto", sep = "_"),
               paste(paste("rpkm", genotype, stage, "allAuto", sep = "_"),"norm", sep = "."),
               sep = " "))
}

#apply function to all genotypes and stages
mapply(subset_counts_by_genotype_and_stage_allAuto, expand.grid(genotypes, stages)$Var1, expand.grid(genotypes, stages)$Var2)
```

<h3>Expression Levels for each Stage/Genotype </h3>
```{r rpkm-by-cell-definitions}
#create genotype x stage rpkm dataframes (with the different normalization methods)
cell_type_list <- c("BBBB.*SP", "BBSS.*SP", "SSSS.*SP", "BBBB.*LZ", "BBSS.*LZ", "SSSS.*LZ", "BBBB.*DIP", "BBSS.*DIP", "SSSS.*DIP", "BBBB.*RS", "SSSS.*RS")
cell_type_names <- c("BBBB_SP", "BBSS_SP", "SSSS_SP", "BBBB_LZ", "BBSS_LZ", "SSSS_LZ", "BBBB_DIP", "BBSS_DIP", "SSSS_DIP", "BBBB_RS", "SSSS_RS")

#un-normalized rpkm
#set up empty dataframe
rpkm_by_cell_type <- data.frame(chr = annotations[annotations$gene %in% rownames(dgeHamster$genes),]$chr, 
                                gene = annotations[annotations$gene %in% rownames(dgeHamster$genes),]$gene,
                                X_category = annotations[annotations$gene %in% rownames(dgeHamster$genes),]$X_category,
                                row.names = annotations[annotations$gene %in% rownames(dgeHamster$genes),]$gene)

#define a function to get the average expression of each gene within a given cell type and append it to the rpkm_by_cell_type dataframe 
get_cell_type_mean_rpkm <- function(cell_type) {
  rpkm_by_cell_type <<- cbind(rpkm_by_cell_type, 
                              as.data.frame(rowMeans(rpkm.nonLog[,grep(cell_type, colnames(rpkm.nonLog))])))
}

#apply the get_cell_type_mean_rpkm across all cell types then rename colnames
invisible(lapply(cell_type_list, get_cell_type_mean_rpkm))
colnames(rpkm_by_cell_type) <- c("chr", "gene", "X_category", cell_type_names)
rpkm_by_cell_type_long <- data.frame(pivot_longer(rpkm_by_cell_type, 
                                                  cols = grep("^[B,S]", colnames(rpkm_by_cell_type)),
                                                  names_to = "cell_type",
                                                  values_to = "rpkm"))
rpkm_by_cell_type_long <- separate(rpkm_by_cell_type_long, 
                                   col = c("cell_type"), 
                                   into = c("genotype", "stage"), 
                                   remove = FALSE)
rpkm_by_cell_type_long$genotype <- factor(rpkm_by_cell_type_long$genotype,
                                          c("BBBB", "BBSS", "SSSS"))
rpkm_by_cell_type_long$stage <- factor(rpkm_by_cell_type_long$stage,
                                       c("SP", "LZ", "DIP", "RS"))
rpkm_by_cell_type_long$X_category <- factor(rpkm_by_cell_type_long$X_category,
                                            c("Auto", "X"))




#log-normalized rpkm
#set up empty dataframe
rpkm_log_by_cell_type <- data.frame(chr = annotations[annotations$gene %in% rownames(dgeHamster$genes),]$chr, 
                                    gene = annotations[annotations$gene %in% rownames(dgeHamster$genes),]$gene,
                                    X_category = annotations[annotations$gene %in% rownames(dgeHamster$genes),]$X_category,
                                    row.names = annotations[annotations$gene %in% rownames(dgeHamster$genes),]$gene)

#define a function to get the average expression of each gene within a given cell type and append it to the rpkm_log_by_cell_type dataframe 
get_cell_type_mean_rpkm_log <- function(cell_type) {
  rpkm_log_by_cell_type <<- cbind(rpkm_log_by_cell_type, 
                                  as.data.frame(rowMeans(rpkm.log[,grep(cell_type, colnames(rpkm.log))])))
}

#apply the get_cell_type_mean_rpkm_log across all cell types then rename colnames
invisible(lapply(cell_type_list, get_cell_type_mean_rpkm_log))
colnames(rpkm_log_by_cell_type) <- c("chr", "gene", "X_category", cell_type_names)
rpkm_log_by_cell_type_long <- data.frame(pivot_longer(rpkm_log_by_cell_type, 
                                                      cols = grep("^[B,S]", colnames(rpkm_log_by_cell_type)),
                                                      names_to = "cell_type",
                                                      values_to = "rpkm"))
rpkm_log_by_cell_type_long <- separate(rpkm_log_by_cell_type_long, 
                                       col = c("cell_type"), 
                                       into = c("genotype", "stage"), 
                                       remove = FALSE)
rpkm_log_by_cell_type_long$genotype <- factor(rpkm_log_by_cell_type_long$genotype,
                                              c("BBBB", "BBSS", "SSSS"))
rpkm_log_by_cell_type_long$stage <- factor(rpkm_log_by_cell_type_long$stage,
                                           c("SP", "LZ", "DIP", "RS"))
rpkm_log_by_cell_type_long$X_category <- factor(rpkm_log_by_cell_type_long$X_category,
                                                c("Auto", "X"))


#decostand normalized rpkm
#set up empty dataframe
rpkm_norm_by_cell_type <- data.frame(chr = annotations[annotations$gene %in% rownames(dgeHamster$genes),]$chr, 
                                     gene = annotations[annotations$gene %in% rownames(dgeHamster$genes),]$gene,
                                     X_category = annotations[annotations$gene %in% rownames(dgeHamster$genes),]$X_category,
                                     row.names = annotations[annotations$gene %in% rownames(dgeHamster$genes),]$gene)

#define a function to get the average expression of each gene within a given cell type and append it to the rpkm_norm_by_cell_type dataframe 
get_cell_type_mean_rpkm_norm <- function(cell_type) {
  rpkm_norm_by_cell_type <<- cbind(rpkm_norm_by_cell_type, 
                                   as.data.frame(rowMeans(rpkm.norm[,grep(cell_type, colnames(rpkm.norm))])))
}

#apply the get_cell_type_mean_rpkm across all cell types then rename colnames
invisible(lapply(cell_type_list, get_cell_type_mean_rpkm_norm))
colnames(rpkm_norm_by_cell_type) <- c("chr", "gene", "X_category", cell_type_names)
rpkm_norm_by_cell_type_long <- data.frame(pivot_longer(rpkm_norm_by_cell_type, 
                                                       cols = grep("^[B,S]", colnames(rpkm_norm_by_cell_type)),
                                                       names_to = "cell_type",
                                                       values_to = "rpkm"))
rpkm_norm_by_cell_type_long <- separate(rpkm_norm_by_cell_type_long, 
                                        col = c("cell_type"), 
                                        into = c("genotype", "stage"), 
                                        remove = FALSE)
rpkm_norm_by_cell_type_long$genotype <- factor(rpkm_norm_by_cell_type_long$genotype,
                                               c("BBBB", "BBSS", "SSSS"))
rpkm_norm_by_cell_type_long$stage <- factor(rpkm_norm_by_cell_type_long$stage,
                                            c("SP", "LZ", "DIP", "RS"))
rpkm_norm_by_cell_type_long$X_category <- factor(rpkm_norm_by_cell_type_long$X_category,
                                                 c("Auto", "X"))


rpkm_by_cell_type_density_plot <- ggplot(rpkm_by_cell_type_long, aes(x=rpkm, group = cell_type, color = stage, linetype = genotype)) +
  geom_density() + 
  theme_minimal() + 
  scale_color_manual(values = c(sp_color, lz_color, dip_color, rs_color)) +
  ylab("Mean Un-normalized FPKM across cell types") +
  theme(axis.title.x=element_blank(), 
        text = element_text(size = text_size*1.5), 
        plot.title = element_text(hjust = 0.5),
        panel.grid.minor = element_blank(),
        panel.grid.major = element_blank(),
        axis.line.x.bottom = element_line(color = "black"),
        axis.line.y.left = element_line(color = "black")) 

rpkm_log_by_cell_type_density_plot <- ggplot(rpkm_log_by_cell_type_long, aes(x=rpkm, group = cell_type, color = stage, linetype = genotype)) +
  geom_density() + 
  theme_minimal() + 
  scale_color_manual(values = c(sp_color, lz_color, dip_color, rs_color)) +
  ylab("Mean Log-normalized FPKM across cell types") +
  theme(axis.title.x=element_blank(), 
        text = element_text(size = text_size*1.5), 
        plot.title = element_text(hjust = 0.5),
        panel.grid.minor = element_blank(),
        panel.grid.major = element_blank(),
        axis.line.x.bottom = element_line(color = "black"),
        axis.line.y.left = element_line(color = "black")) 

rpkm_norm_by_cell_type_long_density_plot <- ggplot(rpkm_norm_by_cell_type_long, aes(x=rpkm, group = cell_type, color = stage, linetype = genotype)) +
  geom_density() + 
  theme_minimal() + 
  scale_color_manual(values = c(sp_color, lz_color, dip_color, rs_color)) +
  ylab("Mean Decostand-normalized FPKM across cell types") +
  theme(axis.title.x=element_blank(), 
        text = element_text(size = text_size*1.5), 
        plot.title = element_text(hjust = 0.5),
        panel.grid.minor = element_blank(),
        panel.grid.major = element_blank(),
        axis.line.x.bottom = element_line(color = "black"),
        axis.line.y.left = element_line(color = "black")) 

rpkm_by_cell_type_density_plot
rpkm_log_by_cell_type_density_plot
rpkm_norm_by_cell_type_long_density_plot
```

<h3>Defining cell population-specific expressed genes (for use in hypergeometric tests of DE enrichment)</h3>
```{r defining-sets-of-expressed-genes}
#get expressed genes meeting filtering requirements from those DGEs you just set up
#expressed across all samples
all_exp <- dgeHamster$genes[c("gene", "mouse_ensemble_ID")] # 21848
#expressed across parents
BBBB_SSSS_all_exp <- rownames(rpkm.nonLog[rowSums(rpkm.nonLog[,grep("BS.", colnames(rpkm.nonLog), value = TRUE, invert = TRUE)] > rpkm.cutoff)  >= min.reps, ]) #21040
#expressed within P. campbelli
BBBB_all_exp <- dgeBBBB$genes[c("gene", "mouse_ensemble_ID")] # 19857
#expressed within hybrids
Hyb_all_exp <- dgeBBSS$genes[c("gene", "mouse_ensemble_ID")] # 19209
#expressed within P. sungorus
SSSS_all_exp <- dgeSSSS$genes[c("gene", "mouse_ensemble_ID")] # 20025

#expressed within hybrids for each stage
Hyb_SP_exp <- annotations[annotations$gene %in% 
                            rownames(rpkm.nonLog[rowSums(rpkm.nonLog[,grep("BBSS.*SP", colnames(rpkm.nonLog))] 
                                                         > rpkm.cutoff) >= min.reps, ]),c("gene", "mouse_ensemble_ID")] # 16087
Hyb_LZ_exp <- annotations[annotations$gene %in% 
                            rownames(rpkm.nonLog[rowSums(rpkm.nonLog[,grep("BBSS.*LZ", colnames(rpkm.nonLog))] 
                                                         > rpkm.cutoff) >= min.reps, ]),c("gene", "mouse_ensemble_ID")]  # 15715
Hyb_DIP_exp <- annotations[annotations$gene %in% 
                             rownames(rpkm.nonLog[rowSums(rpkm.nonLog[,grep("BBSS.*DIP", colnames(rpkm.nonLog))] 
                                                          > rpkm.cutoff) >= min.reps, ]),c("gene", "mouse_ensemble_ID")]  # 14625

#expressed within P. campbelli for each stage
BBBB_SP_exp <- annotations[annotations$gene %in% 
                             rownames(rpkm.nonLog[rowSums(rpkm.nonLog[,grep("BBBB*SP", colnames(rpkm.nonLog))] 
                                                          > rpkm.cutoff) >= min.reps, ]),c("gene", "mouse_ensemble_ID")] # 16232
BBBB_LZ_exp <- annotations[annotations$gene %in% 
                             rownames(rpkm.nonLog[rowSums(rpkm.nonLog[,grep("BBBB*LZ", colnames(rpkm.nonLog))] 
                                                          > rpkm.cutoff) >= min.reps, ]),c("gene", "mouse_ensemble_ID")] # 16746
BBBB_DIP_exp <- annotations[annotations$gene %in% 
                              rownames(rpkm.nonLog[rowSums(rpkm.nonLog[,grep("BBBB*DIP", colnames(rpkm.nonLog))] 
                                                           > rpkm.cutoff) >= min.reps, ]),c("gene", "mouse_ensemble_ID")] # 12413
BBBB_RS_exp <- annotations[annotations$gene %in% 
                             rownames(rpkm.nonLog[rowSums(rpkm.nonLog[,grep("BBBB*RS", colnames(rpkm.nonLog))] 
                                                          > rpkm.cutoff) >= min.reps, ]),c("gene", "mouse_ensemble_ID")] # 14442

#expressed within P. sungorus for each stage
SSSS_SP_exp <- annotations[annotations$gene %in% 
                             rownames(rpkm.nonLog[rowSums(rpkm.nonLog[,grep("SSSS*SP", colnames(rpkm.nonLog))] 
                                                          > rpkm.cutoff) >= min.reps, ]),c("gene", "mouse_ensemble_ID")] # 17023
SSSS_LZ_exp <- annotations[annotations$gene %in% 
                             rownames(rpkm.nonLog[rowSums(rpkm.nonLog[,grep("SSSS*LZ", colnames(rpkm.nonLog))] 
                                                          > rpkm.cutoff) >= min.reps, ]),c("gene", "mouse_ensemble_ID")]  # 16148
SSSS_DIP_exp <- annotations[annotations$gene %in% 
                              rownames(rpkm.nonLog[rowSums(rpkm.nonLog[,grep("SSSS*DIP", colnames(rpkm.nonLog))] 
                                                           > rpkm.cutoff) >= min.reps, ]),c("gene", "mouse_ensemble_ID")]  # 13113
SSSS_RS_exp <- annotations[annotations$gene %in% 
                             rownames(rpkm.nonLog[rowSums(rpkm.nonLog[,grep("SSSS*RS", colnames(rpkm.nonLog))] 
                                                          > rpkm.cutoff) >= min.reps, ]),c("gene", "mouse_ensemble_ID")]  # 13454

#expressed within hybrids and P.campbelli for each stage
Hyb_BB_SP_exp <-  annotations[annotations$gene %in% 
                                rownames(rpkm.nonLog[rowSums(rpkm.nonLog[,grep("BB.*SP", colnames(rpkm.nonLog))] 
                                                             > rpkm.cutoff) >= min.reps, ]),c("gene", "mouse_ensemble_ID")] # 17980
Hyb_BB_LZ_exp <-  annotations[annotations$gene %in% 
                                rownames(rpkm.nonLog[rowSums(rpkm.nonLog[,grep("BB.*LZ", colnames(rpkm.nonLog))] 
                                                             > rpkm.cutoff) >= min.reps, ]),c("gene", "mouse_ensemble_ID")] # 17786
Hyb_BB_DIP_exp <- annotations[annotations$gene %in% 
                                rownames(rpkm.nonLog[rowSums(rpkm.nonLog[,grep("BB.*DIP", colnames(rpkm.nonLog))] 
                                                             > rpkm.cutoff) >= min.reps, ]),c("gene", "mouse_ensemble_ID")] # 15792

#expressed within hybrids and P. sungorus for each stage
Hyb_SS_SP_exp <-  annotations[annotations$gene %in% 
                                rownames(rpkm.nonLog[rowSums(rpkm.nonLog[,grep("SS.*SP", colnames(rpkm.nonLog))] 
                                                             > rpkm.cutoff) >= min.reps, ]),c("gene", "mouse_ensemble_ID")] # 18250
Hyb_SS_LZ_exp <-  annotations[annotations$gene %in% 
                                rownames(rpkm.nonLog[rowSums(rpkm.nonLog[,grep("SS*LZ", colnames(rpkm.nonLog))] 
                                                             > rpkm.cutoff) >= min.reps, ]),c("gene", "mouse_ensemble_ID")] # 17536
Hyb_SS_DIP_exp <- annotations[annotations$gene %in% 
                                rownames(rpkm.nonLog[rowSums(rpkm.nonLog[,grep("SS*DIP", colnames(rpkm.nonLog))] 
                                                             > rpkm.cutoff) >= min.reps, ]),c("gene", "mouse_ensemble_ID")] # 15860

#expressed within both parent species for each stage
BBBB_SSSS_SP_exp <-  annotations[annotations$gene %in% 
                                   rownames(rpkm.nonLog[rowSums(rpkm.nonLog[,grep("SP", grep("BS.", colnames(rpkm.nonLog), value = TRUE, invert = TRUE), value = TRUE)] > rpkm.cutoff)  >= min.reps, ]), c("gene", "mouse_ensemble_ID")] # 18289
BBBB_SSSS_LZ_exp <-  annotations[annotations$gene %in% 
                                   rownames(rpkm.nonLog[rowSums(rpkm.nonLog[,grep("LZ", grep("BS.", colnames(rpkm.nonLog), value = TRUE, invert = TRUE), value = TRUE)] > rpkm.cutoff)  >= min.reps, ]), c("gene", "mouse_ensemble_ID")] # 17893
BBBB_SSSS_DIP_exp <- annotations[annotations$gene %in% 
                                   rownames(rpkm.nonLog[rowSums(rpkm.nonLog[,grep("DIP", grep("BS.", colnames(rpkm.nonLog), value = TRUE, invert = TRUE), value = TRUE)] > rpkm.cutoff)  >= min.reps, ]), c("gene", "mouse_ensemble_ID")] # 14175
BBBB_SSSS_RS_exp <- annotations[annotations$gene %in% 
                                  rownames(rpkm.nonLog[rowSums(rpkm.nonLog[,grep("RS", grep("BS.", colnames(rpkm.nonLog), value = TRUE, invert = TRUE), value = TRUE)] > rpkm.cutoff)  >= min.reps, ]), c("gene", "mouse_ensemble_ID")]  # 15901

#make lists of cell types for hypergeometric tests below
Hyb_BB_cell_exp_list <- c("Hyb_BB_SP_exp", "Hyb_BB_LZ_exp", "Hyb_BB_DIP_exp")
Hyb_SS_cell_exp_list <- c("Hyb_SS_SP_exp", "Hyb_SS_LZ_exp", "Hyb_SS_DIP_exp")
```

<h3>Define sets of induced genes for each stage</h3>
```{r defining-induced}
#define sets of genes "induced" in each cell type; these are genes with a median expression in a given stage that is 2 (induced_cutoff) times higher than the median expression in all other cell types
induced_cutoff <- 2

define_induced <- function(stage) {
  focal_stages <- grep(stage, cell_type_names, value = TRUE)
  nonFocal_stages <- grep(stage, cell_type_names, value = TRUE, invert = TRUE)
  expressed_genes <- get(paste("BBBB_SSSS_", stage, "_exp", sep = ""))[,1]
  
  #finds genes expressed in both parents in a given stage and those that meet the expression cutoff
  example_induced <- rownames(rpkm_by_cell_type[rownames(rpkm_by_cell_type) %in% expressed_genes
                                                & rowMedians(as.matrix(rpkm_by_cell_type[,focal_stages])) >
                                                  rowMedians(as.matrix(rpkm_by_cell_type[,nonFocal_stages]))*induced_cutoff, ])
  example_Auto_induced <- example_induced[!(example_induced %in% Auto_genes)]
  example_X_induced <- example_induced[(example_induced %in% X_genes)]
  
  
  assign(paste(stage, "induced", sep = "_"), example_induced, envir = .GlobalEnv)
  assign(paste(stage, "Auto_induced", sep = "_"), example_Auto_induced, envir = .GlobalEnv)
  assign(paste(stage, "X_induced", sep = "_"), example_X_induced, envir = .GlobalEnv)
  
  write.table(example_induced, 
              file = paste("induced/hamster_", stage ,"_induced_cutoff_", induced_cutoff,".txt", sep = ""), 
              sep = "\t", quote = FALSE, row.names = FALSE, col.names = FALSE)
  write.table(example_Auto_induced, 
              file = paste("induced/hamster_", stage ,"_induced_X_cutoff_", induced_cutoff,".txt", sep = ""), 
              sep = "\t", quote = FALSE, row.names = FALSE, col.names = FALSE)
  write.table(example_X_induced, 
              file = paste("induced/hamster_", stage ,"_induced_Auto_cutoff_", induced_cutoff,".txt", sep = ""), 
              sep = "\t", quote = FALSE, row.names = FALSE, col.names = FALSE)
}

invisible(lapply(stages, define_induced))
```

<h3>Establish hamster/mouse orthology</h3>
```{r establish-orthology}
# #uncomment code to establish list of orthologs between hamsters and house mice
# #if the file "mouse_hamster_orthology_29May23.csv" already exists, proceed from the read.csv command
# 
# #read in hamster-mouse blast results from Moore et al. 2022
ham_mouse_orthology <- read.csv("input_data/psun_mmus_blast.csv")

#create a dictionary of orthologs; recomment on next run
orthology_df <- data.frame()

#find the longest orthologous hamster gene for each mouse gene
find_longest_ortholog <- function(mouseGene) {
  hGene_longestMatch <- order(ham_mouse_orthology[ham_mouse_orthology$Mouse_gene == mouseGene,]$Alignment_Length, decreasing = TRUE)[1]
  hamGene <- ham_mouse_orthology[ham_mouse_orthology$Mouse_gene == mouseGene,][hGene_longestMatch,]$Phodopus_gene
  orthology_df <<- rbind(orthology_df, data.frame(mGene = mouseGene, hGene = hamGene))
  
}

invisible(lapply(unique(ham_mouse_orthology$Mouse_gene), find_longest_ortholog))

#write file to exported_files and also input_files
write.csv(orthology_df, file = "exported_files/mouse_hamster_orthology_13Jun24.csv", row.names = FALSE)
write.csv(orthology_df, file = "input_data/mouse_hamster_orthology_13Jun24.csv", row.names = FALSE)
```

<h3>Compare how orthologous genes differ between stages for which stages they are induced in</h3>
```{r induced-orthology}
#establishes comparable sets of orthologous induced genes
orthology_df <- read.csv("input_data/mouse_hamster_orthology_13Jun24.csv")

#create hamster stage dataframes with only genes with orthologs
#initial set up
ham_SP_induced <-  orthology_df[orthology_df$hGene %in% SP_induced,]
ham_LZ_induced <-  orthology_df[orthology_df$hGene %in% LZ_induced,]
ham_DIP_induced <- orthology_df[orthology_df$hGene %in% DIP_induced,]
ham_RS_induced <-  orthology_df[orthology_df$hGene %in% RS_induced,]

#read in stage dataframes from mouse data
mouse_SP_induced <-  read.table(file = paste("induced/mouse_SP_induced_inducedCutoff_", induced_cutoff,".txt", sep = ""))
mouse_LZ_induced <-  read.table(file = paste("induced/mouse_LZ_induced_inducedCutoff_", induced_cutoff,".txt", sep = ""))
mouse_DIP_induced <- read.table(file = paste("induced/mouse_DIP_induced_inducedCutoff_", induced_cutoff,".txt", sep = ""))
mouse_RS_induced <-  read.table(file = paste("induced/mouse_RS_induced_inducedCutoff_", induced_cutoff,".txt", sep = ""))

colnames(mouse_SP_induced) <- "mGene"
colnames(mouse_LZ_induced) <- "mGene"
colnames(mouse_DIP_induced) <- "mGene"
colnames(mouse_RS_induced) <- "mGene"

#modify mouse stage dataframes to include only genes with orthologs
mouse_SP_induced <-  orthology_df[orthology_df$mGene %in% mouse_SP_induced$mGene,]
mouse_LZ_induced <-  orthology_df[orthology_df$mGene %in% mouse_LZ_induced$mGene,]
mouse_DIP_induced <- orthology_df[orthology_df$mGene %in% mouse_DIP_induced$mGene,]
mouse_RS_induced <-  orthology_df[orthology_df$mGene %in% mouse_RS_induced$mGene,]
```

<h3>Compare sets of overexpressed genes during diplotene</h3>
```{r diplotene-escapee-genes}
#threshold for determining parental and hybrid escapee genes; selecting for top 10% of genes then determining what rpkm threshold corresponds to that
threshold <- 0.90
hamster_parental_threshold <- quantile(rowMeans(rpkm_BBBB_DIP_justX.norm), probs = c(threshold))

#define two sets of genes:
#parental_escapees; the top 10% of genes expressed in parental diplotene (MSCI escapees)
hamster_parental_escapees <- names(rowMeans(rpkm_BBBB_DIP_justX.norm)[rowMeans(rpkm_BBBB_DIP_justX.norm) >
                                                                        hamster_parental_threshold])
hamster_parental_escapees_mGene <- orthology_df[orthology_df$hGene %in% hamster_parental_escapees,]$mGene

#hybrid_escapees; the top 10% of genes expressed in hybrid diplotene (overexpressed X-linked genes)
hamster_hybrid_escapees <- names(rowMeans(rpkm_BBSS_DIP_justX.norm)[rowMeans(rpkm_BBSS_DIP_justX.norm) > hamster_parental_threshold])

hamster_hybrid_escapees_mGene <- orthology_df[orthology_df$hGene %in% hamster_hybrid_escapees,]$mGene

#import parental_escapees and hybrid_escapees from mouse data
mouse_parental_escapees <- as.vector(read.table(file = paste("induced/mouse_parental_escapees_inducedCutoff_", 
                                                             induced_cutoff,".txt", sep = ""))$V1)
mouse_hybrid_escapees <- as.vector(read.table(file = paste("induced/mouse_hybrid_escapees_inducedCutoff_", 
                                                           induced_cutoff,".txt", sep = ""))$V1)

#rename to hamster ortholog
mouse_parental_escapees_hGene <-  orthology_df[orthology_df$mGene %in% mouse_parental_escapees,]$hGene
mouse_hybrid_escapees_hGene <-  orthology_df[orthology_df$mGene %in% mouse_hybrid_escapees,]$hGene

#number of genes that escape in common between both species; only 2 (2.6%) - most of the mouse genes have no homolog in the hamster blast results
both_species_parental_escapees_overlap <- intersect(hamster_parental_escapees, mouse_parental_escapees_hGene)

#number of genes that overachieve in common between both species; 62 or about 9%
both_species_hybrid_escapees_overlap <- intersect(hamster_hybrid_escapees, mouse_hybrid_escapees_hGene)
length(both_species_hybrid_escapees_overlap)

#make a dataframe of all of the mouse and hamster parental and hybrid escapees gene counts
both_species_escapee_df <- data.frame(species = rep(c("ham", "mouse", "both"), 2),
                                      type = c(rep("hybrid_escapees", 3), rep("parental_escapees", 3)),
                                      numGnees = c(length(hamster_hybrid_escapees), 
                                                   length(mouse_hybrid_escapees), 
                                                   length(both_species_hybrid_escapees_overlap), 
                                                   length(hamster_parental_escapees),
                                                   length(mouse_parental_escapees),
                                                   length(both_species_parental_escapees_overlap)))

#distribution of hybrid genes across stages for hamsters then mice
hamster_hybrid_escapees_df <- data.frame(stage = stages[1:4],
                                         species = "ham",
                                         overlap = c(length(intersect(hamster_hybrid_escapees, ham_SP_induced$hGene)),
                                                     length(intersect(hamster_hybrid_escapees, ham_LZ_induced$hGene)),
                                                     length(intersect(hamster_hybrid_escapees, ham_DIP_induced$hGene)),
                                                     length(intersect(hamster_hybrid_escapees, ham_RS_induced$hGene))),
                                         num_escapees = length(hamster_hybrid_escapees))

mouse_hybrid_escapees_df <- data.frame(stage = stages[1:4],
                                       species = "mouse",
                                       overlap = c(length(intersect(mouse_hybrid_escapees, mouse_SP_induced$mGene)),
                                                   length(intersect(mouse_hybrid_escapees, mouse_LZ_induced$mGene)),
                                                   length(intersect(mouse_hybrid_escapees, mouse_DIP_induced$mGene)),
                                                   length(intersect(mouse_hybrid_escapees, mouse_RS_induced$mGene))),
                                       num_escapees = length(mouse_hybrid_escapees))

#formatting
hamster_hybrid_escapees_df$stage <- factor(hamster_hybrid_escapees_df$stage, levels = stages[1:4])
hamster_hybrid_escapees_df$species <- factor(hamster_hybrid_escapees_df$species)
mouse_hybrid_escapees_df$stage <- factor(mouse_hybrid_escapees_df$stage, levels = stages[1:4])
mouse_hybrid_escapees_df$species <- factor(mouse_hybrid_escapees_df$species)

#combine both species dataframes into one
both_species_hybrid_escapees_df <- rbind(hamster_hybrid_escapees_df, mouse_hybrid_escapees_df)
both_species_hybrid_escapees_df$species <- factor(both_species_hybrid_escapees_df$species, levels = c("mouse", "ham"))

#bar plot comparing how hybrid escapee genes are conserved/not conserved across spermatogenic stages between species
both_species_hybrid_escapees_plot <- ggplot(both_species_hybrid_escapees_df, aes(fill=stage, y=overlap, x=species)) + 
  geom_bar(position="stack", stat="identity") +
  ylab(label = "Number of genes") +
  scale_fill_manual(values = c(sp_color, lz_color, dip_color, rs_color)) +
  ggtitle("Induced stage identity of overexpressed X genes in hybrid hamsters") +
  theme_minimal() +
  theme(legend.position = "none", 
        panel.grid.major.x = element_blank()) + 
  labs(y = "Cell affiliation of overexpressed X-linked hybrid diplotene genes") + 
  scale_x_discrete(labels = c("House mouse", "Dwarf hamster")) +
  theme(axis.title.x = element_blank())

if(exportFigs == FALSE) {
  print(both_species_hybrid_escapees_plot)
}

paste(mouse_hybrid_escapees_df$overlap, " ", mouse_hybrid_escapees_df$stage, " genes", sep = "")
paste(hamster_hybrid_escapees_df$overlap, " ", hamster_hybrid_escapees_df$stage, " genes", sep = "")
```

<h3>Plot all samples in a dendrogram based on experssion values</h3>
```{r dendrogram}
#make dendrograms of samples
#choose normalization type
#which_norm <- "norm"
#which_norm <- "nonlog"
which_norm <- "log"

if(exportFigs == TRUE) {
  pdf(file = paste("exported_figures/sample_dendrograms_", which_norm,"RPKM.pdf", sep = ""), width = 8.5, height = 11)
}

par(mfrow = c(3,1), mar = c(0, 3, 1.5, 0))
#clustering samples by all genes
Hamster_TreeCluster_allGenes = hclust(dist(t(get(paste("rpkm.", which_norm, sep = "")))), method = "average");
Hamster_TreeCluster_allGenes_dend <- hang.dendrogram(as.dendrogram(Hamster_TreeCluster_allGenes))

dend_colors <- gsub("SP", sp_color, gsub("LZ", lz_color, gsub("DIP", dip_color, gsub("RS", rs_color, gsub(".*_", "", Hamster_TreeCluster_allGenes_dend %>% labels)))))
dend_shapes <- as.numeric(gsub("BBSS", 4, gsub("BBBB", 24, gsub("SSSS", 25, gsub("_.*", "", Hamster_TreeCluster_allGenes_dend %>% labels)))))

Hamster_TreeCluster_allGenes_dend %>% 
  set("leaves_pch", dend_shapes) %>% 
  set("leaves_col", dend_colors) %>% 
  set("leaves_bg", dend_colors) %>% 
  set("leaves_cex", c(2)) %>% 
  set("labels", "") %>% 
  plot(main = "All sample dendrogram based on all genes")


#clustering samples by all autosomal genes
Hamster_TreeCluster_allAuto = hclust(dist(t(get(paste("rpkm_allAuto.", which_norm, sep = "")))), method = "average");
Hamster_TreeCluster_allAuto_dend <- hang.dendrogram(as.dendrogram(Hamster_TreeCluster_allAuto))

#set colors
dend_colors <- gsub("SP", sp_color, gsub("LZ", lz_color, gsub("DIP", dip_color, gsub("RS", rs_color, gsub(".*_", "", Hamster_TreeCluster_allAuto_dend %>% labels)))))
dend_shapes <- as.numeric(gsub("BBSS", 4, gsub("BBBB", 24, gsub("SSSS", 25, gsub("_.*", "", Hamster_TreeCluster_allAuto_dend %>% labels)))))

#plot dendrogram for autosomal genes
suppressWarnings(Hamster_TreeCluster_allAuto_dend %>% 
                   set("leaves_pch", dend_shapes) %>% 
                   set("leaves_col", dend_colors) %>% 
                   set("leaves_bg", dend_colors) %>% 
                   set("leaves_cex", c(2)) %>% 
                   set("labels", "") %>% 
                   plot(main = "All sample dendrogram based on all Auto genes"))

#clustering samples by X genes
Hamster_TreeCluster_justX = hclust(dist(t(get(paste("rpkm_justX.", which_norm, sep = "")))), method = "average");
Hamster_TreeCluster_justX_dend <- hang.dendrogram(as.dendrogram(Hamster_TreeCluster_justX))

#set colors
dend_colors <- gsub("SP", sp_color, gsub("LZ", lz_color, gsub("DIP", dip_color, gsub("RS", rs_color, gsub(".*_", "", Hamster_TreeCluster_justX_dend %>% labels)))))
dend_shapes <- as.numeric(gsub("BBSS", 4, gsub("BBBB", 24, gsub("SSSS", 25, gsub("_.*", "", Hamster_TreeCluster_justX_dend %>% labels)))))

#plot dendrogram for autosomal genes
suppressWarnings(Hamster_TreeCluster_justX_dend %>% 
                   set("leaves_pch", dend_shapes) %>% 
                   set("leaves_col", dend_colors) %>% 
                   set("leaves_bg", dend_colors) %>% 
                   set("leaves_cex", c(2)) %>% 
                   set("labels", "") %>% 
                   plot(main = "All sample dendrogram based on all X genes"))

if(exportFigs == TRUE) {
  dev.off()
}
```

<h3>MDS Plots of each dataset and stage</h3>
```{r mds-plots-multi-pane-figure-by-stage}
#set MDS plotting parameters
c_value <- 2
l_value <- 1
#number of genes to use in MDS analysis
t_value <- 500

#set colors
dev_stage_colors <- color_list

#function that adjusts MDS plot labels to be more readable
mds_labels <- function(dge_of_interest) {
  label_vector <- colnames(dge_of_interest$counts)
  label_vector <- gsub("_.*","",label_vector)
  label_vector <- gsub("SSSS","Psun",label_vector)
  label_vector <- gsub("BBBB","Pcam",label_vector)
  label_vector <- gsub("BBSS","hybrid",label_vector)
  return(label_vector)
}

#make an MDS plot showing minimal batch effects between library types
china_samples <- c("BBBB_198-7M_SP", "BBSS_90-2M_SP", "BBSS_95-3M_DIP", "BBSS_95-3M_LZ", "SSSS_188-4M_RS", "SSSS_188-5M_SP")

#set point shapes to be used in MDS plot
shapes_by_cross_and_library <- as.numeric(gsub("hybrid", 4, gsub("Pcam", 2, gsub("Psun", 6, mds_labels(dgeHamster)))))
shapes_by_cross_and_library[intersect(which(rownames(dgeHamster$samples) %in% china_samples),which(mds_labels(dgeHamster) == "Pcam"))] <- 24
shapes_by_cross_and_library[intersect(which(rownames(dgeHamster$samples) %in% china_samples),which(mds_labels(dgeHamster) == "Psun"))] <- 25
shapes_by_cross_and_library[intersect(which(rownames(dgeHamster$samples) %in% china_samples),which(mds_labels(dgeHamster) == "hybrid"))] <- 7

#MDS plot of all Dev samples colored by library preparation
if(exportFigs == TRUE) {
  pdf(file = "exported_figures/MDS_dev_samples_colored_by_library_prep.pdf", width=5, height=4, pointsize=12)
  plotMDS(dgeHamster, top = t_value, lwd = l_value, col = dev_stage_colors[gsub(".*-", "", dgeHamster$samples$group)], 
          bg = dev_stage_colors[gsub(".*-", "", dgeHamster$samples$group)], 
          pch = shapes_by_cross_and_library, cex = c_value*0.7, main = "All samples by library preparation", plot = TRUE)
  dev.off()
} else {
  plotMDS(dgeHamster, top = t_value, lwd = l_value, col = dev_stage_colors[gsub(".*-", "", dgeHamster$samples$group)], 
          bg = dev_stage_colors[gsub(".*-", "", dgeHamster$samples$group)], 
          pch = shapes_by_cross_and_library, cex = c_value*0.7, main = "All samples by library preparation", plot = TRUE)
}

#grid of MDS plots of SP, LZ, DIP, and RS samples
if (exportFigs == TRUE) {
  pdf(file = paste("exported_figures/MDS_plots_by_stage_outlines_cex", c_value, "_top", t_value, ".pdf", sep = ""), width = 6.5, height = 5, pointsize = 12)
}
layout(matrix(c(1,2,3,4), ncol = 2, byrow = TRUE))

#SP MDS
par(mar=c(3, 3, 2, 1), mgp=c(1.5, 0.5, 0))
metadata$stage <- factor(metadata$stage, levels = stages)
stage_colors <- color_list[metadata$stage]
plotMDS(dgeSP, top = t_value, lwd = l_value, col = color_list["SP"], pch = as.numeric(gsub("hybrid", 4, gsub("Pcam", 2, gsub("Psun", 6, mds_labels(dgeSP))))), cex = c_value, main = "Mitosis (SP)", plot = TRUE, cex.lab = 0.75, cex.main = 1, cex.axis = 0.75)

#LZ MDS
plotMDS(dgeLZ, top = t_value, lwd = l_value, col = color_list["LZ"], pch = as.numeric(gsub("hybrid", 4, gsub("Pcam", 2, gsub("Psun", 6, mds_labels(dgeLZ))))), cex = c_value, main = "Meiosis-Before X-Inact. (LZ)", plot = TRUE, cex.lab = 0.75, cex.main = 1, cex.axis = 0.75)

#DIP MDS
plotMDS(dgeDIP, top = t_value, lwd = l_value, col = color_list["DIP"], pch = as.numeric(gsub("hybrid", 4, gsub("Pcam", 2, gsub("Psun", 6, mds_labels(dgeDIP))))), cex = c_value, main = "Meiosis-After X-Inact. (DIP)", plot = TRUE, cex.lab = 0.75, cex.main = 1, cex.axis = 0.75)

#RS MDS
plotMDS(dgeRS, top = t_value, lwd = l_value, col = color_list["RS"], pch = as.numeric(gsub("hybrid", 4, gsub("Pcam", 2, gsub("Psun", 6, mds_labels(dgeRS))))), cex = c_value, main = "Postmeiosis (RS)", plot = TRUE, cex.lab = 0.75, cex.main = 1, cex.axis = 0.75)

if (exportFigs == TRUE) {
  dev.off()
}

rm(dgeSP, dgeLZ, dgeDIP, dgeRS)
rm(stage_colors)
```

<h3>X Chromosome MDSPlots</h3>
```{r mds-plots-of-X-and-Auto}
#MDS plot of X-linked expression for all samples
#set MDS plotting parameters
c_value <- 2
l_value <- 1
#number of genes to use in MDS analysis
t_value <- 500

#make grid of X-linked and Autosomal gene MDS plots for dwarf hamster
#X-linked
if(exportFigs == TRUE) {
  pdf(file = "exported_figures/MDS_X-linked_genes_colored_by_genotype_with_symbols.pdf", width=5, height=4, pointsize=12)
}

plotMDS(dgeX, top = t_value, lwd =2, 
        col= color_list[gsub(".*-", "", dgeX$samples$group)], 
        pch = as.numeric(gsub("hybrid", 4, gsub("Pcam", 2, gsub("Psun", 6, mds_labels(dgeX))))), 
        cex = c_value, plot = TRUE)
title("X MDS - hamster")

if(exportFigs == TRUE) {
  dev.off()
}

#Auto
if(exportFigs == TRUE) {
  pdf(file = "exported_figures/MDS_Autosomal_genes_colored_by_genotype_with_symbols.pdf", width=5, height=4, pointsize=12)
}
plotMDS(dgeAuto, top = t_value, lwd =2, 
        col= color_list[gsub(".*-", "", dgeAuto$samples$group)], 
        pch = as.numeric(gsub("hybrid", 4, gsub("Pcam", 2, gsub("Psun", 6, mds_labels(dgeAuto))))), 
        cex = c_value, plot = TRUE)
title("Autosomal MDS - hamster")

if(exportFigs == TRUE) {
  dev.off()
}
```

<h3>Heatmaps of Log(Normalized(RPKM)) for the autosomes and X chromosome from all data</h3>
```{r heatmaps-of-x-genes}
#heatmap of normalized rpkm values for x-linked genes from all stages; un-normalized rpkm is not useful for visualizing differences
#choose colors
colfunc <- colorRampPalette(c("#FFFFFF", "#2F2323"))
gap_value <- 4
total_number <- 255 + (15*gap_value)
browns_mod <- colfunc(total_number)
browns_mod <- browns_mod[c(seq(1,1+(15*gap_value),gap_value),seq(2+(15*gap_value),total_number,1))]

#auto heatmap; uncomment if desired
if(exportFigs == TRUE) {
  pdf(file="exported_figures/Heatmap_hamster_allAuto_all-genotypes_all_stages.pdf", width=10, height=8, pointsize=12)
}
Heatmap(as.matrix(rpkm_allAuto.norm), name = "Autosomal expression (rpkm)", column_order = Full_heatmap_order, column_names_gp = gpar(fontsize=4), show_row_names = FALSE, col = browns_mod, show_row_dend = FALSE, cluster_columns = FALSE, use_raster = FALSE)
if(exportFigs == TRUE) {
  dev.off()
}

#x-linked heatmap with genes ordered by clustering
if(exportFigs == TRUE) {
  pdf(file="exported_figures/Heatmap_hamster_justX_all-genotypes_all_stages.pdf", width=10, height=8, pointsize=12)
}
Heatmap(as.matrix(rpkm_justX.norm), name = "X-linked expression (rpkm)", column_order = Full_heatmap_order, column_names_gp = gpar(fontsize=4), show_row_names = FALSE, col = browns_mod, show_row_dend = FALSE, cluster_columns = FALSE)
if(exportFigs == TRUE) {
  dev.off()
}
```

<h3>Determine which parental stages hybrid diplotene samples are most similar to</h3>
```{r which-stage-does-hybDIP-correlate-to}
#the goal of this section is to see which parental stages the hybrid diplotene samples have the highest expression correlations to; then to see how those parental stage correlations differ between house mice and dwarf hamsters

#calculate rho of X-linked correlations
hyb_v_BBBB_SP_X_corr <- cor.test(rowMeans(rpkm_BBSS_DIP_justX.norm), rowMeans(rpkm_BBBB_SP_justX.norm))
hyb_v_BBBB_LZ_X_corr <- cor.test(rowMeans(rpkm_BBSS_DIP_justX.norm), rowMeans(rpkm_BBBB_LZ_justX.norm))
hyb_v_BBBB_DIP_X_corr <- cor.test(rowMeans(rpkm_BBSS_DIP_justX.norm), rowMeans(rpkm_BBBB_DIP_justX.norm))
hyb_v_BBBB_RS_X_corr <- cor.test(rowMeans(rpkm_BBSS_DIP_justX.norm), rowMeans(rpkm_BBBB_RS_justX.norm))

#calculate rho of autosomal correlations
hyb_v_BBBB_SP_Auto_corr <- cor.test(rowMeans(rpkm_BBSS_DIP_allAuto.norm), rowMeans(rpkm_BBBB_SP_allAuto.norm))
hyb_v_BBBB_LZ_Auto_corr <- cor.test(rowMeans(rpkm_BBSS_DIP_allAuto.norm), rowMeans(rpkm_BBBB_LZ_allAuto.norm))
hyb_v_BBBB_DIP_Auto_corr <- cor.test(rowMeans(rpkm_BBSS_DIP_allAuto.norm), rowMeans(rpkm_BBBB_DIP_allAuto.norm))
hyb_v_BBBB_RS_Auto_corr <- cor.test(rowMeans(rpkm_BBSS_DIP_allAuto.norm), rowMeans(rpkm_BBBB_RS_allAuto.norm))

#begin bootstrap analysis of correlation coefficient
reps <- 100

bootstrap_correlation_analysis <- function(comparison) {
  example_stage <- unlist(strsplit(comparison, split = "_"))[1]
  example_X_cat <- unlist(strsplit(comparison, split = "_"))[2]
  
  example_X_cat_genes <- get(paste("filtered_", example_X_cat,"_genes", sep = ""))
  
  hybrid_rpkm_rowMeans <- rowMeans(get(paste("rpkm_BBSS_DIP_", example_X_cat, ".norm", sep = "")))
  parent_rpkm_rowMeans <- rowMeans(get(paste("rpkm_BBBB_", example_stage, "_", example_X_cat, ".norm", sep = "")))
  
  combined_rpkm_df <- data.frame(hyb = hybrid_rpkm_rowMeans, par = parent_rpkm_rowMeans)
  
  sample_combined_rpkm_df <- function(example_rpkm_df) {
    sampled_combined_rpkm_df <- example_rpkm_df[sample(rownames(example_rpkm_df), 
                                                       length(rownames(example_rpkm_df)), replace = TRUE),]
    
    sampled_corr_test_estimate <- cor.test(sampled_combined_rpkm_df$hyb, 
                                           sampled_combined_rpkm_df$par, 
                                           method = "pearson")$estimate
    return(sampled_corr_test_estimate)
  }
  
  sampled_corr_test_estimates <- replicate(reps, sample_combined_rpkm_df(combined_rpkm_df))
  
  return(data.frame(stage = example_stage, X_cat = example_X_cat, corr = sampled_corr_test_estimates))
}

bootstap_correlations_list <- lapply(c("SP_justX", "LZ_justX", "DIP_justX", "RS_justX",
                                       "SP_allAuto", "LZ_allAuto", "DIP_allAuto", "RS_allAuto"), 
                                     bootstrap_correlation_analysis)

bootstap_correlations_df <- do.call(rbind, bootstap_correlations_list)

colnames(bootstap_correlations_df) <- c("stage", "X_cat" ,"rho")
bootstap_correlations_df$stage <- factor(bootstap_correlations_df$stage, levels = c("SP", "LZ", "DIP", "RS"))
bootstap_correlations_df$X_cat <- factor(bootstap_correlations_df$X_cat, levels = c("justX", "allAuto"))


bootstrap_calculate_CI <- function(comparison) {
  example_stage <- unlist(strsplit(comparison, split = "_"))[1]
  example_X_cat <- unlist(strsplit(comparison, split = "_"))[2]
  
  exaxmple_ci <- round(quantile(bootstap_correlations_df[bootstap_correlations_df$stage == example_stage & 
                                                           bootstap_correlations_df$X_cat == example_X_cat,]$rho, 
                                probs = c(0.025, 0.975)), 4)
  return(c(example_stage, example_X_cat, exaxmple_ci))
}

bootstap_confidence_intervals_df <- data.frame(do.call(rbind, lapply(c("SP_justX", "LZ_justX", "DIP_justX", "RS_justX",
                                                                       "SP_allAuto", "LZ_allAuto", "DIP_allAuto", "RS_allAuto"),
                                                                     bootstrap_calculate_CI)))

colnames(bootstap_confidence_intervals_df) <- c("stage", "X_cat", "CI_lower", "CI_upper")


#collate all correlation data then fdr correct actual (non-bootstrap) X-linked correlation rhos
hybDIP_corr_table_ham <- data.frame(stage = bootstap_confidence_intervals_df[1],
                                    
                                    pVals = c(hyb_v_BBBB_SP_X_corr$p.value, 
                                              hyb_v_BBBB_LZ_X_corr$p.value, 
                                              hyb_v_BBBB_DIP_X_corr$p.value, 
                                              hyb_v_BBBB_RS_X_corr$p.value,
                                              hyb_v_BBBB_SP_Auto_corr$p.value, 
                                              hyb_v_BBBB_LZ_Auto_corr$p.value, 
                                              hyb_v_BBBB_DIP_Auto_corr$p.value, 
                                              hyb_v_BBBB_RS_Auto_corr$p.value),
                                    
                                    corVals = c(hyb_v_BBBB_SP_X_corr$estimate,
                                                hyb_v_BBBB_LZ_X_corr$estimate,
                                                hyb_v_BBBB_DIP_X_corr$estimate,
                                                hyb_v_BBBB_RS_X_corr$estimate,
                                                hyb_v_BBBB_SP_Auto_corr$estimate,
                                                hyb_v_BBBB_LZ_Auto_corr$estimate,
                                                hyb_v_BBBB_DIP_Auto_corr$estimate,
                                                hyb_v_BBBB_RS_Auto_corr$estimate),
                                    
                                    comp = c("Ham-X",
                                             "Ham-X",
                                             "Ham-X",
                                             "Ham-X",
                                             "Ham-Auto",
                                             "Ham-Auto",
                                             "Ham-Auto",
                                             "Ham-Auto"),
                                    
                                    species = c("Ham",
                                                "Ham",
                                                "Ham",
                                                "Ham",
                                                "Ham",
                                                "Ham",
                                                "Ham",
                                                "Ham"),
                                    
                                    CI_lower = as.numeric(unlist(bootstap_confidence_intervals_df[3])),
                                    
                                    CI_upper = as.numeric(unlist(bootstap_confidence_intervals_df[4])))

#formatting data frame
hybDIP_corr_table_ham$stage   <- factor(hybDIP_corr_table_ham$stage, levels = c("SP", "LZ", "DIP", "RS"))
hybDIP_corr_table_ham$comp    <- factor(hybDIP_corr_table_ham$comp, levels = c("Ham-Auto", "Ham-X"))
hybDIP_corr_table_ham$comb_comp <- paste(hybDIP_corr_table_ham$comp, hybDIP_corr_table_ham$stage, sep = "-")
hybDIP_corr_table_ham$comb_comp    <- factor(hybDIP_corr_table_ham$comb_comp, levels = c("Ham-Auto-SP", "Ham-X-SP",
                                                                                         "Ham-Auto-LZ", "Ham-X-LZ",
                                                                                         "Ham-Auto-DIP", "Ham-X-DIP",
                                                                                         "Ham-Auto-RS", "Ham-X-RS"))
hybDIP_corr_table_ham$species <- factor(hybDIP_corr_table_ham$species, levels = c("Mouse", "Ham"))
#the FDR correction
hybDIP_corr_table_ham$pAdj    <- p.adjust(hybDIP_corr_table_ham$pVals, method = "fdr")

#add column indicating if correlation is significant for use in plot coloring
hybDIP_corr_table_ham$sig <- ""
if(sum(hybDIP_corr_table_ham$pAdj < 0.05) > 0){
  hybDIP_corr_table_ham[hybDIP_corr_table_ham$pAdj < 0.05,]$sig <- "yes"
}
if(sum(hybDIP_corr_table_ham$pAdj > 0.05) > 0){
  hybDIP_corr_table_ham[hybDIP_corr_table_ham$pAdj > 0.05,]$sig <- "no"
}

#print summary of hamster X-linked rho stats
print(paste("hybDIP_BBBB_SP_X rho X: ", round(hyb_v_BBBB_SP_X_corr$estimate, 4), "; p = ",
            round(hybDIP_corr_table_ham$pAdj[1], 4), 
            sep = ""))
print(paste("hybDIP_BBBB_LZ_X rho X: ", round(hyb_v_BBBB_LZ_X_corr$estimate, 4), "; p = ",
            round(hybDIP_corr_table_ham$pAdj[2], 3), 
            sep = ""))
print(paste("hybDIP_BBBB_RS_X rho X: ", round(hyb_v_BBBB_RS_X_corr$estimate, 4), "; p = ",
            round(hybDIP_corr_table_ham$pAdj[4], 3), 
            sep = ""))

#print summary of hamster autosomal rho stats
print(paste("hybDIP_BBBB_SP_Auto rho Auto: ", round(hyb_v_BBBB_SP_Auto_corr$estimate, 4), "; p = ",
            round(hybDIP_corr_table_ham$pAdj[5], 3), 
            sep = ""))
print(paste("hybDIP_BBBB_LZ_Auto rho Auto: ", round(hyb_v_BBBB_LZ_Auto_corr$estimate, 4), "; p = ",
            round(hybDIP_corr_table_ham$pAdj[6], 3), 
            sep = ""))
print(paste("hybDIP_BBBB_RS_Auto rho Auto: ", round(hyb_v_BBBB_RS_Auto_corr$estimate, 4), "; p = ",
            round(hybDIP_corr_table_ham$pAdj[8], 3), 
            sep = ""))


#read in mouse correlations and bootstraps then combine with hamsters
hybDIP_corr_table_mouse <- read.csv(file = "exported_files/hybDIP_correlation_mouse.csv", header = TRUE, stringsAsFactors = TRUE)[,-1]
hybDIP_corr_table_mouse$stage <- factor(hybDIP_corr_table_mouse$stage, levels = c("SP", "LZ", "DIP", "RS"))
hybDIP_corr_table_mouse$comb_comp <- paste(hybDIP_corr_table_mouse$comp, hybDIP_corr_table_mouse$stage, sep = "-")
hybDIP_corr_table_mouse$comb_comp    <- factor(hybDIP_corr_table_mouse$comb_comp, levels = c("Mouse-Auto-SP", "Mouse-X-SP",
                                                                                             "Mouse-Auto-LZ", "Mouse-X-LZ",
                                                                                             "Mouse-Auto-DIP", "Mouse-X-DIP",
                                                                                             "Mouse-Auto-RS", "Mouse-X-RS"))

#plot correlations for dwarf hamster X-linked and Autosomal hybrid diplotene genes
hamster_correlation_plot <- ggplot(data = subset(hybDIP_corr_table_ham, hybDIP_corr_table_ham$stage != "DIP"), 
                                   aes(x = stage, y = corVals, col = stage)) +
  geom_point(size = 1) +
  theme_minimal() +
  theme(legend.position = "none",
        strip.background =element_rect(fill=gray_1),
        panel.grid.minor = element_blank()) +
  facet_wrap(vars(comp), ncol = 2, scales = "free") +
  geom_hline(yintercept = 0, linewidth = 0.5) +
  labs(y = "Pearson's correlation (rho)", x = "Parental Stage") + 
  ylim(-0.375, 0.375) +
  scale_color_manual(values = c(sp_color_dark, lz_color_dark, rs_color_dark)) +
  geom_errorbar(aes(ymin=CI_lower, ymax=CI_upper), width=.2,
                position=position_dodge(0.05))

#plot correlations for house mouse X-linked and Autosomal hybrid diplotene genes
mouse_correlation_plot <- ggplot(data = subset(hybDIP_corr_table_mouse, hybDIP_corr_table_mouse$stage != "DIP"), 
                                 aes(x = stage, y = corVals, col = stage)) +
  geom_point(size = 1) +
  theme_minimal() +
  theme(legend.position = "none",
        strip.background =element_rect(fill=gray_1),
        panel.grid.minor = element_blank()) +
  facet_wrap(vars(comp), ncol = 2, scales = "free") +
  geom_hline(yintercept = 0, linewidth = 0.5) +
  labs(y = "Pearson's correlation (rho)", x = "Parental Stage") + 
  ylim(-0.375, 0.375) +
  scale_color_manual(values = c(sp_color, lz_color, rs_color)) +
  geom_errorbar(aes(ymin=CI_lower, ymax=CI_upper), width=.2,
                position=position_dodge(0.05))

if(exportFigs == TRUE) {
  ggsave(filename = "exported_figures/hybDIP_correlations.pdf", plot_grid(mouse_correlation_plot, hamster_correlation_plot), width = 6, height = 8)
} else {
  plot_grid(mouse_correlation_plot, hamster_correlation_plot)
}
```

<h3>Cell Purity of sorted cells and whole testes</h3>
```{r purity}
#the goal of this analysis is to assess the relative purity of each of the parental sorted cell populations to confirm that FACS worked to isolate the right populations
# #import marker genes; here

#marker genes are primarily sourced from Raymond et al. 2000, Green et al. 2018, Hermann et al. 2018, and Murat et al. 2023).
gene.data <- read.table("input_data/curated_purity_genes_hamster.csv", header = TRUE, stringsAsFactors = FALSE, sep = ",")

#plot marker gene expression for each condition
assess_marker_gene_expression <- function(focal) {
  
  #get marker gene expression in each cell type for parent: BBBB
  fpkm <- as.numeric(rpkm_BBBB_SP[focal, ])
  condition <- gsub("_[0-9]+.-[0-9]M_","_",names(rpkm_BBBB_SP[focal, ]))
  
  fpkm <- append(fpkm, as.numeric(rpkm_BBBB_LZ[focal, ]))
  condition <- append(condition, gsub("_[0-9]+.-[0-9]M_","_",names(rpkm_BBBB_LZ[focal, ])))
  
  fpkm <- append(fpkm, as.numeric(rpkm_BBBB_DIP[focal, ]))
  condition <- append(condition, gsub("_[0-9]+.-[0-9]M_","_",names(rpkm_BBBB_DIP[focal, ])))
  
  fpkm <- append(fpkm, as.numeric(rpkm_BBBB_RS[focal, ]))
  condition <- append(condition, gsub("_[0-9]+.-[0-9]M_","_",names(rpkm_BBBB_RS[focal, ])))
  
  #get marker gene expression in each cell type for parent: SSSS
  fpkm <- append(fpkm, as.numeric(rpkm_SSSS_SP[focal, ]))
  condition <- append(condition, gsub("_[0-9]+.-[0-9]M_","_",names(rpkm_SSSS_SP[focal, ])))
  
  fpkm <- append(fpkm, as.numeric(rpkm_SSSS_LZ[focal, ]))
  condition <- append(condition, gsub("_[0-9]+.-[0-9]M_","_",names(rpkm_SSSS_LZ[focal, ])))
  
  fpkm <- append(fpkm, as.numeric(rpkm_SSSS_DIP[focal, ]))
  condition <- append(condition, gsub("_[0-9]+.-[0-9]M_","_",names(rpkm_SSSS_DIP[focal, ])))
  
  fpkm <- append(fpkm, as.numeric(rpkm_SSSS_RS[focal, ]))
  condition <- append(condition, gsub("_[0-9]+.-[0-9]M_","_",names(rpkm_SSSS_RS[focal, ])))
  
  #get marker gene expression in each cell type for hybrid: BBSS
  fpkm <- append(fpkm, as.numeric(rpkm_BBSS_SP[focal, ]))
  condition <- append(condition, gsub("_[0-9]+.-[0-9]M_","_",names(rpkm_BBSS_SP[focal, ])))
  
  fpkm <- append(fpkm, as.numeric(rpkm_BBSS_LZ[focal, ]))
  condition <- append(condition, gsub("_[0-9]+.-[0-9]M_","_",names(rpkm_BBSS_LZ[focal, ])))
  
  fpkm <- append(fpkm, as.numeric(rpkm_BBSS_DIP[focal, ]))
  condition <- append(condition, gsub("_[0-9]+.-[0-9]M_","_",names(rpkm_BBSS_DIP[focal, ])))
  
  rpkm_matrix <- data.frame(fpkm, condition)
  
  rpkm_matrix$condition <- factor(rpkm_matrix$condition, levels = c("BBBB_SP", "BBBB_LZ", "BBBB_DIP", "BBBB_RS", "SSSS_SP", "SSSS_LZ", "SSSS_DIP", "SSSS_RS", "BBSS_SP", "BBSS_LZ", "BBSS_DIP"))
  rpkm_matrix$gene <- rep(gene.data[gene.data$Geneid == focal,]$gene, dim(rpkm_matrix)[1])
  rpkm_matrix$stage <- as.factor(gene.data[gene.data$Geneid == focal,]$CellType)
  
  
  #kruskal wallis test followed by dunn's posthoc test - BBBB
  example_gene_BBBB_fpkm_kw <- kruskal.test(fpkm ~ condition, 
                                            data = rpkm_matrix[grep("BBBB", rpkm_matrix$condition),])
  example_gene_BBBB_fpkm_dt <- dunnTest(fpkm ~ condition, 
                                        data = rpkm_matrix[grep("BBBB", rpkm_matrix$condition),])
  
  #assign Dunn's test object for use in stats printing later
  assign(paste(unique(rpkm_matrix$gene), "_BBBB_fpkm_dt", sep = ""), 
         example_gene_BBBB_fpkm_dt, envir = .GlobalEnv)
  
  #adjust p-values for multi-species comparison
  example_gene_BBBB_fpkm_padj <- example_gene_BBBB_fpkm_dt$res$P.adj
  names(example_gene_BBBB_fpkm_padj) <- gsub(" ","", example_gene_BBBB_fpkm_dt$res$Comparison)
  
  #convert p-values to symbols for graphing
  example_gene_BBBB_fpkm_letters <- data.frame(multcompView::multcompLetters(example_gene_BBBB_fpkm_padj)$Letters)
  example_gene_BBBB_fpkm_letters$Strain <- rownames(example_gene_BBBB_fpkm_letters)
  colnames(example_gene_BBBB_fpkm_letters) <- c("Letters", "Stage")
  example_gene_BBBB_fpkm_padj <- sig_conversion(example_gene_BBBB_fpkm_padj)
  
  example_gene_BBBB_fpkm_letters$Stage <- factor(example_gene_BBBB_fpkm_letters$Stage,
                                                 levels = c("BBBB_SP", "BBBB_LZ",
                                                            "BBBB_DIP", "BBBB_RS"))
  setorder(example_gene_BBBB_fpkm_letters, Stage)
  
  #assign Dunn's test object for use in stats printing later
  assign(paste(unique(rpkm_matrix$gene), "_BBBB_fpkm_letters", sep = ""), 
         example_gene_BBBB_fpkm_letters, envir = .GlobalEnv)
  #assign Dunn's test object for use in stats printing later
  assign(paste(unique(rpkm_matrix$gene), "_BBBB_fpkm_padj", sep = ""), 
         example_gene_BBBB_fpkm_padj, envir = .GlobalEnv)
  
  
  #kruskal wallis test followed by dunn's posthoc test - SSSS
  example_gene_SSSS_fpkm_kw <- kruskal.test(fpkm ~ condition, 
                                            data = rpkm_matrix[grep("SSSS", rpkm_matrix$condition),])
  example_gene_SSSS_fpkm_dt <- dunnTest(fpkm ~ condition, 
                                        data = rpkm_matrix[grep("SSSS", rpkm_matrix$condition),])
  
  #assign Dunn's test object for use in stats printing later
  assign(paste(unique(rpkm_matrix$gene), "_SSSS_fpkm_dt", sep = ""), 
         example_gene_SSSS_fpkm_dt, envir = .GlobalEnv)
  
  #adjust p-values for multi-species comparison
  example_gene_SSSS_fpkm_padj <- example_gene_SSSS_fpkm_dt$res$P.adj
  names(example_gene_SSSS_fpkm_padj) <- gsub(" ","", example_gene_SSSS_fpkm_dt$res$Comparison)
  
  #convert p-values to symbols for graphing
  example_gene_SSSS_fpkm_letters <- data.frame(multcompView::multcompLetters(example_gene_SSSS_fpkm_padj)$Letters)
  example_gene_SSSS_fpkm_letters$Strain <- rownames(example_gene_SSSS_fpkm_letters)
  colnames(example_gene_SSSS_fpkm_letters) <- c("Letters", "Stage")
  example_gene_SSSS_fpkm_padj <- sig_conversion(example_gene_SSSS_fpkm_padj)
  
  example_gene_SSSS_fpkm_letters$Stage <- factor(example_gene_SSSS_fpkm_letters$Stage,
                                                 levels = c("SSSS_SP", "SSSS_LZ",
                                                            "SSSS_DIP", "SSSS_RS"))
  setorder(example_gene_SSSS_fpkm_letters, Stage)
  
  #assign Dunn's test object for use in stats printing later
  assign(paste(unique(rpkm_matrix$gene), "_SSSS_fpkm_letters", sep = ""), 
         example_gene_SSSS_fpkm_letters, envir = .GlobalEnv)
  #assign Dunn's test object for use in stats printing later
  assign(paste(unique(rpkm_matrix$gene), "_SSSS_fpkm_padj", sep = ""), 
         example_gene_SSSS_fpkm_padj, envir = .GlobalEnv)
  
  individual_BBBB_marker_plot <-   ggplot(data = rpkm_matrix[grep("BBBB", rpkm_matrix$condition),],
                                          aes(x = condition, y = fpkm, color = condition)) +
    stat_boxplot(geom ='errorbar') +
    geom_boxplot(outlier.colour = "black",outlier.shape = NA, notch = FALSE) +
    geom_quasirandom(aes(col = condition), alpha = 0.6, size = 2) + 
    facet_wrap(vars(gene), ncol = 4, scales = "free_y") +
    theme_classic() +
    scale_color_manual(values = c(sp_color, lz_color, dip_color, rs_color)) +
    theme(axis.title.x = element_blank(),
          axis.line = element_blank(),
          panel.spacing.x=unit(0.5, "lines"),panel.spacing.y = unit(0, "lines"),
          axis.text=element_text(size=8),
          strip.text = element_text(face = "italic", size = 8),
          strip.background = element_blank(),
          panel.border = element_rect(linetype = "solid", fill = NA),
          legend.position = "none")  +
    ylab("Expression level (FPKM)") +
    scale_x_discrete(labels = c("SP", "LZ", "DIP", "RS")) + 
    expand_limits(y = 0)    + 
    geom_signif(
      y_position = max(rpkm_matrix$fpkm)*1.2, 
      xmin = c(1,2,3,4), 
      xmax = c(1,2,3,4), 
      annotation = example_gene_BBBB_fpkm_letters$Letters, 
      tip_length = 0, textsize = sigText_size, size = 0, color = "black") +
    ylim(0, max(rpkm_matrix$fpkm)*1.3)
  
  individual_SSSS_marker_plot <-   ggplot(data = rpkm_matrix[grep("SSSS", rpkm_matrix$condition),],
                                          aes(x = condition, y = fpkm, color = condition)) +
    stat_boxplot(geom ='errorbar') +
    geom_boxplot(outlier.colour = "black",outlier.shape = NA, notch = FALSE) +
    geom_quasirandom(aes(col = condition), alpha = 0.6, size = 2) + 
    facet_wrap(vars(gene), ncol = 4, scales = "free_y") +
    theme_classic() +
    scale_color_manual(values = c(sp_color, lz_color, dip_color, rs_color)) +
    theme(axis.title.x = element_blank(),
          axis.line = element_blank(),
          panel.spacing.x=unit(0.5, "lines"),panel.spacing.y = unit(0, "lines"),
          axis.text=element_text(size=8),
          strip.text = element_text(face = "italic", size = 8),
          strip.background = element_blank(),
          panel.border = element_rect(linetype = "solid", fill = NA),
          legend.position = "none")  +
    ylab("Expression level (FPKM)") +
    scale_x_discrete(labels = c("SP", "LZ", "DIP", "RS")) + 
    expand_limits(y = 0)    + 
    geom_signif(
      y_position = max(rpkm_matrix$fpkm)*1.2, 
      xmin = c(1,2,3,4), 
      xmax = c(1,2,3,4), 
      annotation = example_gene_SSSS_fpkm_letters$Letters, 
      tip_length = 0, textsize = sigText_size, size = 0, color = "black") +
    ylim(0, max(rpkm_matrix$fpkm)*1.3)
  
  assign(paste(gene.data[gene.data$Geneid == focal,]$gene, 
               "_BBBB_marker_plot", sep = ""), 
         individual_BBBB_marker_plot, 
         envir = .GlobalEnv)
  
  
  assign(paste(gene.data[gene.data$Geneid == focal,]$gene, 
               "_SSSS_marker_plot", sep = ""), 
         individual_SSSS_marker_plot, 
         envir = .GlobalEnv)
  
  fname <- paste("exported_figures/purity-test_", 
                 paste(gene.data[gene.data$Geneid == focal,]$gene, 
                       gene.data[gene.data$Geneid == focal,]$CellType, 
                       sep = "_"), ".pdf", sep = "")
  
  #saves boxplot of expression of a single marker gene across all parental samples/stages
  if(exportFigs == TRUE) {
    ggsave(filename = fname, plot_grid(individual_BBBB_marker_plot, 
                                       individual_SSSS_marker_plot, nrow = 1))
  }
}

invisible(lapply(gene.data$Geneid, assess_marker_gene_expression))



bbbb_markers <- plot_grid(Dmrt1_BBBB_marker_plot, Ccnb1ip1_BBBB_marker_plot,
                          Aurka_BBBB_marker_plot, Cabyr_BBBB_marker_plot,
                          Hells_BBBB_marker_plot, Adad2_BBBB_marker_plot, 
                          Tank_BBBB_marker_plot, Acrv1_BBBB_marker_plot,
                          nrow = 2)

ssss_markers <- plot_grid(Dmrt1_SSSS_marker_plot, Ccnb1ip1_SSSS_marker_plot,
                          Aurka_SSSS_marker_plot, Cabyr_SSSS_marker_plot,
                          Hells_SSSS_marker_plot, Adad2_SSSS_marker_plot, 
                          Tank_SSSS_marker_plot, Acrv1_SSSS_marker_plot,
                          nrow = 2)

#change exportFigs to TRUE to export as PDF if desired
if(exportFigs == TRUE) {
  ggsave(filename = "exported_figures/BBBB_purity_markers.pdf", plot = bbbb_markers, width = 6.5, height = 3.5)
  ggsave(filename = "exported_figures/SSSS_purity_markers.pdf", plot = ssss_markers, width = 6.5, height = 3.5)
} else if (exportFigs == FALSE) {
  print("Marker gene expression in P. campbelli")
  print(bbbb_markers)
  print("Marker gene expression in P. sungorus")
  print(ssss_markers)
}
```

<h1>Hybrid Differential Expression</h1>
<h3>Hybrid Design set-up</h3>
```{r hybrid-de-design-matrix}
#set up differential expression experimental design and contrasts
design <- cbind(rownames(metadata),
                metadata[c("cross", "stage", "condition_name")])
design$condition_name <- factor(design$condition_name)
colnames(design) <- c("sample_ID", "cross", "stage", "condition_name")
design$condition_name <- gsub("-",".",design$condition_name)
design$condition_name <- as.factor(design$condition_name)

#create DE contrast matrix
designMatrix  <-  model.matrix(~0+design$condition_name)
colnames(designMatrix)  <-  levels(design$condition_name)
rownames(designMatrix)  <-  design$sample_ID
```

<h3>Defining hybrid contrasts</h3>
```{r hybrid-de-defining contrasts}
#set up DE contrasts
contrast.list <- makeContrasts(
  # compare cell types between hybrids and campbelli
  Hyb.BB.SP  =  BBSS.SP - BBBB.SP,  
  Hyb.BB.LZ  =  BBSS.LZ - BBBB.LZ, 
  Hyb.BB.DIP  =  BBSS.DIP - BBBB.DIP,
  
  # compare cell types between hybrids and campbelli
  Hyb.SS.SP   =  BBSS.SP -  SSSS.SP,  
  Hyb.SS.LZ   =  BBSS.LZ -  SSSS.LZ, 
  Hyb.SS.DIP  =  BBSS.DIP - SSSS.DIP,
  
  # parents
  BBBB.SSSS.SP  =   BBBB.SP -  SSSS.SP, 
  BBBB.SSSS.LZ  =   BBBB.LZ -  SSSS.LZ, 
  BBBB.SSSS.DIP  =  BBBB.DIP - SSSS.DIP, 
  BBBB.SSSS.RS  =   BBBB.RS  - SSSS.RS, 
  
  levels  =  designMatrix
)
contrast.list
```

<h3>Dispersion estimates of main datasets</h3>
```{r hybrid-bcv-calculations}
#calculate BCV (biological coefficient of variation) for main DE analysis; this is a calculation of the intra-replicate variability
dgeHamster <- estimateGLMCommonDisp(dgeHamster, designMatrix, verbose = TRUE) 
dgeHamster <- estimateGLMTrendedDisp(dgeHamster, designMatrix) 
dgeHamster <- estimateGLMTagwiseDisp(dgeHamster, designMatrix) 

#generate BCV plots
if(exportFigs == TRUE) {
  pdf(file="exported_figures/Dispersion_plots_Dev_dataset.pdf", width=10, height=8, pointsize=12)
  plotBCV(dgeHamster, main = paste("Dev dataset (all genes); BCV = ",round(sqrt(dgeHamster$common.dispersion), 3)), ylim = c(0, 12), col.common = "black", col.trend = gray_6, xlab = "Average log(CPM)", xlim = c(-4,13), col = alpha(gray_4, 0.15), cex = 0.5)
  dev.off()
} else {
  plotBCV(dgeHamster, main = paste("Dispersion estimates for Dev dataset; BCV = ",round(sqrt(dgeHamster$common.dispersion), 3)), ylim = c(0, 12), col.common = "black", col.trend = gray_6, xlab = "Average log(CPM)", xlim = c(-4,13), col = alpha(gray_4, 0.15), cex = 0.5)
}
```

<h3>Fit model and calculate Hybrid DE genes</h3>
```{r hybrid-calculate-DE-genes}
#set the false discovery rate
FDR_cutoff <- 0.05
logFC_cutoff <- 1.25

contrasts_Hamster <- colnames(contrast.list)

retrieve_contrast_DEs <- function(example_contrast) {
  #perform LRT
  example_lrt <- glmLRT(glmFit(dgeHamster,  designMatrix), 
                        contrast = contrast.list[,example_contrast])
  assign(paste(example_contrast, "LRT", sep = "_"), example_lrt, envir = .GlobalEnv)
  
  #extract DE genes
  example_tt <- topTags(example_lrt, n = nrow(example_lrt))
  example_de <- example_tt$table[example_tt$table$FDR <= FDR_cutoff &
                                   abs(example_tt$table$logFC) >= logFC_cutoff, ]
  
  setorder(example_tt$table, -logFC)
  
  assign(paste(example_contrast, "tt", sep = "."), example_tt, envir = .GlobalEnv)
  assign(paste(example_contrast, "de", sep = "."), example_de, envir = .GlobalEnv)
  
  return(data.frame(non_DE = sum(example_tt$table$FDR > 0.05),
                    stage_up_DE = sum(example_tt$table$FDR <= 0.05 & abs(example_tt$table$logFC) < logFC_cutoff),
                    stage_down_DE = sum(example_tt$table$FDR <= 0.05 & abs(example_tt$table$logFC) > logFC_cutoff)))
}

#calculate DE genes for each stages
sapply(contrasts_Hamster, retrieve_contrast_DEs)

#create a dataframe of genes that are DE between hybrids and BOTH parents
Hyb.Comb.SP.de <- merge(Hyb.BB.SP.de, Hyb.SS.SP.de, by = c("Length", "chr", "start", "stop", "gene", "X_category", "scaffold", "mouse_ensemble_ID", "mouse_gene_symbol", "position_cM", "size_bp"))
Hyb.Comb.LZ.de <- merge(Hyb.BB.LZ.de, Hyb.SS.LZ.de, by = c("Length", "chr", "start", "stop", "gene", "X_category", "scaffold", "mouse_ensemble_ID", "mouse_gene_symbol", "position_cM", "size_bp"))
Hyb.Comb.DIP.de <- merge(Hyb.BB.DIP.de, Hyb.SS.DIP.de, by = c("Length", "chr", "start", "stop", "gene", "X_category", "mouse_ensemble_ID", "scaffold", "mouse_gene_symbol", "position_cM", "size_bp"))

#create a dataframe of all genes with logFC info for both parents
Hyb.Comb.SP.tt <- merge(Hyb.BB.SP.tt$table, Hyb.SS.SP.tt$table, by = c("Length", "chr", "start", "stop", "gene", "X_category", "scaffold", "mouse_ensemble_ID", "mouse_gene_symbol", "position_cM", "size_bp"))
Hyb.Comb.LZ.tt <- merge(Hyb.BB.LZ.tt$table, Hyb.SS.LZ.tt$table, by = c("Length", "chr", "start", "stop", "gene", "X_category", "scaffold", "mouse_ensemble_ID", "mouse_gene_symbol", "position_cM", "size_bp"))
Hyb.Comb.DIP.tt <- merge(Hyb.BB.DIP.tt$table, Hyb.SS.DIP.tt$table, by = c("Length", "chr", "start", "stop", "gene", "X_category", "scaffold", "mouse_ensemble_ID", "mouse_gene_symbol", "position_cM", "size_bp"))
Hyb.Comb.SP.tt$stage <- "SP"
Hyb.Comb.LZ.tt$stage <- "LZ"
Hyb.Comb.DIP.tt$stage <- "DIP"

Hyb.Comb.SP.tt$species <- "hamster"
Hyb.Comb.LZ.tt$species <- "hamster"
Hyb.Comb.DIP.tt$species <- "hamster"

#write out parent DE gene files to supplement; the hybrid DE genes are written out in a chunk below after the DE categories have been defined
write.csv(BBBB.SSSS.SP.de,  paste("exported_files/parent_hamster_SP_DE_Genes_logFC_", logFC_cutoff, ".csv", sep = ""))
write.csv(BBBB.SSSS.LZ.de,  paste("exported_files/parent_hamster_LZ_DE_Genes_logFC_", logFC_cutoff, ".csv", sep = ""))
write.csv(BBBB.SSSS.DIP.de, paste("exported_files/parent_hamster_DIP_DE_Genes_logFC_", logFC_cutoff, ".csv", sep = ""))
write.csv(BBBB.SSSS.RS.de,  paste("exported_files/parent_hamster_RS_DE_Genes_logFC_", logFC_cutoff, ".csv", sep = ""))
```
```{r eligible-DE-expressed genes}
#define background sets of genes for tests of overlap between Hamster and hamster based on if gene had a calculatable logFC in DE contrasts for a given stage 
Hyb_Comb_SP_exp <- Hyb.Comb.SP.tt[Hyb.Comb.SP.tt$logFC.x != 0 & Hyb.Comb.SP.tt$logFC.y != 0 & !is.na(Hyb.Comb.SP.tt$mouse_ensemble_ID),c("gene", "mouse_ensemble_ID")] # 20246
Hyb_Comb_LZ_exp <- Hyb.Comb.LZ.tt[Hyb.Comb.LZ.tt$logFC.x != 0 & Hyb.Comb.LZ.tt$logFC.y != 0 & !is.na(Hyb.Comb.LZ.tt$mouse_ensemble_ID),c("gene", "mouse_ensemble_ID")] # 20189
Hyb_Comb_DIP_exp <- Hyb.Comb.DIP.tt[Hyb.Comb.DIP.tt$logFC.x != 0 & Hyb.Comb.DIP.tt$logFC.y != 0 & !is.na(Hyb.Comb.DIP.tt$mouse_ensemble_ID),c("gene", "mouse_ensemble_ID")]  # 20181

Hyb_Comb_SP_X_exp <- Hyb_Comb_SP_exp[Hyb_Comb_SP_exp$gene %in% X_genes,] # 632
Hyb_Comb_LZ_X_exp <- Hyb_Comb_LZ_exp[Hyb_Comb_LZ_exp$gene %in% X_genes,] # 628
Hyb_Comb_DIP_X_exp <- Hyb_Comb_DIP_exp[Hyb_Comb_DIP_exp$gene %in% X_genes,]  # 828

#export the genes expressed across all three crosses in each stage
write.csv(Hyb_Comb_SP_exp,  row.names = FALSE,
          "exported_files/Hamster_SP_all_exp_genes.csv")
write.csv(Hyb_Comb_LZ_exp,  row.names = FALSE,
          "exported_files/Hamster_LZ_all_exp_genes.csv")
write.csv(Hyb_Comb_DIP_exp, row.names = FALSE,
          "exported_files/Hamster_DIP_all_exp_genes.csv")

#export the genes expressed across all three crosses in each stage on the X chromosome
write.csv(Hyb_Comb_SP_X_exp,  row.names = FALSE,
          "exported_files/Hamster_SP_X_exp_genes.csv") #632
write.csv(Hyb_Comb_LZ_X_exp,  row.names = FALSE,
          "exported_files/Hamster_LZ_X_exp_genes.csv") #628
write.csv(Hyb_Comb_DIP_X_exp,  row.names = FALSE,
          "exported_files/Hamster_DIP_X_exp_genes.csv") #628

Hyb_Comb_cell_exp_list <- c("Hyb_Comb_SP_exp", "Hyb_Comb_LZ_exp", "Hyb_Comb_DIP_exp")
```

```{r transgressive-expression}
define_transressive_expression_and_plot <- function (example_stage) {
  #get relavant data frame for stage
  example_tt <- get(paste("Hyb.Comb.", example_stage, ".tt", sep = ""))
  
  #add column indicating whether genes in hybrids were DE relative to both parents
  example_tt$sig_in_both <- "no"
  example_tt[example_tt$FDR.x < 0.05 & 
               example_tt$FDR.y < 0.05 &
               abs(example_tt$logFC.x) > logFC_cutoff & 
               abs(example_tt$logFC.y) > logFC_cutoff,]$sig_in_both <- "yes" 
  
  #add column indicating whether genes in hybrids were DE relative to only Pcam
  example_tt$sig_in_sameX <- "no"
  example_tt[example_tt$FDR.x < 0.05 &
               abs(example_tt$logFC.x) > logFC_cutoff,]$sig_in_sameX <- "yes" 
  
  #add column indicating whether genes in hybrids were DE relative to only Psun
  example_tt$sig_in_diffX <- "no"
  example_tt[example_tt$FDR.y < 0.05 &
               abs(example_tt$logFC.y) > logFC_cutoff,]$sig_in_diffX <- "yes" 
  
  #classify genes as not_different, sameX_only, diffX_only, intermediate, or transgressive
  
  #by default, genes are classified as not different from parentals if not DE in both directions
  example_tt$category <- "not_different"
  
  #genes are classified as intermediate if DE in both directions, and less than one parent and greater than the other; this is wrong
  example_tt[example_tt$sig_in_diffX == "yes" &
               example_tt$sig_in_sameX == "no",]$category <- "diffX_only"
  
  example_tt[example_tt$sig_in_sameX == "yes" &
               example_tt$sig_in_diffX == "no",]$category <- "sameX_only"
  
  
  #genes are classified as transgressive if DE in both directions, and either greater than both parents, or less than both parents
  example_tt[example_tt$sig_in_both == "yes" &
               example_tt$logFC.x > logFC_cutoff &
               example_tt$logFC.y > logFC_cutoff,]$category <- "transgressive"
  
  example_tt[example_tt$sig_in_both == "yes" &
               example_tt$logFC.x < -logFC_cutoff &
               example_tt$logFC.y < -logFC_cutoff,]$category <- "transgressive"
  
  
  #genes are classified as intermediate if DE in both directions, and less than one parent and greater than the other; this is wrong
  example_tt[example_tt$sig_in_both == "yes" &
               example_tt$logFC.x > logFC_cutoff &
               example_tt$logFC.y < -logFC_cutoff,]$category <- "intermediate"
  
  if(sum(example_tt$sig_in_both == "yes" &
         example_tt$logFC.x < -logFC_cutoff &
         example_tt$logFC.y > logFC_cutoff) > 0){
    example_tt[example_tt$sig_in_both == "yes" &
                 example_tt$logFC.x < -logFC_cutoff &
                 example_tt$logFC.y > logFC_cutoff,]$category <- "intermediate"}
  
  example_tt$category <- factor(example_tt$category, levels = c("transgressive",
                                                                "intermediate", 
                                                                "sameX_only", 
                                                                "diffX_only",
                                                                "not_different"))
  
  setorder(example_tt, -category)
  
  
  
  example_category_plot <- ggplot(data = example_tt, aes(x = logFC.x, y = logFC.y, color = category)) +
    geom_hline(yintercept = 0, linewidth = 0.5) +
    geom_vline(xintercept = 0, linewidth = 0.5) +
    geom_point(alpha = 0.7, size = 0.4) +
    scale_color_manual(values = unname(c(color_list_darkest[example_stage], 
                                         color_list[example_stage], 
                                         gray_5, 
                                         gray_4, 
                                         gray_2))) +
    theme_bw() +
    xlab("log2FC(cam x sun hybrid) - log2FC(cam)") +
    ylab("log2FC(cam x sun hybrid) - log2FC(sun)") +
    theme(legend.position = "none",
          text = element_text(size = 4)) +
    ylim(-12.5, 12.5) +
    xlim(-12.5, 12.5)
  
  assign(paste(example_stage, "_DE_category_plot", sep = ""), example_category_plot, envir = .GlobalEnv)
  assign(paste("Hyb.Comb.", example_stage, ".tt", sep = ""), example_tt, envir = .GlobalEnv)
}

invisible(lapply(stages[1:3], define_transressive_expression_and_plot))

all_stages_DE_category_plot <- plot_grid(SP_DE_category_plot,
                                         LZ_DE_category_plot,
                                         DIP_DE_category_plot,
                                         nrow = 1)

if(exportFigs == TRUE) {
  ggsave(filename = paste("exported_figures/transgressive_expression_categories_hamster_logFC_", 
                          logFC_cutoff ,".pdf", sep = ""), 
         all_stages_DE_category_plot,
         width = 4.875, height = 2)
} else {
  all_stages_DE_category_plot
}
```

```{r graphing-DE-categories}
hamster_DE_categories <- data.frame(rbind(table(Hyb.Comb.SP.tt$category), table(Hyb.Comb.LZ.tt$category)))
hamster_DE_categories <- rbind(hamster_DE_categories, table(Hyb.Comb.DIP.tt$category))

hamster_DE_categories$stage <- c("SP", "LZ", "DIP")
hamster_DE_categories$species <- c(rep("hamster", 3))  
hamster_DE_categories$perc_transgressive <- round(100*hamster_DE_categories$transgressive/
                                                    rowSums(hamster_DE_categories[,1:2]),1)
hamster_DE_categories$perc_intermediate <- round(100*hamster_DE_categories$intermediate/
                                                   rowSums(hamster_DE_categories[,1:2]),1)

#melt the DE categories df for plotting
hamster_DE_categories

hamster_DE_categories_number_long <- pivot_longer(hamster_DE_categories, cols = 1:4, names_to = "category", values_to = "number_DE")

write.csv(hamster_DE_categories, 
          paste("exported_files/hamster_DE_gene_category_counts_logFC_", logFC_cutoff,".csv", sep = ""))

#import the mouse DE categories df for plotting, match the current logFC cutoff
mouse_DE_categories <- read.csv(paste("exported_files/mouse_DE_gene_category_counts_logFC_",
                                      logFC_cutoff,".csv", sep = ""),
                                row.names = 1)

mouse_DE_categories_number_long <- data.frame(pivot_longer(mouse_DE_categories, cols = 1:4, names_to = "category", values_to = "number_DE"))

comb_DE_categories_number_long <- rbind(hamster_DE_categories_number_long, mouse_DE_categories_number_long)

#add in empty hamster RS rows for plotting
comb_DE_categories_number_long <- rbind(comb_DE_categories_number_long, 
                                        c(0, "RS", "hamster", 0, 0, 0, 0, "transgressive", 0))


comb_DE_categories_number_long$stage <- factor(comb_DE_categories_number_long$stage, levels = c("SP", "LZ", "DIP", "RS"))

comb_DE_categories_number_long$category <- factor(comb_DE_categories_number_long$category ,
                                                  levels = c("diffX_only", "sameX_only", "intermediate", "transgressive"))

comb_DE_categories_number_long$number_DE <- as.numeric(comb_DE_categories_number_long$number_DE)
comb_DE_categories_number_long$stage_x_species <- paste(comb_DE_categories_number_long$species,
                                                        comb_DE_categories_number_long$stage,
                                                        sep = "_")
comb_DE_categories_number_long$stage_x_species <- factor(comb_DE_categories_number_long$stage_x_species,
                                                         levels = c("mouse_SP", "hamster_SP",
                                                                    "mouse_LZ", "hamster_LZ",
                                                                    "mouse_DIP", "hamster_DIP",
                                                                    "mouse_RS", "hamster_RS"))



mouse_v_hamster_categories_plot <- ggplot(mapping = aes(x = stage_x_species, y = number_DE)) +
  
  geom_bar(position="stack", stat="identity", 
           data = comb_DE_categories_number_long[comb_DE_categories_number_long$stage == "SP" &
                                                   comb_DE_categories_number_long$species == "mouse",], 
           aes(fill = category)) +
  scale_fill_manual(values = c(gray_4, gray_5, sp_color, sp_color_darkest)) +
  new_scale_fill() + 
  #hamster SP
  geom_bar(position="stack", stat="identity", 
           data = comb_DE_categories_number_long[comb_DE_categories_number_long$stage == "SP" &
                                                   comb_DE_categories_number_long$species == "hamster",], 
           aes(fill = category)) +
  scale_fill_manual(values = c(gray_4, gray_5, sp_color, sp_color_darkest)) +
  new_scale_fill() + 
  #mouse LZ
  geom_bar(position="stack", stat="identity", 
           data = comb_DE_categories_number_long[comb_DE_categories_number_long$stage == "LZ" &
                                                   comb_DE_categories_number_long$species == "mouse",], 
           aes(fill = category)) +
  scale_fill_manual(values = c(gray_4, gray_5, lz_color, lz_color_darkest)) +
  new_scale_fill() + 
  #hamster LZ
  geom_bar(position="stack", stat="identity", 
           data = comb_DE_categories_number_long[comb_DE_categories_number_long$stage == "LZ" &
                                                   comb_DE_categories_number_long$species == "hamster",], 
           aes(fill = category)) +
  scale_fill_manual(values = c(gray_4, gray_5, lz_color, lz_color_darkest)) +
  new_scale_fill() + 
  #mouse DIP
  geom_bar(position="stack", stat="identity", 
           data = comb_DE_categories_number_long[comb_DE_categories_number_long$stage == "DIP" &
                                                   comb_DE_categories_number_long$species == "mouse",], 
           aes(fill = category)) +
  scale_fill_manual(values = c(gray_4, gray_5, dip_color, dip_color_darkest)) +
  new_scale_fill() + 
  
  #hamster DIP
  geom_bar(position="stack", stat="identity", 
           data = comb_DE_categories_number_long[comb_DE_categories_number_long$stage == "DIP" &
                                                   comb_DE_categories_number_long$species == "hamster",], 
           aes(fill = category)) +
  scale_fill_manual(values = c(gray_4, gray_5, dip_color, dip_color_darkest)) +
  new_scale_fill() + 
  
  #mouse RS
  geom_bar(position="stack", stat="identity", 
           data = comb_DE_categories_number_long[comb_DE_categories_number_long$stage == "RS" &
                                                   comb_DE_categories_number_long$species == "mouse",], 
           aes(fill = category)) +
  scale_fill_manual(values = c(gray_4, gray_5, rs_color, rs_color_darkest)) +
  #hamster RS
  geom_bar(position="stack", stat="identity", 
           data = comb_DE_categories_number_long[comb_DE_categories_number_long$stage == "RS" &
                                                   comb_DE_categories_number_long$species == "hamster",], 
           aes(fill = category)) +
  scale_fill_manual(values = c(gray_4, gray_5, rs_color, rs_color_darkest)) +
  #general formatting
  theme_bw() + 
  scale_x_discrete(labels= rep(c("Mouse", "Hamster"), 4)) +
  theme(text = element_text(size = text_size*3),
        panel.grid.minor = element_blank(),
        panel.grid.major.x = element_blank(),
        axis.title.x = element_blank(),
        axis.text.x = element_text(angle = 45, hjust = 1),
        legend.position = "none") +
  labs(y = "Number of all DE genes", x = "Stage of spermatogenesis")
```

```{r all-DE-category-plot}
#write DE gene categories out to files
write.csv(Hyb.Comb.SP.tt,  paste("exported_files/hybrid_hamster_Comb_SP_DE_Genes_wCategories_logFC_", logFC_cutoff ,".csv", sep = ""))
write.csv(Hyb.Comb.LZ.tt,  paste("exported_files/hybrid_hamster_Comb_LZ_DE_Genes_wCategories_logFC_", logFC_cutoff ,".csv", sep = ""))
write.csv(Hyb.Comb.DIP.tt, paste("exported_files/hybrid_hamster_Comb_DIP_DE_Genes_wCategories_logFC_", logFC_cutoff ,".csv", sep = ""))


###assign hamster
Hamster.SP.DE <- Hyb.Comb.SP.tt[Hyb.Comb.SP.tt$sig_in_both == "yes",]
Hamster.LZ.DE <- Hyb.Comb.LZ.tt[Hyb.Comb.LZ.tt$sig_in_both == "yes",]
Hamster.DIP.DE <- Hyb.Comb.DIP.tt[Hyb.Comb.DIP.tt$sig_in_both == "yes",]

Hamster.SP.trans <- Hyb.Comb.SP.tt[Hyb.Comb.SP.tt$category == "transgressive",]
Hamster.LZ.trans <- Hyb.Comb.LZ.tt[Hyb.Comb.LZ.tt$category == "transgressive",]
Hamster.DIP.trans <- Hyb.Comb.DIP.tt[Hyb.Comb.DIP.tt$category == "transgressive",]

Hamster.SP.inter <- Hyb.Comb.SP.tt[Hyb.Comb.SP.tt$category == "intermediate",]
Hamster.LZ.inter <- Hyb.Comb.LZ.tt[Hyb.Comb.LZ.tt$category == "intermediate",]
Hamster.DIP.inter <- Hyb.Comb.DIP.tt[Hyb.Comb.DIP.tt$category == "intermediate",]

#write hamster trans DE gene info out to files for post-analysis in processing
write.csv(Hamster.SP.trans, "circos_processing_scripts/hamster_circos_hybrids/data/hybrid_hamster_trans_SP_DE_Genes.csv")
write.csv(Hamster.LZ.trans, "circos_processing_scripts/hamster_circos_hybrids/data/hybrid_hamster_trans_LZ_DE_Genes.csv")
write.csv(Hamster.DIP.trans, "circos_processing_scripts/hamster_circos_hybrids/data/hybrid_hamster_trans_DIP_DE_Genes.csv")

#print DE summary stats; specifically, numbers of DE genes for different contrasts
#transgressive DEs
print(paste("There are ", dim(Hamster.SP.trans)[1],
            " genes transgressively differentially expressed in spermatogonia between hybrids and both parents", 
            sep = ""))
print(paste("There are ", dim(Hamster.LZ.trans)[1],
            " genes transgressively differentially expressed in leptotene/zygotene between hybrids and both parents", 
            sep = ""))
print(paste("There are ", dim(Hamster.DIP.trans)[1],
            " genes transgressively differentially expressed in diplotene between hybrids and both parents", 
            sep = ""))

####assign mouse
Mouse.SP.tt <- read.csv(paste("exported_files/hybrid_mouse_Comb_SP_DE_Genes_wCategories_logFC_", logFC_cutoff ,".csv", sep = ""), row.names = 1)
Mouse.LZ.tt <- read.csv(paste("exported_files/hybrid_mouse_Comb_LZ_DE_Genes_wCategories_logFC_", logFC_cutoff, ".csv", sep = ""), row.names = 1)
Mouse.DIP.tt <- read.csv(paste("exported_files/hybrid_mouse_Comb_DIP_DE_Genes_wCategories_logFC_", logFC_cutoff, ".csv", sep = ""), row.names = 1)
Mouse.RS.tt <- read.csv(paste("exported_files/hybrid_mouse_Comb_RS_DE_Genes_wCategories_logFC_", logFC_cutoff, ".csv", sep = ""), row.names = 1)

Mouse.SP.DE <- Mouse.SP.tt[Mouse.SP.tt$sig_in_both == "yes",]
Mouse.LZ.DE <- Mouse.LZ.tt[Mouse.LZ.tt$sig_in_both == "yes",]
Mouse.DIP.DE <- Mouse.DIP.tt[Mouse.DIP.tt$sig_in_both == "yes",]
Mouse.RS.DE <- Mouse.RS.tt[Mouse.RS.tt$sig_in_both == "yes",]

Mouse.SP.trans <- Mouse.SP.tt[Mouse.SP.tt$category == "transgressive",]
Mouse.LZ.trans <- Mouse.LZ.tt[Mouse.LZ.tt$category == "transgressive",]
Mouse.DIP.trans <- Mouse.DIP.tt[Mouse.DIP.tt$category == "transgressive",]
Mouse.RS.trans <- Mouse.RS.tt[Mouse.RS.tt$category == "transgressive",]

Mouse.SP.inter <- Mouse.SP.tt[Mouse.SP.tt$category == "intermediate",]
Mouse.LZ.inter <- Mouse.LZ.tt[Mouse.LZ.tt$category == "intermediate",]
Mouse.DIP.inter <- Mouse.DIP.tt[Mouse.DIP.tt$category == "intermediate",]
Mouse.RS.inter <- Mouse.RS.tt[Mouse.RS.tt$category == "intermediate",]
```

<h3>How does the number of DE genes over time change?</h3>
```{r DE-over-time-ham-and-mouse}
#this code generates a plot of the number of DE genes between hybrids and parents with the same X chromosome across the different stages of spermatogenesis
DE_over_time_df <- data.frame(species = c(rep("mouse", 4), rep("ham", 3)),
                              stage = c("SP", "LZ", "DIP", "RS", "SP", "LZ", "DIP"),
                              DE_genes = c(dim(Mouse.SP.DE)[1],
                                           dim(Mouse.LZ.DE)[1],
                                           dim(Mouse.DIP.DE)[1],
                                           dim(Mouse.RS.DE)[1],
                                           dim(Hamster.SP.DE)[1],
                                           dim(Hamster.LZ.DE)[1],
                                           dim(Hamster.DIP.DE)[1]),
                              trans_genes = c(dim(Mouse.SP.trans)[1],
                                              dim(Mouse.LZ.trans)[1],
                                              dim(Mouse.DIP.trans)[1],
                                              dim(Mouse.RS.trans)[1],
                                              dim(Hamster.SP.trans)[1],
                                              dim(Hamster.LZ.trans)[1],
                                              dim(Hamster.DIP.trans)[1]),
                              inter_genes = c(dim(Mouse.SP.inter)[1],
                                              dim(Mouse.LZ.inter)[1],
                                              dim(Mouse.DIP.inter)[1],
                                              dim(Mouse.RS.inter)[1],
                                              dim(Hamster.SP.inter)[1],
                                              dim(Hamster.LZ.inter)[1],
                                              dim(Hamster.DIP.inter)[1]),
                              exp_genes = c(dim(read.csv("exported_files/Mouse_SP_exp_genes.csv"))[1],
                                            dim(read.csv("exported_files/Mouse_LZ_exp_genes.csv"))[1],
                                            dim(read.csv("exported_files/Mouse_DIP_exp_genes.csv"))[1],
                                            dim(read.csv("exported_files/Mouse_RS_exp_genes.csv"))[1],
                                            dim(Hyb_Comb_SP_exp)[1],
                                            dim(Hyb_Comb_LZ_exp)[1],
                                            dim(Hyb_Comb_DIP_exp)[1]))


DE_over_time_df$stage <- factor(DE_over_time_df$stage, levels = c("SP", "LZ", "DIP", "RS"))

DE_over_time_df$percent <- DE_over_time_df$DE_genes/DE_over_time_df$exp_genes

DE_over_time_df$comp <- paste(DE_over_time_df$species, DE_over_time_df$stage, sep = "_")
DE_over_time_df$comp <- factor(DE_over_time_df$comp, levels = c("mouse_SP", "ham_SP", 
                                                                "mouse_LZ", "ham_LZ",
                                                                "mouse_DIP", "ham_DIP",
                                                                "mouse_RS"))

#this plot compares both and is in the supplemental material of the manuscript
DE_over_time_DE_plot_comparison <- ggplot(data = DE_over_time_df, aes(x=stage, y=DE_genes, group=species, color=stage, shape = species)) +
  geom_line(color = "black", linewidth = 1) +
  scale_color_manual(values=c(sp_color, lz_color, dip_color, rs_color)) + 
  scale_alpha_manual(values = c(0.3, 1)) +
  geom_point(size = 5) +
  theme_bw() + 
  scale_x_discrete(labels= c("Spermatogonia", "Leptotene/zygotene", "Diplotene", "Round spermatids")) +
  theme(text = element_text(size = text_size*3),
        panel.grid.minor = element_blank(),
        panel.grid.major.x = element_blank(),
        axis.title.x = element_blank(),
        axis.text.x = element_text(angle = 45, hjust = 1)) +
  labs(y = "Number of all DE genes", x = "Stage of spermatogenesis")

trans_over_time_DE_plot_comparison <- ggplot(data = DE_over_time_df, aes(x=stage, y=trans_genes, group=species, color=stage, shape = species)) +
  geom_line(color = "black", linewidth = 1) +
  scale_color_manual(values=c(sp_color, lz_color, dip_color, rs_color)) + 
  scale_alpha_manual(values = c(0.3, 1)) +
  geom_point(size = 5) +
  theme_bw() + 
  scale_x_discrete(labels= c("SP", "LZ", "DIP", "RS")) +
  theme(text = element_text(size = text_size*3),
        panel.grid.minor = element_blank(),
        panel.grid.major.x = element_blank(),
        axis.title.x = element_blank(),
        axis.text.x = element_text(angle = 45, hjust = 1)) +
  labs(y = "Number of transgressive DE genes", x = "Stage of spermatogenesis")

inter_over_time_DE_plot_comparison <- ggplot(data = DE_over_time_df, aes(x=stage, y=inter_genes, group=species, color=stage, shape = species)) +
  geom_line(color = "black", linewidth = 1) +
  scale_color_manual(values=c(sp_color, lz_color, dip_color, rs_color)) + 
  scale_alpha_manual(values = c(0.3, 1)) +
  geom_point(size = 5) +
  theme_bw() + 
  scale_x_discrete(labels= c("Spermatogonia", "Leptotene/zygotene", "Diplotene", "Round spermatids")) +
  theme(text = element_text(size = text_size*3),
        panel.grid.minor = element_blank(),
        panel.grid.major.x = element_blank(),
        axis.title.x = element_blank(),
        axis.text.x = element_text(angle = 45, hjust = 1)) +
  labs(y = "Number of intermediate DE genes", x = "Stage of spermatogenesis")

if(exportFigs == TRUE) {
  ggsave(filename = "exported_figures/DE_over_time_DE_plot_both_comparison.pdf", 
         DE_over_time_DE_plot_comparison, width = 7, height = 4)
  ggsave(filename = "exported_figures/trans_over_time_DE_plot_both_comparison.pdf",
         trans_over_time_DE_plot_comparison, width = 7, height = 4)
  ggsave(filename = "exported_figures/inter_over_time_DE_plot_both_comparison.pdf", 
         inter_over_time_DE_plot_comparison, width = 7, height = 4)
} else {
  DE_over_time_DE_plot_comparison
  trans_over_time_DE_plot_comparison
  inter_over_time_DE_plot_comparison
}
```


```{r graphing-DE-categories-hybrids}
hamster_DE_categories_percent_long <- data.frame(pivot_longer(hamster_DE_categories, cols = grep("perc", colnames(hamster_DE_categories)), names_to = "category", values_to = "percent_DE"))

mouse_DE_categories_percent_long <- data.frame(pivot_longer(mouse_DE_categories, cols = grep("perc", colnames(mouse_DE_categories)), names_to = "category", values_to = "percent_DE"))

comb_DE_categories_percent_long <- rbind(hamster_DE_categories_percent_long, mouse_DE_categories_percent_long)

#add in empty hamster RS rows for plotting
comb_DE_categories_percent_long <- rbind(comb_DE_categories_percent_long, 
                                         c(0, 0, 0, 0, 0, "RS", "hamster", "perc_transgressive", 0))


comb_DE_categories_percent_long$stage <- factor(comb_DE_categories_percent_long$stage, levels = c("SP", "LZ", "DIP", "RS"))

comb_DE_categories_percent_long$category <- gsub("perc_", "", comb_DE_categories_percent_long$category)
comb_DE_categories_percent_long$category <- factor(comb_DE_categories_percent_long$category ,
                                                   levels = c("sameX_only", "diffX_only", "intermediate", "transgressive"))

comb_DE_categories_percent_long$percent_DE <- as.numeric(comb_DE_categories_percent_long$percent_DE)
comb_DE_categories_percent_long$stage_x_species <- paste(comb_DE_categories_percent_long$species,
                                                         comb_DE_categories_percent_long$stage,
                                                         sep = "_")
comb_DE_categories_percent_long$stage_x_species <- factor(comb_DE_categories_percent_long$stage_x_species,
                                                          levels = c("mouse_SP", "hamster_SP",
                                                                     "mouse_LZ", "hamster_LZ",
                                                                     "mouse_DIP", "hamster_DIP",
                                                                     "mouse_RS", "hamster_RS"))

comb_DE_categories_number_long$species <- factor(comb_DE_categories_number_long$species, levels = c("mouse", "hamster"))

comb_DE_categories_number_long$category <- factor(comb_DE_categories_number_long$category ,
                                                  levels = c("sameX_only", "diffX_only", "intermediate", "transgressive"))

setorder(comb_DE_categories_number_long, category)

hybrid_DE_area_plot <- ggplot(comb_DE_categories_number_long[!(comb_DE_categories_number_long$stage == "RS" &
                                                                 comb_DE_categories_number_long$species == "hamster"),], 
                              aes(x = stage, y = number_DE, group = category, fill = category)) +
  geom_area() +
  scale_fill_manual(values = c(gray_1, gray_3, gray_4, gray_5)) +
  facet_wrap(vars(species)) +
  theme_minimal() +
  scale_x_discrete(labels= rep(c("SP", "LZ", "DIP", "RS"),2)) +
  ylab("Number of hybrid DE genes") +
  theme(panel.grid.minor.y = element_blank(),
        panel.grid.minor.x = element_blank(),
        panel.grid.major.x = element_blank(),
        legend.position = "bottom",
        axis.text.x = element_text(angle = 45, vjust = 0.75),
        text = element_text(size = text_size*1.5)) +
  ylim(0, max(comb_DE_categories_number_long$number_DE)*1.3)

```

```{r graphing-DE-categories-parents}
parent_hamster_DE_categories <- data.frame(sameX_down = c(dim(BBBB.SSSS.SP.de[BBBB.SSSS.SP.de$logFC < 0,])[1],
                                                          dim(BBBB.SSSS.LZ.de[BBBB.SSSS.LZ.de$logFC < 0,])[1],
                                                          dim(BBBB.SSSS.DIP.de[BBBB.SSSS.DIP.de$logFC < 0,])[1],
                                                          dim(BBBB.SSSS.RS.de[BBBB.SSSS.RS.de$logFC < 0,])[1]),
                                           
                                           sameX_up = c(dim(BBBB.SSSS.SP.de[BBBB.SSSS.SP.de$logFC > 0,])[1],
                                                        dim(BBBB.SSSS.LZ.de[BBBB.SSSS.LZ.de$logFC > 0,])[1],
                                                        dim(BBBB.SSSS.DIP.de[BBBB.SSSS.DIP.de$logFC > 0,])[1],
                                                        dim(BBBB.SSSS.RS.de[BBBB.SSSS.RS.de$logFC > 0,])[1]),
                                           
                                           stage = stages,
                                           
                                           species = rep("hamster", 4))

parent_hamster_DE_categories$sameX_down_percent <- 100*parent_hamster_DE_categories$sameX_down/
  (parent_hamster_DE_categories$sameX_down + parent_hamster_DE_categories$sameX_up)
parent_hamster_DE_categories$sameX_up_percent <- 100*parent_hamster_DE_categories$sameX_up/
  (parent_hamster_DE_categories$sameX_down + parent_hamster_DE_categories$sameX_up)



MusP.DomP.SP.de <- read.csv(paste("exported_files/parent_mouse_SP_DE_Genes_logFC_", logFC_cutoff, ".csv", sep = ""))
MusP.DomP.LZ.de <- read.csv(paste("exported_files/parent_mouse_LZ_DE_Genes_logFC_", logFC_cutoff, ".csv", sep = ""))
MusP.DomP.DIP.de <- read.csv(paste("exported_files/parent_mouse_DIP_DE_Genes_logFC_", logFC_cutoff, ".csv", sep = ""))
MusP.DomP.RS.de <- read.csv(paste("exported_files/parent_mouse_RS_DE_Genes_logFC_", logFC_cutoff, ".csv", sep = ""))


parent_mouse_DE_categories <- data.frame(sameX_down = c(dim(MusP.DomP.SP.de[MusP.DomP.SP.de$logFC < 0,])[1],
                                                        dim(MusP.DomP.LZ.de[MusP.DomP.LZ.de$logFC < 0,])[1],
                                                        dim(MusP.DomP.DIP.de[MusP.DomP.DIP.de$logFC < 0,])[1],
                                                        dim(MusP.DomP.RS.de[MusP.DomP.RS.de$logFC < 0,])[1]),
                                         
                                         sameX_up = c(dim(MusP.DomP.SP.de[MusP.DomP.SP.de$logFC > 0,])[1],
                                                      dim(MusP.DomP.LZ.de[MusP.DomP.LZ.de$logFC > 0,])[1],
                                                      dim(MusP.DomP.DIP.de[MusP.DomP.DIP.de$logFC > 0,])[1],
                                                      dim(MusP.DomP.RS.de[MusP.DomP.RS.de$logFC > 0,])[1]),
                                         
                                         stage = stages,
                                         
                                         species = rep("mouse", 4))

parent_mouse_DE_categories$sameX_down_percent <- 100*parent_mouse_DE_categories$sameX_down/
  (parent_mouse_DE_categories$sameX_down + parent_mouse_DE_categories$sameX_up)
parent_mouse_DE_categories$sameX_up_percent <- 100*parent_mouse_DE_categories$sameX_up/
  (parent_mouse_DE_categories$sameX_down + parent_mouse_DE_categories$sameX_up)

parent_combined_DE_categories <- rbind(parent_hamster_DE_categories, parent_mouse_DE_categories)

parent_combined_DE_categories$stage <- factor(parent_combined_DE_categories$stage,
                                              levels = c("SP", "LZ", "DIP", "RS"))
parent_combined_DE_categories$species <- factor(parent_combined_DE_categories$species, levels = c("mouse", "hamster"))

parent_combined_DE_categories_long <- data.frame(pivot_longer(parent_combined_DE_categories, cols = c(1, 2, 5, 6), names_to = "direction", values_to = "number_or_percent_DE"))

parent_DE_area_plot <- ggplot(parent_combined_DE_categories_long[grep("percent",                                                                       parent_combined_DE_categories_long$direction, 
                                                                      invert = TRUE),], 
                              aes(x = stage, y = number_or_percent_DE, 
                                  group = direction, fill = direction)) +
  geom_area() +
  scale_fill_manual(values = c(gray_1, gray_3)) +
  facet_wrap(vars(species)) +
  theme_minimal() +
  scale_x_discrete(labels= rep(c("SP", "LZ", "DIP", "RS"), 2)) +
  ylab("Number of parental DE genes") +
  theme(panel.grid.minor.y = element_blank(),
        panel.grid.minor.x = element_blank(),
        panel.grid.major.x = element_blank(),
        legend.position = "bottom",
        axis.text.x = element_text(angle = 45, vjust = 0.5),
        text = element_text(size = text_size*1.5)) +
  ylim(0, max(comb_DE_categories_number_long$number_DE)*1.3)
```


```{r gProfileR-hamster-DE-highlighted-terms}
FDR_cutoff <- 0.05

#Hyb.Comb.SP_trans go term enrichment
go_terms_Hyb.Comb.SP_trans_highlight <- gost(query = Hyb.Comb.SP.tt[Hyb.Comb.SP.tt$category == "transgressive",]$mouse_ensemble_ID, 
                                             organism = "mmusculus", ordered_query = FALSE, 
                                             multi_query = FALSE, significant = TRUE, exclude_iea = FALSE, 
                                             measure_underrepresentation = FALSE, evcodes = FALSE, 
                                             user_threshold = FDR_cutoff, correction_method = "g_SCS", 
                                             domain_scope = "custom", highlight = TRUE,
                                             custom_bg = annotations[annotations$gene %in%
                                                                       Hyb_Comb_SP_exp$gene,]$mouse_ensemble_ID, 
                                             numeric_ns = "", sources = c("GO:BP", "GO:CC", "GO:MF"), 
                                             as_short_link = FALSE)

#Hyb.Comb.LZ_trans go term enrichment
go_terms_Hyb.Comb.LZ_trans_highlight <- gost(query = Hyb.Comb.LZ.tt[Hyb.Comb.LZ.tt$category == "transgressive",]$mouse_ensemble_ID, 
                                             organism = "mmusculus", ordered_query = FALSE, 
                                             multi_query = FALSE, significant = TRUE, exclude_iea = FALSE, 
                                             measure_underrepresentation = FALSE, evcodes = FALSE, 
                                             user_threshold = FDR_cutoff, correction_method = "g_SCS", 
                                             domain_scope = "custom", highlight = TRUE,
                                             custom_bg = annotations[annotations$gene %in%
                                                                       Hyb_Comb_LZ_exp$gene,]$mouse_ensemble_ID, 
                                             numeric_ns = "", sources = c("GO:BP", "GO:CC", "GO:MF"), 
                                             as_short_link = FALSE)

#Hyb.Comb.DIP_trans go term enrichment
go_terms_Hyb.Comb.DIP_trans_highlight <- gost(query = Hyb.Comb.DIP.tt[Hyb.Comb.DIP.tt$category == "transgressive",]$mouse_ensemble_ID, 
                                              organism = "mmusculus", ordered_query = FALSE, 
                                              multi_query = FALSE, significant = TRUE, exclude_iea = FALSE, 
                                              measure_underrepresentation = FALSE, evcodes = FALSE, 
                                              user_threshold = FDR_cutoff, correction_method = "g_SCS", 
                                              domain_scope = "custom", highlight = TRUE,
                                              custom_bg = annotations[annotations$gene %in%
                                                                        Hyb_Comb_DIP_exp$gene,]$mouse_ensemble_ID, 
                                              numeric_ns = "", sources = c("GO:BP", "GO:CC", "GO:MF"), 
                                              as_short_link = FALSE)

go_terms_Hyb.Comb.SP_trans_highlight_results <- go_terms_Hyb.Comb.SP_trans_highlight$result
go_terms_Hyb.Comb.SP_trans_highlight_results$exp <- go_terms_Hyb.Comb.SP_trans_highlight_results$query_size*(go_terms_Hyb.Comb.SP_trans_highlight_results$term_size/go_terms_Hyb.Comb.SP_trans_highlight_results$effective_domain_size)
go_terms_Hyb.Comb.SP_trans_highlight_results$fold.enrichment <- go_terms_Hyb.Comb.SP_trans_highlight_results$intersection_size/go_terms_Hyb.Comb.SP_trans_highlight_results$exp

#plot significant and highlight GO terms for SP
SP_GO_highlight_plot <- ggplot(
  data = go_terms_Hyb.Comb.SP_trans_highlight_results[go_terms_Hyb.Comb.SP_trans_highlight_results$source == "GO:BP" &
                                                        go_terms_Hyb.Comb.SP_trans_highlight_results$highlighted,], 
  aes(x = fold.enrichment, y = reorder(term_name, fold.enrichment), 
      size = intersection_size, color = source)) +
  geom_point() +
  theme_minimal() + 
  scale_color_manual(values = c(sp_color)) +
  labs(x = "Fold enrichment", y = "") + 
  scale_size_continuous(name = "# Genes")

go_terms_Hyb.Comb.LZ_trans_highlight_results <- go_terms_Hyb.Comb.LZ_trans_highlight$result
go_terms_Hyb.Comb.LZ_trans_highlight_results$exp <- go_terms_Hyb.Comb.LZ_trans_highlight_results$query_size*(go_terms_Hyb.Comb.LZ_trans_highlight_results$term_size/go_terms_Hyb.Comb.LZ_trans_highlight_results$effective_domain_size)
go_terms_Hyb.Comb.LZ_trans_highlight_results$fold.enrichment <- go_terms_Hyb.Comb.LZ_trans_highlight_results$intersection_size/go_terms_Hyb.Comb.LZ_trans_highlight_results$exp

#plot significant and highlight GO terms for LZ
LZ_GO_highlight_plot <- ggplot(
  data = go_terms_Hyb.Comb.LZ_trans_highlight_results[go_terms_Hyb.Comb.LZ_trans_highlight_results$source == "GO:BP" &
                                                        go_terms_Hyb.Comb.LZ_trans_highlight_results$highlighted,], 
  aes(x = fold.enrichment, y = reorder(term_name, fold.enrichment), 
      size = intersection_size, color = source)) +
  geom_point() +
  theme_minimal() + 
  scale_color_manual(values = c(lz_color)) +
  labs(x = "Fold enrichment", y = "") + 
  scale_size_continuous(name = "# Genes")


go_terms_Hyb.Comb.DIP_trans_highlight_results <- go_terms_Hyb.Comb.DIP_trans_highlight$result
go_terms_Hyb.Comb.DIP_trans_highlight_results$exp <- go_terms_Hyb.Comb.DIP_trans_highlight_results$query_size*(go_terms_Hyb.Comb.DIP_trans_highlight_results$term_size/go_terms_Hyb.Comb.DIP_trans_highlight_results$effective_domain_size)
go_terms_Hyb.Comb.DIP_trans_highlight_results$fold.enrichment <- go_terms_Hyb.Comb.DIP_trans_highlight_results$intersection_size/go_terms_Hyb.Comb.DIP_trans_highlight_results$exp

#plot significant and highlight GO terms for DIP
DIP_GO_highlight_plot <- ggplot(
  data = go_terms_Hyb.Comb.DIP_trans_highlight_results[go_terms_Hyb.Comb.DIP_trans_highlight_results$source == "GO:BP" &
                                                         go_terms_Hyb.Comb.DIP_trans_highlight_results$highlighted,], 
  aes(x = fold.enrichment, y = reorder(term_name, fold.enrichment), 
      size = intersection_size, color = source)) +
  geom_point() +
  theme_minimal() + 
  scale_color_manual(values = c(dip_color)) +
  labs(x = "Fold enrichment", y = "") + 
  scale_size_continuous(name = "# Genes")


SP_GO_highlight_plot
LZ_GO_highlight_plot
DIP_GO_highlight_plot

#save full hamster GO term lists to files
write.csv(x = data.frame(go_terms_Hyb.Comb.SP_trans_highlight_results[,c("p_value", "term_size", "query_size",
                                                                         "intersection_size", "term_id", "source",
                                                                         "term_name", "effective_domain_size",
                                                                         "highlighted", "exp", "fold.enrichment")]),
          file = "exported_files/hamster_transgressive_DE_SP_GO_terms.csv", row.names = FALSE)

write.csv(x = data.frame(go_terms_Hyb.Comb.LZ_trans_highlight_results[,c("p_value", "term_size", "query_size",
                                                                         "intersection_size", "term_id", "source",
                                                                         "term_name", "effective_domain_size",
                                                                         "highlighted", "exp", "fold.enrichment")]),
          file = "exported_files/hamster_transgressive_DE_LZ_GO_terms.csv", row.names = FALSE)

write.csv(x = data.frame(go_terms_Hyb.Comb.DIP_trans_highlight_results[,c("p_value", "term_size", "query_size",
                                                                          "intersection_size", "term_id", "source",
                                                                          "term_name", "effective_domain_size",
                                                                          "highlighted", "exp", "fold.enrichment")]),
          file = "exported_files/hamster_transgressive_DE_DIP_GO_terms.csv", row.names = FALSE)
```

```{r gProfileR-hybrid-DE-all-terms}
FDR_cutoff <- 0.05

#Hyb.Comb.SP_trans go term enrichment
go_terms_Hyb.Comb.SP_trans_all <- gost(query = Hyb.Comb.SP.tt[Hyb.Comb.SP.tt$category == "transgressive",]$mouse_ensemble_ID, 
                                       organism = "mmusculus", ordered_query = FALSE, 
                                       multi_query = FALSE, significant = TRUE, exclude_iea = FALSE, 
                                       measure_underrepresentation = FALSE, evcodes = FALSE, 
                                       user_threshold = FDR_cutoff, correction_method = "g_SCS", 
                                       domain_scope = "custom", highlight = FALSE,
                                       custom_bg = annotations[annotations$gene %in%
                                                                 Hyb_Comb_SP_exp$gene,]$mouse_ensemble_ID, 
                                       numeric_ns = "", sources = c("GO:BP", "GO:CC", "GO:MF"), 
                                       as_short_link = FALSE)

#Hyb.Comb.LZ_trans go term enrichment
go_terms_Hyb.Comb.LZ_trans_all <- gost(query = Hyb.Comb.LZ.tt[Hyb.Comb.LZ.tt$category == "transgressive",]$mouse_ensemble_ID, 
                                       organism = "mmusculus", ordered_query = FALSE, 
                                       multi_query = FALSE, significant = TRUE, exclude_iea = FALSE, 
                                       measure_underrepresentation = FALSE, evcodes = FALSE, 
                                       user_threshold = FDR_cutoff, correction_method = "g_SCS", 
                                       domain_scope = "custom", highlight = FALSE,
                                       custom_bg = annotations[annotations$gene %in%
                                                                 Hyb_Comb_LZ_exp$gene,]$mouse_ensemble_ID, 
                                       numeric_ns = "", sources = c("GO:BP", "GO:CC", "GO:MF"), 
                                       as_short_link = FALSE)

#Hyb.Comb.DIP_trans go term enrichment
go_terms_Hyb.Comb.DIP_trans_all <- gost(query = Hyb.Comb.DIP.tt[Hyb.Comb.DIP.tt$category == "transgressive",]$mouse_ensemble_ID, 
                                        organism = "mmusculus", ordered_query = FALSE, 
                                        multi_query = FALSE, significant = TRUE, exclude_iea = FALSE, 
                                        measure_underrepresentation = FALSE, evcodes = FALSE, 
                                        user_threshold = FDR_cutoff, correction_method = "g_SCS", 
                                        domain_scope = "custom", highlight = FALSE,
                                        custom_bg = annotations[annotations$gene %in%
                                                                  Hyb_Comb_DIP_exp$gene,]$mouse_ensemble_ID, 
                                        numeric_ns = "", sources = c("GO:BP", "GO:CC", "GO:MF"), 
                                        as_short_link = FALSE)

go_terms_Hyb.Comb.SP_trans_all_results <- go_terms_Hyb.Comb.SP_trans_all$result
go_terms_Hyb.Comb.SP_trans_all_results$exp <- go_terms_Hyb.Comb.SP_trans_all_results$query_size*(go_terms_Hyb.Comb.SP_trans_all_results$term_size/go_terms_Hyb.Comb.SP_trans_all_results$effective_domain_size)
go_terms_Hyb.Comb.SP_trans_all_results$fold.enrichment <- go_terms_Hyb.Comb.SP_trans_all_results$intersection_size/go_terms_Hyb.Comb.SP_trans_all_results$exp

#plot significant and all GO terms for SP
SP_GO_all_plot <- ggplot(
  data = go_terms_Hyb.Comb.SP_trans_all_results[go_terms_Hyb.Comb.SP_trans_all_results$source == "GO:BP",], 
  aes(x = fold.enrichment, y = reorder(term_name, fold.enrichment), 
      size = intersection_size, color = source)) +
  geom_point() +
  theme_minimal() + 
  scale_color_manual(values = c(sp_color)) +
  labs(x = "Fold enrichment", y = "") + 
  scale_size_continuous(name = "# Genes")

go_terms_Hyb.Comb.LZ_trans_all_results <- go_terms_Hyb.Comb.LZ_trans_all$result
go_terms_Hyb.Comb.LZ_trans_all_results$exp <- go_terms_Hyb.Comb.LZ_trans_all_results$query_size*(go_terms_Hyb.Comb.LZ_trans_all_results$term_size/go_terms_Hyb.Comb.LZ_trans_all_results$effective_domain_size)
go_terms_Hyb.Comb.LZ_trans_all_results$fold.enrichment <- go_terms_Hyb.Comb.LZ_trans_all_results$intersection_size/go_terms_Hyb.Comb.LZ_trans_all_results$exp

#plot significant and all GO terms for LZ
LZ_GO_all_plot <- ggplot(
  data = go_terms_Hyb.Comb.LZ_trans_all_results[go_terms_Hyb.Comb.LZ_trans_all_results$source == "GO:BP",], 
  aes(x = fold.enrichment, y = reorder(term_name, fold.enrichment), 
      size = intersection_size, color = source)) +
  geom_point() +
  theme_minimal() + 
  scale_color_manual(values = c(lz_color)) +
  labs(x = "Fold enrichment", y = "") + 
  scale_size_continuous(name = "# Genes")


go_terms_Hyb.Comb.DIP_trans_all_results <- go_terms_Hyb.Comb.DIP_trans_all$result
go_terms_Hyb.Comb.DIP_trans_all_results$exp <- go_terms_Hyb.Comb.DIP_trans_all_results$query_size*(go_terms_Hyb.Comb.DIP_trans_all_results$term_size/go_terms_Hyb.Comb.DIP_trans_all_results$effective_domain_size)
go_terms_Hyb.Comb.DIP_trans_all_results$fold.enrichment <- go_terms_Hyb.Comb.DIP_trans_all_results$intersection_size/go_terms_Hyb.Comb.DIP_trans_all_results$exp

#plot significant and all GO terms for DIP
DIP_GO_all_plot <- ggplot(
  data = go_terms_Hyb.Comb.DIP_trans_all_results[go_terms_Hyb.Comb.DIP_trans_all_results$source == "GO:BP",], 
  aes(x = fold.enrichment, y = reorder(term_name, fold.enrichment), 
      size = intersection_size, color = source)) +
  geom_point() +
  theme_minimal() + 
  scale_color_manual(values = c(dip_color)) +
  labs(x = "Fold enrichment", y = "") + 
  scale_size_continuous(name = "# Genes")


SP_GO_all_plot
LZ_GO_all_plot
DIP_GO_all_plot
```

```{r gProfileR-mouse-DE-highlighted-terms}
FDR_cutoff <- 0.05

#Mouse.SP_trans go term enrichment
go_terms_Mouse.SP_trans_all <- gost(query = Mouse.SP.trans$gene, 
                                    organism = "mmusculus", ordered_query = FALSE, 
                                    multi_query = FALSE, significant = FALSE, exclude_iea = FALSE, 
                                    measure_underrepresentation = FALSE, evcodes = FALSE, 
                                    user_threshold = FDR_cutoff, correction_method = "g_SCS", 
                                    domain_scope = "custom", highlight = TRUE,
                                    custom_bg = read.csv("exported_files/Mouse_SP_exp_genes.csv")$gene, 
                                    numeric_ns = "", sources = c("GO:BP", "GO:CC", "GO:MF"), 
                                    as_short_link = FALSE)

#Mouse.LZ_trans go term enrichment
go_terms_Mouse.LZ_trans_all <- gost(query = Mouse.LZ.trans$gene, 
                                    organism = "mmusculus", ordered_query = FALSE, 
                                    multi_query = FALSE, significant = FALSE, exclude_iea = FALSE, 
                                    measure_underrepresentation = FALSE, evcodes = FALSE, 
                                    user_threshold = FDR_cutoff, correction_method = "g_SCS", 
                                    domain_scope = "custom", highlight = FALSE,
                                    custom_bg = read.csv("exported_files/Mouse_LZ_exp_genes.csv")$gene, 
                                    numeric_ns = "", sources = c("GO:BP", "GO:CC", "GO:MF"), 
                                    as_short_link = FALSE)

#Mouse.DIP_trans go term enrichment
go_terms_Mouse.DIP_trans_all <- gost(query = Mouse.LZ.trans$gene, 
                                     organism = "mmusculus", ordered_query = FALSE, 
                                     multi_query = FALSE, significant = FALSE, exclude_iea = FALSE, 
                                     measure_underrepresentation = FALSE, evcodes = FALSE, 
                                     user_threshold = FDR_cutoff, correction_method = "g_SCS", 
                                     domain_scope = "custom", highlight = FALSE,
                                     custom_bg = read.csv("exported_files/Mouse_DIP_exp_genes.csv")$gene, 
                                     numeric_ns = "", sources = c("GO:BP", "GO:CC", "GO:MF"), 
                                     as_short_link = FALSE)

#Mouse.RS_trans go term enrichment
go_terms_Mouse.RS_trans_all <- gost(query = Mouse.LZ.trans$gene, 
                                    organism = "mmusculus", ordered_query = FALSE, 
                                    multi_query = FALSE, significant = FALSE, exclude_iea = FALSE, 
                                    measure_underrepresentation = FALSE, evcodes = FALSE, 
                                    user_threshold = FDR_cutoff, correction_method = "g_SCS", 
                                    domain_scope = "custom", highlight = FALSE,
                                    custom_bg = read.csv("exported_files/Mouse_RS_exp_genes.csv")$gene, 
                                    numeric_ns = "", sources = c("GO:BP", "GO:CC", "GO:MF"), 
                                    as_short_link = FALSE)


go_terms_Mouse.SP_trans_all_results <- go_terms_Mouse.SP_trans_all$result
go_terms_Mouse.SP_trans_all_results$exp <- go_terms_Mouse.SP_trans_all_results$query_size*(go_terms_Mouse.SP_trans_all_results$term_size/go_terms_Mouse.SP_trans_all_results$effective_domain_size)
go_terms_Mouse.SP_trans_all_results$fold.enrichment <- go_terms_Mouse.SP_trans_all_results$intersection_size/go_terms_Mouse.SP_trans_all_results$exp

#plot significant and highlight GO terms for SP
SP_Mouse_GO_all_plot <- ggplot(
  data = go_terms_Mouse.SP_trans_all_results[go_terms_Mouse.SP_trans_all_results$significant == TRUE,], 
  aes(x = fold.enrichment, y = reorder(term_name, fold.enrichment), 
      size = intersection_size, color = source)) +
  geom_point() +
  theme_minimal() + 
  scale_color_manual(values = c(sp_color)) +
  labs(x = "Fold enrichment", y = "") + 
  scale_size_continuous(name = "# Genes")

go_terms_Mouse.LZ_trans_all_results <- go_terms_Mouse.LZ_trans_all$result
go_terms_Mouse.LZ_trans_all_results$exp <- go_terms_Mouse.LZ_trans_all_results$query_size*(go_terms_Mouse.LZ_trans_all_results$term_size/go_terms_Mouse.LZ_trans_all_results$effective_domain_size)
go_terms_Mouse.LZ_trans_all_results$fold.enrichment <- go_terms_Mouse.LZ_trans_all_results$intersection_size/go_terms_Mouse.LZ_trans_all_results$exp

#plot significant and highlight GO terms for LZ
LZ_Mouse_GO_all_plot <- ggplot(
  data = go_terms_Mouse.LZ_trans_all_results[go_terms_Mouse.LZ_trans_all_results$significant == TRUE,], 
  aes(x = fold.enrichment, y = reorder(term_name, fold.enrichment), 
      size = intersection_size, color = source)) +
  geom_point() +
  theme_minimal() + 
  scale_color_manual(values = c(lz_color)) +
  labs(x = "Fold enrichment", y = "") + 
  scale_size_continuous(name = "# Genes")


go_terms_Mouse.DIP_trans_all_results <- go_terms_Mouse.DIP_trans_all$result
go_terms_Mouse.DIP_trans_all_results$exp <- go_terms_Mouse.DIP_trans_all_results$query_size*(go_terms_Mouse.DIP_trans_all_results$term_size/go_terms_Mouse.DIP_trans_all_results$effective_domain_size)
go_terms_Mouse.DIP_trans_all_results$fold.enrichment <- go_terms_Mouse.DIP_trans_all_results$intersection_size/go_terms_Mouse.DIP_trans_all_results$exp

#plot significant and highlight GO terms for DIP
DIP_Mouse_GO_all_plot <- ggplot(
  data = go_terms_Mouse.DIP_trans_all_results[go_terms_Mouse.DIP_trans_all_results$significant == TRUE,], 
  aes(x = fold.enrichment, y = reorder(term_name, fold.enrichment), 
      size = intersection_size, color = source)) +
  geom_point() +
  theme_minimal() + 
  scale_color_manual(values = c(dip_color)) +
  labs(x = "Fold enrichment", y = "") + 
  scale_size_continuous(name = "# Genes")

go_terms_Mouse.RS_trans_all_results <- go_terms_Mouse.RS_trans_all$result
go_terms_Mouse.RS_trans_all_results$exp <- go_terms_Mouse.RS_trans_all_results$query_size*(go_terms_Mouse.RS_trans_all_results$term_size/go_terms_Mouse.RS_trans_all_results$effective_domain_size)
go_terms_Mouse.RS_trans_all_results$fold.enrichment <- go_terms_Mouse.RS_trans_all_results$intersection_size/go_terms_Mouse.RS_trans_all_results$exp

#plot significant and highlight GO terms for RS
RS_Mouse_GO_all_plot <- ggplot(
  data = go_terms_Mouse.RS_trans_all_results[go_terms_Mouse.RS_trans_all_results$significant == TRUE,], 
  aes(x = fold.enrichment, y = reorder(term_name, fold.enrichment), 
      size = intersection_size, color = source)) +
  geom_point() +
  theme_minimal() + 
  scale_color_manual(values = c(rs_color)) +
  labs(x = "Fold enrichment", y = "") + 
  scale_size_continuous(name = "# Genes")


SP_Mouse_GO_all_plot
LZ_Mouse_GO_all_plot
DIP_Mouse_GO_all_plot
RS_Mouse_GO_all_plot

#save full hamster GO term lists to files
write.csv(x = data.frame(go_terms_Mouse.SP_trans_all_results[,c("p_value", "term_size", "query_size",
                                                                "intersection_size", "term_id", "source",
                                                                "term_name", "effective_domain_size",
                                                                "exp", "fold.enrichment")]),
          file = "exported_files/mouse_transgressive_DE_SP_GO_terms.csv")

write.csv(x = data.frame(go_terms_Mouse.LZ_trans_all_results[,c("p_value", "term_size", "query_size",
                                                                "intersection_size", "term_id", "source",
                                                                "term_name", "effective_domain_size",
                                                                "exp", "fold.enrichment")]),
          file = "exported_files/mouse_transgressive_DE_LZ_GO_terms.csv")

write.csv(x = data.frame(go_terms_Mouse.DIP_trans_all_results[,c("p_value", "term_size", "query_size",
                                                                 "intersection_size", "term_id", "source",
                                                                 "term_name", "effective_domain_size",
                                                                 "exp", "fold.enrichment")]),
          file = "exported_files/mouse_transgressive_DE_DIP_GO_terms.csv")

write.csv(x = data.frame(go_terms_Mouse.RS_trans_all_results[,c("p_value", "term_size", "query_size",
                                                                "intersection_size", "term_id", "source",
                                                                "term_name", "effective_domain_size",
                                                                "exp", "fold.enrichment")]),
          file = "exported_files/mouse_transgressive_DE_RS_GO_terms.csv")

```

<h3>Compare the different extents of X-linked overexpression between species</h3>
```{r X-over-expression-versus-in-diplotene} 
#the goal of this analysis is to calculate the average (+/- SEM) relative overexpression of autosomal and X linked genes for each stage in hybrids and in parents
#set up rowMeans for stage x genotype x chromosome type for BBSS
rpkm_BBSS_SP_allAuto.norm.mean <- data.frame(rowMeans(rpkm_BBSS_SP_allAuto.norm))
colnames(rpkm_BBSS_SP_allAuto.norm.mean) <- "mean_FPKM"
rpkm_BBSS_SP_justX.norm.mean <- data.frame(rowMeans(rpkm_BBSS_SP_justX.norm))
colnames(rpkm_BBSS_SP_justX.norm.mean) <- "mean_FPKM"

rpkm_BBSS_LZ_allAuto.norm.mean <- data.frame(rowMeans(rpkm_BBSS_LZ_allAuto.norm))
colnames(rpkm_BBSS_LZ_allAuto.norm.mean) <- "mean_FPKM"
rpkm_BBSS_LZ_justX.norm.mean <- data.frame(rowMeans(rpkm_BBSS_LZ_justX.norm))
colnames(rpkm_BBSS_LZ_justX.norm.mean) <- "mean_FPKM"

rpkm_BBSS_DIP_allAuto.norm.mean <- data.frame(rowMeans(rpkm_BBSS_DIP_allAuto.norm))
colnames(rpkm_BBSS_DIP_allAuto.norm.mean) <- "mean_FPKM"
rpkm_BBSS_DIP_justX.norm.mean <- data.frame(rowMeans(rpkm_BBSS_DIP_justX.norm))
colnames(rpkm_BBSS_DIP_justX.norm.mean) <- "mean_FPKM"


#set up rowMeans for stage x genotype x chromosome type for BBBB
rpkm_BBBB_SP_allAuto.norm.mean <- data.frame(rowMeans(rpkm_BBBB_SP_allAuto.norm))
colnames(rpkm_BBBB_SP_allAuto.norm.mean) <- "mean_FPKM"
rpkm_BBBB_SP_justX.norm.mean <- data.frame(rowMeans(rpkm_BBBB_SP_justX.norm))
colnames(rpkm_BBBB_SP_justX.norm.mean) <- "mean_FPKM"

rpkm_BBBB_LZ_allAuto.norm.mean <- data.frame(rowMeans(rpkm_BBBB_LZ_allAuto.norm))
colnames(rpkm_BBBB_LZ_allAuto.norm.mean) <- "mean_FPKM"
rpkm_BBBB_LZ_justX.norm.mean <- data.frame(rowMeans(rpkm_BBBB_LZ_justX.norm))
colnames(rpkm_BBBB_LZ_justX.norm.mean) <- "mean_FPKM"

rpkm_BBBB_DIP_allAuto.norm.mean <- data.frame(rowMeans(rpkm_BBBB_DIP_allAuto.norm))
colnames(rpkm_BBBB_DIP_allAuto.norm.mean) <- "mean_FPKM"
rpkm_BBBB_DIP_justX.norm.mean <- data.frame(rowMeans(rpkm_BBBB_DIP_justX.norm))
colnames(rpkm_BBBB_DIP_justX.norm.mean) <- "mean_FPKM"

rpkm_BBBB_RS_allAuto.norm.mean <- data.frame(rowMeans(rpkm_BBBB_RS_allAuto.norm))
colnames(rpkm_BBBB_RS_allAuto.norm.mean) <- "mean_FPKM"
rpkm_BBBB_RS_justX.norm.mean <- data.frame(rowMeans(rpkm_BBBB_RS_justX.norm))
colnames(rpkm_BBBB_RS_justX.norm.mean) <- "mean_FPKM"


#set up rowMeans for stage x genotype x chromosome type for SSSS
rpkm_SSSS_SP_allAuto.norm.mean <- data.frame(rowMeans(rpkm_SSSS_SP_allAuto.norm))
colnames(rpkm_SSSS_SP_allAuto.norm.mean) <- "mean_FPKM"
rpkm_SSSS_SP_justX.norm.mean <- data.frame(rowMeans(rpkm_SSSS_SP_justX.norm))
colnames(rpkm_SSSS_SP_justX.norm.mean) <- "mean_FPKM"

rpkm_SSSS_LZ_allAuto.norm.mean <- data.frame(rowMeans(rpkm_SSSS_LZ_allAuto.norm))
colnames(rpkm_SSSS_LZ_allAuto.norm.mean) <- "mean_FPKM"
rpkm_SSSS_LZ_justX.norm.mean <- data.frame(rowMeans(rpkm_SSSS_LZ_justX.norm))
colnames(rpkm_SSSS_LZ_justX.norm.mean) <- "mean_FPKM"

rpkm_SSSS_DIP_allAuto.norm.mean <- data.frame(rowMeans(rpkm_SSSS_DIP_allAuto.norm))
colnames(rpkm_SSSS_DIP_allAuto.norm.mean) <- "mean_FPKM"
rpkm_SSSS_DIP_justX.norm.mean <- data.frame(rowMeans(rpkm_SSSS_DIP_justX.norm))
colnames(rpkm_SSSS_DIP_justX.norm.mean) <- "mean_FPKM"

rpkm_SSSS_RS_allAuto.norm.mean <- data.frame(rowMeans(rpkm_SSSS_RS_allAuto.norm))
colnames(rpkm_SSSS_RS_allAuto.norm.mean) <- "mean_FPKM"
rpkm_SSSS_RS_justX.norm.mean <- data.frame(rowMeans(rpkm_SSSS_RS_justX.norm))
colnames(rpkm_SSSS_RS_justX.norm.mean) <- "mean_FPKM"

#calculate the variance in X-linked overexpression; produces Figure S9
#creates a data frame that contains the average relative amount of X-linked overexpression in each diplotene individual
avg_overExpression_DIP_justX <- data.frame(overExp = 
                                             c(colMeans(rpkm_BBBB_DIP_justX.norm)/mean(rpkm_BBBB_DIP_justX.norm.mean$mean_FPKM),
                                               colMeans(rpkm_BBSS_DIP_justX.norm)/mean(rpkm_BBBB_DIP_justX.norm.mean$mean_FPKM),
                                               colMeans(rpkm_SSSS_DIP_justX.norm)/mean(rpkm_BBBB_DIP_justX.norm.mean$mean_FPKM)))
avg_overExpression_DIP_justX$ID <- gsub("_DIP", "", rownames(avg_overExpression_DIP_justX))
avg_overExpression_DIP_justX$Strain <- factor(gsub("_.*", "", avg_overExpression_DIP_justX$ID), levels = c("BBBB", "BBSS", "SSSS"))

#creates a data frame that contains the SEM of the relative amount of X-linked overexpression in each diplotene individual
var_overExpression_DIP_justX <- 
  data.frame(Strain = c("BBBB", "BBSS", "SSSS"),
             Var = c(calculate_sem(subset(avg_overExpression_DIP_justX,
                                          avg_overExpression_DIP_justX$Strain == "BBBB")$overExp),
                     calculate_sem(subset(avg_overExpression_DIP_justX, 
                                          avg_overExpression_DIP_justX$Strain == "BBSS")$overExp),
                     calculate_sem(subset(avg_overExpression_DIP_justX, 
                                          avg_overExpression_DIP_justX$Strain == "SSSS")$overExp)))

#plots the amount of X-linked over-expression in each diplotene sample
overExp_justX_variance_plot <- 
  ggplot(avg_overExpression_DIP_justX, aes(x=Strain, y = overExp, 
                                           shape = Strain, col = Strain, fill = Strain)) + 
  geom_quasirandom(aes(fill = Strain, color = Strain, shape = Strain), alpha = 1, size = 0.75) +
  geom_boxplot(width=0.1, outlier.shape = NA, lwd = 0.2, alpha = 0.2, color = "black") +
  scale_fill_manual(values=c(gray_5, gray_4, gray_2)) + 
  scale_color_manual(values=c(gray_5, gray_4, gray_2)) + 
  scale_shape_manual(values=c(24,4,25)) + 
  theme_bw() + 
  theme(legend.position = "none", 
        text = element_text(size = 6),
        axis.title.x=element_blank(), 
        axis.title.y = element_text(size = 6), 
        axis.text.x = element_text(angle = 45, vjust = 0.8, size = 6), 
        plot.margin = unit(c(0.4, 0.07, 0, 0.07), "cm"), 
        panel.grid.major.x = element_blank(), 
        panel.grid.minor = element_blank(), 
        axis.line.y = element_line(colour = "black", linewidth = 0.2), 
        panel.border = element_blank(), 
        axis.ticks.length.x = unit(0, "cm")) +
  scale_x_discrete(labels = c("Pcam", "Hybrid", "Psun")) + ylab("Relative X Overexpression") + 
  geom_signif(
    y_position = c(summary(avg_overExpression_DIP_justX[avg_overExpression_DIP_justX$Strain == "BBBB",]$overExp)[6]+2,
                   summary(avg_overExpression_DIP_justX[avg_overExpression_DIP_justX$Strain == "BBSS",]$overExp)[6]+2,
                   summary(avg_overExpression_DIP_justX[avg_overExpression_DIP_justX$Strain == "SSSS",]$overExp)[6]+2), 
    xmin = c(1,2,3), 
    xmax = c(1,2,3), 
    annotation = c(paste("SEM = ", round(var_overExpression_DIP_justX[1,2],2)),
                   paste("SEM = ", round(var_overExpression_DIP_justX[2,2],2)),
                   paste("SEM = ", round(var_overExpression_DIP_justX[3,2],2))), 
    tip_length = 0, textsize = sigText_size, size = 0) +
  ylim(values = c(0,round(1.1*max(avg_overExpression_DIP_justX$overExp))))


#calculate the variance in autosomal overexpression; produces Figure S9
#creates a data frame that contains the average relative amount of autosomal overexpression in each diplotene individual
avg_overExpression_DIP_allAuto <- data.frame(overExp = 
                                               c(colMeans(rpkm_BBBB_DIP_allAuto.norm)/mean(rpkm_BBBB_DIP_allAuto.norm.mean$mean_FPKM),
                                                 colMeans(rpkm_BBSS_DIP_allAuto.norm)/mean(rpkm_BBBB_DIP_allAuto.norm.mean$mean_FPKM),
                                                 colMeans(rpkm_SSSS_DIP_allAuto.norm)/mean(rpkm_BBBB_DIP_allAuto.norm.mean$mean_FPKM)))

avg_overExpression_DIP_allAuto$ID <- gsub("_DIP", "", rownames(avg_overExpression_DIP_allAuto))
avg_overExpression_DIP_allAuto$Strain <- factor(gsub("_.*", "", avg_overExpression_DIP_allAuto$ID), levels = c("BBBB", "BBSS", "SSSS"))

#creates a data frame that contains the SEM of the relative amount of autosomal overexpression in each diplotene individual
var_overExpression_DIP_allAuto <- 
  data.frame(Strain = c("BBBB", "BBSS", "SSSS"),
             Var = c(calculate_sem(subset(avg_overExpression_DIP_allAuto,
                                          avg_overExpression_DIP_allAuto$Strain == "BBBB")$overExp),
                     calculate_sem(subset(avg_overExpression_DIP_allAuto, 
                                          avg_overExpression_DIP_allAuto$Strain == "BBSS")$overExp),
                     calculate_sem(subset(avg_overExpression_DIP_allAuto, 
                                          avg_overExpression_DIP_allAuto$Strain == "SSSS")$overExp)))

#plots the amount of autosomal over-expression in each diplotene sample
overExp_allAuto_variance_plot <- 
  ggplot(avg_overExpression_DIP_allAuto, aes(x=Strain, y = overExp, 
                                             shape = Strain, col = Strain, fill = Strain)) + 
  geom_quasirandom(aes(fill = Strain, color = Strain, shape = Strain), alpha = 1, size = 0.75) +
  geom_boxplot(width=0.1, outlier.shape = NA, lwd = 0.2, alpha = 0.2, color = "black") +
  scale_fill_manual(values=c(gray_5, gray_4, gray_2)) + 
  scale_color_manual(values=c(gray_5, gray_4, gray_2)) + 
  scale_shape_manual(values=c(24,4,25)) + 
  theme_bw() + 
  theme(legend.position = "none", 
        text = element_text(size = 6),
        axis.title.x=element_blank(), 
        axis.title.y = element_text(size = 6), 
        axis.text.x = element_text(angle = 45, vjust = 0.8, size = 6), 
        plot.margin = unit(c(0.4, 0.07, 0, 0.07), "cm"), 
        panel.grid.major.x = element_blank(), 
        panel.grid.minor = element_blank(), 
        axis.line.y = element_line(colour = "black", linewidth = 0.2), 
        panel.border = element_blank(), 
        axis.ticks.length.x = unit(0, "cm")) +
  scale_x_discrete(labels = c("Pcam", "Hybrid", "Psun")) + ylab("Relative Auto Overexpression") + 
  geom_signif(
    y_position = c(summary(avg_overExpression_DIP_allAuto[avg_overExpression_DIP_allAuto$Strain == "BBBB",]$overExp)[6]+2,
                   summary(avg_overExpression_DIP_allAuto[avg_overExpression_DIP_allAuto$Strain == "BBSS",]$overExp)[6]+2,
                   summary(avg_overExpression_DIP_allAuto[avg_overExpression_DIP_allAuto$Strain == "SSSS",]$overExp)[6]+2), 
    xmin = c(1,2,3), 
    xmax = c(1,2,3), 
    annotation = c(paste("SEM = ", round(var_overExpression_DIP_allAuto[1,2],2)),
                   paste("SEM = ", round(var_overExpression_DIP_allAuto[2,2],2)),
                   paste("SEM = ", round(var_overExpression_DIP_allAuto[3,2],2))), 
    tip_length = 0, textsize = sigText_size, size = 0) +
  ylim(values = c(0,round(1.1*max(avg_overExpression_DIP_justX$overExp))))

if(exportFigs == TRUE) {
  ggsave(filename = "exported_figures/X_and_auto_overexpression_hamster.pdf", 
         plot_grid(overExp_allAuto_variance_plot, overExp_justX_variance_plot, nrow = 1),
         width = 4, height = 0.75*half_plot_height)
} else {
  plot_grid(overExp_allAuto_variance_plot, overExp_justX_variance_plot, nrow = 1)
}

#print out the relative overexpression of the X during diplotene in hybrids relative to parents
avg_overExpression_DIP_justX %>% group_by(Strain) %>%
  summarize_at('overExp',mean)


#set up logFC expressed dataframes to test if X is statistically overexpressed relative to autosomes
#BBSS - BBBB
BBSS_BBBB_SP_Dev_allAuto.logFC.diff <- data.frame(logFC = Hyb.BB.SP.tt$table[Hyb.BB.SP.tt$table$chr != "X",]$logFC)
BBSS_BBBB_SP_Dev_justX.logFC.diff <-   data.frame(logFC = Hyb.BB.SP.tt$table[Hyb.BB.SP.tt$table$chr == "X",]$logFC)

BBSS_BBBB_LZ_Dev_allAuto.logFC.diff <- data.frame(logFC = Hyb.BB.LZ.tt$table[Hyb.BB.LZ.tt$table$chr != "X",]$logFC)
BBSS_BBBB_LZ_Dev_justX.logFC.diff <-   data.frame(logFC = Hyb.BB.LZ.tt$table[Hyb.BB.LZ.tt$table$chr == "X",]$logFC)

BBSS_BBBB_DIP_Dev_allAuto.logFC.diff <- data.frame(logFC = Hyb.BB.DIP.tt$table[Hyb.BB.DIP.tt$table$chr != "X",]$logFC)
BBSS_BBBB_DIP_justX.logFC.diff <-   data.frame(logFC = Hyb.BB.DIP.tt$table[Hyb.BB.DIP.tt$table$chr == "X",]$logFC)

#wilcoxon rank sum tests
x_vs_auto_pvalues <- data.frame(names = c("BBSS_BBBB_SP", "BBSS_BBBB_LZ", "BBSS_BBBB_DIP"),
                                pValues = c(wilcox.test(BBSS_BBBB_SP_Dev_justX.logFC.diff$logFC, 
                                                        BBSS_BBBB_SP_Dev_allAuto.logFC.diff$logFC)$p.value,
                                            wilcox.test(BBSS_BBBB_LZ_Dev_justX.logFC.diff$logFC, 
                                                        BBSS_BBBB_LZ_Dev_allAuto.logFC.diff$logFC)$p.value,
                                            wilcox.test(BBSS_BBBB_DIP_justX.logFC.diff$logFC,
                                                        BBSS_BBBB_DIP_Dev_allAuto.logFC.diff$logFC)$p.value))

x_vs_auto_pvalues$pValues <- signif(p.adjust(x_vs_auto_pvalues$pValues, method = "fdr"), 2)
x_vs_auto_pvalues$significant <- "yes"
x_vs_auto_pvalues[x_vs_auto_pvalues$pValues > 0.05,]$significant <- "no"
```

<h3>Compare the logFC of Differentially Expressed Genes between Hybrids and Parents with Same X Chromosome for Both Mouse and Hamster across Spermatogenesis</h3>
```{r hybrid-DE-logFC-graphs-BBBB}
#this section produces figure 3 in the manuscript and demonstrates how patterns of up and downregulation across stages in hybrids differs between systems
#combine all stages within each species
Mouse.All.trans <- rbind(Mouse.SP.trans, Mouse.LZ.trans, Mouse.DIP.trans)
Hamster.All.trans <- rbind(Hamster.SP.trans, Hamster.LZ.trans, Hamster.DIP.trans)

#restrict hamster df to colnames shared with mus
Hamster.All.trans <- Hamster.All.trans[,intersect(colnames(Hamster.All.trans), colnames(Mouse.All.trans))]
Hamster.All.trans$gene_name <- NA

#combine mouse and hamster dataframes
Hyb.All.trans <- rbind(Mouse.All.trans, Hamster.All.trans)

#make plotting columns into factors
Hyb.All.trans$comparison <- paste(Hyb.All.trans$species, Hyb.All.trans$stage, sep = "_")
Hyb.All.trans$comparison <- factor(Hyb.All.trans$comparison, 
                                   levels = c("mouse_SP", "hamster_SP", 
                                              "mouse_LZ", "hamster_LZ", 
                                              "mouse_DIP", "hamster_DIP"))
Hyb.All.trans$stage <- factor(Hyb.All.trans$stage, levels = c("SP", "LZ", "DIP"))
Hyb.All.trans$species <- factor(Hyb.All.trans$species, levels = c("mouse", "hamster"))


#set up chi square function
calculate_chisq <- function(comp) {
  comparison <- paste(unlist(strsplit(comp, split = "_"))[1:2], collapse = "_")
  X_cat <- unlist(strsplit(comp, split = "_"))[3]
  logFC_df <- Hyb.All.trans[Hyb.All.trans$category == "transgressive" & Hyb.All.trans$chr != "Y",]
  
  #get observed up and down regulated DE gene numbers
  obs_chisq <- table(sign(logFC_df[logFC_df$sig_in_both == "yes" & 
                                     logFC_df$comparison == comparison & 
                                     logFC_df$X_category == X_cat,]$logFC.x))
  if(is.na(obs_chisq["1"])) {
    obs_chisq["1"] <- 0
  } else if(is.na(obs_chisq["-1"])) {
    obs_chisq["-1"] <- 0
  }
  names(obs_chisq) <- gsub("1", "up", gsub("-1", "down", names(obs_chisq)))
  chisq_results <- chisq.test(obs_chisq, p = c(0.5, 0.5))
  
  return(c(comparison, X_cat, chisq_results$parameter, sum(chisq_results$observed), round(chisq_results$statistic,4), chisq_results$p.value, obs_chisq["up"] > obs_chisq["down"], paste("d = ", obs_chisq["down"], sep = ""), paste("u = ", obs_chisq["up"], sep = "")))
}

#calculate chi_square of over vs under expressed transgressive DE genes for each species * stage * X_category
chi_square_table_BBBB_df <- data.frame(do.call(rbind, lapply(c("mouse_SP_Auto", 
                                                               "mouse_SP_X",
                                                               "mouse_LZ_Auto", 
                                                               "mouse_LZ_X",
                                                               "mouse_DIP_Auto", 
                                                               "mouse_DIP_X",
                                                               
                                                               "hamster_SP_Auto", 
                                                               "hamster_SP_X",
                                                               "hamster_LZ_Auto", 
                                                               "hamster_LZ_X",
                                                               "hamster_DIP_Auto", 
                                                               "hamster_DIP_X"), 
                                                             calculate_chisq)))

#reformat chi_square data frame for plotting
colnames(chi_square_table_BBBB_df) <- c("comparison", "X_category", "chisq_parameter", "observed_sum", "chisq_statistic", "p_value", "more_ups", "num_downs", "num_ups")

#calculate whether mouse and hamster differs in logFC of transgressive DE genes for each stage * X_category using a wilcoxin signed rank test
calculate_wilcoxin <- function(stage, X_cat) {
  #perform the wilcoxin and get the p_value
  comparison_p.value <- wilcox.test(Hyb.All.trans[Hyb.All.trans$category == "transgressive" &
                                                    Hyb.All.trans$comparison == paste("mouse", stage, sep = "_") & 
                                                    Hyb.All.trans$X_category == X_cat,]$logFC.x, 
                                    Hyb.All.trans[Hyb.All.trans$category == "transgressive" &
                                                    Hyb.All.trans$comparison == paste("hamster", stage, sep = "_") & 
                                                    Hyb.All.trans$X_category == X_cat,]$logFC.x)$p.value
  
  #does mouse have the higher logFC for the contrast?
  mouse_higher <- mean(Hyb.All.trans[Hyb.All.trans$category == "transgressive" &
                                       Hyb.All.trans$comparison == paste("mouse", stage, sep = "_") & 
                                       Hyb.All.trans$X_category == X_cat,]$logFC.x) >
    mean(Hyb.All.trans[Hyb.All.trans$category == "transgressive" &
                         Hyb.All.trans$comparison == paste("hamster", stage, sep = "_") & 
                         Hyb.All.trans$X_category == X_cat,]$logFC.x)
  
  return(c(stage, X_cat, comparison_p.value, mouse_higher))
}

#apply the wilcoxin function and save results to a data frame
wilcoxin_table_BBBB_df <- data.frame(t(mapply(calculate_wilcoxin,
                                              c("SP", "LZ", "DIP", "SP", "LZ", "DIP"),
                                              c("Auto", "Auto", "Auto", "X", "X", "X"))))

#format wilcoxin results
colnames(wilcoxin_table_BBBB_df) <- c("stage", "X_category", "p_value", "mouse_higher")

#reformatting for plotting
wilcoxin_table_BBBB_df$start <- wilcoxin_table_BBBB_df$stage
wilcoxin_table_BBBB_df$start <- gsub("^", "mouse_", wilcoxin_table_BBBB_df$start)
wilcoxin_table_BBBB_df$end <- wilcoxin_table_BBBB_df$stage
wilcoxin_table_BBBB_df$end <- gsub("^", "hamster_", wilcoxin_table_BBBB_df$end)

wilcoxin_table_BBBB_df$sig <- sig_conversion(wilcoxin_table_BBBB_df$p_value)

#plot the logFC of each gene by species * stage, with X_category as the facet
logFC_BBBB_trans_comps_plot <- ggplot(data = Hyb.All.trans[Hyb.All.trans$category == "transgressive" & Hyb.All.trans$chr != "Y",], aes(x = comparison, y = logFC.x)) +
  geom_hline(yintercept = 0, linewidth = 0.5) +
  geom_quasirandom(aes(col = comparison), alpha = 0.3, size = 2*point_size, varwidth = TRUE, stroke = NA, width = 0.2) + 
  scale_color_manual(values=c(sp_color, sp_color_dark,
                              lz_color, lz_color_dark,
                              dip_color, dip_color_dark)) + 
  new_scale_color() + 
  geom_violin(aes(col = comparison), scale = "width", fill = NA, width = 0.5) + 
  geom_boxplot(width=boxplot_width, outlier.shape = NA, alpha = 0.8) + 
  theme_minimal() +
  facet_wrap(vars(X_category), ncol = 4, scales = "free_y") + 
  scale_color_manual(values=c(sp_color_dark, sp_color_darkest,
                              lz_color_dark, lz_color_darkest,
                              dip_color_dark, dip_color_darkest)) +
  ylab("Differential expression (logFC)") +
  ylim(-11.5, 16)  +
  ylim(-13, 16) +
  geom_signif(
    data = wilcoxin_table_BBBB_df,
    aes(xmin = start, xmax = end, annotations = sig, 
        y_position = max(Hyb.All.trans$logFC.x)*1.3),
    vjust = 0.4, tip_length = 0.01, textsize = text_size*0.5, 
    size = 0.5,
    manual = TRUE) +
  geom_signif(
    data = chi_square_table_BBBB_df,
    aes(xmin = comparison, xmax = comparison, annotations = num_ups, 
        y_position = max(Hyb.All.trans$logFC.x)*1.15), 
    vjust = -0.2, tip_length = 0.00, size = 0, textsize = text_size*0.3,
    manual = TRUE) +  
  geom_signif(
    data = chi_square_table_BBBB_df,
    aes(xmin = comparison, xmax = comparison, annotations = num_downs, 
        y_position = max(Hyb.All.trans$logFC.x)*1.05),
    vjust = -0.2, tip_length = 0.00, size = 0, textsize = text_size*0.3,
    manual = TRUE) +
  scale_x_discrete(labels= rep(c("Mouse", "Hamster"), 3)) +
  theme(legend.position = "none",
        panel.grid.minor = element_blank(),
        panel.grid.major.x = element_blank(),
        axis.title.x = element_blank(),
        axis.text.x = element_text(angle = 45, vjust = 1.5),
        plot.title = element_text(hjust = 0.5),
        text = element_text(size = text_size))

#save figure or print if saving not desired
if(exportFigs == TRUE) {
  ggsave(filename ="exported_figures/hybrid_logFC_BBBB_both_comps_mouse_v_hamster.pdf", 
         logFC_BBBB_trans_comps_plot, width=full_plot_width*1.5, height=3*1.5)
} else {
  logFC_BBBB_trans_comps_plot
}

#print average logFC for hybrid mouse and ham DE genes across spermatogenesis
Hyb.All.trans[Hyb.All.trans$X_category == "X",] %>% group_by(comparison) %>%
  summarize_at('logFC.x',mean)

Hyb.All.trans[Hyb.All.trans$X_category == "Auto",] %>% group_by(comparison) %>%
  summarize_at('logFC.x',mean)

#bind logFC graph with DE over time to produce Figure 4
figure_4_plot <- plot_grid(plot_grid(parent_DE_area_plot, hybrid_DE_area_plot, nrow = 1),
                           logFC_BBBB_trans_comps_plot, nrow = 2, rel_heights = c(1, 1.5))

#save figure or print if saving not desired
if(exportFigs == TRUE) {
  ggsave(filename ="exported_figures/figure_4_DE_and_logFC_over_time.pdf", 
         figure_4_plot, width=full_plot_width*1.25, height=7)
} else {
  figure_4_plot
}
```

```{r hybrid-DE-logFC-graphs-SSSS}
#this section produces figure SX in the manuscript and demonstrates how patterns of up and downregulation across stages in hybrids differs between systems
#set up chi square function
calculate_chisq <- function(comp) {
  comparison <- paste(unlist(strsplit(comp, split = "_"))[1:2], collapse = "_")
  X_cat <- unlist(strsplit(comp, split = "_"))[3]
  logFC_df <- Hyb.All.trans[Hyb.All.trans$category == "transgressive" & Hyb.All.trans$chr != "Y",]
  
  #get observed up and down regulated DE gene numbers
  obs_chisq <- table(sign(logFC_df[logFC_df$sig_in_both == "yes" & 
                                     logFC_df$comparison == comparison & 
                                     logFC_df$X_category == X_cat,]$logFC.y))
  if(is.na(obs_chisq["1"])) {
    obs_chisq["1"] <- 0
  } else if(is.na(obs_chisq["-1"])) {
    obs_chisq["-1"] <- 0
  }
  names(obs_chisq) <- gsub("1", "up", gsub("-1", "down", names(obs_chisq)))
  chisq_results <- chisq.test(obs_chisq, p = c(0.5, 0.5))
  
  return(c(comparison, X_cat, chisq_results$parameter, sum(chisq_results$observed), round(chisq_results$statistic,4), chisq_results$p.value, obs_chisq["up"] > obs_chisq["down"], paste("d = ", obs_chisq["down"], sep = ""), paste("u = ", obs_chisq["up"], sep = "")))
}

#calculate chi_square of over vs under expressed transgressive DE genes for each species * stage * X_category
chi_square_table_SSSS_df <- data.frame(do.call(rbind, lapply(c("mouse_SP_Auto", 
                                                               "mouse_SP_X",
                                                               "mouse_LZ_Auto", 
                                                               "mouse_LZ_X",
                                                               "mouse_DIP_Auto", 
                                                               "mouse_DIP_X",
                                                               
                                                               "hamster_SP_Auto", 
                                                               "hamster_SP_X",
                                                               "hamster_LZ_Auto", 
                                                               "hamster_LZ_X",
                                                               "hamster_DIP_Auto", 
                                                               "hamster_DIP_X"), 
                                                             calculate_chisq)))

#reformat chi_square data frame for plotting
colnames(chi_square_table_SSSS_df) <- c("comparison", "X_category", "chisq_parameter", "observed_sum", "chisq_statistic", "p_value", "more_ups", "num_downs", "num_ups")

#calculate whether mouse and hamster differs in logFC of transgressive DE genes for each stage * X_category using a wilcoxin signed rank test
calculate_wilcoxin <- function(stage, X_cat) {
  #perform the wilcoxin and get the p_value
  comparison_p.value <- wilcox.test(Hyb.All.trans[Hyb.All.trans$category == "transgressive" &
                                                    Hyb.All.trans$comparison == paste("mouse", stage, sep = "_") & 
                                                    Hyb.All.trans$X_category == X_cat,]$logFC.y, 
                                    Hyb.All.trans[Hyb.All.trans$category == "transgressive" &
                                                    Hyb.All.trans$comparison == paste("hamster", stage, sep = "_") & 
                                                    Hyb.All.trans$X_category == X_cat,]$logFC.y)$p.value
  
  #does mouse have the higher logFC for the contrast?
  mouse_higher <- mean(Hyb.All.trans[Hyb.All.trans$category == "transgressive" &
                                       Hyb.All.trans$comparison == paste("mouse", stage, sep = "_") & 
                                       Hyb.All.trans$X_category == X_cat,]$logFC.y) >
    mean(Hyb.All.trans[Hyb.All.trans$category == "transgressive" &
                         Hyb.All.trans$comparison == paste("hamster", stage, sep = "_") & 
                         Hyb.All.trans$X_category == X_cat,]$logFC.y)
  
  return(c(stage, X_cat, comparison_p.value, mouse_higher))
}

#apply the wilcoxin function and save results to a data frame
wilcoxin_table_SSSS_df <- data.frame(t(mapply(calculate_wilcoxin,
                                              c("SP", "LZ", "DIP", "SP", "LZ", "DIP"),
                                              c("Auto", "Auto", "Auto", "X", "X", "X"))))

#format wilcoxin results
colnames(wilcoxin_table_SSSS_df) <- c("stage", "X_category", "p_value", "mouse_higher")

#reformatting for plotting
wilcoxin_table_SSSS_df$start <- wilcoxin_table_SSSS_df$stage
wilcoxin_table_SSSS_df$start <- gsub("^", "mouse_", wilcoxin_table_SSSS_df$start)
wilcoxin_table_SSSS_df$end <- wilcoxin_table_SSSS_df$stage
wilcoxin_table_SSSS_df$end <- gsub("^", "hamster_", wilcoxin_table_SSSS_df$end)

wilcoxin_table_SSSS_df$sig <- sig_conversion(wilcoxin_table_SSSS_df$p_value)

#plot the logFC of each gene by species * stage, with X_category as the facet
logFC_SSSS_trans_comps_plot <- ggplot(data = Hyb.All.trans[Hyb.All.trans$category == "transgressive" & Hyb.All.trans$chr != "Y",], aes(x = comparison, y = logFC.y)) +
  geom_hline(yintercept = 0, linewidth = 0.5) +
  geom_quasirandom(aes(col = comparison), alpha = 0.3, size = 2*point_size, varwidth = TRUE, stroke = NA, width = 0.2) + 
  scale_color_manual(values=c(sp_color, sp_color_dark,
                              lz_color, lz_color_dark,
                              dip_color, dip_color_dark)) + 
  new_scale_color() + 
  geom_violin(aes(col = comparison), scale = "width", fill = NA, width = 0.5) + 
  geom_boxplot(width=boxplot_width, outlier.shape = NA, alpha = 0.8) + 
  theme_minimal() +
  facet_wrap(vars(X_category), ncol = 4, scales = "free_y") + 
  scale_color_manual(values=c(sp_color_dark, sp_color_darkest,
                              lz_color_dark, lz_color_darkest,
                              dip_color_dark, dip_color_darkest)) +
  ylab("Differential expression (logFC)") +
  ylim(-11.5, 16)  +
  ylim(-13, 16) +
  geom_signif(
    data = wilcoxin_table_SSSS_df,
    aes(xmin = start, xmax = end, annotations = sig, 
        y_position = max(Hyb.All.trans$logFC.y)*1.3),
    vjust = 0.4, tip_length = 0.01, textsize = text_size*0.5, 
    size = 0.5,
    manual = TRUE) +
  geom_signif(
    data = chi_square_table_SSSS_df,
    aes(xmin = comparison, xmax = comparison, annotations = num_ups, 
        y_position = max(Hyb.All.trans$logFC.y)*1.15), 
    vjust = -0.2, tip_length = 0.00, size = 0, textsize = text_size*0.3,
    manual = TRUE) +  
  geom_signif(
    data = chi_square_table_SSSS_df,
    aes(xmin = comparison, xmax = comparison, annotations = num_downs, 
        y_position = max(Hyb.All.trans$logFC.y)*1.05),
    vjust = -0.2, tip_length = 0.00, size = 0, textsize = text_size*0.3,
    manual = TRUE) +
  scale_x_discrete(labels= rep(c("Mouse", "Hamster"), 3)) +
  theme(legend.position = "none",
        panel.grid.minor = element_blank(),
        panel.grid.major.x = element_blank(),
        axis.title.x = element_blank(),
        axis.text.x = element_text(angle = 45, vjust = 1.5),
        plot.title = element_text(hjust = 0.5),
        text = element_text(size = text_size))

#save figure or print if saving not desired
if(exportFigs == TRUE) {
  ggsave(filename ="exported_figures/hybrid_logFC_SSSS_both_comps_mouse_v_hamster.pdf", 
         logFC_SSSS_trans_comps_plot, width=full_plot_width*1.5, height=3*1.5)
} else {
  logFC_SSSS_trans_comps_plot
}
```

<h3>Hybrid hypergeometric tests to test if different chromosomes are enriched for DEs</h3>
```{r hybrid-DE-hypergeometric-tests}
enrichment_FDR <- 0.05

#set the list of chromosomes to iterate over
chr_list <<- chrom_key$chr

#create two functions: the first performs a hypergeometric test for the distribution of 
test_each_genomic_interval <- function(chromosome) {
  
  #hypergeometric tests should take the form of:
  #phyper(Overlap, group2, Total-group2, group1,  lower.tail= TRUE) for depletion
  #phyper(Overlap - 1, group2, Total-group2, group1,  lower.tail= FALSE) for enrichment
  
  #where:
  #Total = expressed genes on all chromosomes for appropriate contrast: exp_genes_on_chr
  #Group1 = expressed genes on given chromosome for appropriate contrast: exp_genes_on_chr
  #Group2 = all DE genes across chromosomes for appropriate contrast: all_DE_genes
  #Overlap = DE genes on given chromosome for appropriate contrast: DE_on_chr
  DE_on_chr <- dim(example_de[example_de$chr == chromosome,])[1]
  all_DE_genes <- dim(example_de)[1]
  all_expressed_genes <- length(example_exp)
  non_DE_genes <- all_expressed_genes - all_DE_genes
  exp_genes_on_chr <- sum(example_exp %in% annotations[annotations$chr == chromosome,]$gene)
  
  #run phyper tests
  under.p <- phyper(DE_on_chr, all_DE_genes, non_DE_genes, exp_genes_on_chr, lower.tail = TRUE)
  over.p <- phyper(DE_on_chr - 1, all_DE_genes, non_DE_genes, exp_genes_on_chr, lower.tail = FALSE)
  
  #round p-values to 5 digits
  under <- round(under.p, digits = 5)
  over <- round(over.p, digits = 5)
  
  #calculate number of observed and expected genes
  exp <- round((all_DE_genes/all_expressed_genes)*exp_genes_on_chr,digits = 0)
  obs <- DE_on_chr
  
  #return the depletion and enrichment pvalues plus the exp and obs number of genes
  return(c(under, over, exp, obs))
}

test_for_DE_enrichment <- function(stage, DE_type) {
  
  #get the relevant DE object for the given stage and type of DE gene (all, transgressive, intermediate)
  example_de <<- get(ls(pattern = paste("Hamster", stage, DE_type, sep = "\\."), envir = .GlobalEnv))
  
  #retrieve the list of genes expressed in the given contrast as the background
  example_exp <<- get(ls(pattern = paste("Hyb_Comb_", stage, "_exp", sep = ""), envir = .GlobalEnv))[,1]
  
  #call chr interval
  results_df <- data.frame(do.call(rbind,lapply(chrom_key$chr, test_each_genomic_interval)))
  
  #add formatting to the hypergeometric test data frame
  colnames(results_df) <- c("under_p_adjust", "over_p_adjust", "exp", "obs")
  rownames(results_df) <- chr_list
  results_df$chr <- chr_list
  
  #assign the data frame to a globally accessible object
  assign(paste("Hamster", stage, DE_type, "enrichment_df", sep = "_"), results_df, envir = .GlobalEnv)
  
  #print the results about which chromosomes were depleted or enriched
  return(writeLines(paste("For ", DE_type," genes during " ,stage, ": enriched chromosomes include: ", 
                          paste(rownames(results_df)[results_df$over_p_adjust < 0.05], collapse = ", "), 
                          "\n", "For ", DE_type," genes during " ,stage, ": depleted chromosomes include: ", 
                          paste(rownames(results_df)[results_df$under_p_adjust < 0.05], collapse = ", "), "\n", 
                          sep = "")))
}

#apply hypergeometric tests to DE genes from all stages
mapply(test_for_DE_enrichment,
       c("SP", "LZ", "DIP", "SP", "LZ", "DIP", "SP", "LZ", "DIP"),
       c("DE", "DE", "DE", "trans", "trans", "trans", "inter", "inter", "inter"))

#a function to reformat an enrichment data frame to be compatibile with the processing circos diagram script
reformat_enrichment_df_for_processing <- function(enrich_df) {
  #retrieve data frame
  enrichment_df <- get(enrich_df)
  
  #initialize blank columns
  enrichment_df$enriched <- ""
  enrichment_df$direction <- ""
  enrichment_df$sig <- ""
  enrichment_df$ringText <- ""
  
  #column indicating if there is enrichment or depletion
  if(sum(enrichment_df$under_p_adjust < 0.05 | enrichment_df$over_p_adjust < 0.05) > 0) {
    enrichment_df[enrichment_df$under_p_adjust < 0.05 | enrichment_df$over_p_adjust < 0.05,]$enriched <- "yes"
  }
  
  #if depletion exists, then add in all the information about significance
  if(sum(enrichment_df$under_p_adjust < 0.05) > 0) {
    enrichment_df[enrichment_df$under_p_adjust < 0.05 & enrichment_df$enriched == "yes",]$direction <- "(-)"
    enrichment_df[enrichment_df$direction == "(-)",]$sig <- sig_conversion(enrichment_df[enrichment_df$direction == "(-)",]$under_p_adjust)
    enrichment_df[enrichment_df$direction == "(-)",]$ringText <- gsub("\\*", "- ", enrichment_df[enrichment_df$direction == "(-)",]$sig)
  }
  
  #if enrichment exists, then add in all the information about significance
  if(sum(enrichment_df$over_p_adjust < 0.05) > 0) {
    enrichment_df[enrichment_df$under_p_adjust > 0.05 & enrichment_df$enriched == "yes",]$direction <- "(+)"
    enrichment_df[enrichment_df$direction == "(+)",]$sig <- sig_conversion(enrichment_df[enrichment_df$direction == "(+)",]$over_p_adjust)
    enrichment_df[enrichment_df$direction == "(+)",]$ringText <- gsub("\\*", "+", enrichment_df[enrichment_df$direction == "(+)",]$sig)
  }
  
  #remove the extra space that the depletion code ads
  enrichment_df$ringText <- gsub(" $", "", enrichment_df$ringText)
  
  
  #uncomment if you want to create a new data frame with the reformatting
  #assign(paste(enrich_df, "_for_processing", sep = ""), enrichment_df, envir = .GlobalEnv)
  
  #save to file for use in processing scripts
  write.csv(enrichment_df, 
            file = paste("circos_processing_scripts/hamster_circos_hybrids/data/", 
                         paste(unlist(strsplit(enrich_df, split = "_"))[1:2], collapse = "."),
                         "_sig.csv", sep = ""), row.names = FALSE)
}

#apply reformatting to all enrichment data frames
invisible(lapply(c("Hamster_SP_trans_enrichment_df",
                   "Hamster_LZ_trans_enrichment_df",
                   "Hamster_DIP_trans_enrichment_df"), 
                 reformat_enrichment_df_for_processing))
```

```{r overlap-in-DE-between-hamster-and-mouse}
enrichment_FDR <- 0.05

test_for_overlap_enrichment <- function(stage) {
  
  Mouse_exp <- read.csv(paste("exported_files/Mouse_", stage ,"_all_exp_genes.csv", sep = ""))$gene
  Hamster_exp <- get(paste("Hyb_Comb_", stage, "_exp", sep = ""))[,2]
  
  #a list of genes that are expressed in the given stage in either system
  exp_in_both <- intersect(Mouse_exp, Hamster_exp)
  
  #a list of the trangressively expressed mouse genes that are expressed in either system
  Mouse.trans <- get(paste("Mouse.", stage, ".trans", sep = ""))
  Mouse.trans <- Mouse.trans[Mouse.trans$gene %in% exp_in_both,]
  
  #a list of the transgressively expressed hamster genes
  Hamster.trans <- get(paste("Hamster.", stage, ".trans", sep = ""))
  Hamster.trans <- Hamster.trans[unique(Hamster.trans$mouse_ensemble_ID) %in% exp_in_both,]
  
  #a list of the transgressively expressed genes in both hybrid systems
  Both.trans <- intersect(Mouse.trans$gene,
                          Hamster.trans$mouse_ensemble_ID)
  
  #run phyper tests
  under.p <- phyper(length(Both.trans), 
                    length(unique(Hamster.trans$mouse_ensemble_ID)), 
                    length(exp_in_both) - length(unique(Hamster.trans$mouse_ensemble_ID)), 
                    length(Mouse.trans$gene), 
                    lower.tail = TRUE)
  
  over.p <- phyper(length(Both.trans) - 1, 
                   length(unique(Hamster.trans$mouse_ensemble_ID)), 
                   length(exp_in_both) - length(unique(Hamster.trans$mouse_ensemble_ID)), 
                   length(Mouse.trans$gene), 
                   lower.tail = FALSE)
  
  #round p-values to 5 digits
  under <- round(under.p, digits = 5)
  over <- round(over.p, digits = 5)
  
  #calculate number of observed and expected genes
  exp <- round((length(unique(Hamster.trans$mouse_ensemble_ID))/
                  (length(exp_in_both) - length(unique(Hamster.trans$mouse_ensemble_ID)))*dim(Mouse.trans)[1]),
               digits = 0)
  obs <- length(Both.trans)
  
  #here
  overlap_go <- gost(query = intersect(Mouse.trans$gene,
                                       Hamster.trans$mouse_ensemble_ID), 
                     organism = "mmusculus", ordered_query = FALSE, 
                     multi_query = FALSE, significant = TRUE, exclude_iea = FALSE, 
                     measure_underrepresentation = FALSE, evcodes = FALSE, 
                     user_threshold = FDR_cutoff, correction_method = "g_SCS", 
                     domain_scope = "custom", highlight = FALSE, sources = c("GO:BP", "GO:CC", "GO:MF"), 
                     custom_bg = unique(c(Mouse.trans$gene,
                                          Hamster.trans$mouse_ensemble_ID)), 
                     numeric_ns = "", as_short_link = FALSE)
  
  assign(paste(stage, "_overlap_GO_results", sep = ""), overlap_go$result, envir = .GlobalEnv)
  
  return(c(stage, under, over, exp, obs, 
           length(Mouse.trans$gene), length(unique(Hamster.trans$mouse_ensemble_ID)),
           paste(intersect(Mouse.trans$gene,
                           Hamster.trans$mouse_ensemble_ID), 
                 collapse = ",")))
}

#apply hypergeometric tests to DE genes from all stages
overlap_df <- data.frame(do.call(rbind,lapply(c("SP", "LZ", "DIP"), test_for_overlap_enrichment)))
colnames(overlap_df) <- c("stage", "depletion_pvalue", "enrichment_pvalue",
                          "number_expected", "number_observed",
                          "number_mouse_DE", "number_hamster_DE",
                          "number_overlap_DE")
write.csv(file = "exported_files/transgressive_DE_overlap_enrichment_stats.csv", 
          x = overlap_df[-length(colnames(overlap_df))], row.names = FALSE)



#look up gene descriptions for overlap genes in each stage
#SP
overlap_definitions_SP_df <- data.frame(cbind(queryMany(unlist(strsplit(overlap_df$number_overlap_DE[1], split = ",")),
                                                        scopes="ensembl.gene", species="mouse"), "SP"))
colnames(overlap_definitions_SP_df) <- gsub("X.SP.", "stage", colnames(overlap_definitions_SP_df))

#LZ
overlap_definitions_LZ_df <- data.frame(cbind(queryMany(unlist(strsplit(overlap_df$number_overlap_DE[2], split = ",")),
                                                        scopes="ensembl.gene", species="mouse"), "LZ"))
colnames(overlap_definitions_LZ_df) <- gsub("X.LZ.", "stage", colnames(overlap_definitions_LZ_df))

#DIP
overlap_definitions_DIP_df <- data.frame(cbind(queryMany(unlist(strsplit(overlap_df$number_overlap_DE[3], split = ",")),
                                                         scopes="ensembl.gene", species="mouse"), "DIP"))
colnames(overlap_definitions_DIP_df) <- gsub("X.DIP.", "stage", colnames(overlap_definitions_DIP_df))

#combine dataframes
overlap_definitions_df <- data.frame(do.call(rbind, lapply(ls(pattern = "overlap_definitions"), function(x)
  if(exists(x, .GlobalEnv)) get(x) else NULL)))
overlap_definitions_df <- overlap_definitions_df[,c("query", "name", "symbol", "stage")]
colnames(overlap_definitions_df) <- c("Ensembl_ID", "gene_description", "gene_symbol", "overlap_stage")
write.csv(file = "exported_files/transgressive_DE_overlap_genes.csv", x = overlap_definitions_df, row.names = FALSE)

overlap_definitions_df
```



```{r x_overlap-in-DE-between-hamster-and-mouse}
test_for_x_overlap_enrichment <- function(stage) {
  
  Mouse_exp <- read.csv(paste("exported_files/Mouse_", stage ,"_all_exp_genes.csv", sep = ""))$gene
  Hamster_exp <- get(paste("Hyb_Comb_", stage, "_exp", sep = ""))[,2]
  
  #a list of genes that are expressed in the given stage in either system
  exp_in_both <- intersect(Mouse_exp, Hamster_exp)
  
  #a list of the trangressively expressed mouse genes that are expressed in either system
  Mouse.trans <- get(paste("Mouse.", stage, ".trans", sep = ""))
  Mouse.trans <- Mouse.trans[Mouse.trans$gene %in% exp_in_both,]
  
  #a list of the transgressively expressed hamster genes
  Hamster.trans <- get(paste("Hamster.", stage, ".trans", sep = ""))
  Hamster.trans <- Hamster.trans[unique(Hamster.trans$mouse_ensemble_ID) %in% exp_in_both,]
  
  #a list of the transgressively expressed genes in both hybrid systems
  Both.trans <- intersect(Mouse.trans$gene,
                          Hamster.trans$mouse_ensemble_ID)
  
  Mouse_X_exp <- read.csv(paste("exported_files/Mouse_", stage ,"_X_exp_genes.csv", sep = ""))$gene
  Hamster_X_exp <- get(paste("Hyb_Comb_", stage, "_X_exp", sep = ""))[,2]
  exp_X_in_both <- intersect(Mouse_X_exp, Hamster_X_exp)
  Both.X.trans <- Both.trans[Both.trans %in% exp_X_in_both]
  
  #run phyper tests
  under.p <- phyper(length(Both.X.trans), 
                    length(exp_X_in_both), 
                    length(exp_in_both) - length(exp_X_in_both), 
                    length(Both.trans), 
                    lower.tail = TRUE)
  
  over.p <- phyper(length(Both.X.trans) - 1, 
                   length(exp_X_in_both), 
                   length(exp_in_both) - length(exp_X_in_both), 
                   length(Both.trans), 
                   lower.tail = FALSE)
  
  #round p-values to 5 digits
  under <- round(under.p, digits = 5)
  over <- round(over.p, digits = 5)
  
  #calculate number of observed and expected genes
  exp <- round((length(unique(Hamster.trans$mouse_ensemble_ID))/
                  (length(exp_in_both)))*dim(Mouse.trans)[1],
               digits = 0)
  
  obs <- length(Both.X.trans)
  
  return(c(stage, under, over, exp, obs, 
           length(Mouse.trans$gene), length(unique(Hamster.trans$mouse_ensemble_ID)),
           paste(intersect(Mouse.trans$gene,
                           Hamster.trans$mouse_ensemble_ID), 
                 collapse = ",")))
}

#apply hypergeometric tests to DE genes from all stages
x_overlap_df <- data.frame(do.call(rbind,lapply(c("SP", "LZ", "DIP"), test_for_x_overlap_enrichment)))
colnames(x_overlap_df) <- c("stage", "depletion_pvalue", "enrichment_pvalue",
                            "number_expected", "number_observed",
                            "number_mouse_DE", "number_hamster_DE",
                            "number_x_overlap_DE")
write.csv(file = "exported_files/transgressive_DE_x_overlap_enrichment_stats.csv", 
          x = x_overlap_df[-length(colnames(x_overlap_df))], row.names = FALSE)
```

<h3>Plot GO term enrichment from DE genes</h3>
```{r go-enrichment-bp-visualization}
#produces supplement figure S10 and main figure 5D
if(exportFigs == TRUE) {
  ggsave("exported_figures/hybrid_sp_go_enrichment.pdf", SP_GO_all_plot, width = 1.5*full_plot_width, height = 2.5*full_plot_height)
  ggsave("exported_figures/hybrid_lz_go_enrichment.pdf", LZ_GO_all_plot, width = 1.5*full_plot_width, height = 2.5*full_plot_height)
  ggsave("exported_figures/hybrid_dip_go_enrichment.pdf", DIP_GO_all_plot, width = 1.5*full_plot_width, height = 2.5*full_plot_height)
  
}else if (exportFigs == FALSE) {
  SP_GO_all_plot
  LZ_GO_all_plot
  DIP_GO_all_plot
}

#producing the figure 5 multipanel with the dip go results
hybDIP_corr_multiPanel <-  
  plot_grid(
    plot_grid(NULL, plot_grid(mouse_correlation_plot, hamster_correlation_plot, labels = c('A', 'B'), nrow = 2), both_species_hybrid_escapees_plot, rel_widths = c(0.5, 5, 2), labels = c('', '', 'C'), nrow = 1),
    DIP_GO_highlight_plot, labels = c('', 'D'), rel_heights = c(2,1), nrow = 2)

if(exportFigs == TRUE) {
  ggsave(filename = "exported_figures/hybrid_DIP_correlation_multiPanel_plot.pdf", hybDIP_corr_multiPanel, width = 10, height = 8)
} else if(exportFigs == FALSE) {
  print(hybDIP_corr_multiPanel)
}
```

<h3>PAR Behavior in Hybrid versus Parental Dwarf Hamsters</h3>
```{r par-behavior}
#The starting position of the dwarf hamster PAR as reported by Moore et al. 2022
PAR_coord <- 115350000
#create a data frame of the genes located in the PAR
par_genes <- annotations[annotations$chr == "X" & annotations$start > PAR_coord,]

#a function to help make the output table more readible for inclusion in the supplement
convert_DE_labels <- function(true_false_column) {
  if(sum(true_false_column != "TRUE") != length(true_false_column)){
    true_false_column[true_false_column != "TRUE"] <- ""
    true_false_column[true_false_column == "TRUE"] <- "X"
  } else {
    true_false_column <- ""
  }
  return(true_false_column)
}

#set up an empty data frame to contain par gene expression values/ DE status
par_gene_behavior <- data.frame()

#add DE status for both species for each ortholog for each stage
par_gene_behavior <- cbind(par_genes,       SP_BB_DE =    par_genes$gene %in% Hyb.BB.SP.de$gene)
par_gene_behavior <- cbind(par_gene_behavior, SP_SS_DE =  par_genes$gene  %in% Hyb.SS.SP.de$gene)
par_gene_behavior <- cbind(par_gene_behavior, LZ_BB_DE =  par_genes$gene  %in% Hyb.BB.LZ.de$gene)
par_gene_behavior <- cbind(par_gene_behavior, LZ_SS_DE =  par_genes$gene  %in% Hyb.SS.LZ.de$gene)
par_gene_behavior <- cbind(par_gene_behavior, DIP_BB_DE = par_genes$gene %in% Hyb.BB.DIP.de$gene)
par_gene_behavior <- cbind(par_gene_behavior, DIP_SS_DE = par_genes$gene %in% Hyb.SS.DIP.de$gene)

par_gene_behavior <- cbind(par_gene_behavior, SP_trans_DE = par_genes$gene %in% Hamster.SP.trans$gene)
par_gene_behavior <- cbind(par_gene_behavior, SP_inter_DE = par_genes$gene %in% Hamster.SP.inter$gene)
par_gene_behavior <- cbind(par_gene_behavior, LZ_trans_DE = par_genes$gene %in% Hamster.LZ.trans$gene)
par_gene_behavior <- cbind(par_gene_behavior, LZ_inter_DE = par_genes$gene %in% Hamster.LZ.inter$gene)
par_gene_behavior <- cbind(par_gene_behavior, DIP_trans_DE = par_genes$gene %in% Hamster.DIP.trans$gene)
par_gene_behavior <- cbind(par_gene_behavior, DIP_inter_DE = par_genes$gene %in% Hamster.DIP.inter$gene)

par_gene_behavior$SP_BB_DE <- convert_DE_labels(par_gene_behavior$SP_BB_DE)
par_gene_behavior$SP_SS_DE <- convert_DE_labels(par_gene_behavior$SP_SS_DE)
par_gene_behavior$LZ_BB_DE <- convert_DE_labels(par_gene_behavior$LZ_BB_DE)
par_gene_behavior$LZ_SS_DE <- convert_DE_labels(par_gene_behavior$LZ_SS_DE)
par_gene_behavior$DIP_BB_DE <- convert_DE_labels(par_gene_behavior$DIP_BB_DE)
par_gene_behavior$DIP_SS_DE <- convert_DE_labels(par_gene_behavior$DIP_SS_DE)

par_gene_behavior$SP_trans_DE <-  convert_DE_labels(par_gene_behavior$SP_trans_DE)
par_gene_behavior$SP_inter_DE <-  convert_DE_labels(par_gene_behavior$SP_inter_DE)
par_gene_behavior$LZ_trans_DE <-  convert_DE_labels(par_gene_behavior$LZ_trans_DE)
par_gene_behavior$LZ_inter_DE <-  convert_DE_labels(par_gene_behavior$LZ_inter_DE)
par_gene_behavior$DIP_trans_DE <- convert_DE_labels(par_gene_behavior$DIP_trans_DE)
par_gene_behavior$DIP_inter_DE <- convert_DE_labels(par_gene_behavior$DIP_inter_DE)

#print table to html file and save to exported files
kable(par_gene_behavior, format = "simple")
write.csv(x = par_gene_behavior, file = "exported_files/Hamster_PAR_genes.csv")

#details to be included in text
print(paste("There are ", sum(table(par_gene_behavior[par_gene_behavior$Length != "NA",]$mouse_gene_symbol) == 1), " MSCI genes present as single-copy", sep = ""))
print(paste("There are ", sum(table(par_gene_behavior[par_gene_behavior$Length != "NA",]$mouse_gene_symbol) != 1), " MSCI genes present as multi-copy", sep = ""))
print(paste("There are ", sum(table(par_gene_behavior[par_gene_behavior$Length == "NA",]$mouse_gene_symbol) == 1), " MSCI genes not present", sep = ""))

#summarizing expression data for plotting; generates Figure 6 in main text
par_genes_rpkm.norm <- rpkm.norm[rownames(rpkm.norm) %in% par_genes$gene,]
par_genes_rpkm.norm$Psun_ID <- rownames(par_genes_rpkm.norm)
par_genes_rpkm.norm <- pivot_longer(par_genes_rpkm.norm, cols = 1:(dim(par_genes_rpkm.norm)[2]-1), names_to = "ID", values_to = "RPKM")
par_genes_rpkm.norm$CrossXStage <- gsub("[0-9].*M_", "", par_genes_rpkm.norm$ID)

par_genes_rpkm.norm$CrossXStage <- factor(par_genes_rpkm.norm$CrossXStage, levels =
                                            c("BBBB_SP", "BBSS_SP", "SSSS_SP", 
                                              "BBBB_LZ", "BBSS_LZ", "SSSS_LZ", 
                                              "BBBB_DIP", "BBSS_DIP", "SSSS_DIP", 
                                              "BBBB_RS", "SSSS_RS"))
par_genes_rpkm.norm$cross <- gsub("_.*", "", par_genes_rpkm.norm$CrossXStage)
par_genes_rpkm.norm$cross <- factor(par_genes_rpkm.norm$cross, levels =
                                      c("BBBB", "BBSS", "SSSS"))
par_genes_rpkm.norm$stage <- gsub(".*_", "", par_genes_rpkm.norm$CrossXStage)
par_genes_rpkm.norm$stage <- factor(par_genes_rpkm.norm$stage, levels =
                                      c("SP", "LZ", "DIP", "RS"))

par_genes_rpkm.norm$Mouse_ID <- par_genes_rpkm.norm$Psun_ID

for (hamsterGene in unique(par_genes_rpkm.norm$Psun_ID)) {
  if(!is.na(par_genes[grep(hamsterGene, par_genes$gene),]$mouse_gene_symbol)) {
    par_genes_rpkm.norm$Mouse_ID <- gsub(hamsterGene, 
                                         par_genes[grep(hamsterGene, par_genes$gene),]$mouse_gene_symbol,
                                         par_genes_rpkm.norm$Mouse_ID)
  }
}

par_genes_rpkm.norm$Psun_ID <- factor(par_genes_rpkm.norm$Psun_ID)
par_genes_rpkm.norm$Mouse_ID <- factor(par_genes_rpkm.norm$Mouse_ID)


PAR_average_exp_df <- data.frame()

for(gene in unique(par_genes_rpkm.norm$Mouse_ID)){
  gene_tmp_df <- as.data.frame(subset(par_genes_rpkm.norm,par_genes_rpkm.norm$Mouse_ID == gene))
  PAR_sem_exp <- c()
  for(condition in unique(par_genes_rpkm.norm$CrossXStage)) {
    crossXstage_tmp <- subset(gene_tmp_df, gene_tmp_df$CrossXStage == condition)
    PAR_sem_exp <- c(PAR_sem_exp, calculate_sem(crossXstage_tmp$RPKM))
  }
  PAR_average_exp_df <- rbind(PAR_average_exp_df, cbind(aggregate(gene_tmp_df$RPKM, list(gene_tmp_df$CrossXStage), FUN=mean), PAR_sem_exp, rep(gene, length(unique(gene_tmp_df$CrossXStage)))))
}
PAR_average_exp_df <- separate(PAR_average_exp_df, Group.1, into = c("cross", "stage"))
colnames(PAR_average_exp_df) <- c("cross", "stage", "mean_RPKM", "sem", "gene")
PAR_average_exp_df$stage <- factor(PAR_average_exp_df$stage, levels = c("SP", "LZ", "DIP", "RS"))

#first plots expression without the gene trends collapsed on top of one another
PAR_trends_plot_byGene <- ggplot(data = subset(PAR_average_exp_df, PAR_average_exp_df$stage != "RS"), aes(x = stage, y = mean_RPKM, group = cross, col = cross, shape = cross, fill = cross)) +
  geom_hline(yintercept = 0) +
  geom_line(linewidth = 1) +
  geom_errorbar(aes(ymin=mean_RPKM-sem, ymax=mean_RPKM+sem), width=.1, color = "black") +
  geom_point(color = "black") +
  theme_minimal() +
  theme(panel.grid.minor.y = element_blank()) +
  scale_color_manual(values=c(gray_5, gray_4, gray_2)) + 
  scale_fill_manual(values=c(gray_5, gray_4, gray_2)) + 
  scale_shape_manual(values=c(24,4,25)) + 
  facet_wrap(vars(gene, cross), nrow = length(unique(PAR_average_exp_df$gene)))

cross.labs <- c("P. campbelli", "Hybrid", "P. sungorus")
names(cross.labs) <- c("BBBB", "BBSS", "SSSS")

colorBrownLightest  <- "#D4CCC4";
colorBrownLight     <- "#BAADA1";
colorBrownMedium    <- "#A08F7E";
colorBrownMediumest <- "#81705F";
colorBrownDark      <- "#584D41";
colorBrownDarkest   <- "#3A332C";
colorBrownBlack     <- "#231F1A";

#then plots expression with the gene trends collapsed on top of one another (version in manuscript)
PAR_trends_plot_collapsed <- ggplot(data = subset(PAR_average_exp_df, PAR_average_exp_df$stage != "RS"), aes(x = stage, y = mean_RPKM, group = gene, col = gene)) +
  geom_hline(yintercept = 0) +
  geom_line(linewidth = 0.75, aes(linetype = gene, color = gene), position=position_jitter(w=0.05)) +
  theme_minimal() +
  theme(panel.grid.minor.y = element_blank()) +
  scale_color_manual(values = c(colorBrownLightest, colorBrownLight, colorBrownMedium, colorBrownMediumest, colorBrownDark, colorBrownDarkest, colorBrownBlack)) + 
  scale_linetype_manual(values = c("solid", "dashed", "solid", "dashed", "solid", "dashed", "solid")) +
  labs(x = "Stage of spermatogenesis", y = "Average expression of PAR genes (normalized RPKM)") +
  facet_wrap(vars(cross), nrow = 1, labeller = labeller(cross = cross.labs))

if(exportFigs == TRUE) {
  ggsave(filename = "exported_figures/hamster_PAR_trends_byGene_plot.pdf", PAR_trends_plot_byGene, width = 1.25*full_plot_width, height = 1.5*full_plot_height)
} else {
  print(PAR_trends_plot_byGene)
}

if(exportFigs == TRUE) {
  ggsave(filename = "exported_figures/hamster_PAR_trends_collapsed_plot.pdf", PAR_trends_plot_collapsed, height = 5, width = full_plot_width*1.5)
} else {
  print(PAR_trends_plot_collapsed)
}
```

<h3>Calculate BCV for each Species/Hybrid</h3>
```{r calculate-BCV-for-each-hamster-cross}
#the purpose of this analysis is to see how variation within hybrids and within parents compares
#variation across hybrids for SP/LZ/DIP
dgeBBSSDev <- DGEList(just_counts[,grep("BBSS_.*_[D,S,L]",colnames(just_counts))], genes=annotations, group=grep("BBSS-[D,S,L]",metadata$condition_name, value = TRUE))
dgeBBSSDev <- normalize_dge(dgeBBSSDev)

#variation across hybrids for SP/LZ/DIP
design_BBSSDev <- design[grep("BBSS\\.[D,L,S]", design$condition_name),]
design_BBSSDev$condition_name <- factor(design_BBSSDev$condition_name)

design_BBSSDev_Matrix  <-  model.matrix(~0+design_BBSSDev$condition_name)
colnames(design_BBSSDev_Matrix)  <-  levels(design_BBSSDev$condition_name)
rownames(design_BBSSDev_Matrix)  <-  design_BBSSDev$sample_ID

dgeBBSSDev <-  estimateGLMCommonDisp(dgeBBSSDev, design_BBSSDev_Matrix, verbose = TRUE)
dgeBBSSDev <- estimateGLMTrendedDisp(dgeBBSSDev, design_BBSSDev_Matrix) 
dgeBBSSDev <- estimateGLMTagwiseDisp(dgeBBSSDev, design_BBSSDev_Matrix)  # estimate genewise dispersion


#change exportFigs to TRUE to export as PDF if desired
if(exportFigs == TRUE) {
  pdf(file="exported_figures/Dispersion_plots_justBBSS_dataset.pdf", width=10, height=8, pointsize=12)
  plotBCV(dgeBBSSDev, main = paste("Dispersion estimates for BBSS dataset; BCV = ",round(sqrt(dgeBBSSDev$common.dispersion), 3)), ylim = c(0, 12), col.common = "black", col.trend = gray_6, xlab = "Average log(CPM)", xlim = c(-4,13), col = alpha(gray_4, 0.15), cex = 0.5)
  dev.off()
}
plotBCV(dgeBBSSDev, main = paste("Dispersion estimates for BBSS dataset; BCV = ",round(sqrt(dgeBBSSDev$common.dispersion), 3)), ylim = c(0, 12), col.common = "black", col.trend = gray_6, xlab = "Average log(CPM)", xlim = c(-4,13), col = alpha(gray_4, 0.15), cex = 0.5)


#variation across P. cam for SP/LZ/DIP
dgeBBBBDev <- DGEList(just_counts[,grep("BBBB_.*_[D,S,L]",colnames(just_counts))], genes=annotations, group=grep("BBBB-[D,S,L]",metadata$condition_name, value = TRUE))
dgeBBBBDev <- normalize_dge(dgeBBBBDev)

#variation across P. cam for SP/LZ/DIP
design_BBBBDev <- design[grep("BBBB\\.[D,L,S]", design$condition_name),]
design_BBBBDev$condition_name <- factor(design_BBBBDev$condition_name)

design_BBBBDev_Matrix  <-  model.matrix(~0+design_BBBBDev$condition_name)
colnames(design_BBBBDev_Matrix)  <-  levels(design_BBBBDev$condition_name)
rownames(design_BBBBDev_Matrix)  <-  design_BBBBDev$sample_ID

dgeBBBBDev <-  estimateGLMCommonDisp(dgeBBBBDev, design_BBBBDev_Matrix, verbose = TRUE)
dgeBBBBDev <- estimateGLMTrendedDisp(dgeBBBBDev, design_BBBBDev_Matrix) 
dgeBBBBDev <- estimateGLMTagwiseDisp(dgeBBBBDev, design_BBBBDev_Matrix)  # estimate genewise dispersion


#change exportFigs to TRUE to export as PDF if desired
if(exportFigs == TRUE) {
  pdf(file="exported_figures/Dispersion_plots_justBBBB_dataset.pdf", width=10, height=8, pointsize=12)
  plotBCV(dgeBBBBDev, main = paste("Dispersion estimates for BBBB dataset; BCV = ",round(sqrt(dgeBBBBDev$common.dispersion), 3)), ylim = c(0, 12), col.common = "black", col.trend = gray_6, xlab = "Average log(CPM)", xlim = c(-4,13), col = alpha(gray_4, 0.15), cex = 0.5)
  dev.off()
}
plotBCV(dgeBBBBDev, main = paste("Dispersion estimates for BBBB dataset; BCV = ",round(sqrt(dgeBBBBDev$common.dispersion), 3)), ylim = c(0, 12), col.common = "black", col.trend = gray_6, xlab = "Average log(CPM)", xlim = c(-4,13), col = alpha(gray_4, 0.15), cex = 0.5)


#variation across P. sun for SP/LZ/DIP
dgeSSSSDev <- DGEList(just_counts[,grep("SSSS_.*_[D,S,L]",colnames(just_counts))], genes=annotations, group=grep("SSSS-[D,S,L]",metadata$condition_name, value = TRUE))
dgeSSSSDev <- normalize_dge(dgeSSSSDev)

#variation across P. sun for SP/LZ/DIP
design_SSSSDev <- design[grep("SSSS\\.[D,L,S]", design$condition_name),]
design_SSSSDev$condition_name <- factor(design_SSSSDev$condition_name)

design_SSSSDev_Matrix  <-  model.matrix(~0+design_SSSSDev$condition_name)
colnames(design_SSSSDev_Matrix)  <-  levels(design_SSSSDev$condition_name)
rownames(design_SSSSDev_Matrix)  <-  design_SSSSDev$sample_ID

dgeSSSSDev <-  estimateGLMCommonDisp(dgeSSSSDev, design_SSSSDev_Matrix, verbose = TRUE)
dgeSSSSDev <- estimateGLMTrendedDisp(dgeSSSSDev, design_SSSSDev_Matrix) 
dgeSSSSDev <- estimateGLMTagwiseDisp(dgeSSSSDev, design_SSSSDev_Matrix)  # estimate genewise dispersion


#change exportFigs to TRUE to export as PDF if desired
if(exportFigs == TRUE) {
  pdf(file="exported_figures/Dispersion_plots_justSSSS_dataset.pdf", width=10, height=8, pointsize=12)
  plotBCV(dgeSSSSDev, main = paste("Dispersion estimates for SSSS dataset; BCV = ",round(sqrt(dgeSSSSDev$common.dispersion), 3)), ylim = c(0, 12), col.common = "black", col.trend = gray_6, xlab = "Average log(CPM)", xlim = c(-4,13), col = alpha(gray_4, 0.15), cex = 0.5)
  dev.off()
}
plotBCV(dgeSSSSDev, main = paste("Dispersion estimates for SSSS dataset; BCV = ",round(sqrt(dgeSSSSDev$common.dispersion), 3)), ylim = c(0, 12), col.common = "black", col.trend = gray_6, xlab = "Average log(CPM)", xlim = c(-4,13), col = alpha(gray_4, 0.15), cex = 0.5)
```
