---
title: "Hamster Hybrid Testes RNASeq"
author: "Kelsie E. Hunnicutt"
output:
  html_document: default
  pdf_document: default
---

<h3>Loading libraries and importing data</h3>
```{r setup}
#load packages; make sure these are installed on your machine
suppressMessages(library("edgeR")) #for differential expression analysis
suppressMessages(library("limma")) #for differential expression analysis
suppressMessages(library("ComplexHeatmap")) #for heatmap generation
suppressMessages(library("reshape2")) #for creating the conditions dataframe 
suppressMessages(library("data.table")) #for creating and manipulating data frames 
suppressMessages(library("ggplot2")) #for data visualizing
suppressMessages(library("vegan")) # for decostand count normalization method
suppressMessages(library("matrixStats")) #for calculating median/max values across rows of data frames
suppressMessages(library("ggbeeswarm")) #for offsetting overlapping points in boxplots and violins
suppressMessages(library("tidyr")) #for transforming data frames from wide to long format or long to wide format
suppressMessages(library("FSA")) #for dunn's test
suppressMessages(library("magick")) #for better Heatmap rasterization
suppressMessages(library("ggsignif")) #add significance annotation data to ggplot graphs
suppressMessages(library("dplyr")) #for recoding when doing orthology assignments
suppressMessages(library("knitr")) #to display tables with kable
suppressMessages(library("ggalluvial")) #for alluvial diagrams
suppressMessages(library("dendextend")) #for dendogram plotting
suppressMessages(library("cowplot")) # for multipanel plot organization

#set working directory and get today's date; set to the directory corresponding to where you downloaded this repository
knitr::opts_knit$set(root.dir = "~/Library/CloudStorage/Dropbox/1_Denver/Research/Hamsters/Sterile_hamster_hybrids/")
getwd()
options(scipen = 999)

todays_date <- gsub("^[A-z][A-z][A-z] *([A-z][A-z][A-z]) *([0-9]+).*20([0-9][0-9])", "\\2\\1\\3", date())
todays_date

sessionInfo()
```

```{r import-data}
#select a counts file then read in raw counts; no multimapping was allowed for the main text figures but the multi-mapped files are provided for both dwarf hamster and mouse if desired; general patterns are the same
count_file.path <- "input_data/Hamsters_psun_14Mar23_singleMapping_featureCounts.csv"
count_data <- read.table(count_file.path, header = TRUE, row.names=NULL, sep = "\t")

#select a metadata file then read in info
metadata_file.path <- "input_data/Hamsters_psun_14Mar23_singleMapping_featureCounts.csv.summary"
metadata <- read.table(metadata_file.path, header = TRUE, row.names=1, sep = "\t")

#reads in annotation data from Moore et al. 2022 and reformats
other_tissue_taus <- read.csv("input_data/psun_annotations.csv", header = TRUE)
annotations <- other_tissue_taus[c("Length", "Chr", "Start", "End", "Geneid", "Gene_name")]
rownames(annotations) <- annotations$Geneid
colnames(annotations) <- c("Length", "chr", "start", "stop", "gene", "mouse_gene")
annotations[annotations$mouse_gene == "#N/A",]$mouse_gene <- NA
annotations$chr <- as.factor(annotations$chr)

#renames the main scaffolds to chromosome name for scaffolds that were successfully assigned to linkage groups; for chromosomes with more than one long scaffold, give scaffolds unique chromosome names
chrom_key <- read.csv("input_data/psun_chromosome_scaffold_key.csv")
chrom_key$scaffold.name <- gsub(";HRSC", "_HRSC", chrom_key$scaffold.name)
chrom_sizes <- read.csv("input_data/psun_chrom_sizes.csv", header = TRUE)
for (i in 1:length(chrom_key$scaffold.name)) {
  search_keyword <- chrom_key[i,3]
  replace_keyword <- chrom_key[i,5]
  annotations$chr <- gsub(search_keyword, replace_keyword, annotations$chr)
  chrom_sizes$scaffold_name <- gsub(search_keyword, replace_keyword, chrom_sizes$scaffold_name)
}
rm(i, replace_keyword, search_keyword)
chrom_sizes <- chrom_sizes[grep("Scmy", chrom_sizes$scaffold_name, invert = TRUE),]
setorder(chrom_sizes, scaffold_name)
#list both X-linked scaffolds as separate chromosomes
chrom_sizes$start <- 1
colnames(chrom_sizes) <- c("chromosome", "end", "start")
chrom_sizes <- chrom_sizes[,c(1,3,2)]

#reorder data frame by chromosome name
chrom_sizes$chromosome <- factor(chrom_sizes$chromosome, levels = c("1.1", "1.2", "1.3", "2.1", "2.2", "2.3", "2.4", "3.1", "3.2", "3.3", "4.1", "4.2", "5.1", "5.2", "5.3", "5.4", "5.5", "5.6", "6.1", "6.2", "6.3", "6.4", "7.1", "8.1", "8.2", "9.1", "9.2", "9.3", "10.1", "10.2", "11.1", "11.2", "12.1", "13.1", "13.2", "X", "X.2"))
chrom_sizes <- chrom_sizes[order(chrom_sizes$chromosome),]

#make a list of just the chromosomes/scaffolds over 1000000 for hypergeometric tests
chrom_sizes_large <- subset(chrom_sizes, chrom_sizes$end > 1000000)
head(chrom_sizes_large)

#write information about chromosome sizes (both all scaffolds and just the long scaffolds) to files for circos diagrams of hybrid and parent DE genes (Supplemental Figures S7 and S8)
write.csv(chrom_sizes, file = "circos_processing_scripts/hamster_circos_hybrids/data/psun_chromosomes.csv", row.names = FALSE)
write.csv(chrom_sizes_large, file = "circos_processing_scripts/hamster_circos_hybrids/data/psun_chromosomes_longChromosomes.csv", row.names = FALSE)
```

<h3>Reformatting condition names</h3>
```{r renaming-treatments-across-dfs}
#edit metadata to include condition type (condition_name) for each sample
#transpose, rename columns, delete redundant row
metadata <- t(metadata)
rownames(metadata) <- gsub(".*ers.sorted\\.", "", rownames(metadata))
rownames(metadata) <- gsub('\\.', '-', rownames(metadata))
rownames(metadata) <- gsub("\\.([D,R,L,W,S])", "-\\1",rownames(metadata))
rownames(metadata) <- gsub('M-', '-', rownames(metadata))
rownames(metadata) <- gsub("^([A-Z]+)_([0-9A-Z]*-[0-9A-Z]*)_([A-Z]+).*", "\\1_\\2_\\3", rownames(metadata))
head(rownames(metadata))

#split status into meaningful info (which species/hybrid, individual ID, and stage of spermatogenesis) then create condition names for DE analysis later and bind data frame to metadata
conditions_df <- colsplit(row.names(metadata), pattern = "_", names = c("cross","male_number","stage"))
conditions_df$sample_name <- paste(conditions_df$cross, conditions_df$male_number, conditions_df$stage, sep = ".")
conditions_df$condition_name <- paste(conditions_df$cross, conditions_df$stage, sep = "-") 
metadata <- cbind(metadata, conditions_df)

#in preparation for DGE creation, make a dataframe of just gene name and counts
just_counts <- count_data[7:dim(count_data)[2]]
row.names(just_counts) <- count_data[,1]
#match order of count dataframe to annotations
just_counts <- just_counts[rownames(annotations), ]
head(colnames(just_counts))

#match just_counts IDs with metadata IDs (i.e. remove extraneous info from featureCount filenames)
colnames(just_counts) <- gsub(".*ers.sorted\\.", "", colnames(just_counts))
colnames(just_counts) <- gsub('\\.', '-', colnames(just_counts))
colnames(just_counts) <- gsub("\\.([D,R,L,W,S])", "-\\1",colnames(just_counts))
colnames(just_counts) <- gsub('M-', '-', colnames(just_counts))
colnames(just_counts) <- gsub("^([A-Z]+)_([0-9A-Z]*-[0-9A-Z]*)_([A-Z]+).*", "\\1_\\2_\\3", colnames(just_counts))

head(colnames(just_counts))
rm(count_data)
```

<h3>Color Set-Up</h3>
```{r color-setup}
#coloring indicates stage and colors used here are used throughout
#Red = spermatogonia, Yellow = leptotene-zygotene, Green = diplotene, and Blue = round spermatids
sp_color <- "#E07A5F"
sp_color_dark <- "#D85531"
sp_color_darkest <- "#AB3F21"
lz_color <- "#EEBC6D"
lz_color_dark <- "#E7A336"
lz_color_darkest <- "#DB911A"
dip_color <- "#81B29A"
dip_color_dark <- "#5D987B"
dip_color_darkest <- "#46725D"
rs_color <- "#6C719D"
rs_color_dark <- "#3D405B"
rs_color_darkest <- "#292B3D"

gray_1 <- "#EBEBEB"
gray_2 <- "#D6D6D6"
gray_3 <- "#C2C2C2"
gray_4 <- "#999999"
gray_5 <- "#646464"

#combining colors into vectors based on the color saturation
color_list <- c(SP = sp_color, 
                LZ = lz_color, 
                DIP = dip_color, 
                RS = rs_color)
color_list_dark <- c(SP = sp_color_dark, 
                     LZ = lz_color_dark, 
                     DIP = dip_color_dark, 
                     RS = rs_color_dark)
color_list_darkest <- c(SP = sp_color_darkest, 
                        LZ = lz_color_darkest, 
                        DIP = dip_color_darkest, 
                        RS = rs_color_darkest)

#a function to convert a vector from raw p-values to significance symbols
sig_conversion <- function(misc_dataframe) {
  triple <- misc_dataframe <= 0.001
  double <- misc_dataframe >= 0.001 & misc_dataframe<0.01 
  single <- misc_dataframe >= 0.01 & misc_dataframe<0.05 
  nonsig <- misc_dataframe >= 0.05 
  misc_dataframe[single] <- "*"
  misc_dataframe[double] <- "**"
  misc_dataframe[triple] <- "***"
  misc_dataframe[nonsig] <- "n.s."
  return(misc_dataframe)
}

#a function for calculating standard error of the mean from an input vector of data
calculate_sem <- function(vector_of_interest) {
  sd(vector_of_interest)/sqrt(length((vector_of_interest)))
}

#set up chi square function
calculate_chisq <- function(logFC_df, stage, cat) {
  cat_chisq <- cat
  
  #get observed up and down regulated DE gene numbers
  obs_chisq <- table(sign(logFC_df[logFC_df$treatment == stage,]$logFC))
  if(is.na(obs_chisq["1"])) {
    obs_chisq["1"] <- 0
  } else if(is.na(obs_chisq["-1"])) {
    obs_chisq["-1"] <- 0
  }
  names(obs_chisq) <- gsub("1", "up", gsub("-1", "down", names(obs_chisq)))
  chisq_results <- chisq.test(obs_chisq, p = c(0.5, 0.5))
  
  return(c(paste(stage, cat, sep = " "), chisq_results$parameter, sum(chisq_results$observed), round(chisq_results$statistic,4), chisq_results$p.value))
}


#graphing parameters
point_size <- 0.5
text_size <- 6
boxplot_width <- 0.1
vjust_value <- 0.8
boxplot_lwd <- 0.2
axis_size <- 0.2
sigText_size <- 2
sigLine_size <- 0.25

#set figure export dimensions
half_plot_width <- 3.25
full_plot_width <- 6.75
half_plot_height <- 3
full_plot_height <- 6
```

<h3>Gene expression settings</h3>
```{r settings-and-base-expression-stats}
#set thresholds for determining if a gene is "expressed"; needs to have an rpkm of at least 1 in at least 3 replicates
rpkm.cutoff <- 1
min.reps <- 3

#methods used for normalizing data; RLE used in manuscript
norm.method <- "RLE"
#norm.method <- "TMM"

#should figures should be written to files or plotted in R?
exportFigs <- FALSE

#define orders of datasets, stages, and genotypes for factor purposes; WT not used in this manuscript but is present in countfiles
#BBBB = parental P. campbelli, BBSS = F1 campbelli x sungorus males, SSSS = parental P. sungorus
stages <- c("SP", "LZ", "DIP", "RS", "WT")
genotypes <- c("BBBB", "SSSS","BBSS")

#printing for record keeping purposes then cleanup unused values
paste("countfile used:", count_file.path, sep = " ")
paste("metadata file used:", metadata_file.path, sep = " ")
rm(count_file.path, metadata_file.path)

#printing for record keeping purposes
print("Parameters Used:")
paste("All expressed genes have a minimum rpkm of:", rpkm.cutoff, sep = " ")
paste("An expressed gene must be expressed in at least", min.reps, "samples", sep = " ")
paste("Counts are normalized with:", norm.method, sep = " ")

#make DGEs for sorted cell data (i.e. remove whole testes samples which are not used in hybrid analyses)
dgeDev <- DGEList(just_counts[,grep("WT",colnames(just_counts), invert = TRUE)], genes=annotations, group=grep("WT",metadata$condition_name, value = TRUE, invert = TRUE))
paste("Expressed genes in sorted cell dataset (no cutoffs):", nrow(dgeDev), sep = " ")

#create function normalize_dge which reads in a dge, applies the rpkm and minimum replicate cutoffs for determining if a gene is expressed, then it re-normalizes the data using the normalization method specified above
normalize_dge <- function(dge_name) {
  keep <- rowSums(rpkm(dge_name) >= rpkm.cutoff) >= min.reps 
  dge_name <- dge_name[keep, keep.lib.sizes=FALSE]
  dge_name <- calcNormFactors(dge_name, method = norm.method)
}

#filter DGEs with genes meeting minimum requirements using normalize_dge function defined above; Dev throughout refers to dataset consisting of sorted developmental stages (i.e., not including whole testes samples)
dgeDev <- normalize_dge(dgeDev)

#print expressed genes for each DGE- post filter
paste("Expressed genes in sorted cell dataset (with filter):", nrow(dgeDev), sep = " ")
```

<h3>Hamster fertility data</h3>
```{r fertility-analysis}
#read in fertility data, select columns of interest, then rename column names
fertility <- read.csv("input_data/Hamster_fertility_phenotype_data_7Mar22.csv", header = TRUE)
fertility <- fertility[c("Age", "Count.Rows", "Count.Total", "Dilution", "EarLength", "FootLength", "HamsterLength", "Motility.Mot", "Motility.Non.Mot", "RelativeSVWeight..mg.g.", "RelativeTestesWeight..mg.g.", "SampleID", "Strain", "SVWeight.g.", "TailLength", "TestesOnly.G.", "Weight..g.")]
colnames(fertility) <- c("Age", "SpermRows", "SpermCount", "SpermDilution", "Ear", "Foot", "BodyLength", "MotileSperm", "NonMotileSperm", "RelativeSVWeight", "RelativeTestesWeight", "ID", "Strain", "SVWeight", "Tail", "TestesWeight", "BodyWeight")

#reformat IDs and strain to match metadata
fertility$ID <- gsub("^([B|S][B|S]).([B|S][B|S])", "\\1\\2_", fertility$ID)
fertility$ID <- gsub("\\.", "-", fertility$ID)
fertility$Strain <- gsub("\\.", "", fertility$Strain)
fertility$Strain <- factor(fertility$Strain, levels = c("BBBB", "BBSS", "SSSS"))

#hamster age; set NA and recast as numeric
fertility$Age[fertility$Age == "#VALUE!"] <- NA
fertility$Age <- as.numeric(fertility$Age)

#calculate sperm motility (number motile sperm/ number motile + nonmotile sperm)
fertility$MotileSperm <- gsub("TNTC", NA, fertility$MotileSperm)
fertility$MotileSperm <- as.numeric(fertility$MotileSperm)
fertility$NonMotileSperm <- gsub("TNTC", NA, fertility$NonMotileSperm)
fertility$NonMotileSperm <- as.numeric(fertility$NonMotileSperm)
fertility$Motility <- fertility$MotileSperm/(fertility$MotileSperm + fertility$NonMotileSperm)
fertility[is.nan(fertility$Motility),]$Motility <- NA

#Calculate normalized sperm count (number sperm counted/number rows counted * sperm dilution )
fertility$SpermCount <- gsub("TNTC", NA, fertility$SpermCount)
fertility$SpermRows <- gsub("TNTC", NA, fertility$SpermRows)
fertility$NormSpermCount <- (5*as.numeric(fertility$SpermCount)) / (as.numeric(fertility$SpermRows)*as.numeric(fertility$SpermDilution))

#fixing record errors in original phenotype datasheet
#fixing misrecorded body measurements
fertility[fertility$ID == "BBBB_197-3M",]$Tail <- 8
fertility[fertility$ID == "SSSS_188-5M",]$Tail <- 7
fertility[fertility$ID == "BBSS_90-2M", ]$BodyLength <- 85

#set all ruptured SVs to NA
fertility[fertility$ID == "BBBB_195-4M",]$RelativeSVWeight <- NA
fertility[fertility$ID == "BBBB_295A-3M",]$RelativeSVWeight <- NA
fertility[fertility$ID == "SSSS_155-6M",]$RelativeSVWeight <- NA
fertility[fertility$ID == "SSSS_182-5M",]$RelativeSVWeight <- NA

#incorrect body weight for BBSS_100-2M because measured without organs; corrected with the addition of testes weight and SV weight (missing liver weight)
fertility[fertility$ID == "BBSS_100-2M",]$BodyWeight <- fertility[fertility$ID == "BBSS_100-2M",]$BodyWeight + fertility[fertility$ID == "BBSS_100-2M",]$TestesWeight + fertility[fertility$ID == "BBSS_100-2M",]$SVWeight
fertility[fertility$ID == "BBSS_100-2M",]$RelativeTestesWeight <- 100*fertility[fertility$ID == "BBSS_100-2M",]$TestesWeight/fertility[fertility$ID == "BBSS_100-2M",]$BodyWeight
fertility[fertility$ID == "BBSS_100-2M",]$RelativeSVWeight <- 100*fertility[fertility$ID == "BBSS_100-2M",]$SVWeight/fertility[fertility$ID == "BBSS_100-2M",]$BodyWeight

#hamster not weighed
fertility[fertility$ID == "SSSS_188-4M",]$BodyWeight <- NA
fertility[fertility$ID == "SSSS_188-4M",]$RelativeTestesWeight <- NA
fertility[fertility$ID == "SSSS_188-4M",]$RelativeSVWeight <- NA

#ear/foot swapped on datasheet
fertility[fertility$ID == "BBSS_7-1M",]$Foot <- 9.86
fertility[fertility$ID == "BBSS_7-1M",]$Ear <- 12.98

#incorrect motility
fertility[fertility$ID == "SSSS_152-4M",]$Motility <- NA

#toggle to exclude WT from analysis (collected at a different time than cell sorts); manuscript includes WT individuals
wt_individuals <- gsub("_WT", "", grep("WT", colnames(just_counts), value = TRUE))
#fertility <- fertility[!(fertility$ID %in% wt_individuals),]

#tests if testes weight and SV weight are correlated with body weight to determine if relative testes weight and SV weight should be used instead; they are both correlated so relative weight used in manuscript
#first investigate testes weight correlation
TW_corr <- cor.test(fertility$BodyWeight, fertility$TestesWeight, method = c("pearson"))
TW_rho <- TW_corr$estimate
TW_pValue <- TW_corr$p.value

#correlation plot between body weight and testes weight
bw_v_tw_plot <- ggplot(fertility[!is.na(fertility$BodyWeight),], 
                       aes(x=BodyWeight, y = TestesWeight, shape = Strain, col = Strain, fill = Strain)) +
  geom_point(size=4, stroke = 1) +  
  scale_color_manual(values=c(gray_5, gray_4, gray_2)) + 
  scale_fill_manual(values=c(gray_5, gray_4, gray_2)) + 
  scale_shape_manual(values=c(24, 4, 25)) +
  theme_minimal() +
  xlab("Body weight (g)") +
  ylab("Testes weight (mg)") + 
  annotate(geom="text", x=20, y = 1.7, label=paste("rho = ", round(TW_rho, 3), sep = "")) + 
  annotate(geom="text", x=20, y = 1.5, label=paste("p = ", round(TW_pValue, 3), sep = ""))

#save or print correlation graph
if(exportFigs == TRUE) {
  ggsave(file = "exported_figures/hamster_bodyWeight_by_testesWeight.pdf", bw_v_tw_plot)
} else {
  if(TW_pValue < 0.05) {
    print(paste("Testes weight is correlated with body weight (Pearson’s r(", TW_corr$parameter, ") = ", round(TW_rho, 3), ", p = ", round(TW_pValue, 3), sep = ""))
  } else {
    print(paste("Testes weight is NOT correlated with body weight (Pearson’s r(", TW_corr$parameter, ") = ", round(TW_rho, 3), ", p = ", round(TW_pValue, 3), sep = ""))
  }
  bw_v_tw_plot
}

#second, investigate SV weight correlation
SVW_corr <- cor.test(fertility$BodyWeight, fertility$SVWeight, method = c("pearson"))
SVW_rho <- SVW_corr$estimate
SVW_pValue <- SVW_corr$p.value

#SV weight correlation plot
bw_v_svw_plot <- ggplot(fertility[!is.na(fertility$BodyWeight) & !is.na(fertility$SVWeight),], aes(x=BodyWeight, y = SVWeight, shape = Strain, col = Strain, fill = Strain)) +
  geom_point(size=4, stroke = 1) +  
  scale_color_manual(values=c(gray_5, gray_4, gray_2)) + 
  scale_fill_manual(values=c(gray_5, gray_4, gray_2)) + 
  scale_shape_manual(values=c(24, 4, 25)) +
  theme_minimal() +
  xlab("Body weight (g)") +
  ylab("Seminal vesicle weight (mg)") + 
  annotate(geom="text", x=20, y = 0.8, label=paste("rho = ", round(SVW_rho, 3), sep = "")) + 
  annotate(geom="text", x=20, y = 0.7, label=paste("p = ", round(SVW_pValue, 3), sep = ""))

#either save or print SV correlation graph
if(exportFigs == TRUE) {
  ggsave(file = "exported_figures/hamster_bodyWeight_by_SVWeight.pdf", bw_v_svw_plot)
} else {
  if(SVW_pValue < 0.05) {
    print(paste("SV weight is correlated with body weight (Pearson’s r(", SVW_corr$parameter, ") = ", round(SVW_rho, 3), ", p = ", round(SVW_pValue, 3), sep = ""))
  } else {
    print(paste("SV weight is NOT correlated with body weight (Pearson’s r(", SVW_corr$parameter, ") = ", round(SVW_rho, 3), ", p = ", round(SVW_pValue, 3), sep = ""))
  }
  bw_v_svw_plot
}


#plot focal fertility phenotypes by strain type and test for statistical differences between parents and hybrids
#all analyses consist of the following steps:
#kruskal wallis test followed by dunn's posthoc test
#adjust p-values for multi-species comparison
#convert p-values to symbols for graphing
#plot fertility metric for each species/hybrid

#BodyWeight
bodyWeight_kw <- kruskal.test(BodyWeight ~ Strain, data = fertility[!is.na(fertility$BodyWeight),])
bodyWeight_dt <- dunnTest(BodyWeight ~ Strain, data = fertility[!is.na(fertility$BodyWeight),], method="bonferroni")
bodyWeight_padj <- bodyWeight_dt$res$P.adj
names(bodyWeight_padj) <- gsub(" ","", bodyWeight_dt$res$Comparison)
bodyWeight_letters <- data.frame(multcompView::multcompLetters(bodyWeight_padj)$Letters)
bodyWeight_letters$Strain <- rownames(bodyWeight_letters)
colnames(bodyWeight_letters) <- c("Letters", "Strain")
bodyWeight_padj <- sig_conversion(bodyWeight_padj)

bw_plot <- ggplot(fertility[!is.na(fertility$BodyWeight),], aes(x = Strain, y = BodyWeight)) + 
  geom_quasirandom(aes(fill = Strain, color = Strain, shape = Strain), alpha = 1, size = point_size) +
  geom_boxplot(width=boxplot_width, outlier.shape = NA, lwd = boxplot_lwd, alpha = 0.2) +
  scale_fill_manual(values=c(gray_5, gray_4, gray_2)) + 
  scale_color_manual(values=c(gray_5, gray_4, gray_2)) + 
  scale_shape_manual(values=c(24,4,25)) + 
  theme_bw() + 
  theme(legend.position = "none", 
        text = element_text(size = text_size),
        axis.title.x=element_blank(), 
        axis.title.y = element_text(size = text_size), 
        axis.text.x = element_text(angle = 45, vjust = vjust_value, size = text_size), 
        plot.margin = unit(c(0.4, 0.07, 0, 0.07), "cm"), 
        panel.grid.major.x = element_blank(), 
        panel.grid.minor = element_blank(), 
        axis.line.y = element_line(colour = "black", linewidth = axis_size), 
        panel.border = element_blank(), 
        axis.ticks.length.x = unit(0, "cm")) +
  scale_x_discrete(labels = c("P. campbelli", "Hybrid", "P. sungorus")) + ylab("Relative testis weight (mg/g)") + 
  geom_signif(
    y_position = c(summary(fertility[fertility$Strain == "BBBB",]$BodyWeight)[6]*1.18,
                   summary(fertility[fertility$Strain == "BBBB",]$BodyWeight)[6]*1.10,
                   summary(fertility[fertility$Strain == "BBBB",]$BodyWeight)[6]*1.02), 
    xmin = c(1,1,2), 
    xmax = c(3,2,3), 
    annotation = c(bodyWeight_padj[2],bodyWeight_padj[1],bodyWeight_padj[3]), 
    tip_length = 0, textsize = sigText_size, size = sigLine_size) +
  ylim(0, summary(fertility[fertility$Strain == "BBBB",]$BodyWeight)[6]*1.2)

#RelativeTestesWeight
relTW_kw <- kruskal.test(RelativeTestesWeight ~ Strain, data = fertility[!is.na(fertility$RelativeTestesWeight),])
relTW_dt <- dunnTest(RelativeTestesWeight ~ Strain, data = fertility[!is.na(fertility$RelativeTestesWeight),], method="bonferroni")
relTW_padj <- relTW_dt$res$P.adj
names(relTW_padj) <- gsub(" ","", relTW_dt$res$Comparison)
relTW_letters <- data.frame(multcompView::multcompLetters(relTW_padj)$Letters)
relTW_letters$Strain <- rownames(relTW_letters)
colnames(relTW_letters) <- c("Letters", "Strain")
relTW_padj <- sig_conversion(relTW_padj)

relTW_plot <- ggplot(fertility[!is.na(fertility$RelativeTestesWeight),], aes(x = Strain, y = RelativeTestesWeight)) + 
  geom_quasirandom(aes(fill = Strain, color = Strain, shape = Strain), alpha = 1, size = point_size) +
  geom_boxplot(width=boxplot_width, outlier.shape = NA, lwd = boxplot_lwd, alpha = 0.2) +
  scale_fill_manual(values=c(gray_5, gray_4, gray_2)) + 
  scale_color_manual(values=c(gray_5, gray_4, gray_2)) + 
  scale_shape_manual(values=c(24,4,25)) + 
  theme_bw() + 
  theme(legend.position = "none", 
        text = element_text(size = text_size),
        axis.title.x=element_blank(), 
        axis.title.y = element_text(size = text_size), 
        axis.text.x = element_text(angle = 45, vjust = vjust_value, size = text_size), 
        plot.margin = unit(c(0.4, 0.07, 0, 0.07), "cm"), 
        panel.grid.major.x = element_blank(), 
        panel.grid.minor = element_blank(), 
        axis.line.y = element_line(colour = "black", linewidth = axis_size), 
        panel.border = element_blank(), 
        axis.ticks.length.x = unit(0, "cm")) +
  scale_x_discrete(labels = c("P. campbelli", "Hybrid", "P. sungorus")) + ylab("Relative testis weight (mg/g)") + 
  geom_signif(
    y_position = c(summary(fertility[fertility$Strain == "BBBB",]$RelativeTestesWeight)[6]*1.18,
                   summary(fertility[fertility$Strain == "BBBB",]$RelativeTestesWeight)[6]*1.10,
                   summary(fertility[fertility$Strain == "BBBB",]$RelativeTestesWeight)[6]*1.02), 
    xmin = c(1,1,2), 
    xmax = c(3,2,3), 
    annotation = c(relTW_padj[2],relTW_padj[1],relTW_padj[3]), 
    tip_length = 0, textsize = sigText_size, size = sigLine_size) +
  ylim(0, summary(fertility[fertility$Strain == "BBBB",]$RelativeTestesWeight)[6]*1.2)

#Absolute testes weight plots for comparison to Ishishita et al. 2015 hybrids
absTW_plot <- ggplot(fertility[!is.na(fertility$TestesWeight),], aes(x = Strain, y = TestesWeight)) + 
  geom_quasirandom(aes(fill = Strain, color = Strain, shape = Strain), alpha = 1, size = point_size*4) +
  geom_boxplot(width=boxplot_width, outlier.shape = NA, lwd = boxplot_lwd, alpha = 0.2) +
  scale_fill_manual(values=c(gray_5, gray_4, gray_2)) + 
  scale_color_manual(values=c(gray_5, gray_4, gray_2)) + 
  scale_shape_manual(values=c(24,4,25)) + 
  theme_bw() + 
  theme(legend.position = "none", 
        text = element_text(size = text_size*4),
        axis.title.x=element_blank(), 
        axis.title.y = element_text(size = text_size*2), 
        axis.text.x = element_text(angle = 45, vjust = vjust_value, size = text_size*4), 
        plot.margin = unit(c(0.4, 0.07, 0, 0.07), "cm"), 
        panel.grid.major.x = element_blank(), 
        panel.grid.minor = element_blank(), 
        axis.line.y = element_line(colour = "black", linewidth = axis_size), 
        panel.border = element_blank(), 
        axis.ticks.length.x = unit(0, "cm")) +
  scale_x_discrete(labels = c("P. campbelli", "Hybrid", "P. sungorus")) + ylab("Absolute testis weight (mg/g)") + 
  geom_text(x = 1, y = max(fertility[fertility$Strain == "BBBB",]$TestesWeight)*1.15, label = paste(round(mean(subset(fertility, fertility$Strain == "BBBB")$TestesWeight), 3), " +/- ", round(sd(subset(fertility, fertility$Strain == "BBBB")$TestesWeight),4), "g", sep = "")) + 
  geom_text(x = 2, y = max(fertility[fertility$Strain == "BBSS",]$TestesWeight)*1.15, label = paste(round(mean(subset(fertility, fertility$Strain == "BBSS")$TestesWeight), 3), " +/- ", round(sd(subset(fertility, fertility$Strain == "BBSS")$TestesWeight),4), "g", sep = "")) + 
  geom_text(x = 3, y = max(fertility[fertility$Strain == "SSSS",]$TestesWeight)*1.15, label = paste(round(mean(subset(fertility, fertility$Strain == "SSSS")$TestesWeight), 3), " +/- ", round(sd(subset(fertility, fertility$Strain == "SSSS")$TestesWeight),4), "g", sep = ""))  

#RelativeSVWeight
relSVW_kw <- kruskal.test(RelativeSVWeight ~ Strain, data = fertility[!is.na(fertility$RelativeSVWeight),])
relSVW_dt <- dunnTest(RelativeSVWeight ~ Strain, data = fertility[!is.na(fertility$RelativeSVWeight),], method="bonferroni")
relSVW_padj <- relSVW_dt$res$P.adj
names(relSVW_padj) <- gsub(" ","", relSVW_dt$res$Comparison)
relSVW_letters <- data.frame(multcompView::multcompLetters(relSVW_padj)$Letters)
relSVW_letters$Strain <- rownames(relSVW_letters)
colnames(relSVW_letters) <- c("Letters", "Strain")
relSVW_padj <- sig_conversion(relSVW_padj)

relSVW_plot <- ggplot(fertility[!is.na(fertility$RelativeSVWeight),], aes(x = Strain, y = RelativeSVWeight)) + 
  geom_quasirandom(aes(fill = Strain, color = Strain, shape = Strain), alpha = 1, size = point_size) +
  geom_boxplot(width=boxplot_width, outlier.shape = NA, lwd = boxplot_lwd, alpha = 0.2) +
  scale_fill_manual(values=c(gray_5, gray_4, gray_2)) + 
  scale_color_manual(values=c(gray_5, gray_4, gray_2)) + 
  scale_shape_manual(values=c(24,4,25)) + 
  theme_bw() + 
  theme(legend.position = "none", 
        text = element_text(size = text_size),
        axis.title.x=element_blank(), 
        axis.title.y = element_text(size = text_size), 
        axis.text.x = element_text(angle = 45, vjust = vjust_value, size = text_size), 
        plot.margin = unit(c(0.4, 0.07, 0, 0.07), "cm"), 
        panel.grid.major.x = element_blank(), 
        panel.grid.minor = element_blank(), 
        axis.line.y = element_line(colour = "black", linewidth = axis_size), 
        panel.border = element_blank(), 
        axis.ticks.length.x = unit(0, "cm")) +
  scale_x_discrete(labels = c("P. campbelli", "Hybrid", "P. sungorus")) + ylab("Relative seminal vesicle weight (mg/g)") + 
  geom_signif(
    y_position = c(summary(fertility[fertility$Strain == "BBBB",]$RelativeSVWeight)[6]*1.18,
                   summary(fertility[fertility$Strain == "BBBB",]$RelativeSVWeight)[6]*1.10,
                   summary(fertility[fertility$Strain == "BBBB",]$RelativeSVWeight)[6]*1.02), 
    xmin = c(1,1,2), 
    xmax = c(3,2,3), 
    annotation = c(relSVW_padj[2],relSVW_padj[1],relSVW_padj[3]), 
    tip_length = 0, textsize = sigText_size, size = sigLine_size) +
  ylim(0, summary(fertility[fertility$Strain == "BBBB",]$RelativeSVWeight)[6]*1.2)

#Sperm motility; only for parents because almost all hybrids don't produce sperm
motility_kw <- kruskal.test(Motility ~ Strain, data = fertility[fertility$Strain != "BBSS" & !is.na(fertility$Motility),])
motility_dt <- dunnTest(Motility ~ Strain, data = fertility[fertility$Strain != "BBSS" & !is.na(fertility$Motility),], method="bonferroni")
motility_padj <- motility_dt$res$P.adj
names(motility_padj) <- gsub(" ","", motility_dt$res$Comparison)
motility_letters <- data.frame(multcompView::multcompLetters(motility_padj)$Letters)
motility_letters$Strain <- rownames(motility_letters)
colnames(motility_letters) <- c("Letters", "Strain")
motility_padj <- sig_conversion(motility_padj)

motility_plot <- ggplot(fertility[!is.na(fertility$Motility),], aes(x = Strain, y = Motility)) + 
  geom_quasirandom(aes(fill = Strain, color = Strain, shape = Strain), alpha = 1, size = point_size) +
  geom_boxplot(width=boxplot_width, outlier.shape = NA, lwd = boxplot_lwd, alpha = 0.2) +
  scale_fill_manual(values=c(gray_5, gray_4, gray_2)) + 
  scale_color_manual(values=c(gray_5, gray_4, gray_2)) + 
  scale_shape_manual(values=c(24,4,25)) + 
  theme_bw() + 
  theme(legend.position = "none", 
        text = element_text(size = text_size),
        axis.title.x=element_blank(), 
        axis.title.y = element_text(size = text_size), 
        axis.text.x = element_text(angle = 45, vjust = vjust_value, size = text_size), 
        plot.margin = unit(c(0.4, 0.07, 0, 0.07), "cm"), 
        panel.grid.major.x = element_blank(), 
        panel.grid.minor = element_blank(), 
        axis.line.y = element_line(colour = "black", linewidth = axis_size), 
        panel.border = element_blank(), 
        axis.ticks.length.x = unit(0, "cm")) +
  scale_x_discrete(labels = c("P. campbelli", "Hybrid", "P. sungorus")) + ylab("Proportion motile sperm")  + 
  geom_signif(
    y_position = summary(fertility$Motility)[6]*1.18, 
    xmin = c(1), 
    xmax = c(3), 
    annotation = motility_padj, 
    tip_length = 0, textsize = sigText_size, size = sigLine_size) +
  ylim(0, summary(fertility[fertility$Strain == "BBBB",]$Motility)[6]*1.2)

#Normalized sperm counts
spermCount_kw <- kruskal.test(NormSpermCount ~ Strain, data = fertility[!is.na(fertility$NormSpermCount),])
spermCount_dt <- dunnTest(NormSpermCount ~ Strain, data = fertility[!is.na(fertility$NormSpermCount),], method="bonferroni")
spermCount_padj <- spermCount_dt$res$P.adj
names(spermCount_padj) <- gsub(" ","", spermCount_dt$res$Comparison)
spermCount_letters <- data.frame(multcompView::multcompLetters(spermCount_padj)$Letters)
spermCount_letters$Strain <- rownames(spermCount_letters)
colnames(spermCount_letters) <- c("Letters", "Strain")
spermCount_padj <- sig_conversion(spermCount_padj)

spermCount_plot <- ggplot(fertility[!is.na(fertility$NormSpermCount),], aes(x = Strain, y = NormSpermCount)) + 
  geom_quasirandom(aes(fill = Strain, color = Strain, shape = Strain), alpha = 1, size = point_size) +
  geom_boxplot(width=boxplot_width, outlier.shape = NA, lwd = boxplot_lwd, alpha = 0.2) +
  scale_fill_manual(values=c(gray_5, gray_4, gray_2)) + 
  scale_color_manual(values=c(gray_5, gray_4, gray_2)) + 
  scale_shape_manual(values=c(24,4,25)) + 
  theme_bw() + 
  theme(legend.position = "none", 
        text = element_text(size = text_size),
        axis.title.x=element_blank(), 
        axis.title.y = element_text(size = text_size), 
        axis.text.x = element_text(angle = 45, vjust = vjust_value, size = text_size), 
        plot.margin = unit(c(0.4, 0.07, 0, 0.07), "cm"), 
        panel.grid.major.x = element_blank(), 
        panel.grid.minor = element_blank(), 
        axis.line.y = element_line(colour = "black", linewidth = axis_size), 
        panel.border = element_blank(), 
        axis.ticks.length.x = unit(0, "cm")) +
  scale_x_discrete(labels = c("P. campbelli", "Hybrid", "P. sungorus")) + ylab("Sperm count (1 x 10^6)")  + 
  geom_signif(
    y_position = c(summary(fertility[fertility$Strain == "BBBB",]$NormSpermCount)[6]*1.18,
                   summary(fertility[fertility$Strain == "BBBB",]$NormSpermCount)[6]*1.10,
                   summary(fertility[fertility$Strain == "BBBB",]$NormSpermCount)[6]*1.02), 
    xmin = c(1,1,2), 
    xmax = c(3,2,3), 
    annotation = c(spermCount_padj[2],spermCount_padj[1],spermCount_padj[3]), 
    tip_length = 0, textsize = sigText_size, size = sigLine_size) +
  ylim(0, summary(fertility[fertility$Strain == "BBBB",]$NormSpermCount)[6]*1.2)

#print stats about focal fertility metrics
#testes weight (absolute)
BBBB_TW_mean <- mean(fertility[fertility$Strain == "BBBB" & !is.na(fertility$TestesWeight),]$TestesWeight)
BBBB_TW_sem <- calculate_sem(fertility[fertility$Strain == "BBBB" & !is.na(fertility$TestesWeight),]$TestesWeight)
BBSS_TW_mean <- mean(fertility[fertility$Strain == "BBSS" & !is.na(fertility$TestesWeight),]$TestesWeight)
BBSS_TW_sem <- calculate_sem(fertility[fertility$Strain == "BBSS" & !is.na(fertility$TestesWeight),]$TestesWeight)
SSSS_TW_mean <- mean(fertility[fertility$Strain == "SSSS" & !is.na(fertility$TestesWeight),]$TestesWeight)
SSSS_TW_sem <- calculate_sem(fertility[fertility$Strain == "SSSS" & !is.na(fertility$TestesWeight),]$TestesWeight)

#SV weight (absolute)
BBBB_SV_mean <- mean(fertility[fertility$Strain == "BBBB" & !is.na(fertility$SVWeight),]$SVWeight)
BBBB_SV_sem <- calculate_sem(fertility[fertility$Strain == "BBBB" & !is.na(fertility$SVWeight),]$SVWeight)
BBSS_SV_mean <- mean(fertility[fertility$Strain == "BBSS" & !is.na(fertility$SVWeight),]$SVWeight)
BBSS_SV_sem <- calculate_sem(fertility[fertility$Strain == "BBSS" & !is.na(fertility$SVWeight),]$SVWeight)
SSSS_SV_mean <- mean(fertility[fertility$Strain == "SSSS" & !is.na(fertility$SVWeight),]$SVWeight)
SSSS_SV_sem <- calculate_sem(fertility[fertility$Strain == "SSSS" & !is.na(fertility$SVWeight),]$SVWeight)

#Normalized sperm counts
BBBB_sperm_mean <- mean(fertility[fertility$Strain == "BBBB" & !is.na(fertility$NormSpermCount),]$NormSpermCount)
BBBB_sperm_sem <- calculate_sem(fertility[fertility$Strain == "BBBB" & !is.na(fertility$NormSpermCount),]$NormSpermCount)
BBSS_sperm_mean <- mean(fertility[fertility$Strain == "BBSS" & !is.na(fertility$NormSpermCount),]$NormSpermCount)
BBSS_sperm_sem <- calculate_sem(fertility[fertility$Strain == "BBSS" & !is.na(fertility$NormSpermCount),]$NormSpermCount)
SSSS_sperm_mean <- mean(fertility[fertility$Strain == "SSSS" & !is.na(fertility$NormSpermCount),]$NormSpermCount)
SSSS_sperm_sem <- calculate_sem(fertility[fertility$Strain == "SSSS" & !is.na(fertility$NormSpermCount),]$NormSpermCount)

#Sperm motiltity
BBBB_mot_mean <- mean(fertility[fertility$Strain == "BBBB" & !is.na(fertility$Motility),]$Motility)
BBBB_mot_sem <- calculate_sem(fertility[fertility$Strain == "BBBB" & !is.na(fertility$Motility),]$Motility)
BBSS_mot_mean <- mean(fertility[fertility$Strain == "BBSS" & !is.na(fertility$Motility),]$Motility)
BBSS_mot_sem <- calculate_sem(fertility[fertility$Strain == "BBSS" & !is.na(fertility$Motility),]$Motility)
SSSS_mot_mean <- mean(fertility[fertility$Strain == "SSSS" & !is.na(fertility$Motility),]$Motility)
SSSS_mot_sem <- calculate_sem(fertility[fertility$Strain == "SSSS" & !is.na(fertility$Motility),]$Motility)

#print summary stats for inclusion in results
paste("P. sungorus males had smaller testes and seminal vesicles (testes = ",
      round(SSSS_TW_mean,3), "g +/- ", round(SSSS_TW_sem,3),
      "; seminal vesicles = ",
      round(SSSS_SV_mean,3), "g +/- ", round(SSSS_SV_sem,3),
      "; mean +/- se) than P. campbelli males (testes = ",
      round(BBBB_TW_mean,3), "g +/- ", round(BBBB_TW_sem,3),
      "; seminal vesicles = ",
      round(BBBB_SV_mean,3), "g +/- ", round(BBBB_SV_sem,3),
      "; mean +/- se)(Dunn's Test relative testes weight p = ",
      round(relTW_dt$res[2,4], 4),
      " and relative seminal vesicle weight p = ",
      round(relSVW_dt$res[2,4], 4),
      "; figure X).",
      sep = "")

paste("Parental males did not significantly differ in normalized sperm counts (P. campbelli = ",
      round(BBBB_sperm_mean,3), " +/- ", round(BBBB_sperm_sem,3), " x 10^6",
      "; P. sungorus = ",
      round(SSSS_sperm_mean,3), " +/- ", round(SSSS_sperm_sem,3), " x 10^6",
      "); p = ",
      round(spermCount_dt$res[2,4], 4),
      ") or in sperm motility (P. campbelli = ",
      round(BBBB_mot_mean,3)*100, " +/- ", round(BBBB_mot_sem,3)*100, "%",
      "; P. sungorus = ",
      round(SSSS_mot_mean,3)*100, " +/- ", round(SSSS_mot_sem,3)*100, "%",
      "); p = ",
      round(motility_dt$res[4], 4),
      ").",
      sep = "")

paste("Hybrid males had smaller testes (",
      round(BBSS_TW_mean,3), "g +/- ", round(BBSS_TW_sem,3), "; p = ",
      round(relTW_dt$res[1,4], 6),
      ") and seminal vesicles (",
      round(BBSS_SV_mean,3), "g +/- ", round(BBSS_SV_sem,3), "; p = ",
      round(relSVW_dt$res[1,4], 4),
      ") than male P. campbelli hamsters and produced almost no mature spermatozoa.",
      sep = "")


#saving fertility plots
#either save or print figure 1
if(exportFigs == TRUE) {
  ggsave(file = "exported_figures/sterility_phenotypes.pdf", plot_grid(
    relTW_plot,
    relSVW_plot,
    spermCount_plot,
    motility_plot, 
    nrow = 1), 
    width = full_plot_width, height = 2)
} else {
  plot_grid(
    relTW_plot,
    relSVW_plot,
    spermCount_plot,
    motility_plot, 
    nrow = 1)
}

if(exportFigs == TRUE) {
  ggsave(file = "exported_figures/hamster_absolute_TW_by_cross.pdf", absTW_plot)
  ggsave(file = "exported_figures/hamster_bodyWeight_by_cross.pdf", bw_plot)
  ggsave(file = "exported_figures/hamster_relativeTestesWeight_by_cross.pdf", relTW_plot)
  ggsave(file = "exported_figures/hamster_relativeSVWeight_by_cross.pdf", relSVW_plot)
  ggsave(file = "exported_figures/hamster_motility_by_cross.pdf", motility_plot)
  ggsave(file = "exported_figures/hamster_spermCount_by_cross.pdf", spermCount_plot)
}

#exporting cleaned data for table S1
cleaned_fertility_data <- data.frame(Strain = gsub("_.*", "", fertility$ID),
                                     Family = gsub(".*_(.*)-.*", "\\1", fertility$ID),
                                     Individual = gsub(".*-", "", fertility$ID),
                                     Age = fertility$Age,
                                     BodyWeight = round(fertility$BodyWeight, 1),
                                     TestesWeight = round(fertility$TestesWeight, 2),
                                     RelativeTestesWeight = round(fertility$RelativeTestesWeight, 2),
                                     SVWeight = round(fertility$SVWeight, 2),
                                     RelativeSVWeight = round(fertility$RelativeSVWeight, 2),
                                     Motility = round(fertility$Motility, 2),
                                     NormalizedSpermCount = round(fertility$NormSpermCount, 0))

write.csv(cleaned_fertility_data, file = "exported_files/cleaned_hamster_ferility_data.csv")

#cleanup; remove unused objects and plots
rm(absTW_plot, bw_plot, bw_v_svw_plot, bw_v_tw_plot, motility_plot, relSVW_plot, relTW_plot, spermCount_plot)
rm(BBBB_mot_sem, BBBB_sperm_sem, BBBB_SV_sem, BBBB_TW_sem, BBSS_mot_sem, BBSS_sperm_sem, BBSS_SV_sem, BBSS_TW_sem, SSSS_mot_sem, SSSS_sperm_sem, SSSS_SV_sem, SSSS_TW_sem)
rm(BBBB_mot_mean, BBBB_sperm_mean, BBBB_SV_mean, BBBB_TW_mean, BBSS_mot_mean, BBSS_sperm_mean, BBSS_SV_mean, BBSS_TW_mean, SSSS_mot_mean, SSSS_sperm_mean, SSSS_SV_mean, SSSS_TW_mean)
rm(bodyWeight_padj, motility_padj, relSVW_padj, relTW_padj, spermCount_padj)
rm(SVW_pValue, TW_pValue)
rm(SVW_rho, TW_rho)
rm(SVW_corr, TW_corr)
rm(bodyWeight_dt, motility_dt, relSVW_dt, relTW_dt, spermCount_dt)
rm(bodyWeight_kw, motility_kw, relSVW_kw, relTW_kw, spermCount_kw)
rm(bodyWeight_letters, motility_letters, relSVW_letters, relTW_letters, spermCount_letters)
```

<h3>Categorize expressed genes by chromosome</h3>
```{r categorize-genes-by-chromosome}
#subset count data into counts for genes on all autosomes (counts are not normalized)
just_auto_counts <- just_counts[grep("X",annotations$chr, invert = TRUE),]
auto_genes <- annotations[rownames(annotations[grep("X",annotations$chr, invert = TRUE),]),]

#subset count data into counts for genes on the X chromosome
just_X_counts <- just_counts[rownames(just_counts) %in% annotations[grep("X",annotations$chr),]$gene,]
X_genes <- annotations[grepl("^X",annotations$chr),]

#summary table of expressed genes by chr
summary_df <- data.frame(dataset =      c("raw_counts", "filtered_counts"),
                         tota_genes =   c(length(rownames(annotations)), length(dgeDev$genes$gene)),
                         auto_genes =   c(length(rownames(auto_genes)), sum(auto_genes$gene %in% dgeDev$genes$gene)),
                         X_genes =      c(length(rownames(X_genes)), sum(X_genes$gene  %in% dgeDev$genes$gene)),
                         prop_X_genes = c(round(100*length(rownames(X_genes))/length(rownames(annotations)), 2), 
                                          round(sum(X_genes$gene %in% dgeDev$genes$gene)/length(dgeDev$genes$gene)*100,2)))
kable(summary_df, "simple")

#cleanup
rm(summary_df)
```

<h3>Make all subsetted DGEs for MDSplots using all genes</h3>
```{r dge-subsets}
#make all subsetted DGEs for MDSplots by specifying which countfiles, annotations, and grouping info to use for each dge then normalize that dge with the above normalize_dge function

#subsetted by spermatogenic stage
dgeSP <- DGEList(just_counts[,grep("SP",colnames(just_counts))], genes=annotations, group=grep("SP",metadata$condition_name, value = TRUE))
dgeSP <- normalize_dge(dgeSP)

dgeLZ <- DGEList(just_counts[,grep("LZ",colnames(just_counts))], genes=annotations, group=grep("LZ",metadata$condition_name, value = TRUE))
dgeLZ <- normalize_dge(dgeLZ)

dgeDIP <- DGEList(just_counts[,grep("DIP",colnames(just_counts))], genes=annotations, group=grep("DIP",metadata$condition_name, value = TRUE))
dgeDIP <- normalize_dge(dgeDIP)

dgeRS <- DGEList(just_counts[,grep("RS",colnames(just_counts))], genes=annotations, group=grep("RS",metadata$condition_name, value = TRUE))
dgeRS <- normalize_dge(dgeRS)

#subsetted by genotype
dgeSSSS <- DGEList(just_counts[,grep("SSSS.*_[D,L,S,R]",colnames(just_counts))], genes=annotations, group=grep("SSSS-[D,L,S,R]",metadata$condition_name, value = TRUE))
dgeSSSS <- normalize_dge(dgeSSSS)

dgeBBSS <- DGEList(just_counts[,grep("BBSS.*_[D,L,S,R]",colnames(just_counts))], genes=annotations, group=grep("BBSS-[D,L,S,R]",metadata$condition_name, value = TRUE))
dgeBBSS <- normalize_dge(dgeBBSS)

dgeBBBB <- DGEList(just_counts[,grep("BBBB.*_[D,L,S,R]",colnames(just_counts))], genes=annotations, group=grep("BBBB-[D,L,S,R]",metadata$condition_name, value = TRUE))
dgeBBBB <- normalize_dge(dgeBBBB)

#sex chromosome/autosome DGEs
dgeX <- DGEList(just_X_counts[,grep("WT",colnames(just_X_counts), invert = TRUE)], genes=X_genes, group=grep("WT",metadata$condition_name, value = TRUE, invert = TRUE))
dgeX <- normalize_dge(dgeX)

dgeAuto <- DGEList(just_auto_counts[,grep("WT",colnames(just_auto_counts), invert = TRUE)], genes=auto_genes, group=grep("WT",metadata$condition_name, value = TRUE, invert = TRUE))
dgeAuto <- normalize_dge(dgeAuto)
```

<h3>Define the base order of the heatmap columns</h3>
```{r heatmap-settings}
#formatting metadata
metadata$cross <- factor(metadata$cross, levels = genotypes)
genotype_order <- seq(1,3)[metadata$cross]
metadata$stage <- factor(metadata$stage, levels = stages)
stage_order <- seq(1,5)[metadata$stage]

#heatmap order for all samples from Ham dataset and just sorted cells
Full_heatmap_order <- rownames(metadata[order(stage_order, genotype_order),])
Dev_heatmap_order <- grep("WT", Full_heatmap_order, invert = TRUE, value = TRUE)
rm(genotype_order, stage_order)

#heatmap order by genotype
SSSS_heatmap_order <- grep("SSSS", Dev_heatmap_order, value = TRUE)
BBBB_heatmap_order <- grep("BBBB", Dev_heatmap_order, value = TRUE)
BBSS_heatmap_order <- grep("BBSS", Dev_heatmap_order, value = TRUE)

#heatmap order by stage
SP_heatmap_order <- grep("SP", Dev_heatmap_order, value = TRUE)
LZ_heatmap_order <- grep("LZ", Dev_heatmap_order, value = TRUE)
DIP_heatmap_order <- grep("DIP", Dev_heatmap_order, value = TRUE)
RS_heatmap_order <- grep("RS", Dev_heatmap_order, value = TRUE)

#define color palette for all heatmaps
colfunc <- colorRampPalette(c("#F6F3F3", "#231A1A"))
browns <- colfunc(255)
rm(colfunc)
```

<h3>Set up CPM, RPKM, and Normalized FPKM data frames - not subsetted</h3>
```{r create-count-dfs-not-subsetted}
#counts per million dataframes for the whole dataset and by sex chromosome
cpm_Dev  <- cpm(dgeDev, log = TRUE)
cpm_Dev_justX <- subset(cpm_Dev, rownames(cpm_Dev) %in% rownames(X_genes))
cpm_Dev_allAuto <- subset(cpm_Dev, !rownames(cpm_Dev) %in% rownames(X_genes))

#reads of kilobase of transcript, per million mapped reads (rpkm) dataframes for each of the three datasets (Ham, dev, wt) and for each by sex chromosome
rpkm_Dev  <- as.data.frame(rpkm(dgeDev,  normalized.lib.sizes = TRUE, log = FALSE))
rpkm_Dev_justX <- subset(rpkm_Dev, rownames(rpkm_Dev) %in% rownames(X_genes))
rpkm_Dev_allAuto <- subset(rpkm_Dev, !rownames(rpkm_Dev) %in% rownames(X_genes))

#log2 reads of kilobase of transcript, per million mapped reads (log2(rpkm)) dataframes for each of the three datasets (Ham, dev, wt) and for eac dataset by sex chromosome
rpkm_Dev.log  <- as.data.frame(rpkm(dgeDev,  normalized.lib.sizes = TRUE, log = TRUE))
rpkm_Dev_justX.log <- subset(rpkm_Dev.log, rownames(rpkm_Dev.log) %in% rownames(X_genes))
rpkm_Dev_allAuto.log <- subset(rpkm_Dev.log, !rownames(rpkm_Dev.log) %in% rownames(X_genes))

#normalized reads of kilobase of transcript, per million mapped reads so that sum of squares is 1 (normalized rpkm) dataframes for each of the three datasets (Ham, dev, wt) and for each dataset by sex chromosome
rpkm_Dev.norm  <- decostand(as.data.frame(rpkm(dgeDev,  normalized.lib.sizes = TRUE)), "normalize")
rpkm_Dev_justX.norm <- subset(rpkm_Dev.norm, rownames(rpkm_Dev.norm) %in% rownames(X_genes))
rpkm_Dev_allAuto.norm <- subset(rpkm_Dev.norm, !rownames(rpkm_Dev.norm) %in% rownames(X_genes))

write.csv(rpkm_Dev.log, file = "exported_files/log_norm_Ham_counts_for_WGCNA.csv")
```

<h3>Set up CPM, RPKM, and Normalized FPKM data frames - subsetted by stage</h3>
```{r create-count-dfs-subsetted-by-stage}
#make cpm, rpkm, and rpkm.norm dataframes for each stage across genotypes
for (f in 1:length(stages[1:4])){
  cpm <- cpm_Dev[,grep(stages[f], colnames(cpm_Dev))]
  rpkm <- rpkm_Dev[,grep(stages[f], colnames(rpkm_Dev))]
  rpkm.norm <- rpkm_Dev.norm[,grep(stages[f], colnames(rpkm_Dev.norm))]
  assign(paste("cpm", "Dev", stages[f], sep = "_"), cpm)
  assign(paste("rpkm", "Dev", stages[f], sep = "_"), rpkm)
  assign(paste(paste("rpkm", "Dev", stages[f], sep = "_"),"norm", sep = "."), rpkm.norm)
}

rm(cpm, rpkm, rpkm.norm)
```

<h3>Set up CPM, RPKM, and Normalized FPKM data frames - subsetted by genotype and stage but not by chromosome</h3>
```{r create-count-dfs-subsetted-by-cross-and-stage}
#make cpm, rpkm, and rpkm.norm dataframes for each stage type for each genotype across all genes
for (r in 1:length(genotypes)) {
  #subsets for each genotype ACROSS stages
  cpmAll <- cpm_Dev[,grep(genotypes[r], colnames(cpm_Dev))]
  rpkmAll <- rpkm_Dev[,grep(genotypes[r], colnames(cpm_Dev))]
  rpkmAll.norm <- rpkm_Dev.norm[,grep(genotypes[r], colnames(cpm_Dev))]
  
  #assign to relevant variable name
  assign(paste("cpm", "Dev", genotypes[r], sep = "_"), cpmAll)
  assign(paste("rpkm", "Dev", genotypes[r], sep = "_"), rpkmAll)
  assign(paste(paste("rpkm", "Dev", genotypes[r], sep = "_"),"norm", sep = "."), rpkmAll.norm)
  
  for (q in 1:length(stages[1:4])){
    #print genotype plus stage combination
    combo_name <- paste(genotypes[r], stages[q], sep = ".*")
    #print(paste(genotypes[r], stages[q], sep = ".*"))
    
    #subsets for each genotype FOR each stage
    cpm <- cpm_Dev[,grep(combo_name, colnames(cpm_Dev))]
    rpkm <- rpkm_Dev[,grep(combo_name, colnames(cpm_Dev))]
    rpkm.norm <- rpkm_Dev.norm[,grep(combo_name, colnames(cpm_Dev))]
    
    #assign to relevant variable name
    assign(paste("cpm", "Dev", genotypes[r], stages[q], sep = "_"), cpm)
    assign(paste("rpkm", "Dev", genotypes[r], stages[q], sep = "_"), rpkm)
    assign(paste(paste("rpkm", "Dev", genotypes[r], stages[q], sep = "_"),"norm", sep = "."), rpkm.norm)
  }
}

rm(cpmAll, rpkmAll, rpkmAll.norm)
rm(cpm, rpkm, rpkm.norm)
```

<h3>Create sex chromosomes subsets by genotype using just sorted cell dataset</h3>
```{r create-count-dfs-x-and-auto-by-cross-and-stage-Dev}
#make cpm, rpkm, and rpkm.norm dataframes for each stage type for each genotype across all X chromosome genes
for (r in 1:length(genotypes)) {
  #subsets for each genotype across stages
  cpmAll <- cpm_Dev_justX[,grep(genotypes[r], colnames(cpm_Dev_justX))]
  rpkmAll <- rpkm_Dev_justX[,grep(genotypes[r], colnames(rpkm_Dev_justX))]
  rpkmAll.norm <- rpkm_Dev_justX.norm[,grep(genotypes[r], colnames(rpkm_Dev_justX.norm))]
  
  #assign to relevant variable name
  assign(paste("cpm", genotypes[r], "Dev", "justX", sep = "_"), cpmAll)
  assign(paste("rpkm", genotypes[r], "Dev", "justX", sep = "_"), rpkmAll)
  assign(paste(paste("rpkm", genotypes[r], "Dev", "justX", sep = "_"),"norm", sep = "."), rpkmAll.norm)
  
  for (q in 1:length(stages[1:4])){
    #print genotype plus stage combination
    combo_name <- paste(genotypes[r], stages[q], sep = ".*")
    #print(paste(genotypes[r], stages[q], sep = ".*"))
    
    #subsets for each genotype for each stage
    cpm <- cpm_Dev_justX[,grep(combo_name, colnames(cpm_Dev_justX))]
    rpkm <- rpkm_Dev_justX[,grep(combo_name, colnames(rpkm_Dev_justX))]
    rpkm.norm <- rpkm_Dev_justX.norm[,grep(combo_name, colnames(rpkm_Dev_justX.norm))]
    
    #assign to relevant variable name
    assign(paste("cpm", genotypes[r], stages[q], "Dev", "justX", sep = "_"), cpm)
    assign(paste("rpkm", genotypes[r], stages[q], "Dev", "justX", sep = "_"), rpkm)
    assign(paste(paste("rpkm", genotypes[r], stages[q], "Dev", "justX", sep = "_"),"norm", sep = "."), rpkm.norm)
  }
}

#make cpm, rpkm, and rpkm.norm dataframes for each stage type for each genotype across all autosomal genes
for (r in 1:length(genotypes)) {
  #subsets for each genotype across stages
  cpmAll <- cpm_Dev_allAuto[,grep(genotypes[r], colnames(cpm_Dev_allAuto))]
  rpkmAll <- rpkm_Dev_allAuto[,grep(genotypes[r], colnames(rpkm_Dev_allAuto))]
  rpkmAll.norm <- rpkm_Dev_allAuto.norm[,grep(genotypes[r], colnames(rpkm_Dev_allAuto.norm))]
  
  #assign to relevant variable name
  assign(paste("cpm", genotypes[r], "Dev", "allAuto", sep = "_"), cpmAll)
  assign(paste("rpkm", genotypes[r], "Dev", "allAuto", sep = "_"), rpkmAll)
  assign(paste(paste("rpkm", genotypes[r], "Dev", "allAuto", sep = "_"),"norm", sep = "."), rpkmAll.norm)
  
  for (q in 1:length(stages[1:4])){
    #print genotype plus stage combination
    combo_name <- paste(genotypes[r], stages[q], sep = ".*")
    #print(paste(genotypes[r], stages[q], sep = ".*"))
    
    #subsets for each genotype for each stage
    cpm <- cpm_Dev_allAuto[,grep(combo_name, colnames(cpm_Dev_allAuto))]
    rpkm <- rpkm_Dev_allAuto[,grep(combo_name, colnames(rpkm_Dev_allAuto))]
    rpkm.norm <- rpkm_Dev_allAuto.norm[,grep(combo_name, colnames(rpkm_Dev_allAuto.norm))]
    
    #assign to relevant variable name
    assign(paste("cpm", genotypes[r], stages[q], "Dev", "allAuto", sep = "_"), cpm)
    assign(paste("rpkm", genotypes[r], stages[q], "Dev", "allAuto", sep = "_"), rpkm)
    assign(paste(paste("rpkm", genotypes[r], stages[q], "Dev", "allAuto", sep = "_"),"norm", sep = "."), rpkm.norm)
  }
}

rm(cpmAll, rpkmAll, rpkmAll.norm)
rm(cpm, rpkm, rpkm.norm)
```

<h3>Expression Levels for each Stage/Genotype </h3>
```{r rpkm-by-cell-definitions}
#create genotype x stage rpkm dataframes (with the different normalization methods)
cell.list <- c("BBBB.*SP", "BBSS.*SP", "SSSS.*SP", "BBBB.*LZ", "BBSS.*LZ", "SSSS.*LZ", "BBBB.*DIP", "BBSS.*DIP", "SSSS.*DIP", "BBBB.*RS", "SSSS.*RS")
cell.names <- c("BBBB_SP", "BBSS_SP", "SSSS_SP", "BBBB_LZ", "BBSS_LZ", "SSSS_LZ", "BBBB_DIP", "BBSS_DIP", "SSSS_DIP", "BBBB_RS", "SSSS_RS")

#unnormalized rpkm values
rpkm.bycell <- data.frame("chr" = annotations[rownames(annotations) %in%  rownames(dgeDev$genes),]$chr, row.names = rownames(annotations[rownames(annotations) %in%  rownames(dgeDev$genes),]))

for (cell in 1:length(cell.list)) {
  rpkm.bycell <- cbind(rpkm.bycell, as.data.frame(rowMeans(rpkm_Dev[,grep(cell.list[cell],colnames(rpkm_Dev))])))
}

colnames(rpkm.bycell) <- c("chr", cell.names)

#normalized RPKM; sum of squares
rpkm.norm.bycell <- data.frame("chr" = annotations[rownames(annotations) %in%  rownames(dgeDev$genes),]$chr, row.names = rownames(annotations[rownames(annotations) %in%  rownames(dgeDev$genes),]))
for (cell in 1:length(cell.list)) {
  rpkm.norm.bycell <- cbind(rpkm.norm.bycell, as.data.frame(rowMeans(rpkm_Dev.norm[,grep(cell.list[cell],colnames(rpkm_Dev.norm))])))
}
colnames(rpkm.norm.bycell) <- c("chr", cell.names)

#normalized log(RPKM)
rpkm.log2.bycell <- data.frame("chr" = annotations[rownames(annotations) %in%  rownames(dgeDev$genes),]$chr, row.names = rownames(annotations[rownames(annotations) %in%  rownames(dgeDev$genes),]))
for (cell in 1:length(cell.list)) {
  rpkm.log2.bycell <- cbind(rpkm.log2.bycell, as.data.frame(rowMeans(rpkm_Dev.log[,grep(cell.list[cell],colnames(rpkm_Dev.log))])))
}
colnames(rpkm.log2.bycell) <- c("chr", cell.names)


#plot RPKM density graphs for each genotype x stage to check if any genotype x stage combination is an outlier (hybrid diplotene is the major outlier; discussed in manuscript)
colors <- c(sp_color, sp_color, sp_color, lz_color, lz_color, lz_color, dip_color, dip_color, dip_color, rs_color, rs_color)
lines <- c(1, 2, 3, 1, 2, 3, 1, 2, 3, 1, 3)
#change exportFigs to TRUE to export as PDF if desired
if(exportFigs == TRUE) {
  pdf(file="exported_figures/RPKM_density_all_genes.pdf", width=10, height=8, pointsize=12)
  for (sample in 1:(ncol(rpkm.log2.bycell)-1)){
    plot(density(rpkm.log2.bycell[, sample+1]), col = colors[sample], lty = lines[sample], ylab = "", xaxt = "n", xlab = "", yaxt = "n", main = "", ylim = c(0, 0.18), xlim = c(-10, 13), lwd = 3)
    par(new = TRUE)
  }
  axis(side = 2)
  axis(side = 1)
  title(xlab = "Mean RPKM after filtering - all genes", ylab = "Density")
  legend(-11, 0.19, c("Spermatogonia", "Leptotene/zygotene", "Diplotene", "Round spermatid"),
         lty = c(1), col = colors[c(1,4,7,10,12)], lwd = 3, bty = "n")
  dev.off()
} else {
  for (sample in 1:(ncol(rpkm.log2.bycell)-1)){
    plot(density(rpkm.log2.bycell[, sample+1]), col = colors[sample], lty = lines[sample], ylab = "", xaxt = "n", xlab = "", yaxt = "n", main = "", ylim = c(0, 0.18), xlim = c(-10, 13), lwd = 3)
    par(new = TRUE)
  }
  axis(side = 2)
  axis(side = 1)
  title(xlab = "Mean RPKM after filtering - all genes", ylab = "Density")
  legend(-11, 0.19, c("Spermatogonia", "Leptotene/zygotene", "Diplotene", "Round spermatid"),
         lty = c(1), col = colors[c(1,4,7,10,12)], lwd = 3, bty = "n")
}
```

<h3>Defining cell population-specific expressed genes (for use in hypergeometric tests of DE enrichment)</h3>
```{r defining-sets-of-expressed-genes}
#get expressed genes meeting filtering requirements from those DGEs you just set up
#expressed across all samples
all.exp <- dgeDev$genes$gene # 21826
#expressed across parents
BBBB.SSSS.all.exp <- rownames(rpkm_Dev[rowSums(rpkm_Dev[,grep("BS.", colnames(rpkm_Dev), value = TRUE, invert = TRUE)] > rpkm.cutoff)  >= min.reps, ]) #21011
#expressed within P. campbelli
BBBB.all.exp <- dgeBBBB$genes$gene # 19758
#expressed within hybrids
Hyb.all.exp <- dgeBBSS$genes$gene # 19148
#expressed within P. sungorus
SSSS.all.exp <- dgeSSSS$genes$gene # 20060

#expressed within hybrids for each stage
Hyb.SP.exp <- rownames(rpkm_Dev[rowSums(rpkm_Dev[,grep("BBSS.*SP", colnames(rpkm_Dev))] > rpkm.cutoff)  >= min.reps, ]) # 15984
Hyb.LZ.exp <- rownames(rpkm_Dev[rowSums(rpkm_Dev[,grep("BBSS.*LZ", colnames(rpkm_Dev))] > rpkm.cutoff)  >= min.reps, ]) # 15613
Hyb.DIP.exp <- rownames(rpkm_Dev[rowSums(rpkm_Dev[,grep("BBSS.*DIP", colnames(rpkm_Dev))] > rpkm.cutoff)  >= min.reps, ]) # 14511

#expressed within P. campbelli for each stage
BBBB.SP.exp <- rownames(rpkm_Dev[rowSums(rpkm_Dev[,grep("BBBB.*SP", colnames(rpkm_Dev))] > rpkm.cutoff)  >= min.reps, ]) # 16155
BBBB.LZ.exp <- rownames(rpkm_Dev[rowSums(rpkm_Dev[,grep("BBBB.*LZ", colnames(rpkm_Dev))] > rpkm.cutoff)  >= min.reps, ]) # 16651
BBBB.DIP.exp <- rownames(rpkm_Dev[rowSums(rpkm_Dev[,grep("BBBB.*DIP", colnames(rpkm_Dev))] > rpkm.cutoff)  >= min.reps, ]) # 12353
BBBB.RS.exp <- rownames(rpkm_Dev[rowSums(rpkm_Dev[,grep("BBBB.*RS", colnames(rpkm_Dev))] > rpkm.cutoff)  >= min.reps, ]) # 14344

#expressed within P. sungorus for each stage
SSSS.SP.exp <- rownames(rpkm_Dev[rowSums(rpkm_Dev[,grep("SSSS.*SP", colnames(rpkm_Dev))] > rpkm.cutoff)  >= min.reps, ]) # 17025
SSSS.LZ.exp <- rownames(rpkm_Dev[rowSums(rpkm_Dev[,grep("SSSS.*LZ", colnames(rpkm_Dev))] > rpkm.cutoff)  >= min.reps, ]) # 16124
SSSS.DIP.exp <- rownames(rpkm_Dev[rowSums(rpkm_Dev[,grep("SSSS.*DIP", colnames(rpkm_Dev))] > rpkm.cutoff)  >= min.reps, ]) # 13091
SSSS.RS.exp <- rownames(rpkm_Dev[rowSums(rpkm_Dev[,grep("SSSS.*RS", colnames(rpkm_Dev))] > rpkm.cutoff)  >= min.reps, ]) # 13436

#expressed within hybrids and P.campbelli for each stage
Hyb.BB.SP.exp <-  rownames(rpkm_Dev[rowSums(rpkm_Dev[,grep("BB.*SP", colnames(rpkm_Dev))] > rpkm.cutoff)  >= min.reps, ]) # 17919
Hyb.BB.LZ.exp <-  rownames(rpkm_Dev[rowSums(rpkm_Dev[,grep("BB.*LZ", colnames(rpkm_Dev))] > rpkm.cutoff)  >= min.reps, ]) # 17698
Hyb.BB.DIP.exp <- rownames(rpkm_Dev[rowSums(rpkm_Dev[,grep("BB.*DIP", colnames(rpkm_Dev))] > rpkm.cutoff)  >= min.reps, ]) # 15695

#expressed within hybrids and P. sungorus for each stage
Hyb.SS.SP.exp <-  rownames(rpkm_Dev[rowSums(rpkm_Dev[,grep("SS.*SP", colnames(rpkm_Dev))] > rpkm.cutoff)  >= min.reps, ]) # 18187
Hyb.SS.LZ.exp <-  rownames(rpkm_Dev[rowSums(rpkm_Dev[,grep("SS.*LZ", colnames(rpkm_Dev))] > rpkm.cutoff)  >= min.reps, ]) # 17492
Hyb.SS.DIP.exp <- rownames(rpkm_Dev[rowSums(rpkm_Dev[,grep("SS.*DIP", colnames(rpkm_Dev))] > rpkm.cutoff)  >= min.reps, ]) # 15739

#expressed in all samples for each stage
Hyb.Comb.SP.exp <-  rownames(rpkm_Dev[rowSums(rpkm_Dev[,grep("SP", colnames(rpkm_Dev))] > rpkm.cutoff)  >= min.reps, ]) # 18843
Hyb.Comb.LZ.exp <-  rownames(rpkm_Dev[rowSums(rpkm_Dev[,grep("LZ", colnames(rpkm_Dev))] > rpkm.cutoff)  >= min.reps, ]) # 18372
Hyb.Comb.DIP.exp <- rownames(rpkm_Dev[rowSums(rpkm_Dev[,grep("DIP", colnames(rpkm_Dev))] > rpkm.cutoff)  >= min.reps, ]) # 16295

#expressed within both parent species for each stage
BBBB.SSSS.SP.exp <-  rownames(rpkm_Dev[rowSums(rpkm_Dev[,grep("SP", grep("BS.", colnames(rpkm_Dev), value = TRUE, invert = TRUE), value = TRUE)] > rpkm.cutoff)  >= min.reps, ]) # 18241
BBBB.SSSS.LZ.exp <-  rownames(rpkm_Dev[rowSums(rpkm_Dev[,grep("LZ", grep("BS.", colnames(rpkm_Dev), value = TRUE, invert = TRUE), value = TRUE)] > rpkm.cutoff)  >= min.reps, ]) # 17811
BBBB.SSSS.DIP.exp <- rownames(rpkm_Dev[rowSums(rpkm_Dev[,grep("DIP", grep("BS.", colnames(rpkm_Dev), value = TRUE, invert = TRUE), value = TRUE)] > rpkm.cutoff)  >= min.reps, ]) # 14078
BBBB.SSSS.RS.exp <-  rownames(rpkm_Dev[rowSums(rpkm_Dev[,grep("RS", grep("BS.", colnames(rpkm_Dev), value = TRUE, invert = TRUE), value = TRUE)] > rpkm.cutoff)  >= min.reps, ]) # 15837

#make lists of cell types for hypergeometric tests below
Hyb.Comb.cell.exp.list <- c("Hyb.Comb.SP.exp", "Hyb.Comb.LZ.exp", "Hyb.Comb.DIP.exp")
Hyb.BB.cell.exp.list <- c("Hyb.BB.SP.exp", "Hyb.BB.LZ.exp", "Hyb.BB.DIP.exp")
Hyb.SS.cell.exp.list <- c("Hyb.SS.SP.exp", "Hyb.SS.LZ.exp", "Hyb.SS.DIP.exp")
par.cell.exp.list <- c("BBBB.SSSS.SP.exp", "BBBB.SSSS.LZ.exp", "BBBB.SSSS.DIP.exp", "BBBB.SSSS.RS.exp")

print(paste("Only ",
            round(100*sum(BBBB.SSSS.DIP.exp %in% X_genes$gene)/length(BBBB.SSSS.DIP.exp), 3), 
            "% of expressed genes in parental DIP are X-linked, while ", 
            round(100*sum(BBBB.SSSS.SP.exp %in% X_genes$gene)/length(BBBB.SSSS.SP.exp), 3), 
            "% and ", 
            round(100*sum(BBBB.SSSS.LZ.exp %in% X_genes$gene)/length(BBBB.SSSS.LZ.exp), 3), 
            "% in spermatogonia and leptotene/zygotene respectively are X-linked.", 
            sep = ""))

print(paste("In contrast to parental dwarf hamsters ",
            round(100*sum(Hyb.DIP.exp %in% X_genes$gene)/length(Hyb.DIP.exp), 3), 
            "% of expressed genes in hybrid DIP are X-linked, while ", 
            round(100*sum(Hyb.SP.exp %in% X_genes$gene)/length(Hyb.SP.exp), 3), 
            "% and ", 
            round(100*sum(Hyb.LZ.exp %in% X_genes$gene)/length(Hyb.LZ.exp), 3), 
            "% in spermatogonia and leptotene/zygotene respectively are X-linked.", 
            sep = ""))
```

<h3>Define sets of induced genes for each stage</h3>
```{r defining-induced}
#define sets of "induced" genes for each cell population for all genes, X-linked genes, and autosomal genes
#these are genes that in a given cell-type have a median expression (normalized FPKM) greater than two times (induced_cutoff) its median expression across all other sorted cell populations (following Kousathanas et al. 2014)
#after defining sets of induced genes, write them out to files
induced_cutoff <- 2

#SP
SP.induced <- rownames(rpkm.bycell[rownames(rpkm.bycell) %in% BBBB.SSSS.SP.exp
                                   & rowMedians(as.matrix(rpkm.bycell[,c("BBBB_SP","SSSS_SP")])) >
                                     rowMedians(as.matrix(rpkm.bycell[,c("BBBB_LZ","BBBB_DIP","BBBB_RS","SSSS_LZ","SSSS_DIP","SSSS_RS")]))*induced_cutoff, ])
SP_Auto.induced <- SP.induced[!(SP.induced %in% rownames(X_genes))]
SP_X.induced <- SP.induced[SP.induced %in% rownames(X_genes)]
write.table(SP.induced, file = paste("induced/hamster_SP_induced_inducedCutoff_", induced_cutoff,".txt", sep = ""), 
            sep = "\t", quote = FALSE, row.names = FALSE, col.names = FALSE)
write.table(SP_X.induced, file = paste("induced/hamster_SP_induced_X_inducedCutoff_", induced_cutoff,".txt", sep = ""), 
            sep = "\t", quote = FALSE, row.names = FALSE, col.names = FALSE)

#LZ
LZ.induced <- rownames(rpkm.bycell[rownames(rpkm.bycell) %in% BBBB.SSSS.LZ.exp
                                   & rowMedians(as.matrix(rpkm.bycell[,c("BBBB_LZ","SSSS_LZ")])) >
                                     rowMedians(as.matrix(rpkm.bycell[,c("BBBB_SP","BBBB_DIP","BBBB_RS","SSSS_SP","SSSS_DIP","SSSS_RS")]))*induced_cutoff, ])
LZ_Auto.induced <- LZ.induced[!(LZ.induced %in% rownames(X_genes))]
LZ_X.induced <- LZ.induced[LZ.induced %in% rownames(X_genes)]
write.table(LZ.induced, file = paste("induced/hamster_LZ_induced_inducedCutoff_", induced_cutoff,".txt", sep = ""), 
            sep = "\t", quote = FALSE, row.names = FALSE, col.names = FALSE)
write.table(LZ_X.induced, file = paste("induced/hamster_LZ_induced_X_inducedCutoff_", induced_cutoff,".txt", sep = ""), 
            sep = "\t", quote = FALSE, row.names = FALSE, col.names = FALSE)

#DIP
DIP.induced <- rownames(rpkm.bycell[rownames(rpkm.bycell) %in% BBBB.SSSS.DIP.exp
                                    & rowMedians(as.matrix(rpkm.bycell[,c("BBBB_DIP","SSSS_DIP")])) >
                                      rowMedians(as.matrix(rpkm.bycell[,c("BBBB_SP","BBBB_LZ","BBBB_RS","SSSS_SP","SSSS_LZ","SSSS_RS")]))*induced_cutoff, ])
DIP_Auto.induced <- DIP.induced[!(DIP.induced %in% rownames(X_genes))]
DIP_X.induced <- DIP.induced[DIP.induced %in% rownames(X_genes)]
write.table(DIP.induced, file = paste("induced/hamster_DIP_induced_inducedCutoff_", induced_cutoff,".txt", sep = ""), 
            sep = "\t", quote = FALSE, row.names = FALSE, col.names = FALSE)
write.table(DIP_X.induced, file = paste("induced/hamster_DIP_induced_X_inducedCutoff_", induced_cutoff,".txt", sep = ""), 
            sep = "\t", quote = FALSE, row.names = FALSE, col.names = FALSE)

#RS
RS.induced <- rownames(rpkm.bycell[rownames(rpkm.bycell) %in% BBBB.SSSS.RS.exp
                                   & rowMedians(as.matrix(rpkm.bycell[,c("BBBB_RS","SSSS_RS")])) >
                                     rowMedians(as.matrix(rpkm.bycell[,c("BBBB_SP","BBBB_LZ","BBBB_DIP","SSSS_SP","SSSS_LZ","SSSS_DIP")]))*induced_cutoff, ]) 
RS_Auto.induced <- RS.induced[!(RS.induced %in% rownames(X_genes))]
RS_X.induced <- RS.induced[RS.induced %in% rownames(X_genes)]
write.table(RS.induced, file = paste("induced/hamster_RS_induced_inducedCutoff_", induced_cutoff,".txt", sep = ""), 
            sep = "\t", quote = FALSE, row.names = FALSE, col.names = FALSE)
write.table(RS_X.induced, file = paste("induced/hamster_RS_induced_X_inducedCutoff_", induced_cutoff,".txt", sep = ""), 
            sep = "\t", quote = FALSE, row.names = FALSE, col.names = FALSE)


```

<h3>Establish hamster/mouse orthology</h3>
```{r establish-orthology}
# #uncomment code to establish list of orthologs between hamsters and house mice
# #if the file "mouse_hamster_orthology_29May23.csv" already exists, proceed from the read.csv command
# 
# ham_mouse_orthology <- read.csv("input_data/psun_mmus_blast.csv")
# 
# #create a dictionary of orthologs; recomment on next run
# orthology_df <- data.frame()
# 
# for(mouseGene in unique(ham_mouse_orthology$Mouse_gene)){
#   #print(mouseGene)
#   hGene_longestMatch <- order(ham_mouse_orthology[ham_mouse_orthology$Mouse_gene == mouseGene,]$X, decreasing = TRUE)[1]
#   hamGene <- ham_mouse_orthology[ham_mouse_orthology$Mouse_gene == mouseGene,][hGene_longestMatch,]$Phodopus_gene
#   #print(hamGene)
#   orthology_df <- rbind(orthology_df, data.frame(mGene = mouseGene, hGene = hamGene))
# }
# #write file to exported_files and also input_files
# write.csv(orthology_df, file = "exported_files/mouse_hamster_orthology_29May23.csv", row.names = FALSE)
# write.csv(orthology_df, file = "input_data//mouse_hamster_orthology_29May23.csv", row.names = FALSE)
```

<h3>Compare how orthologous genes differ between stages for which stages they are induced in</h3>
```{r induced-orthology}
#establishes comparable sets of orthologous induced genes
orthology_df <- read.csv("input_data/mouse_hamster_orthology_29May23.csv")

#create hamster stage dataframes with only genes that have mouse orthologs
#initial set up
ham_SP_induced <-  data.frame(hGene = SP.induced)
ham_LZ_induced <-  data.frame(hGene = LZ.induced)
ham_DIP_induced <- data.frame(hGene = DIP.induced)
ham_RS_induced <-  data.frame(hGene = RS.induced)

#renaming based on orthology dictionary
ham_SP_induced_conv <- data.frame(hGene = ham_SP_induced, 
                                  mGene = recode(ham_SP_induced$hGene, 
                                                 !!!setNames(as.character(orthology_df$mGene), orthology_df$hGene)))
ham_LZ_induced_conv <- data.frame(hGene = ham_LZ_induced, 
                                  mGene = recode(ham_LZ_induced$hGene, 
                                                 !!!setNames(as.character(orthology_df$mGene), orthology_df$hGene)))
ham_DIP_induced_conv <- data.frame(hGene = ham_DIP_induced, 
                                   mGene = recode(ham_DIP_induced$hGene, 
                                                  !!!setNames(as.character(orthology_df$mGene), orthology_df$hGene)))
ham_RS_induced_conv <- data.frame(hGene = ham_RS_induced, 
                                  mGene = recode(ham_RS_induced$hGene, 
                                                 !!!setNames(as.character(orthology_df$mGene), orthology_df$hGene)))

#get rid of hamster records with no mouse ortholog
ham_SP_induced_conv <- ham_SP_induced_conv[grep("ENS", ham_SP_induced_conv$mGene),]
ham_LZ_induced_conv <- ham_LZ_induced_conv[grep("ENS", ham_LZ_induced_conv$mGene),]
ham_DIP_induced_conv <- ham_DIP_induced_conv[grep("ENS", ham_DIP_induced_conv$mGene),]
ham_RS_induced_conv <- ham_RS_induced_conv[grep("ENS", ham_RS_induced_conv$mGene),]

#read in stage dataframes from mouse data; these files are also generated by the mouse script in this github repository
mouse_SP_induced <-  read.table(file = paste("induced/mouse_SP_induced_inducedCutoff_", induced_cutoff,".txt", sep = ""))
mouse_LZ_induced <-  read.table(file = paste("induced/mouse_LZ_induced_inducedCutoff_", induced_cutoff,".txt", sep = ""))
mouse_DIP_induced <- read.table(file = paste("induced/mouse_DIP_induced_inducedCutoff_", induced_cutoff,".txt", sep = ""))
mouse_RS_induced <-  read.table(file = paste("induced/mouse_RS_induced_inducedCutoff_", induced_cutoff,".txt", sep = ""))

colnames(mouse_SP_induced) <- "mGene"
colnames(mouse_LZ_induced) <- "mGene"
colnames(mouse_DIP_induced) <- "mGene"
colnames(mouse_RS_induced) <- "mGene"

#set up stage dataframes wtih both species IDs
mouse_SP_induced_conv <- data.frame(mGene = mouse_SP_induced, 
                                    hGene = recode(mouse_SP_induced$mGene, 
                                                   !!!setNames(as.character(orthology_df$hGene), orthology_df$mGene)))
mouse_LZ_induced_conv <- data.frame(mGene = mouse_LZ_induced, 
                                    hGene = recode(mouse_LZ_induced$mGene, 
                                                   !!!setNames(as.character(orthology_df$hGene), orthology_df$mGene)))
mouse_DIP_induced_conv <- data.frame(mGene = mouse_DIP_induced, 
                                     hGene = recode(mouse_DIP_induced$mGene, 
                                                    !!!setNames(as.character(orthology_df$hGene), orthology_df$mGene)))
mouse_RS_induced_conv <- data.frame(mGene = mouse_RS_induced, 
                                    hGene = recode(mouse_RS_induced$mGene, 
                                                   !!!setNames(as.character(orthology_df$hGene), orthology_df$mGene)))

#get rid of mouse records with no hamster ortholog
mouse_SP_induced_conv <- mouse_SP_induced_conv[grep("Psun", mouse_SP_induced_conv$hGene),]
mouse_LZ_induced_conv <- mouse_LZ_induced_conv[grep("Psun", mouse_LZ_induced_conv$hGene),]
mouse_DIP_induced_conv <- mouse_DIP_induced_conv[grep("Psun", mouse_DIP_induced_conv$hGene),]
mouse_RS_induced_conv <- mouse_RS_induced_conv[grep("Psun", mouse_RS_induced_conv$hGene),]
```

<h3>Compare sets of overexpressed genes during diplotene</h3>
```{r diplotene-escapee-genes}
#threshold for determining parental and hybrid escapee genes; selecting for top 10% of genes then determining what rpkm threshold corresponds to that
threshold <- 0.90
hamster_parental_threshold <- quantile(rowMeans(rpkm_BBBB_DIP_Dev_justX.norm), probs = c(threshold))

#define two sets of genes:
#parental_escapees; the top 10% of genes expressed in parental diplotene (MSCI escapees)
hamster_parental_escapees <- names(rowMeans(rpkm_BBBB_DIP_Dev_justX.norm)[rowMeans(rpkm_BBBB_DIP_Dev_justX.norm) >
                                                                            hamster_parental_threshold])

#hybrid_escapees; the top 10% of genes expressed in hybrid diplotene (overexpressed X-linked genes)
hamster_hybrid_escapees <- names(rowMeans(rpkm_BBSS_DIP_Dev_justX.norm)[rowMeans(rpkm_BBSS_DIP_Dev_justX.norm) > hamster_parental_threshold])

#import parental_escapees and hybrid_escapees from mouse data
mouse_parental_escapees <- as.vector(read.table(file = paste("induced/mouse_parental_escapees_inducedCutoff_", 
                                                             induced_cutoff,".txt", sep = ""))$V1)
mouse_hybrid_escapees <- as.vector(read.table(file = paste("induced/mouse_hybrid_escapees_inducedCutoff_", 
                                                           induced_cutoff,".txt", sep = ""))$V1)

#number of genes that escape in common between both species
both_species_parental_escapees_overlap <- intersect(hamster_parental_escapees, 
                                                    recode(mouse_parental_escapees, !!!setNames(as.character(orthology_df$hGene),
                                                                                                orthology_df$mGene)))

#number of genes that overachieve in common between both species
both_species_hybrid_escapees_overlap <- intersect(hamster_hybrid_escapees, 
                                                  recode(hamster_hybrid_escapees, !!!setNames(as.character(orthology_df$hGene),
                                                                                              orthology_df$mGene)))

#make a dataframe of all of the mouse and hamster parental and hybrid escapees gene counts
both_species_escapee_df <- data.frame(species = rep(c("ham", "mouse", "both"), 2),
                                      type = c(rep("hybrid_escapees", 3), rep("parental_escapees", 3)),
                                      numGnees = c(length(hamster_hybrid_escapees), 
                                                   length(mouse_hybrid_escapees), 
                                                   length(both_species_hybrid_escapees_overlap), 
                                                   length(hamster_parental_escapees),
                                                   length(mouse_parental_escapees),
                                                   length(both_species_parental_escapees_overlap)))

#distribution of hybrid genes across stages for hamsters then mice
hamster_hybrid_escapees_df <- data.frame(stage = stages[1:4],
                                         species = "ham",
                                         overlap = c(length(intersect(hamster_hybrid_escapees, ham_SP_induced$hGene)),
                                                     length(intersect(hamster_hybrid_escapees, ham_LZ_induced$hGene)),
                                                     length(intersect(hamster_hybrid_escapees, ham_DIP_induced$hGene)),
                                                     length(intersect(hamster_hybrid_escapees, ham_RS_induced$hGene))),
                                         num_escapees = length(hamster_hybrid_escapees))

mouse_hybrid_escapees_df <- data.frame(stage = stages[1:4],
                                       species = "mouse",
                                       overlap = c(length(intersect(mouse_hybrid_escapees, mouse_SP_induced$mGene)),
                                                   length(intersect(mouse_hybrid_escapees, mouse_LZ_induced$mGene)),
                                                   length(intersect(mouse_hybrid_escapees, mouse_DIP_induced$mGene)),
                                                   length(intersect(mouse_hybrid_escapees, mouse_RS_induced$mGene))),
                                       num_escapees = length(mouse_hybrid_escapees))

#formatting
hamster_hybrid_escapees_df$stage <- factor(hamster_hybrid_escapees_df$stage, levels = stages[1:4])
hamster_hybrid_escapees_df$species <- factor(hamster_hybrid_escapees_df$species)
mouse_hybrid_escapees_df$stage <- factor(mouse_hybrid_escapees_df$stage, levels = stages[1:4])
mouse_hybrid_escapees_df$species <- factor(mouse_hybrid_escapees_df$species)

#alluvial plot showing how hybrid escapee genes are conserved/not conserved across spermatogenic stages between species
hamster_hybrid_escapees_plot <- ggplot(hamster_hybrid_escapees_df, aes(fill=stage, y=overlap, x=species)) + 
  geom_bar(position="stack", stat="identity") +
  ylab(label = "Number of genes") +
  scale_fill_manual(values = c(sp_color, lz_color, dip_color, rs_color)) +
  ggtitle("Induced stage identity of overexpressed X genes in hybrid hamsters") +
  theme_minimal() +
  theme(legend.position = "none",
        plot.title = element_text(hjust = 0.5))

#alluvial plot showing how hybrid escapee genes are conserved/not conserved across spermatogenic stages between species
mouse_hybrid_escapees_plot <- ggplot(mouse_hybrid_escapees_df, aes(fill=stage, y=overlap, x=species)) + 
  geom_bar(position="stack", stat="identity") +
  ylab(label = "Number of genes") +
  scale_fill_manual(values = c(sp_color, lz_color, dip_color, rs_color)) +
  ggtitle("Induced stage identity of overexpressed X genes in hybrid hamsters") +
  theme_minimal() +
  theme(legend.position = "none",
        plot.title = element_text(hjust = 0.5))

#combine both species dataframes into one
both_species_hybrid_escapees_df <- rbind(hamster_hybrid_escapees_df, mouse_hybrid_escapees_df)
both_species_hybrid_escapees_df$species <- factor(both_species_hybrid_escapees_df$species, levels = c("mouse", "ham"))

#bar plot comparing how hybrid escapee genes are conserved/not conserved across spermatogenic stages between species
both_species_hybrid_escapees_plot <- ggplot(both_species_hybrid_escapees_df, aes(fill=stage, y=overlap, x=species)) + 
  geom_bar(position="stack", stat="identity") +
  ylab(label = "Number of genes") +
  scale_fill_manual(values = c(sp_color, lz_color, dip_color, rs_color)) +
  ggtitle("Induced stage identity of overexpressed X genes in hybrid hamsters") +
  theme_minimal() +
  theme(legend.position = "none", 
        panel.grid.major.x = element_blank()) + 
  labs(y = "Cell affiliation of overexpressed X-linked hybrid diplotene genes") + 
  scale_x_discrete(labels = c("House mouse", "Dwarf hamster")) +
  theme(axis.title.x = element_blank())

if(exportFigs == FALSE) {
  print(both_species_hybrid_escapees_plot)
}

paste(mouse_hybrid_escapees_df$overlap, " ", mouse_hybrid_escapees_df$stage, " genes", sep = "")
paste(hamster_hybrid_escapees_df$overlap, " ", hamster_hybrid_escapees_df$stage, " genes", sep = "")

#distribution of parental escapee genes across stages for hamsters then mice
hamster_parental_escapees_df <- data.frame(stage = stages[1:4],
                                           species = "ham",
                                           overlap = c(length(intersect(hamster_parental_escapees, ham_SP_induced$hGene)),
                                                       length(intersect(hamster_parental_escapees, ham_LZ_induced$hGene)),
                                                       length(intersect(hamster_parental_escapees, ham_DIP_induced$hGene)),
                                                       length(intersect(hamster_parental_escapees, ham_RS_induced$hGene))),
                                           numparental_escapees = length(hamster_parental_escapees))

mouse_parental_escapees_df <- data.frame(stage = stages[1:4],
                                         species = "mouse",
                                         overlap = c(length(intersect(mouse_parental_escapees, mouse_SP_induced$mGene)),
                                                     length(intersect(mouse_parental_escapees, mouse_LZ_induced$mGene)),
                                                     length(intersect(mouse_parental_escapees, mouse_DIP_induced$mGene)),
                                                     length(intersect(mouse_parental_escapees, mouse_RS_induced$mGene))),
                                         numparental_escapees = length(mouse_parental_escapees))

#formatting
hamster_parental_escapees_df$stage <- factor(hamster_parental_escapees_df$stage, levels = stages[1:4])
hamster_parental_escapees_df$species <- factor(hamster_parental_escapees_df$species)
mouse_parental_escapees_df$stage <- factor(mouse_parental_escapees_df$stage, levels = stages[1:4])
mouse_parental_escapees_df$species <- factor(mouse_parental_escapees_df$species)

#alluvial plot showing how parental escapee genes are conserved/not conserved across spermatogenic stages between species
hamster_parental_escapees_plot <- ggplot(hamster_parental_escapees_df, aes(fill=stage, y=overlap, x=species)) + 
  geom_bar(position="stack", stat="identity") +
  ylab(label = "Number of genes") +
  scale_fill_manual(values = c(sp_color, lz_color, dip_color, rs_color)) +
  ggtitle("Induced stage identity of MSCI escapee genes in parental hamsters") +
  theme_minimal() +
  theme(legend.position = "none",
        plot.title = element_text(hjust = 0.5))

#alluvial plot showing how parental escapee genes are conserved/not conserved across spermatogenic stages between species
mouse_parental_escapees_plot <- ggplot(mouse_parental_escapees_df, aes(fill=stage, y=overlap, x=species)) + 
  geom_bar(position="stack", stat="identity") +
  ylab(label = "Number of genes") +
  scale_fill_manual(values = c(sp_color, lz_color, dip_color, rs_color)) +
  ggtitle("Induced stage identity of MSCI escapee genes in parental hamsters") +
  theme_minimal() +
  theme(legend.position = "none",
        plot.title = element_text(hjust = 0.5))


#combine both species dataframes into one
both_species_parental_escapees_df <- rbind(hamster_parental_escapees_df, mouse_parental_escapees_df)
both_species_parental_escapees_df$species <- factor(both_species_parental_escapees_df$species, levels = c("mouse", "ham"))

#bar plot comparing how parental escapee genes are conserved/not conserved across spermatogenic stages between species
both_species_parental_escapees_plot <- ggplot(both_species_parental_escapees_df, aes(fill=stage, y=overlap, x=species)) + 
  geom_bar(position="stack", stat="identity") +
  ylab(label = "Number of genes") +
  scale_fill_manual(values = c(sp_color, lz_color, dip_color, rs_color)) +
  ggtitle("Induced stage identity of escaped MSCI X-linked genes in parental hamsters") +
  theme_minimal() +
  theme(legend.position = "none", 
        panel.grid.major.x = element_blank()) + 
  labs(y = "Cell affiliation of overexpressed X-linked parental diplotene genes") + 
  scale_x_discrete(labels = c("House mouse", "Dwarf hamster")) +
  theme(axis.title.x = element_blank())

if(exportFigs == FALSE) {
  print(both_species_parental_escapees_plot)
}

paste(mouse_parental_escapees_df$overlap, " ", mouse_parental_escapees_df$stage, " genes", sep = "")
paste(hamster_parental_escapees_df$overlap, " ", hamster_parental_escapees_df$stage, " genes", sep = "")
```

<h3>Plot all samples in a dendrogram based on experssion values</h3>
```{r dendrogram}
#make dendrograms of samples
#choose normalization type
#which_norm <- "norm"
which_norm <- "log"

if(exportFigs == TRUE) {
  pdf(file = paste("exported_figures/sample_dendrograms_", which_norm,"RPKM.pdf", sep = ""), width = 8.5, height = 11)
}

par(mfrow = c(3,1), mar = c(0, 3, 1.5, 0))
#clustering samples by all genes
DevTreeCluster_allGenes = hclust(dist(t(get(paste("rpkm_Dev.", which_norm, sep = "")))), method = "average");
DevTreeCluster_allGenes_dend <- hang.dendrogram(as.dendrogram(DevTreeCluster_allGenes))

dend_colors <- gsub("SP", sp_color, gsub("LZ", lz_color, gsub("DIP", dip_color, gsub("RS", rs_color, gsub(".*_", "", DevTreeCluster_allGenes_dend %>% labels)))))
dend_shapes <- as.numeric(gsub("BBSS", 4, gsub("BBBB", 24, gsub("SSSS", 25, gsub("_.*", "", DevTreeCluster_allGenes_dend %>% labels)))))

DevTreeCluster_allGenes_dend %>% 
  set("leaves_pch", dend_shapes) %>% 
  set("leaves_col", dend_colors) %>% 
  set("leaves_bg", dend_colors) %>% 
  set("leaves_cex", c(2)) %>% 
  set("labels", "") %>% 
  plot(main = "All sample dendrogram based on all genes")


#clustering samples by all autosomal genes
DevTreeCluster_allAuto = hclust(dist(t(get(paste("rpkm_Dev_allAuto.", which_norm, sep = "")))), method = "average");
DevTreeCluster_allAuto_dend <- hang.dendrogram(as.dendrogram(DevTreeCluster_allAuto))

#set colors
dend_colors <- gsub("SP", sp_color, gsub("LZ", lz_color, gsub("DIP", dip_color, gsub("RS", rs_color, gsub(".*_", "", DevTreeCluster_allAuto_dend %>% labels)))))
dend_shapes <- as.numeric(gsub("BBSS", 4, gsub("BBBB", 24, gsub("SSSS", 25, gsub("_.*", "", DevTreeCluster_allAuto_dend %>% labels)))))

#plot dendrogram for autosomal genes
suppressWarnings(DevTreeCluster_allAuto_dend %>% 
                   set("leaves_pch", dend_shapes) %>% 
                   set("leaves_col", dend_colors) %>% 
                   set("leaves_bg", dend_colors) %>% 
                   set("leaves_cex", c(2)) %>% 
                   set("labels", "") %>% 
                   plot(main = "All sample dendrogram based on all Auto genes"))

#clustering samples by X genes
DevTreeCluster_justX = hclust(dist(t(get(paste("rpkm_Dev_justX.", which_norm, sep = "")))), method = "average");
DevTreeCluster_justX_dend <- hang.dendrogram(as.dendrogram(DevTreeCluster_justX))

#set colors
dend_colors <- gsub("SP", sp_color, gsub("LZ", lz_color, gsub("DIP", dip_color, gsub("RS", rs_color, gsub(".*_", "", DevTreeCluster_justX_dend %>% labels)))))
dend_shapes <- as.numeric(gsub("BBSS", 4, gsub("BBBB", 24, gsub("SSSS", 25, gsub("_.*", "", DevTreeCluster_justX_dend %>% labels)))))

#plot dendrogram for autosomal genes
suppressWarnings(DevTreeCluster_justX_dend %>% 
                   set("leaves_pch", dend_shapes) %>% 
                   set("leaves_col", dend_colors) %>% 
                   set("leaves_bg", dend_colors) %>% 
                   set("leaves_cex", c(2)) %>% 
                   set("labels", "") %>% 
                   plot(main = "All sample dendrogram based on all X genes"))

if(exportFigs == TRUE) {
  dev.off()
}
```

<h3>MDS Plots of each dataset and stage</h3>
```{r mds-plots-multi-pane-figure-by-stage}
#set MDS plotting parameters
c_value <- 2
l_value <- 1
#number of genes to use in MDS analysis
t_value <- 500

#set colors
dev_stage_colors <- color_list[grep("WT",metadata$stage, invert = TRUE, value = TRUE)]

#function that adjusts MDS plot labels to be more readable
mds_labels <- function(dge_of_interest) {
  label_vector <- colnames(dge_of_interest$counts)
  label_vector <- gsub("_.*","",label_vector)
  label_vector <- gsub("SSSS","Psun",label_vector)
  label_vector <- gsub("BBBB","Pcam",label_vector)
  label_vector <- gsub("BBSS","hybrid",label_vector)
  return(label_vector)
}

#make an MDS plot showing minimal batch effects between library types
china_samples <- c("BBBB_198-7M_SP", "BBSS_90-2M_SP", "BBSS_95-3M_DIP", "BBSS_95-3M_LZ", "SSSS_188-4M_RS", "SSSS_188-5M_SP")

#set point shapes to be used in MDS plot
shapes_by_cross_and_library <- as.numeric(gsub("hybrid", 4, gsub("Pcam", 2, gsub("Psun", 6, mds_labels(dgeDev)))))
shapes_by_cross_and_library[intersect(which(rownames(dgeDev$samples) %in% china_samples),which(mds_labels(dgeDev) == "Pcam"))] <- 24
shapes_by_cross_and_library[intersect(which(rownames(dgeDev$samples) %in% china_samples),which(mds_labels(dgeDev) == "Psun"))] <- 25
shapes_by_cross_and_library[intersect(which(rownames(dgeDev$samples) %in% china_samples),which(mds_labels(dgeDev) == "hybrid"))] <- 7

#MDS plot of all Dev samples colored by library preparation
if(exportFigs == TRUE) {
  pdf(file = "exported_figures/MDS_dev_samples_colored_by_library_prep.pdf", width=5, height=4, pointsize=12)
  plotMDS(dgeDev, top = t_value, lwd = l_value, col = dev_stage_colors, bg = dev_stage_colors, pch = shapes_by_cross_and_library, cex = c_value*0.7, main = "All samples by library preparation", plot = TRUE)
  dev.off()
} else {
  plotMDS(dgeDev, top = t_value, lwd = l_value, col = dev_stage_colors, bg = dev_stage_colors, pch = shapes_by_cross_and_library, cex = c_value*0.7, main = "All samples by library preparation", plot = TRUE)
}

#grid of MDS plots of SP, LZ, DIP, and RS samples
if (exportFigs == TRUE) {
  pdf(file = paste("exported_figures/MDS_plots_by_stage_outlines_cex", c_value, "_top", t_value, ".pdf", sep = ""), width = 6.5, height = 5, pointsize = 12)
}
layout(matrix(c(1,2,3,4), ncol = 2, byrow = TRUE))

#SP MDS
par(mar=c(3, 3, 2, 1), mgp=c(1.5, 0.5, 0))
metadata$stage <- factor(metadata$stage, levels = stages)
stage_colors <- color_list[metadata$stage]
plotMDS(dgeSP, top = t_value, lwd = l_value, col = color_list["SP"], pch = as.numeric(gsub("hybrid", 4, gsub("Pcam", 2, gsub("Psun", 6, mds_labels(dgeSP))))), cex = c_value, main = "Mitosis (SP)", plot = TRUE, cex.lab = 0.75, cex.main = 1, cex.axis = 0.75)

#LZ MDS
plotMDS(dgeLZ, top = t_value, lwd = l_value, col = color_list["LZ"], pch = as.numeric(gsub("hybrid", 4, gsub("Pcam", 2, gsub("Psun", 6, mds_labels(dgeLZ))))), cex = c_value, main = "Meiosis-Before X-Inact. (LZ)", plot = TRUE, cex.lab = 0.75, cex.main = 1, cex.axis = 0.75)

#DIP MDS
plotMDS(dgeDIP, top = t_value, lwd = l_value, col = color_list["DIP"], pch = as.numeric(gsub("hybrid", 4, gsub("Pcam", 2, gsub("Psun", 6, mds_labels(dgeDIP))))), cex = c_value, main = "Meiosis-After X-Inact. (DIP)", plot = TRUE, cex.lab = 0.75, cex.main = 1, cex.axis = 0.75)

#RS MDS
plotMDS(dgeRS, top = t_value, lwd = l_value, col = color_list["RS"], pch = as.numeric(gsub("hybrid", 4, gsub("Pcam", 2, gsub("Psun", 6, mds_labels(dgeRS))))), cex = c_value, main = "Postmeiosis (RS)", plot = TRUE, cex.lab = 0.75, cex.main = 1, cex.axis = 0.75)

if (exportFigs == TRUE) {
  dev.off()
}

rm(dgeSP, dgeLZ, dgeDIP, dgeRS)
rm(stage_colors)
```

<h3>X Chromosome MDSPlots</h3>
```{r mds-plots-of-X-and-Auto}
#MDS plot of X-linked expression for all samples
#set MDS plotting parameters
c_value <- 2
l_value <- 1
#number of genes to use in MDS analysis
t_value <- 500

#make grid of X-linked and Autosomal gene MDS plots for dwarf hamster
#X-linked
if(exportFigs == TRUE) {
  pdf(file = "exported_figures/MDS_X-linked_genes_colored_by_genotype_with_symbols.pdf", width=5, height=4, pointsize=12)
}

plotMDS(dgeX, top = t_value, lwd =2, 
        col= color_list[gsub(".*-", "", dgeX$samples$group)], 
        pch = as.numeric(gsub("hybrid", 4, gsub("Pcam", 2, gsub("Psun", 6, mds_labels(dgeX))))), 
        cex = c_value, plot = TRUE)
title("X MDS - hamster")

if(exportFigs == TRUE) {
  dev.off()
}

#Auto
if(exportFigs == TRUE) {
  pdf(file = "exported_figures/MDS_Autosomal_genes_colored_by_genotype_with_symbols.pdf", width=5, height=4, pointsize=12)
}
plotMDS(dgeAuto, top = t_value, lwd =2, 
        col= color_list[gsub(".*-", "", dgeAuto$samples$group)], 
        pch = as.numeric(gsub("hybrid", 4, gsub("Pcam", 2, gsub("Psun", 6, mds_labels(dgeAuto))))), 
        cex = c_value, plot = TRUE)
title("Autosomal MDS - hamster")

if(exportFigs == TRUE) {
  dev.off()
}
```

```{r create-orthology-countfiles}
#make and normalize a DGEList of just Psun genes that have Mmus orthology
orthology_df <- read.csv("input_data/mouse_hamster_orthology_29May23.csv")

#MDS analysis of mus samples based only off of orthologous genes with hamster
dgeHam_orthology <- DGEList(counts=just_counts[rownames(just_counts) %in% orthology_df$hGene,
                                               grep("WT", colnames(just_counts), value = FALSE, invert = TRUE)], 
                            genes=annotations[rownames(annotations) %in% orthology_df$hGene,], 
                            group=grep("WT", colnames(just_counts), value = TRUE, invert = TRUE)) 
dgeHam_orthology <- calcNormFactors(dgeHam_orthology)
plotMDS(dgeHam_orthology)

#cpm normalized and log transformed expression dataset
ham_orthology.cpm <- cpm(dgeHam_orthology, log=TRUE, prior.count=1, normalized.lib.sizes=TRUE) 
ham_orthology.rpkm <- as.data.frame(rpkm(dgeHam_orthology,  normalized.lib.sizes = TRUE, log = TRUE))

write.csv(ham_orthology.cpm, file = "exported_files/ham_orthology_cpm_norm_counts.csv")
write.csv(ham_orthology.rpkm, file = "exported_files/ham_orthology_rpkm_norm_counts.csv")
```

<h3>Heatmaps of Log(Normalized(RPKM)) for the autosomes and X chromosome from all data</h3>
```{r heatmaps-of-x-genes}
#heatmap of normalized rpkm values for x-linked genes from all stages; un-normalized rpkm is not useful for visualizing differences
#choose colors
colfunc <- colorRampPalette(c("#FFFFFF", "#2F2323"))
gap_value <- 4
total_number <- 255 + (15*gap_value)
browns_mod <- colfunc(total_number)
browns_mod <- browns_mod[c(seq(1,1+(15*gap_value),gap_value),seq(2+(15*gap_value),total_number,1))]

#auto heatmap; uncomment if desired
if(exportFigs == TRUE) {
  pdf(file="exported_figures/Heatmap_hamster_allAuto_all-genotypes_all_stages.pdf", width=10, height=8, pointsize=12)
}
Heatmap(as.matrix(rpkm_Dev_allAuto.norm), name = "Autosomal expression (rpkm)", column_order = Dev_heatmap_order, column_names_gp = gpar(fontsize=4), show_row_names = FALSE, col = browns_mod, show_row_dend = FALSE, cluster_columns = FALSE, use_raster = FALSE)
if(exportFigs == TRUE) {
  dev.off()
}

#x-linked heatmap with genes ordered by clustering
if(exportFigs == TRUE) {
  pdf(file="exported_figures/Heatmap_hamster_justX_all-genotypes_all_stages.pdf", width=10, height=8, pointsize=12)
}
Heatmap(as.matrix(rpkm_Dev_justX.norm), name = "X-linked expression (rpkm)", column_order = Dev_heatmap_order, column_names_gp = gpar(fontsize=4), show_row_names = FALSE, col = browns_mod, show_row_dend = FALSE, cluster_columns = FALSE)
if(exportFigs == TRUE) {
  dev.off()
}
```

<h3>Determine which parental stages hybrid diplotene samples are most similar to</h3>
```{r which-stage-does-hybDIP-correlate-to}
#the goal of this section is to see which parental stages the hybrid diplotene samples have the highest expression correlations to; then to see how those parental stage correlations differ between house mice and dwarf hamsters
#set up data frames of X-linked correlations between hybDIP and every BBBB stage
hybDIP_BBBB_SP_X_df <- data.frame(hyb = rowMeans(rpkm_BBSS_DIP_Dev_justX.norm), par = rowMeans(rpkm_BBBB_SP_Dev_justX.norm))
hybDIP_BBBB_LZ_X_df <- data.frame(hyb = rowMeans(rpkm_BBSS_DIP_Dev_justX.norm), par = rowMeans(rpkm_BBBB_LZ_Dev_justX.norm))
hybDIP_BBBB_DIP_X_df <- data.frame(hyb = rowMeans(rpkm_BBSS_DIP_Dev_justX.norm), par = rowMeans(rpkm_BBBB_DIP_Dev_justX.norm))
hybDIP_BBBB_RS_X_df <- data.frame(hyb = rowMeans(rpkm_BBSS_DIP_Dev_justX.norm), par = rowMeans(rpkm_BBBB_RS_Dev_justX.norm))

#calculate rho of X-linked correlations
hyb_v_BBBB_SP_X_corr <- cor.test(hybDIP_BBBB_SP_X_df$hyb, hybDIP_BBBB_SP_X_df$par)
hyb_v_BBBB_LZ_X_corr <- cor.test(hybDIP_BBBB_LZ_X_df$hyb, hybDIP_BBBB_LZ_X_df$par)
hyb_v_BBBB_DIP_X_corr <- cor.test(hybDIP_BBBB_DIP_X_df$hyb, hybDIP_BBBB_DIP_X_df$par)
hyb_v_BBBB_RS_X_corr <- cor.test(hybDIP_BBBB_RS_X_df$hyb, hybDIP_BBBB_RS_X_df$par)

#set up data frames of autosomal correlations between hybDIP and every BBBB stage
hybDIP_BBBB_SP_Auto_df <- data.frame(hyb = rowMeans(rpkm_BBSS_DIP_Dev_allAuto.norm), par = rowMeans(rpkm_BBBB_SP_Dev_allAuto.norm))
hybDIP_BBBB_LZ_Auto_df <- data.frame(hyb = rowMeans(rpkm_BBSS_DIP_Dev_allAuto.norm), par = rowMeans(rpkm_BBBB_LZ_Dev_allAuto.norm))
hybDIP_BBBB_DIP_Auto_df <- data.frame(hyb = rowMeans(rpkm_BBSS_DIP_Dev_allAuto.norm), par = rowMeans(rpkm_BBBB_DIP_Dev_allAuto.norm))
hybDIP_BBBB_RS_Auto_df <- data.frame(hyb = rowMeans(rpkm_BBSS_DIP_Dev_allAuto.norm), par = rowMeans(rpkm_BBBB_RS_Dev_allAuto.norm))

#calculate rho of autosomal correlations
hyb_v_BBBB_SP_Auto_corr <- cor.test(hybDIP_BBBB_SP_Auto_df$hyb, hybDIP_BBBB_SP_Auto_df$par)
hyb_v_BBBB_LZ_Auto_corr <- cor.test(hybDIP_BBBB_LZ_Auto_df$hyb, hybDIP_BBBB_LZ_Auto_df$par)
hyb_v_BBBB_DIP_Auto_corr <- cor.test(hybDIP_BBBB_DIP_Auto_df$hyb, hybDIP_BBBB_DIP_Auto_df$par)
hyb_v_BBBB_RS_Auto_corr <- cor.test(hybDIP_BBBB_RS_Auto_df$hyb, hybDIP_BBBB_RS_Auto_df$par)

#bootstrap correlation analysis; randomly sample expression data frames 1000 with replacement and rerun Pearson's correlation analysis
reps <- 1000
hybDIP_BBBB_stages_rho_bs <- data.frame(matrix(ncol = 3, nrow = 0))

for (i in 1:reps){
  #X-linked bootstrapping
  #generate SP_X bootstrap
  boot_SP_X.data <- hybDIP_BBBB_SP_X_df[sample(rownames(hybDIP_BBBB_SP_X_df), dim(hybDIP_BBBB_SP_X_df)[1], replace = TRUE), ]
  hybDIP_BBBB_SP_X.test <- cor.test(boot_SP_X.data$hyb, boot_SP_X.data$par, method = "pearson")
  
  #generate LZ_X bootstrap
  boot_LZ_X.data <- hybDIP_BBBB_LZ_X_df[sample(rownames(hybDIP_BBBB_LZ_X_df), dim(hybDIP_BBBB_LZ_X_df)[1], replace = TRUE), ]
  hybDIP_BBBB_LZ_X.test <- cor.test(boot_LZ_X.data$hyb, boot_LZ_X.data$par, method = "pearson")
  
  #generate DIP_X bootstrap
  boot_DIP_X.data <- hybDIP_BBBB_DIP_X_df[sample(rownames(hybDIP_BBBB_DIP_X_df), dim(hybDIP_BBBB_DIP_X_df)[1], replace = TRUE), ]
  hybDIP_BBBB_DIP_X.test <- cor.test(boot_DIP_X.data$hyb, boot_DIP_X.data$par, method = "pearson")
  
  #generate RS_X bootstrap
  boot_RS_X.data <- hybDIP_BBBB_RS_X_df[sample(rownames(hybDIP_BBBB_RS_X_df), dim(hybDIP_BBBB_RS_X_df)[1], replace = TRUE), ]
  hybDIP_BBBB_RS_X.test <- cor.test(boot_RS_X.data$hyb, boot_RS_X.data$par, method = "pearson")
  
  #Autosomal bootstrapping
  #generate SP_Auto bootstrap
  boot_SP_Auto.data <- hybDIP_BBBB_SP_Auto_df[sample(rownames(hybDIP_BBBB_SP_Auto_df), dim(hybDIP_BBBB_SP_Auto_df)[1], replace = TRUE), ]
  hybDIP_BBBB_SP_Auto.test <- cor.test(boot_SP_Auto.data$hyb, boot_SP_Auto.data$par, method = "pearson")
  
  #generate LZ_Auto bootstrap
  boot_LZ_Auto.data <- hybDIP_BBBB_LZ_Auto_df[sample(rownames(hybDIP_BBBB_LZ_Auto_df), dim(hybDIP_BBBB_LZ_Auto_df)[1], replace = TRUE), ]
  hybDIP_BBBB_LZ_Auto.test <- cor.test(boot_LZ_Auto.data$hyb, boot_LZ_Auto.data$par, method = "pearson")
  
  #generate DIP_Auto bootstrap
  boot_DIP_Auto.data <- hybDIP_BBBB_DIP_Auto_df[sample(rownames(hybDIP_BBBB_DIP_Auto_df), dim(hybDIP_BBBB_DIP_Auto_df)[1], replace = TRUE), ]
  hybDIP_BBBB_DIP_Auto.test <- cor.test(boot_DIP_Auto.data$hyb, boot_DIP_Auto.data$par, method = "pearson")
  
  #generate RS_Auto bootstrap
  boot_RS_Auto.data <- hybDIP_BBBB_RS_Auto_df[sample(rownames(hybDIP_BBBB_RS_Auto_df), dim(hybDIP_BBBB_RS_Auto_df)[1], replace = TRUE), ]
  hybDIP_BBBB_RS_Auto.test <- cor.test(boot_RS_Auto.data$hyb, boot_RS_Auto.data$par, method = "pearson")
  
  #add this round of bootstraps to the bootstraps dataframe
  #add SP_X
  hybDIP_BBBB_stages_rho_bs <- rbind(hybDIP_BBBB_stages_rho_bs, 
                                     c("SP", "X", hybDIP_BBBB_SP_X.test$estimate))
  #add LZ_X
  hybDIP_BBBB_stages_rho_bs <- rbind(hybDIP_BBBB_stages_rho_bs, 
                                     c("LZ", "X", hybDIP_BBBB_LZ_X.test$estimate))
  #add DIP_X
  hybDIP_BBBB_stages_rho_bs <- rbind(hybDIP_BBBB_stages_rho_bs, 
                                     c("DIP", "X", hybDIP_BBBB_DIP_X.test$estimate))
  #add RS_X
  hybDIP_BBBB_stages_rho_bs <- rbind(hybDIP_BBBB_stages_rho_bs, 
                                     c("RS", "X", hybDIP_BBBB_RS_X.test$estimate))
  #add SP_Auto
  hybDIP_BBBB_stages_rho_bs <- rbind(hybDIP_BBBB_stages_rho_bs, 
                                     c("SP", "Auto", hybDIP_BBBB_SP_Auto.test$estimate))
  #add LZ_Auto
  hybDIP_BBBB_stages_rho_bs <- rbind(hybDIP_BBBB_stages_rho_bs, 
                                     c("LZ", "Auto", hybDIP_BBBB_LZ_Auto.test$estimate))
  #add DIP_Auto
  hybDIP_BBBB_stages_rho_bs <- rbind(hybDIP_BBBB_stages_rho_bs, 
                                     c("DIP", "Auto", hybDIP_BBBB_DIP_Auto.test$estimate))
  #add RS_Auto
  hybDIP_BBBB_stages_rho_bs <- rbind(hybDIP_BBBB_stages_rho_bs, 
                                     c("RS", "Auto", hybDIP_BBBB_RS_Auto.test$estimate))
  
}

#formatting the rho bootstrapping dataframes
colnames(hybDIP_BBBB_stages_rho_bs) <- c("stage", "X_cat" ,"rho")
hybDIP_BBBB_stages_rho_bs$stage <- factor(hybDIP_BBBB_stages_rho_bs$stage, levels = c("SP", "LZ", "DIP", "RS"))
hybDIP_BBBB_stages_rho_bs$X_cat <- factor(hybDIP_BBBB_stages_rho_bs$X_cat, levels = c("Auto", "X"))
hybDIP_BBBB_stages_rho_bs$rho <- as.numeric(hybDIP_BBBB_stages_rho_bs$rho)

#calculate 95% CIs on bootstrapped rhos
#SP_X
hybDIP_BBBB_SP_X.ci <- round(quantile(hybDIP_BBBB_stages_rho_bs[hybDIP_BBBB_stages_rho_bs$stage == "SP" & hybDIP_BBBB_stages_rho_bs$X_cat == "X",]$rho, probs = c(0.025, 0.975)),4)

#LZ_X
hybDIP_BBBB_LZ_X.ci <- round(quantile(hybDIP_BBBB_stages_rho_bs[hybDIP_BBBB_stages_rho_bs$stage == "LZ" & hybDIP_BBBB_stages_rho_bs$X_cat == "X",]$rho, probs = c(0.025, 0.975)),4)

#DIP_X
hybDIP_BBBB_DIP_X.ci <- round(quantile(hybDIP_BBBB_stages_rho_bs[hybDIP_BBBB_stages_rho_bs$stage == "DIP" & hybDIP_BBBB_stages_rho_bs$X_cat == "X",]$rho, probs = c(0.025, 0.975)),4)

#RS_X
hybDIP_BBBB_RS_X.ci <- round(quantile(hybDIP_BBBB_stages_rho_bs[hybDIP_BBBB_stages_rho_bs$stage == "RS" & hybDIP_BBBB_stages_rho_bs$X_cat == "X",]$rho, probs = c(0.025, 0.975)),4)

#SP_Auto
hybDIP_BBBB_SP_Auto.ci <- round(quantile(hybDIP_BBBB_stages_rho_bs[hybDIP_BBBB_stages_rho_bs$stage == "SP" & hybDIP_BBBB_stages_rho_bs$X_cat == "Auto",]$rho, probs = c(0.025, 0.975)),4)

#LZ_Auto
hybDIP_BBBB_LZ_Auto.ci <- round(quantile(hybDIP_BBBB_stages_rho_bs[hybDIP_BBBB_stages_rho_bs$stage == "LZ" & hybDIP_BBBB_stages_rho_bs$X_cat == "Auto",]$rho, probs = c(0.025, 0.975)),4)

#DIP_Auto
hybDIP_BBBB_DIP_Auto.ci <- round(quantile(hybDIP_BBBB_stages_rho_bs[hybDIP_BBBB_stages_rho_bs$stage == "DIP" & hybDIP_BBBB_stages_rho_bs$X_cat == "Auto",]$rho, probs = c(0.025, 0.975)),4)

#RS_Auto
hybDIP_BBBB_RS_Auto.ci <- round(quantile(hybDIP_BBBB_stages_rho_bs[hybDIP_BBBB_stages_rho_bs$stage == "RS" & hybDIP_BBBB_stages_rho_bs$X_cat == "Auto",]$rho, probs = c(0.025, 0.975)),4)


#collate all correlation data then fdr correct actual (non-bootstrap) X-linked correlation rhos
hybDIP_corr_table_ham <- data.frame(stage = c("SP", "LZ", "DIP", "RS",
                                              "SP", "LZ", "DIP", "RS"),
                                    
                                    pVals = c(hyb_v_BBBB_SP_X_corr$p.value, 
                                              hyb_v_BBBB_LZ_X_corr$p.value, 
                                              hyb_v_BBBB_DIP_X_corr$p.value, 
                                              hyb_v_BBBB_RS_X_corr$p.value,
                                              hyb_v_BBBB_SP_Auto_corr$p.value, 
                                              hyb_v_BBBB_LZ_Auto_corr$p.value, 
                                              hyb_v_BBBB_DIP_Auto_corr$p.value, 
                                              hyb_v_BBBB_RS_Auto_corr$p.value),
                                    
                                    corVals = c(hyb_v_BBBB_SP_X_corr$estimate,
                                                hyb_v_BBBB_LZ_X_corr$estimate,
                                                hyb_v_BBBB_DIP_X_corr$estimate,
                                                hyb_v_BBBB_RS_X_corr$estimate,
                                                hyb_v_BBBB_SP_Auto_corr$estimate,
                                                hyb_v_BBBB_LZ_Auto_corr$estimate,
                                                hyb_v_BBBB_DIP_Auto_corr$estimate,
                                                hyb_v_BBBB_RS_Auto_corr$estimate),
                                    
                                    comp = c("Ham-X",
                                             "Ham-X",
                                             "Ham-X",
                                             "Ham-X",
                                             "Ham-Auto",
                                             "Ham-Auto",
                                             "Ham-Auto",
                                             "Ham-Auto"),
                                    
                                    species = c("Ham",
                                                "Ham",
                                                "Ham",
                                                "Ham",
                                                "Ham",
                                                "Ham",
                                                "Ham",
                                                "Ham"),
                                    
                                    CI_lower = c(hybDIP_BBBB_SP_X.ci[1],
                                                 hybDIP_BBBB_LZ_X.ci[1],
                                                 hybDIP_BBBB_DIP_X.ci[1],
                                                 hybDIP_BBBB_RS_X.ci[1],
                                                 hybDIP_BBBB_SP_Auto.ci[1],
                                                 hybDIP_BBBB_LZ_Auto.ci[1],
                                                 hybDIP_BBBB_DIP_Auto.ci[1],
                                                 hybDIP_BBBB_RS_Auto.ci[1]),
                                    
                                    CI_upper = c(hybDIP_BBBB_SP_X.ci[2],
                                                 hybDIP_BBBB_LZ_X.ci[2],
                                                 hybDIP_BBBB_DIP_X.ci[2],
                                                 hybDIP_BBBB_RS_X.ci[2],
                                                 hybDIP_BBBB_SP_Auto.ci[2],
                                                 hybDIP_BBBB_LZ_Auto.ci[2],
                                                 hybDIP_BBBB_DIP_Auto.ci[2],
                                                 hybDIP_BBBB_RS_Auto.ci[2])
)

#formatting data frame
hybDIP_corr_table_ham$stage   <- factor(hybDIP_corr_table_ham$stage, levels = c("SP", "LZ", "DIP", "RS"))
hybDIP_corr_table_ham$comp    <- factor(hybDIP_corr_table_ham$comp, levels = c("Ham-Auto", "Ham-X"))
hybDIP_corr_table_ham$comb_comp <- paste(hybDIP_corr_table_ham$comp, hybDIP_corr_table_ham$stage, sep = "-")
hybDIP_corr_table_ham$comb_comp    <- factor(hybDIP_corr_table_ham$comb_comp, levels = c("Ham-Auto-SP", "Ham-X-SP",
                                                                                         "Ham-Auto-LZ", "Ham-X-LZ",
                                                                                         "Ham-Auto-DIP", "Ham-X-DIP",
                                                                                         "Ham-Auto-RS", "Ham-X-RS"))
hybDIP_corr_table_ham$species <- factor(hybDIP_corr_table_ham$species, levels = c("Mouse", "Ham"))
#the FDR correction
hybDIP_corr_table_ham$pAdj    <- p.adjust(hybDIP_corr_table_ham$pVals, method = "fdr")

#add column indicating if correlation is significant for use in plot coloring
hybDIP_corr_table_ham$sig <- ""
if(sum(hybDIP_corr_table_ham$pAdj < 0.05) > 0){
  hybDIP_corr_table_ham[hybDIP_corr_table_ham$pAdj < 0.05,]$sig <- "yes"
}
if(sum(hybDIP_corr_table_ham$pAdj > 0.05) > 0){
  hybDIP_corr_table_ham[hybDIP_corr_table_ham$pAdj > 0.05,]$sig <- "no"
}

#print summary of hamster X-linked rho stats
print(paste("hybDIP_BBBB_SP_X rho X: ", round(hyb_v_BBBB_SP_X_corr$estimate, 4), "; p = ",
            round(hybDIP_corr_table_ham$pAdj[1], 4), 
            sep = ""))
print(paste("hybDIP_BBBB_LZ_X rho X: ", round(hyb_v_BBBB_LZ_X_corr$estimate, 4), "; p = ",
            round(hybDIP_corr_table_ham$pAdj[2], 3), 
            sep = ""))
print(paste("hybDIP_BBBB_RS_X rho X: ", round(hyb_v_BBBB_RS_X_corr$estimate, 4), "; p = ",
            round(hybDIP_corr_table_ham$pAdj[4], 3), 
            sep = ""))

#print summary of hamster autosomal rho stats
print(paste("hybDIP_BBBB_SP_Auto rho Auto: ", round(hyb_v_BBBB_SP_Auto_corr$estimate, 4), "; p = ",
            round(hybDIP_corr_table_ham$pAdj[5], 3), 
            sep = ""))
print(paste("hybDIP_BBBB_LZ_Auto rho Auto: ", round(hyb_v_BBBB_LZ_Auto_corr$estimate, 4), "; p = ",
            round(hybDIP_corr_table_ham$pAdj[6], 3), 
            sep = ""))
print(paste("hybDIP_BBBB_RS_Auto rho Auto: ", round(hyb_v_BBBB_RS_Auto_corr$estimate, 4), "; p = ",
            round(hybDIP_corr_table_ham$pAdj[8], 3), 
            sep = ""))


#read in mouse correlations and bootstraps then combine with hamsters
hybDIP_corr_table_mouse <- read.csv(file = "exported_files/hybDIP_correlation_mouse.csv", header = TRUE, stringsAsFactors = TRUE)[,-1]
hybDIP_corr_table_mouse$stage <- factor(hybDIP_corr_table_mouse$stage, levels = c("SP", "LZ", "DIP", "RS"))
hybDIP_corr_table_mouse$comb_comp <- paste(hybDIP_corr_table_mouse$comp, hybDIP_corr_table_mouse$stage, sep = "-")
hybDIP_corr_table_mouse$comb_comp    <- factor(hybDIP_corr_table_mouse$comb_comp, levels = c("Mouse-Auto-SP", "Mouse-X-SP",
                                                                                             "Mouse-Auto-LZ", "Mouse-X-LZ",
                                                                                             "Mouse-Auto-DIP", "Mouse-X-DIP",
                                                                                             "Mouse-Auto-RS", "Mouse-X-RS"))

#plot correlations for dwarf hamster X-linked and Autosomal hybrid diplotene genes
hamster_correlation_plot <- ggplot(data = subset(hybDIP_corr_table_ham, hybDIP_corr_table_ham$stage != "DIP"), 
                                   aes(x = stage, y = corVals, col = stage)) +
  geom_point(size = 1) +
  theme_minimal() +
  theme(legend.position = "none",
        strip.background =element_rect(fill=gray_1),
        panel.grid.minor = element_blank()) +
  facet_wrap(vars(comp), ncol = 2, scales = "free") +
  geom_hline(yintercept = 0, linewidth = 0.5) +
  labs(y = "Pearson's correlation (rho)", x = "Parental Stage") + 
  ylim(-0.375, 0.375) +
  scale_color_manual(values = c(sp_color_dark, lz_color_dark, rs_color_dark)) +
  geom_errorbar(aes(ymin=CI_lower, ymax=CI_upper), width=.2,
                position=position_dodge(0.05))

#plot correlations for house mouse X-linked and Autosomal hybrid diplotene genes
mouse_correlation_plot <- ggplot(data = subset(hybDIP_corr_table_mouse, hybDIP_corr_table_mouse$stage != "DIP"), 
                                 aes(x = stage, y = corVals, col = stage)) +
  geom_point(size = 1) +
  theme_minimal() +
  theme(legend.position = "none",
        strip.background =element_rect(fill=gray_1),
        panel.grid.minor = element_blank()) +
  facet_wrap(vars(comp), ncol = 2, scales = "free") +
  geom_hline(yintercept = 0, linewidth = 0.5) +
  labs(y = "Pearson's correlation (rho)", x = "Parental Stage") + 
  ylim(-0.375, 0.375) +
  scale_color_manual(values = c(sp_color, lz_color, rs_color)) +
  geom_errorbar(aes(ymin=CI_lower, ymax=CI_upper), width=.2,
                position=position_dodge(0.05))

if(exportFigs == TRUE) {
  ggsave(filename = "exported_figures/hybDIP_correlations.pdf", plot_grid(mouse_correlation_plot, hamster_correlation_plot), width = 6, height = 8)
} else {
  plot_grid(mouse_correlation_plot, hamster_correlation_plot)
}
```
<h3>Cell Purity of sorted cells and whole testes</h3>
```{r purity}
#the goal of this analysis is to assess the relative purity of each of the parental sorted cell populations to confirm that FACS worked to isolate the right populations
# #import marker genes; here

#marker genes are primarily sourced from Raymond et al. 2000, Green et al. 2018, Hermann et al. 2018, and Murat et al. 2023).
gene.data <- read.table("input_data/curated_purity_genes_hamster.csv", header = TRUE, stringsAsFactors = FALSE, sep = ",")

marker_facet_df <- data.frame()

#plot marker gene expression for each condition
for (gene in 1:nrow(gene.data)) {
  focal <- gene.data[gene, 3]
  
  #BBBB
  fpkm <- as.numeric(rpkm_Dev_BBBB_SP[focal, ])
  condition <- gsub("_[0-9]+.-[0-9]M_","_",names(rpkm_Dev_BBBB_SP[focal, ]))
  
  fpkm <- append(fpkm, as.numeric(rpkm_Dev_BBBB_LZ[focal, ]))
  condition <- append(condition, gsub("_[0-9]+.-[0-9]M_","_",names(rpkm_Dev_BBBB_LZ[focal, ])))
  
  fpkm <- append(fpkm, as.numeric(rpkm_Dev_BBBB_DIP[focal, ]))
  condition <- append(condition, gsub("_[0-9]+.-[0-9]M_","_",names(rpkm_Dev_BBBB_DIP[focal, ])))
  
  fpkm <- append(fpkm, as.numeric(rpkm_Dev_BBBB_RS[focal, ]))
  condition <- append(condition, gsub("_[0-9]+.-[0-9]M_","_",names(rpkm_Dev_BBBB_RS[focal, ])))
  
  #SSSS
  fpkm <- append(fpkm, as.numeric(rpkm_Dev_SSSS_SP[focal, ]))
  condition <- append(condition, gsub("_[0-9]+.-[0-9]M_","_",names(rpkm_Dev_SSSS_SP[focal, ])))
  
  fpkm <- append(fpkm, as.numeric(rpkm_Dev_SSSS_LZ[focal, ]))
  condition <- append(condition, gsub("_[0-9]+.-[0-9]M_","_",names(rpkm_Dev_SSSS_LZ[focal, ])))
  
  fpkm <- append(fpkm, as.numeric(rpkm_Dev_SSSS_DIP[focal, ]))
  condition <- append(condition, gsub("_[0-9]+.-[0-9]M_","_",names(rpkm_Dev_SSSS_DIP[focal, ])))
  
  fpkm <- append(fpkm, as.numeric(rpkm_Dev_SSSS_RS[focal, ]))
  condition <- append(condition, gsub("_[0-9]+.-[0-9]M_","_",names(rpkm_Dev_SSSS_RS[focal, ])))
  
  #BBSS
  fpkm <- append(fpkm, as.numeric(rpkm_Dev_BBSS_SP[focal, ]))
  condition <- append(condition, gsub("_[0-9]+.-[0-9]M_","_",names(rpkm_Dev_BBSS_SP[focal, ])))
  
  fpkm <- append(fpkm, as.numeric(rpkm_Dev_BBSS_LZ[focal, ]))
  condition <- append(condition, gsub("_[0-9]+.-[0-9]M_","_",names(rpkm_Dev_BBSS_LZ[focal, ])))
  
  fpkm <- append(fpkm, as.numeric(rpkm_Dev_BBSS_DIP[focal, ]))
  condition <- append(condition, gsub("_[0-9]+.-[0-9]M_","_",names(rpkm_Dev_BBSS_DIP[focal, ])))
  
  rpkm_matrix <- data.frame(fpkm, condition)
  
  rpkm_matrix$condition <- factor(rpkm_matrix$condition, levels = c("BBBB_SP", "BBBB_LZ", "BBBB_DIP", "BBBB_RS", "SSSS_SP", "SSSS_LZ", "SSSS_DIP", "SSSS_RS", "BBSS_SP", "BBSS_LZ", "BBSS_DIP"))
  rpkm_matrix$gene <- rep(gene.data[gene, 1], dim(rpkm_matrix)[1])
  rpkm_matrix$stage <- as.factor(gene.data[gene, 2])
  marker_facet_df <- rbind(marker_facet_df, rpkm_matrix)
  
  #change exportFigs to TRUE to export as SVG if desired
  if(exportFigs == TRUE) {
    marker_alone_plot <- ggplot(rpkm_matrix[grep("BBSS", rpkm_matrix$condition, invert = TRUE),], aes(x = condition, y = fpkm, color = condition)) +
      stat_boxplot(geom ='errorbar') +
      geom_boxplot(outlier.colour = "black",outlier.shape = NA, notch = FALSE) +
      geom_quasirandom(aes(col = condition), alpha = 0.6, size = 3) +
      theme_classic() +
      scale_color_manual(values = rep(c(sp_color, lz_color, dip_color, rs_color), 2)) +
      theme(axis.title.x = element_blank(),
            axis.text.x = element_blank(),
            axis.ticks.x = element_blank(),
            axis.title.y = element_blank(),
            legend.position = "none",
            plot.title = element_text(hjust = 0.5)) +
      ggtitle(paste(gene.data[gene, 1], gene.data[gene, 2], sep = ":"))
    
    print(marker_alone_plot)
    
    
    fname <- paste("exported_figures/purity-test_", paste(gene.data[gene, 1], gene.data[gene,2], sep = "_"), ".pdf", sep = "")
    
    #saves boxplot of expression of a single marker gene across all parental samples/stages
    if(exportFigs == TRUE) {
      ggsave(filename = fname, plot = marker_alone_plot)
    }
  }
}

#just use the final set of marker genes for inclusion in the manuscript
purity_genes <- marker_facet_df[marker_facet_df$gene %in% c("Dmrt1", "Hells", 
                                                            "Ccnb1ip1", "Adad2",
                                                            "Aurka", "Tank",
                                                            "Cabyr", "Acrv1"),]

purity_genes$gene <- factor(purity_genes$gene, levels = c("Dmrt1", "Ccnb1ip1", "Aurka", "Cabyr",
                                                          "Hells", "Adad2", "Tank","Acrv1"))

#final expression plots of marker genes in P. campbelli parental individuals across stages; Figure S4
bbbb_markers <- ggplot(data = purity_genes[grep("BBBB", purity_genes$condition),], aes(x = condition, y = fpkm, color = condition)) +
  stat_boxplot(geom ='errorbar') +
  geom_boxplot(outlier.colour = "black",outlier.shape = NA, notch = FALSE) +
  geom_quasirandom(aes(col = condition), alpha = 0.6, size = 2) + 
  facet_wrap(vars(gene), ncol = 4, scales = "free_y") +
  theme_classic() +
  scale_color_manual(values = c(sp_color, lz_color, dip_color, rs_color)) +
  theme(axis.title.x = element_blank(),
        axis.line = element_blank(),
        panel.spacing.x=unit(0.5, "lines"),panel.spacing.y = unit(0, "lines"),
        axis.text=element_text(size=8),
        strip.text = element_text(face = "italic", size = 8),
        strip.background = element_blank(),
        panel.border = element_rect(linetype = "solid", fill = NA),
        legend.position = "none")  +
  ylab("Expression level (FPKM)") +
  scale_x_discrete(labels = c("SP", "LZ", "DIP", "RS")) + 
  expand_limits(y = 0)

#final expression plots of marker genes in P. sungorus parental individuals across stages; Figure S4
ssss_markers <- ggplot(data = purity_genes[grep("SSSS", purity_genes$condition),], aes(x = condition, y = fpkm, color = condition)) +
  stat_boxplot(geom ='errorbar') +
  geom_boxplot(outlier.colour = "black",outlier.shape = NA, notch = FALSE) +
  geom_quasirandom(aes(col = condition), alpha = 0.6, size = 2) + 
  facet_wrap(vars(gene), ncol = 4, scales = "free_y") +
  theme_classic() +
  scale_color_manual(values = c(sp_color, lz_color, dip_color, rs_color)) +
  theme(axis.title.x = element_blank(),
        axis.line = element_blank(),
        panel.spacing.x=unit(0.5, "lines"),panel.spacing.y = unit(0, "lines"),
        axis.text=element_text(size=8),
        strip.text = element_text(face = "italic", size = 8),
        strip.background = element_blank(),
        panel.border = element_rect(linetype = "solid", fill = NA),
        legend.position = "none")  +
  ylab("Expression level (FPKM)") +
  scale_x_discrete(labels = c("SP", "LZ", "DIP", "RS")) + 
  expand_limits(y = 0)

#change exportFigs to TRUE to export as PDF if desired
if(exportFigs == TRUE) {
  ggsave(filename = "exported_figures/bbbb_purity_markers.pdf", plot = bbbb_markers, width = 6.5, height = 3.5)
  ggsave(filename = "exported_figures/ssss_purity_markers.pdf", plot = ssss_markers, width = 6.5, height = 3.5)
} else if (exportFigs == FALSE) {
  print("Marker gene expression in P. campbelli")
  print(bbbb_markers)
  print("Marker gene expression in P. sungorus")
  print(ssss_markers)
}
```

<h1>Differential Expression</h1>
<h3>Design set-up</h3>
```{r de-design-matrix}
#set up metadata for samples going into differential expression analysis
Dev_metadata <- metadata[grep("WT",metadata$condition_name, invert = TRUE),]
Dev_metadata$stage <- factor(Dev_metadata$stage, levels = stages[1:4])

#set up differential expression experimental design and contrasts
design <- cbind(rownames(Dev_metadata), Dev_metadata[c(grep("cross",colnames(Dev_metadata)),grep("stage",colnames(Dev_metadata)),grep("condition_name",colnames(Dev_metadata)))])
design$condition_name <- factor(design$condition_name)
colnames(design) <- c("sample_ID", "cross", "stage", "condition_name")
design$condition_name <- gsub("-",".",design$condition_name)
design$condition_name <- as.factor(design$condition_name)

#create DE contrast matrix
designMatrix  <-  model.matrix(~0+design$condition_name)
colnames(designMatrix)  <-  levels(design$condition_name)
rownames(designMatrix)  <-  design$sample_ID
```

<h3>Defining contrasts</h3>
```{r de-defining contrasts}
#set up DE contrasts
contrast.list <- makeContrasts(
  # compare cell types between hybrids and campbelli
  Hyb.BB.SP  =  BBSS.SP - BBBB.SP,  
  Hyb.BB.LZ  =  BBSS.LZ - BBBB.LZ, 
  Hyb.BB.DIP  =  BBSS.DIP - BBBB.DIP,
  
  # compare cell types between hybrids and campbelli
  Hyb.SS.SP   =  BBSS.SP -  SSSS.SP,  
  Hyb.SS.LZ   =  BBSS.LZ -  SSSS.LZ, 
  Hyb.SS.DIP  =  BBSS.DIP - SSSS.DIP,
  
  # parents
  BBBB.SSSS.SP  =   BBBB.SP -  SSSS.SP, 
  BBBB.SSSS.LZ  =   BBBB.LZ -  SSSS.LZ, 
  BBBB.SSSS.DIP  =  BBBB.DIP - SSSS.DIP, 
  BBBB.SSSS.RS  =   BBBB.RS  - SSSS.RS, 
  
  levels  =  designMatrix
)
contrast.list
```

<h3>Dispersion estimates of main datasets</h3>
```{r bcv-calculations}
#calculate BCV (biological coefficient of variation) for main DE analysis; this is a calculation of the intra-replicate variability
dgeDev <- estimateGLMCommonDisp(dgeDev, designMatrix, verbose = TRUE) 
dgeDev <- estimateGLMTrendedDisp(dgeDev, designMatrix) 
dgeDev <- estimateGLMTagwiseDisp(dgeDev, designMatrix) 

#generate BCV plots
if(exportFigs == TRUE) {
  pdf(file="exported_figures/Dispersion_plots_Dev_dataset.pdf", width=10, height=8, pointsize=12)
  plotBCV(dgeDev, main = paste("Dev dataset (all genes); BCV = ",round(sqrt(dgeDev$common.dispersion), 3)), ylim = c(0, 12), col.common = "#C2A5A2", col.trend = "#745756", xlab = "Average log(CPM)", xlim = c(-4,13), col = alpha("#6E6E6E", 0.15), cex = 1.25)
  dev.off()
} else {
  plotBCV(dgeDev, main = paste("Dispersion estimates for Dev dataset; BCV = ",round(sqrt(dgeDev$common.dispersion), 3)), ylim = c(0, 12), col.common = "#C2A5A2", col.trend = "#745756", xlab = "Average log(CPM)", xlim = c(-4,13), col = alpha("#6E6E6E", 0.15), cex = 1.25)
}
```

<h3>Fit model and calculate DE genes</h3>
```{r calculate-DE-genes}
#set the false discovery rate
FDR_level <- 0.05

#for each contrast, fit the GLM, perform a likelihood ratio test, then pull out the differential expression stats with topTags
contrasts_Dev <- colnames(contrast.list)
for (f in 1:length(contrasts_Dev)){
  LRT <- glmLRT(glmFit(dgeDev,  designMatrix), contrast = contrast.list[, f])
  assign(paste(contrasts_Dev[f], "Dev", "LRT", sep = "."), LRT)
  
  tt <- topTags(LRT, n = nrow(LRT))
  assign(paste(contrasts_Dev[f], "Dev", "tt", sep = "."), tt)
  assign(paste(contrasts_Dev[f], "Dev", "de", sep = "."), tt$table[tt$table$FDR<= FDR_level, ])
}

#if you want to save the LRTs, but they are large objects that slow down the code
rm(tt, LRT)
rm(list = ls(pattern = "LRT"))

#create a dataframe of genes that are DE between hybrids and BOTH parents
Hyb.Comb.SP.Dev.de <- merge(Hyb.BB.SP.Dev.de, Hyb.SS.SP.Dev.de, by = c("Length", "chr", "start", "stop", "gene"))
Hyb.Comb.LZ.Dev.de <- merge(Hyb.BB.LZ.Dev.de, Hyb.SS.LZ.Dev.de, by = c("Length", "chr", "start", "stop", "gene"))
Hyb.Comb.DIP.Dev.de <- merge(Hyb.BB.DIP.Dev.de, Hyb.SS.DIP.Dev.de, by = c("Length", "chr", "start", "stop", "gene"))


#print DE summary stats; specifically, numbers of DE genes for different contrasts
#BBSS vs BBBB
print(paste("There are ", dim(Hyb.BB.SP.Dev.de[Hyb.BB.SP.Dev.de$FDR < 0.05,])[1]  ," genes differentially expressed in spermatogonia between hybrids and campbelli", sep = ""))
print(paste("There are ", dim(Hyb.BB.LZ.Dev.de[Hyb.BB.LZ.Dev.de$FDR < 0.05,])[1]  ," genes differentially expressed in leptotene/zygotene between hybrids and campbelli", sep = ""))
print(paste("There are ", dim(Hyb.BB.DIP.Dev.de[Hyb.BB.DIP.Dev.de$FDR < 0.05,])[1] ," genes differentially expressed in diplotene between hybrids and campbelli", sep = ""))

#BBSS vs SSSS
print(paste("There are ", dim(Hyb.SS.SP.Dev.de[Hyb.SS.SP.Dev.de$FDR < 0.05,])[1]  ," genes differentially expressed in spermatogonia between hybrids and sungorus", sep = ""))
print(paste("There are ", dim(Hyb.SS.LZ.Dev.de[Hyb.SS.LZ.Dev.de$FDR < 0.05,])[1]  ," genes differentially expressed in leptotene/zygotene between hybrids and sungorus", sep = ""))
print(paste("There are ", dim(Hyb.SS.DIP.Dev.de[Hyb.SS.DIP.Dev.de$FDR < 0.05,])[1] ," genes differentially expressed in diplotene between hybrids and sungorus", sep = ""))

#BBSS vs Comb
print(paste("There are ", dim(Hyb.Comb.SP.Dev.de[Hyb.Comb.SP.Dev.de$FDR.x < 0.05 & Hyb.Comb.SP.Dev.de$FDR.y < 0.05,])[1]  ," genes differentially expressed in spermatogonia between hybrids and both parents", sep = ""))
print(paste("There are ", dim(Hyb.Comb.LZ.Dev.de[Hyb.Comb.LZ.Dev.de$FDR.x < 0.05 & Hyb.Comb.LZ.Dev.de$FDR.y < 0.05,])[1]  ," genes differentially expressed in leptotene/zygotene between hybrids and both parents", sep = ""))
print(paste("There are ", dim(Hyb.Comb.DIP.Dev.de[Hyb.Comb.DIP.Dev.de$FDR.x < 0.05 & Hyb.Comb.DIP.Dev.de$FDR.y < 0.05,])[1] ," genes differentially expressed in diplotene between hybrids and both parents", sep = ""))

#write DE gene info out to files for post-analysis in processing
write.csv(Hyb.BB.SP.Dev.de, "circos_processing_scripts/hamster_circos_hybrids/data/hybrid_hamster_BBBB_SP_DE_Genes.csv")
write.csv(Hyb.BB.LZ.Dev.de, "circos_processing_scripts/hamster_circos_hybrids/data/hybrid_hamster_BBBB_LZ_DE_Genes.csv")
write.csv(Hyb.BB.DIP.Dev.de, "circos_processing_scripts/hamster_circos_hybrids/data/hybrid_hamster_BBBB_DIP_DE_Genes.csv")

write.csv(Hyb.SS.SP.Dev.de, "circos_processing_scripts/hamster_circos_hybrids/data/hybrid_hamster_SSSS_SP_DE_Genes.csv")
write.csv(Hyb.SS.LZ.Dev.de, "circos_processing_scripts/hamster_circos_hybrids/data/hybrid_hamster_SSSS_LZ_DE_Genes.csv")
write.csv(Hyb.SS.DIP.Dev.de, "circos_processing_scripts/hamster_circos_hybrids/data/hybrid_hamster_SSSS_DIP_DE_Genes.csv")

write.csv(Hyb.Comb.SP.Dev.de,  "circos_processing_scripts/hamster_circos_hybrids/data/hybrid_hamster_Comb_SP_DE_Genes.csv")
write.csv(Hyb.Comb.LZ.Dev.de,  "circos_processing_scripts/hamster_circos_hybrids/data/hybrid_hamster_Comb_LZ_DE_Genes.csv")
write.csv(Hyb.Comb.DIP.Dev.de, "circos_processing_scripts/hamster_circos_hybrids/data/hybrid_hamster_Comb_DIP_DE_Genes.csv")

#write DE gene info out to files for supplement
write.csv(Hyb.BB.SP.Dev.de, "exported_files/hybrid_hamster_BBBB_SP_DE_Genes.csv")
write.csv(Hyb.BB.LZ.Dev.de, "exported_files/hybrid_hamster_BBBB_LZ_DE_Genes.csv")
write.csv(Hyb.BB.DIP.Dev.de,"exported_files/hybrid_hamster_BBBB_DIP_DE_Genes.csv")

write.csv(Hyb.SS.SP.Dev.de, "exported_files/hybrid_hamster_SSSS_SP_DE_Genes.csv")
write.csv(Hyb.SS.LZ.Dev.de, "exported_files/hybrid_hamster_SSSS_LZ_DE_Genes.csv")
write.csv(Hyb.SS.DIP.Dev.de,"exported_files/hybrid_hamster_SSSS_DIP_DE_Genes.csv")

write.csv(Hyb.Comb.SP.Dev.de,  "exported_files/hybrid_hamster_Comb_SP_DE_Genes.csv")
write.csv(Hyb.Comb.LZ.Dev.de,  "exported_files/hybrid_hamster_Comb_LZ_DE_Genes.csv")
write.csv(Hyb.Comb.DIP.Dev.de, "exported_files/hybrid_hamster_Comb_DIP_DE_Genes.csv")
```
<h3>How does the number of DE genes over time change?</h3>
```{r DE-over-time-ham-and-mouse}
#this code generates a plot of the number of DE genes between hybrids and parents with the same X chromosome across the different stages of spermatogenesis
DE_over_time_df <- data.frame(species = c(rep("mouse", 4), rep("ham", 3)),
                              stage = c("SP", "LZ", "DIP", "RS", "SP", "LZ", "DIP"),
                              DE_genes = c(dim(read.csv("exported_files/hybrid_mouse_MusP_SP_DE_Genes.csv"))[1],
                                           dim(read.csv("exported_files/hybrid_mouse_MusP_LZ_DE_Genes.csv"))[1],
                                           dim(read.csv("exported_files/hybrid_mouse_MusP_DIP_DE_Genes.csv"))[1],
                                           dim(read.csv("exported_files/hybrid_mouse_MusP_RS_DE_Genes.csv"))[1],
                                           dim(Hyb.BB.SP.Dev.de[Hyb.BB.SP.Dev.de$FDR < 0.05,])[1],
                                           dim(Hyb.BB.LZ.Dev.de[Hyb.BB.LZ.Dev.de$FDR < 0.05,])[1],
                                           dim(Hyb.BB.DIP.Dev.de[Hyb.BB.DIP.Dev.de$FDR < 0.05,])[1]),
                              exp_genes = c(dim(read.csv("exported_files/Hyb.MusP.SP.exp_genes.csv"))[1],
                                            dim(read.csv("exported_files/Hyb.MusP.LZ.exp_genes.csv"))[1],
                                            dim(read.csv("exported_files/Hyb.MusP.DIP.exp_genes.csv"))[1],
                                            dim(read.csv("exported_files/Hyb.MusP.RS.exp_genes.csv"))[1],
                                            length(Hyb.BB.SP.exp),
                                            length(Hyb.BB.LZ.exp),
                                            length(Hyb.BB.DIP.exp)))


DE_over_time_df$stage <- factor(DE_over_time_df$stage, levels = c("SP", "LZ", "DIP", "RS"))

DE_over_time_df$percent <- DE_over_time_df$DE_genes/DE_over_time_df$exp_genes

DE_over_time_df$comp <- paste(DE_over_time_df$species, DE_over_time_df$stage, sep = "_")
DE_over_time_df$comp <- factor(DE_over_time_df$comp, levels = c("mouse_SP", "ham_SP", 
                                                                "mouse_LZ", "ham_LZ",
                                                                "mouse_DIP", "ham_DIP",
                                                                "mouse_RS"))

#all three of these plots display the same data but with a different emphasis for visualization in presentations
#this plot emphasizes the mouse DE genes
DE_over_time_DE_plot_mouse_focus <- 
  ggplot(data = DE_over_time_df, aes(x=comp, y=DE_genes, group=species, color=comp, alpha=species)) +
  geom_line(color = "black", linewidth = 1, aes(alpha=species)) +
  scale_color_manual(values=c(sp_color, gray_1, lz_color, gray_2, dip_color, gray_3, rs_color)) + 
  scale_alpha_manual(values = c(0.3, 1)) +
  geom_point(size = 5) +
  theme_bw() + 
  scale_x_discrete(labels= c(rep(c("Mouse", "Hamster"), 3), "Mouse")) +
  theme(legend.position = "none", 
        text = element_text(size = text_size*3),
        panel.grid.minor = element_blank(),
        panel.grid.major.x = element_blank(),
        axis.title.x = element_blank(),
        axis.text.x = element_text(angle = 45, hjust = 1)) +
  scale_y_continuous(expand = c(0, 0), limits = c(0, max(DE_over_time_df$DE_genes)*1.1)) +
  labs(y = "Number of DE genes", x = "Stage of spermatogenesis") + 
  annotate("text", size = 7, x = 1.5, y = 11750, label= "Mouse", alpha = 1) +
  annotate("text", size = 7, x = 1.5, y = 10750, label= "Hamster", alpha = 0.3)

#this plot emphasizes the hamster DE genes
DE_over_time_DE_plot_ham_focus <- 
  ggplot(data = DE_over_time_df, aes(x=comp, y=DE_genes, group=species, color=comp, alpha=species)) +
  geom_line(color = "black", linewidth = 1, aes(alpha=species)) +
  scale_color_manual(values=c(gray_1, sp_color, gray_2, lz_color, gray_3, dip_color, gray_4)) + 
  scale_alpha_manual(values = c(1, 0.3)) +
  geom_point(size = 5) +
  theme_bw() + 
  theme(legend.position = "none", 
        text = element_text(size = text_size*3),
        panel.grid.minor = element_blank(),
        panel.grid.major.x = element_blank(),
        axis.title.x = element_blank(),
        axis.text.x = element_text(angle = 45, hjust = 1)) +
  scale_y_continuous(expand = c(0, 0), limits = c(0, max(DE_over_time_df$DE_genes)*1.1)) +
  labs(y = "Number of DE genes", x = "Stage of spermatogenesis") + 
  scale_x_discrete(labels= c(rep(c("Mouse", "Hamster"), 3), "Mouse")) +
  annotate("text", size = 7, x = 1.5, y = 11750, label= "Mouse", alpha = 0.3) +
  annotate("text", size = 7, x = 1.5, y = 10750, label= "Hamster", alpha = 1)

#this plot compares both and is in the supplemental material of the manuscript
DE_over_time_DE_plot_comparison <- ggplot(data = DE_over_time_df, aes(x=stage, y=DE_genes, group=species, color=stage, shape = species)) +
  geom_line(color = "black", linewidth = 1) +
  scale_color_manual(values=c(sp_color, lz_color, dip_color, rs_color)) + 
  scale_alpha_manual(values = c(0.3, 1)) +
  geom_point(size = 5) +
  theme_bw() + 
  scale_x_discrete(labels= c("Spermatogonia", "Leptotene/zygotene", "Diplotene", "Round spermatids")) +
  theme(text = element_text(size = text_size*3),
        panel.grid.minor = element_blank(),
        panel.grid.major.x = element_blank(),
        axis.title.x = element_blank(),
        axis.text.x = element_text(angle = 45, hjust = 1)) +
  labs(y = "Number of DE genes", x = "Stage of spermatogenesis")

if(exportFigs == TRUE) {
  ggsave(filename = "exported_figures/DE_over_time_mouse_focus.pdf", DE_over_time_DE_plot_mouse_focus, width = 7, height = 4)
  ggsave(filename = "exported_figures/DE_over_time_DE_plot_ham_focus.pdf", DE_over_time_DE_plot_ham_focus, width = 7, height = 4)
  ggsave(filename = "exported_figures/DE_over_time_DE_plot_both_comparison.pdf", DE_over_time_DE_plot_comparison, width = 7, height = 4)
} else {
  DE_over_time_DE_plot_comparison
}
```

<h3>Compare the different extents of X-linked overexpression between species</h3>
```{r X-over-expression-versus-in-diplotene} 
#the goal of this analysis is to calculate the average (+/- SEM) relative overexpression of autosomal and X linked genes for each stage in hybrids and in parents
#set up rowMeans for stage x genotype x chromosome type for BBSS
rpkm_BBSS_SP_Dev_allAuto.norm.mean <- data.frame(rowMeans(rpkm_BBSS_SP_Dev_allAuto.norm))
colnames(rpkm_BBSS_SP_Dev_allAuto.norm.mean) <- "mean_FPKM"
rpkm_BBSS_SP_Dev_justX.norm.mean <- data.frame(rowMeans(rpkm_BBSS_SP_Dev_justX.norm))
colnames(rpkm_BBSS_SP_Dev_justX.norm.mean) <- "mean_FPKM"

rpkm_BBSS_LZ_Dev_allAuto.norm.mean <- data.frame(rowMeans(rpkm_BBSS_LZ_Dev_allAuto.norm))
colnames(rpkm_BBSS_LZ_Dev_allAuto.norm.mean) <- "mean_FPKM"
rpkm_BBSS_LZ_Dev_justX.norm.mean <- data.frame(rowMeans(rpkm_BBSS_LZ_Dev_justX.norm))
colnames(rpkm_BBSS_LZ_Dev_justX.norm.mean) <- "mean_FPKM"

rpkm_BBSS_DIP_Dev_allAuto.norm.mean <- data.frame(rowMeans(rpkm_BBSS_DIP_Dev_allAuto.norm))
colnames(rpkm_BBSS_DIP_Dev_allAuto.norm.mean) <- "mean_FPKM"
rpkm_BBSS_DIP_Dev_justX.norm.mean <- data.frame(rowMeans(rpkm_BBSS_DIP_Dev_justX.norm))
colnames(rpkm_BBSS_DIP_Dev_justX.norm.mean) <- "mean_FPKM"


#set up rowMeans for stage x genotype x chromosome type for BBBB
rpkm_BBBB_SP_Dev_allAuto.norm.mean <- data.frame(rowMeans(rpkm_BBBB_SP_Dev_allAuto.norm))
colnames(rpkm_BBBB_SP_Dev_allAuto.norm.mean) <- "mean_FPKM"
rpkm_BBBB_SP_Dev_justX.norm.mean <- data.frame(rowMeans(rpkm_BBBB_SP_Dev_justX.norm))
colnames(rpkm_BBBB_SP_Dev_justX.norm.mean) <- "mean_FPKM"

rpkm_BBBB_LZ_Dev_allAuto.norm.mean <- data.frame(rowMeans(rpkm_BBBB_LZ_Dev_allAuto.norm))
colnames(rpkm_BBBB_LZ_Dev_allAuto.norm.mean) <- "mean_FPKM"
rpkm_BBBB_LZ_Dev_justX.norm.mean <- data.frame(rowMeans(rpkm_BBBB_LZ_Dev_justX.norm))
colnames(rpkm_BBBB_LZ_Dev_justX.norm.mean) <- "mean_FPKM"

rpkm_BBBB_DIP_Dev_allAuto.norm.mean <- data.frame(rowMeans(rpkm_BBBB_DIP_Dev_allAuto.norm))
colnames(rpkm_BBBB_DIP_Dev_allAuto.norm.mean) <- "mean_FPKM"
rpkm_BBBB_DIP_Dev_justX.norm.mean <- data.frame(rowMeans(rpkm_BBBB_DIP_Dev_justX.norm))
colnames(rpkm_BBBB_DIP_Dev_justX.norm.mean) <- "mean_FPKM"

rpkm_BBBB_RS_Dev_allAuto.norm.mean <- data.frame(rowMeans(rpkm_BBBB_RS_Dev_allAuto.norm))
colnames(rpkm_BBBB_RS_Dev_allAuto.norm.mean) <- "mean_FPKM"
rpkm_BBBB_RS_Dev_justX.norm.mean <- data.frame(rowMeans(rpkm_BBBB_RS_Dev_justX.norm))
colnames(rpkm_BBBB_RS_Dev_justX.norm.mean) <- "mean_FPKM"


#set up rowMeans for stage x genotype x chromosome type for SSSS
rpkm_SSSS_SP_Dev_allAuto.norm.mean <- data.frame(rowMeans(rpkm_SSSS_SP_Dev_allAuto.norm))
colnames(rpkm_SSSS_SP_Dev_allAuto.norm.mean) <- "mean_FPKM"
rpkm_SSSS_SP_Dev_justX.norm.mean <- data.frame(rowMeans(rpkm_SSSS_SP_Dev_justX.norm))
colnames(rpkm_SSSS_SP_Dev_justX.norm.mean) <- "mean_FPKM"

rpkm_SSSS_LZ_Dev_allAuto.norm.mean <- data.frame(rowMeans(rpkm_SSSS_LZ_Dev_allAuto.norm))
colnames(rpkm_SSSS_LZ_Dev_allAuto.norm.mean) <- "mean_FPKM"
rpkm_SSSS_LZ_Dev_justX.norm.mean <- data.frame(rowMeans(rpkm_SSSS_LZ_Dev_justX.norm))
colnames(rpkm_SSSS_LZ_Dev_justX.norm.mean) <- "mean_FPKM"

rpkm_SSSS_DIP_Dev_allAuto.norm.mean <- data.frame(rowMeans(rpkm_SSSS_DIP_Dev_allAuto.norm))
colnames(rpkm_SSSS_DIP_Dev_allAuto.norm.mean) <- "mean_FPKM"
rpkm_SSSS_DIP_Dev_justX.norm.mean <- data.frame(rowMeans(rpkm_SSSS_DIP_Dev_justX.norm))
colnames(rpkm_SSSS_DIP_Dev_justX.norm.mean) <- "mean_FPKM"

rpkm_SSSS_RS_Dev_allAuto.norm.mean <- data.frame(rowMeans(rpkm_SSSS_RS_Dev_allAuto.norm))
colnames(rpkm_SSSS_RS_Dev_allAuto.norm.mean) <- "mean_FPKM"
rpkm_SSSS_RS_Dev_justX.norm.mean <- data.frame(rowMeans(rpkm_SSSS_RS_Dev_justX.norm))
colnames(rpkm_SSSS_RS_Dev_justX.norm.mean) <- "mean_FPKM"

#calculate the variance in X-linked overexpression; produces Figure S9
#creates a data frame that contains the average relative amount of X-linked overexpression in each diplotene individual
avg_overExpression_DIP_justX <- data.frame(overExp = 
                                             c(colMeans(rpkm_BBBB_DIP_Dev_justX.norm)/mean(rpkm_BBBB_DIP_Dev_justX.norm.mean$mean_FPKM),
                                               colMeans(rpkm_BBSS_DIP_Dev_justX.norm)/mean(rpkm_BBBB_DIP_Dev_justX.norm.mean$mean_FPKM),
                                               colMeans(rpkm_SSSS_DIP_Dev_justX.norm)/mean(rpkm_BBBB_DIP_Dev_justX.norm.mean$mean_FPKM)))
avg_overExpression_DIP_justX$ID <- gsub("_DIP", "", rownames(avg_overExpression_DIP_justX))
avg_overExpression_DIP_justX$Strain <- factor(gsub("_.*", "", avg_overExpression_DIP_justX$ID), levels = c("BBBB", "BBSS", "SSSS"))

#creates a data frame that contains the SEM of the relative amount of X-linked overexpression in each diplotene individual
var_overExpression_DIP_justX <- 
  data.frame(Strain = c("BBBB", "BBSS", "SSSS"),
             Var = c(calculate_sem(subset(avg_overExpression_DIP_justX,
                                          avg_overExpression_DIP_justX$Strain == "BBBB")$overExp),
                     calculate_sem(subset(avg_overExpression_DIP_justX, 
                                          avg_overExpression_DIP_justX$Strain == "BBSS")$overExp),
                     calculate_sem(subset(avg_overExpression_DIP_justX, 
                                          avg_overExpression_DIP_justX$Strain == "SSSS")$overExp)))

#plots the amount of X-linked over-expression in each diplotene sample
overExp_justX_variance_plot <- 
  ggplot(avg_overExpression_DIP_justX, aes(x=Strain, y = overExp, 
                                           shape = Strain, col = Strain, fill = Strain)) + 
  geom_quasirandom(aes(fill = Strain, color = Strain, shape = Strain), alpha = 1, size = 0.75) +
  geom_boxplot(width=0.1, outlier.shape = NA, lwd = 0.2, alpha = 0.2, color = "black") +
  scale_fill_manual(values=c(gray_5, gray_4, gray_2)) + 
  scale_color_manual(values=c(gray_5, gray_4, gray_2)) + 
  scale_shape_manual(values=c(24,4,25)) + 
  theme_bw() + 
  theme(legend.position = "none", 
        text = element_text(size = 6),
        axis.title.x=element_blank(), 
        axis.title.y = element_text(size = 6), 
        axis.text.x = element_text(angle = 45, vjust = 0.8, size = 6), 
        plot.margin = unit(c(0.4, 0.07, 0, 0.07), "cm"), 
        panel.grid.major.x = element_blank(), 
        panel.grid.minor = element_blank(), 
        axis.line.y = element_line(colour = "black", linewidth = 0.2), 
        panel.border = element_blank(), 
        axis.ticks.length.x = unit(0, "cm")) +
  scale_x_discrete(labels = c("Pcam", "Hybrid", "Psun")) + ylab("Relative X Overexpression") + 
  geom_signif(
    y_position = c(summary(avg_overExpression_DIP_justX[avg_overExpression_DIP_justX$Strain == "BBBB",]$overExp)[6]+2,
                   summary(avg_overExpression_DIP_justX[avg_overExpression_DIP_justX$Strain == "BBSS",]$overExp)[6]+2,
                   summary(avg_overExpression_DIP_justX[avg_overExpression_DIP_justX$Strain == "SSSS",]$overExp)[6]+2), 
    xmin = c(1,2,3), 
    xmax = c(1,2,3), 
    annotation = c(paste("SEM = ", round(var_overExpression_DIP_justX[1,2],2)),
                   paste("SEM = ", round(var_overExpression_DIP_justX[2,2],2)),
                   paste("SEM = ", round(var_overExpression_DIP_justX[3,2],2))), 
    tip_length = 0, textsize = sigText_size, size = 0) +
  ylim(values = c(0,round(1.1*max(avg_overExpression_DIP_justX$overExp))))


#calculate the variance in autosomal overexpression; produces Figure S9
#creates a data frame that contains the average relative amount of autosomal overexpression in each diplotene individual
avg_overExpression_DIP_allAuto <- data.frame(overExp = 
                                               c(colMeans(rpkm_BBBB_DIP_Dev_allAuto.norm)/mean(rpkm_BBBB_DIP_Dev_allAuto.norm.mean$mean_FPKM),
                                                 colMeans(rpkm_BBSS_DIP_Dev_allAuto.norm)/mean(rpkm_BBBB_DIP_Dev_allAuto.norm.mean$mean_FPKM),
                                                 colMeans(rpkm_SSSS_DIP_Dev_allAuto.norm)/mean(rpkm_BBBB_DIP_Dev_allAuto.norm.mean$mean_FPKM)))
avg_overExpression_DIP_allAuto$ID <- gsub("_DIP", "", rownames(avg_overExpression_DIP_allAuto))
avg_overExpression_DIP_allAuto$Strain <- factor(gsub("_.*", "", avg_overExpression_DIP_allAuto$ID), levels = c("BBBB", "BBSS", "SSSS"))

#creates a data frame that contains the SEM of the relative amount of autosomal overexpression in each diplotene individual
var_overExpression_DIP_allAuto <- 
  data.frame(Strain = c("BBBB", "BBSS", "SSSS"),
             Var = c(calculate_sem(subset(avg_overExpression_DIP_allAuto,
                                          avg_overExpression_DIP_allAuto$Strain == "BBBB")$overExp),
                     calculate_sem(subset(avg_overExpression_DIP_allAuto, 
                                          avg_overExpression_DIP_allAuto$Strain == "BBSS")$overExp),
                     calculate_sem(subset(avg_overExpression_DIP_allAuto, 
                                          avg_overExpression_DIP_allAuto$Strain == "SSSS")$overExp)))

#plots the amount of autosomal over-expression in each diplotene sample
overExp_allAuto_variance_plot <- 
  ggplot(avg_overExpression_DIP_allAuto, aes(x=Strain, y = overExp, 
                                             shape = Strain, col = Strain, fill = Strain)) + 
  geom_quasirandom(aes(fill = Strain, color = Strain, shape = Strain), alpha = 1, size = 0.75) +
  geom_boxplot(width=0.1, outlier.shape = NA, lwd = 0.2, alpha = 0.2, color = "black") +
  scale_fill_manual(values=c(gray_5, gray_4, gray_2)) + 
  scale_color_manual(values=c(gray_5, gray_4, gray_2)) + 
  scale_shape_manual(values=c(24,4,25)) + 
  theme_bw() + 
  theme(legend.position = "none", 
        text = element_text(size = 6),
        axis.title.x=element_blank(), 
        axis.title.y = element_text(size = 6), 
        axis.text.x = element_text(angle = 45, vjust = 0.8, size = 6), 
        plot.margin = unit(c(0.4, 0.07, 0, 0.07), "cm"), 
        panel.grid.major.x = element_blank(), 
        panel.grid.minor = element_blank(), 
        axis.line.y = element_line(colour = "black", linewidth = 0.2), 
        panel.border = element_blank(), 
        axis.ticks.length.x = unit(0, "cm")) +
  scale_x_discrete(labels = c("Pcam", "Hybrid", "Psun")) + ylab("Relative Auto Overexpression") + 
  geom_signif(
    y_position = c(summary(avg_overExpression_DIP_allAuto[avg_overExpression_DIP_allAuto$Strain == "BBBB",]$overExp)[6]+2,
                   summary(avg_overExpression_DIP_allAuto[avg_overExpression_DIP_allAuto$Strain == "BBSS",]$overExp)[6]+2,
                   summary(avg_overExpression_DIP_allAuto[avg_overExpression_DIP_allAuto$Strain == "SSSS",]$overExp)[6]+2), 
    xmin = c(1,2,3), 
    xmax = c(1,2,3), 
    annotation = c(paste("SEM = ", round(var_overExpression_DIP_allAuto[1,2],2)),
                   paste("SEM = ", round(var_overExpression_DIP_allAuto[2,2],2)),
                   paste("SEM = ", round(var_overExpression_DIP_allAuto[3,2],2))), 
    tip_length = 0, textsize = sigText_size, size = 0) +
  ylim(values = c(0,round(1.1*max(avg_overExpression_DIP_justX$overExp))))

if(exportFigs == TRUE) {
  ggsave(filename = "exported_figures/X_and_auto_overexpression_hamster.pdf", 
         plot_grid(overExp_allAuto_variance_plot, overExp_justX_variance_plot, nrow = 1),
         width = 4, height = 0.75*half_plot_height)
} else {
  plot_grid(overExp_allAuto_variance_plot, overExp_justX_variance_plot, nrow = 1)
}

#print out the relative overexpression of the X during diplotene in hybrids relative to parents
avg_overExpression_DIP_justX %>% group_by(Strain) %>%
  summarize_at('overExp',mean)


#set up logFC expressed dataframes to test if X is statistically overexpressed relative to autosomes
#BBSS - BBBB
BBSS_BBBB_SP_Dev_allAuto.logFC.diff <- data.frame(logFC = Hyb.BB.SP.Dev.tt$table[Hyb.BB.SP.Dev.tt$table$chr != "X",]$logFC)
BBSS_BBBB_SP_Dev_justX.logFC.diff <-   data.frame(logFC = Hyb.BB.SP.Dev.tt$table[Hyb.BB.SP.Dev.tt$table$chr == "X",]$logFC)

BBSS_BBBB_LZ_Dev_allAuto.logFC.diff <- data.frame(logFC = Hyb.BB.LZ.Dev.tt$table[Hyb.BB.LZ.Dev.tt$table$chr != "X",]$logFC)
BBSS_BBBB_LZ_Dev_justX.logFC.diff <-   data.frame(logFC = Hyb.BB.LZ.Dev.tt$table[Hyb.BB.LZ.Dev.tt$table$chr == "X",]$logFC)

BBSS_BBBB_DIP_Dev_allAuto.logFC.diff <- data.frame(logFC = Hyb.BB.DIP.Dev.tt$table[Hyb.BB.DIP.Dev.tt$table$chr != "X",]$logFC)
BBSS_BBBB_DIP_Dev_justX.logFC.diff <-   data.frame(logFC = Hyb.BB.DIP.Dev.tt$table[Hyb.BB.DIP.Dev.tt$table$chr == "X",]$logFC)

#wilcoxon rank sum tests
x_vs_auto_pvalues <- data.frame(names = c("BBSS_BBBB_SP", "BBSS_BBBB_LZ", "BBSS_BBBB_DIP"),
                                pValues = c(wilcox.test(BBSS_BBBB_SP_Dev_justX.logFC.diff$logFC, 
                                                        BBSS_BBBB_SP_Dev_allAuto.logFC.diff$logFC)$p.value,
                                            wilcox.test(BBSS_BBBB_LZ_Dev_justX.logFC.diff$logFC, 
                                                        BBSS_BBBB_LZ_Dev_allAuto.logFC.diff$logFC)$p.value,
                                            wilcox.test(BBSS_BBBB_DIP_Dev_justX.logFC.diff$logFC,
                                                        BBSS_BBBB_DIP_Dev_allAuto.logFC.diff$logFC)$p.value))

x_vs_auto_pvalues$pValues <- signif(p.adjust(x_vs_auto_pvalues$pValues, method = "fdr"), 2)
x_vs_auto_pvalues$significant <- "yes"
x_vs_auto_pvalues[x_vs_auto_pvalues$pValues > 0.05,]$significant <- "no"
```

<h3>Compare the logFC of Differentially Expressed Genes between Hybrids and Parents with Same X Chromosome for Both Mouse and Hamster across Spermatogenesis</h3>
```{r hybrid-DE-logFC-graphs-BBBB}
#this section produces figure 3 in the manuscript and demonstrates how patterns of up and downregulation across stages in hybrids differs between systems
Hyb.Mouse.SP.de <- read.csv("exported_files/hybrid_mouse_MusP_SP_DE_Genes.csv")
Hyb.Mouse.LZ.de <- read.csv("exported_files/hybrid_mouse_MusP_LZ_DE_Genes.csv")
Hyb.Mouse.DIP.de <-read.csv("exported_files/hybrid_mouse_MusP_DIP_DE_Genes.csv")

#set up dataframes of X-linked
logFC_mouse_SP_X_df <- data.frame(logFC =  Hyb.Mouse.SP.de[Hyb.Mouse.SP.de$chr == "X",]$logFC, treatment = "Mouse SP")
logFC_mouse_LZ_X_df <- data.frame(logFC =  Hyb.Mouse.LZ.de[Hyb.Mouse.LZ.de$chr == "X",]$logFC, treatment = "Mouse LZ")
logFC_mouse_DIP_X_df <- data.frame(logFC = Hyb.Mouse.DIP.de[Hyb.Mouse.DIP.de$chr == "X",]$logFC, treatment = "Mouse DIP")

logFC_hamster_SP_BBBB_X_df <-  data.frame(logFC = Hyb.BB.SP.Dev.de[grep("^X", Hyb.BB.SP.Dev.de$chr),]$logFC, treatment  = "Hamster SP")
logFC_hamster_LZ_BBBB_X_df <-  data.frame(logFC = Hyb.BB.LZ.Dev.de[grep("^X", Hyb.BB.LZ.Dev.de$chr),]$logFC, treatment  = "Hamster LZ")
logFC_hamster_DIP_BBBB_X_df <- data.frame(logFC = Hyb.BB.DIP.Dev.de[grep("^X", Hyb.BB.DIP.Dev.de$chr),]$logFC, treatment = "Hamster DIP")

#auto
logFC_mouse_SP_auto_df <- data.frame(logFC =  Hyb.Mouse.SP.de[Hyb.Mouse.SP.de$chr != "X",]$logFC, treatment = "Mouse SP")
logFC_mouse_LZ_auto_df <- data.frame(logFC =  Hyb.Mouse.LZ.de[Hyb.Mouse.LZ.de$chr != "X",]$logFC, treatment = "Mouse LZ")
logFC_mouse_DIP_auto_df <- data.frame(logFC = Hyb.Mouse.DIP.de[Hyb.Mouse.DIP.de$chr != "X",]$logFC, treatment = "Mouse DIP")

logFC_hamster_SP_BBBB_auto_df <-  data.frame(logFC = Hyb.BB.SP.Dev.de[grep("^X", Hyb.BB.SP.Dev.de$chr, invert = TRUE),]$logFC, treatment  = "Hamster SP")
logFC_hamster_LZ_BBBB_auto_df <-  data.frame(logFC = Hyb.BB.LZ.Dev.de[grep("^X", Hyb.BB.LZ.Dev.de$chr, invert = TRUE),]$logFC, treatment  = "Hamster LZ")
logFC_hamster_DIP_BBBB_auto_df <- data.frame(logFC = Hyb.BB.DIP.Dev.de[grep("^X", Hyb.BB.DIP.Dev.de$chr, invert = TRUE),]$logFC, treatment = "Hamster DIP")

#X-linked analysis
logFC_BBBB_X_comps_df <- rbind(logFC_mouse_SP_X_df, logFC_hamster_SP_BBBB_X_df,
                               logFC_mouse_LZ_X_df, logFC_hamster_LZ_BBBB_X_df,
                               logFC_mouse_DIP_X_df, logFC_hamster_DIP_BBBB_X_df)

logFC_BBBB_X_comps_df$treatment <- factor(logFC_BBBB_X_comps_df$treatment, levels = c("Mouse SP", "Hamster SP", 
                                                                                      "Mouse LZ", "Hamster LZ", 
                                                                                      "Mouse DIP", "Hamster DIP"))


logFC_BBBB_auto_comps_df <- rbind(logFC_mouse_SP_auto_df, logFC_hamster_SP_BBBB_auto_df,
                                  logFC_mouse_LZ_auto_df, logFC_hamster_LZ_BBBB_auto_df,
                                  logFC_mouse_DIP_auto_df, logFC_hamster_DIP_BBBB_auto_df)

logFC_BBBB_auto_comps_df$treatment <- factor(logFC_BBBB_auto_comps_df$treatment, levels = c("Mouse SP", "Hamster SP", 
                                                                                            "Mouse LZ", "Hamster LZ", 
                                                                                            "Mouse DIP", "Hamster DIP"))

#paired wilcox tests to test for difference within stage between mouse and hamster that are then FDR corrected
logFC_BBBB_X_pvalues <- data.frame(names = c("SP", "LZ", "DIP"),
                                   pValues = c(wilcox.test(logFC_BBBB_X_comps_df[logFC_BBBB_X_comps_df$treatment == "Mouse SP",]$logFC, 
                                                           logFC_BBBB_X_comps_df[logFC_BBBB_X_comps_df$treatment == "Hamster SP",]$logFC)$p.value,
                                               
                                               wilcox.test(logFC_BBBB_X_comps_df[logFC_BBBB_X_comps_df$treatment == "Mouse LZ",]$logFC, 
                                                           logFC_BBBB_X_comps_df[logFC_BBBB_X_comps_df$treatment == "Hamster LZ",]$logFC)$p.value,
                                               
                                               wilcox.test(logFC_BBBB_X_comps_df[logFC_BBBB_X_comps_df$treatment == "Mouse DIP",]$logFC, 
                                                           logFC_BBBB_X_comps_df[logFC_BBBB_X_comps_df$treatment == "Hamster DIP",]$logFC)$p.value))

logFC_BBBB_X_pvalues$pValues[1:3] <- signif(p.adjust(logFC_BBBB_X_pvalues[1:3,]$pValues, method = "fdr"), 3)
logFC_BBBB_X_pvalues$pSymbols <- sig_conversion(logFC_BBBB_X_pvalues$pValues)

print(paste(
  "(average logFC X-linked genes dwarf hamster: ",
  mean(logFC_BBBB_X_comps_df[logFC_BBBB_X_comps_df$treatment == "Hamster DIP",]$logFC),
  "; average logFC X-linked genes house mice: ",
  mean(logFC_BBBB_X_comps_df[logFC_BBBB_X_comps_df$treatment == "Mouse DIP",]$logFC),
  "; Wilcoxon signed rank test: p < 0.001; Figure 4)",
  sep = ""))


logFC_BBBB_X_comps_plot <- ggplot(logFC_BBBB_X_comps_df, aes(x = treatment, y = logFC)) + 
  geom_hline(yintercept = 0, linewidth = 0.5) +
  geom_quasirandom(aes(col = treatment, fill = treatment), alpha = 0.3, size = point_size, width = 0.4) + 
  geom_violin(aes(col = treatment), width = 1, fill = NA) + 
  geom_boxplot(width=boxplot_width, outlier.shape = NA, alpha = 0.8) + 
  theme_minimal() + 
  scale_color_manual(values=c(sp_color, sp_color_dark,
                              lz_color, lz_color_dark,
                              dip_color, dip_color_dark)) +
  scale_fill_manual(values=c(sp_color, sp_color_dark,
                             lz_color, lz_color_dark,
                             dip_color, dip_color_dark)) +
  ylab("Differential expression (logFC)") + 
  ggtitle("X Chromosome") + 
  scale_x_discrete(labels= rep(c("Mouse", "Hamster"), 3)) +
  ylim(-11.5, 16) +
  geom_signif(
    y_position = c(summary(logFC_BBBB_X_comps_df[grep("SP", logFC_BBBB_X_comps_df$treatment),]$logFC)[6]+3,
                   summary(logFC_BBBB_X_comps_df[grep("LZ", logFC_BBBB_X_comps_df$treatment),]$logFC)[6]+3,
                   summary(logFC_BBBB_X_comps_df[grep("DIP", logFC_BBBB_X_comps_df$treatment),]$logFC)[6]+3), 
    xmin = c(0.8,2.8,4.8), 
    xmax = c(2.2,4.2,6.2), 
    annotation = c(logFC_BBBB_X_pvalues[logFC_BBBB_X_pvalues$names == "SP",]$pSymbols,
                   logFC_BBBB_X_pvalues[logFC_BBBB_X_pvalues$names == "LZ",]$pSymbols,
                   logFC_BBBB_X_pvalues[logFC_BBBB_X_pvalues$names == "DIP",]$pSymbols), 
    tip_length = 0.01, textsize = text_size*0.5, size = 0.5) +
  theme(legend.position = "none", 
        text = element_text(size = text_size*1.5),
        panel.grid.minor = element_blank(),
        panel.grid.major.x = element_blank(),
        axis.title.x = element_blank(),
        axis.text.x = element_text(angle = 45),
        plot.title = element_text(hjust = 0.5)) +
  annotate("text", size = 2, x = 1, y = summary(logFC_BBBB_X_comps_df[grep("SP", logFC_BBBB_X_comps_df$treatment),]$logFC)[6]+2.1, label= paste("u = ", sum(logFC_BBBB_X_comps_df[logFC_BBBB_X_comps_df$treatment == "Mouse SP",]$logFC > 0), sep = "")) +
  annotate("text", size = 2, x = 1, y = summary(logFC_BBBB_X_comps_df[grep("SP", logFC_BBBB_X_comps_df$treatment),]$logFC)[6]+1.3, label= paste("d  = ", sum(logFC_BBBB_X_comps_df[logFC_BBBB_X_comps_df$treatment == "Mouse SP",]$logFC < 0), sep = "")) +
  annotate("text", size = 2, x = 3, y = summary(logFC_BBBB_X_comps_df[grep("LZ", logFC_BBBB_X_comps_df$treatment),]$logFC)[6]+2.1, label= paste("u = ", sum(logFC_BBBB_X_comps_df[logFC_BBBB_X_comps_df$treatment == "Mouse LZ",]$logFC > 0), sep = "")) +
  annotate("text", size = 2, x = 3, y = summary(logFC_BBBB_X_comps_df[grep("LZ", logFC_BBBB_X_comps_df$treatment),]$logFC)[6]+1.3, label= paste("d  = ", sum(logFC_BBBB_X_comps_df[logFC_BBBB_X_comps_df$treatment == "Mouse LZ",]$logFC < 0), sep = "")) +
  annotate("text", size = 2, x = 5, y = summary(logFC_BBBB_X_comps_df[grep("DIP", logFC_BBBB_X_comps_df$treatment),]$logFC)[6]+2.1, label= paste("u = ", sum(logFC_BBBB_X_comps_df[logFC_BBBB_X_comps_df$treatment == "Mouse DIP",]$logFC > 0), sep = "")) +
  annotate("text", size = 2, x = 5, y = summary(logFC_BBBB_X_comps_df[grep("DIP", logFC_BBBB_X_comps_df$treatment),]$logFC)[6]+1.3, label= paste("d  = ", sum(logFC_BBBB_X_comps_df[logFC_BBBB_X_comps_df$treatment == "Mouse DIP",]$logFC < 0), sep = "")) +
  
  annotate("text", size = 2, x = 2, y = summary(logFC_BBBB_X_comps_df[grep("SP", logFC_BBBB_X_comps_df$treatment),]$logFC)[6]+2.1, label= paste("u = ", sum(logFC_BBBB_X_comps_df[logFC_BBBB_X_comps_df$treatment == "Hamster SP",]$logFC > 0), sep = "")) +
  annotate("text", size = 2, x = 2, y = summary(logFC_BBBB_X_comps_df[grep("SP", logFC_BBBB_X_comps_df$treatment),]$logFC)[6]+1.3, label= paste("d  = ", sum(logFC_BBBB_X_comps_df[logFC_BBBB_X_comps_df$treatment == "Hamster SP",]$logFC < 0), sep = "")) +
  annotate("text", size = 2, x = 4, y = summary(logFC_BBBB_X_comps_df[grep("LZ", logFC_BBBB_X_comps_df$treatment),]$logFC)[6]+2.1, label= paste("u = ", sum(logFC_BBBB_X_comps_df[logFC_BBBB_X_comps_df$treatment == "Hamster LZ",]$logFC > 0), sep = "")) +
  annotate("text", size = 2, x = 4, y = summary(logFC_BBBB_X_comps_df[grep("LZ", logFC_BBBB_X_comps_df$treatment),]$logFC)[6]+1.3, label= paste("d  = ", sum(logFC_BBBB_X_comps_df[logFC_BBBB_X_comps_df$treatment == "Hamster LZ",]$logFC < 0), sep = "")) +
  annotate("text", size = 2, x = 6, y = summary(logFC_BBBB_X_comps_df[grep("DIP", logFC_BBBB_X_comps_df$treatment),]$logFC)[6]+2.1, label= paste("u = ", sum(logFC_BBBB_X_comps_df[logFC_BBBB_X_comps_df$treatment == "Hamster DIP",]$logFC > 0), sep = "")) +
  annotate("text", size = 2, x = 6, y = summary(logFC_BBBB_X_comps_df[grep("DIP", logFC_BBBB_X_comps_df$treatment),]$logFC)[6]+1.3, label= paste("d  = ", sum(logFC_BBBB_X_comps_df[logFC_BBBB_X_comps_df$treatment == "Hamster DIP",]$logFC < 0), sep = ""))

#Autosomal analysis
#paired wilcox tests to test for difference within stage between mouse and hamster that are then FDR corrected
logFC_BBBB_auto_pvalues <- 
  data.frame(names = c("SP", "LZ", "DIP"),
             pValues = c(wilcox.test(logFC_BBBB_auto_comps_df[logFC_BBBB_auto_comps_df$treatment == "Mouse SP",]$logFC, 
                                     logFC_BBBB_auto_comps_df[logFC_BBBB_auto_comps_df$treatment == "Hamster SP",]$logFC)$p.value,
                         
                         wilcox.test(logFC_BBBB_auto_comps_df[logFC_BBBB_auto_comps_df$treatment == "Mouse LZ",]$logFC, 
                                     logFC_BBBB_auto_comps_df[logFC_BBBB_auto_comps_df$treatment == "Hamster LZ",]$logFC)$p.value,
                         
                         wilcox.test(logFC_BBBB_auto_comps_df[logFC_BBBB_auto_comps_df$treatment == "Mouse DIP",]$logFC, 
                                     logFC_BBBB_auto_comps_df[logFC_BBBB_auto_comps_df$treatment == "Hamster DIP",]$logFC)$p.value))

logFC_BBBB_auto_pvalues$pValues[1:3] <- signif(p.adjust(logFC_BBBB_auto_pvalues[1:3,]$pValues, method = "fdr"), 3)
logFC_BBBB_auto_pvalues$pSymbols <- sig_conversion(logFC_BBBB_auto_pvalues$pValues)


logFC_BBBB_auto_comps_plot <- ggplot(logFC_BBBB_auto_comps_df, aes(x = treatment, y = logFC)) + 
  geom_hline(yintercept = 0, linewidth = 0.5) +
  geom_quasirandom(aes(col = treatment, fill = treatment), alpha = 0.3, size = point_size, width = 0.4) + 
  geom_violin(aes(col = treatment), width = 1, fill = NA) + 
  geom_boxplot(width=boxplot_width, outlier.shape = NA, alpha = 0.8) + 
  theme_minimal() + 
  scale_color_manual(values=c(sp_color, sp_color_dark,
                              lz_color, lz_color_dark,
                              dip_color, dip_color_dark)) +
  scale_fill_manual(values=c(sp_color, sp_color_dark,
                             lz_color, lz_color_dark,
                             dip_color, dip_color_dark)) +
  ylab("Differential expression (logFC)") + 
  ggtitle("Autosomes") + 
  scale_x_discrete(labels= rep(c("Mouse", "Hamster"), 3)) +
  ylim(-11.5, 16) +
  geom_signif(
    y_position = c(summary(logFC_BBBB_auto_comps_df[grep("SP", logFC_BBBB_auto_comps_df$treatment),]$logFC)[6]+3,
                   summary(logFC_BBBB_auto_comps_df[grep("LZ", logFC_BBBB_auto_comps_df$treatment),]$logFC)[6]+3,
                   summary(logFC_BBBB_auto_comps_df[grep("DIP", logFC_BBBB_auto_comps_df$treatment),]$logFC)[6]+3), 
    xmin = c(0.8,2.8,4.8), 
    xmax = c(2.2,4.2,6.2), 
    annotation = c(logFC_BBBB_auto_pvalues[logFC_BBBB_auto_pvalues$names == "SP",]$pSymbols,
                   logFC_BBBB_auto_pvalues[logFC_BBBB_auto_pvalues$names == "LZ",]$pSymbols,
                   logFC_BBBB_auto_pvalues[logFC_BBBB_auto_pvalues$names == "DIP",]$pSymbols), 
    tip_length = 0.01, textsize = text_size*0.5, size = 0.5) +
  theme(legend.position = "none", 
        text = element_text(size = text_size*1.5),
        panel.grid.minor = element_blank(),
        panel.grid.major.x = element_blank(),
        axis.title.x = element_blank(),
        axis.text.x = element_text(angle = 45),
        plot.title = element_text(hjust = 0.5)) +
  annotate("text", size = 2, x = 1, y = summary(logFC_BBBB_auto_comps_df[grep("SP", logFC_BBBB_auto_comps_df$treatment),]$logFC)[6]+2.1, label= paste("u = ", sum(logFC_BBBB_auto_comps_df[logFC_BBBB_auto_comps_df$treatment == "Mouse SP",]$logFC > 0), sep = "")) +
  annotate("text", size = 2, x = 1, y = summary(logFC_BBBB_auto_comps_df[grep("SP", logFC_BBBB_auto_comps_df$treatment),]$logFC)[6]+1.3, label= paste("d  = ", sum(logFC_BBBB_auto_comps_df[logFC_BBBB_auto_comps_df$treatment == "Mouse SP",]$logFC < 0), sep = "")) +
  annotate("text", size = 2, x = 3, y = summary(logFC_BBBB_auto_comps_df[grep("LZ", logFC_BBBB_auto_comps_df$treatment),]$logFC)[6]+2.1, label= paste("u = ", sum(logFC_BBBB_auto_comps_df[logFC_BBBB_auto_comps_df$treatment == "Mouse LZ",]$logFC > 0), sep = "")) +
  annotate("text", size = 2, x = 3, y = summary(logFC_BBBB_auto_comps_df[grep("LZ", logFC_BBBB_auto_comps_df$treatment),]$logFC)[6]+1.3, label= paste("d  = ", sum(logFC_BBBB_auto_comps_df[logFC_BBBB_auto_comps_df$treatment == "Mouse LZ",]$logFC < 0), sep = "")) +
  annotate("text", size = 2, x = 5, y = summary(logFC_BBBB_auto_comps_df[grep("DIP", logFC_BBBB_auto_comps_df$treatment),]$logFC)[6]+2.1, label= paste("u = ", sum(logFC_BBBB_auto_comps_df[logFC_BBBB_auto_comps_df$treatment == "Mouse DIP",]$logFC > 0), sep = "")) +
  annotate("text", size = 2, x = 5, y = summary(logFC_BBBB_auto_comps_df[grep("DIP", logFC_BBBB_auto_comps_df$treatment),]$logFC)[6]+1.3, label= paste("d  = ", sum(logFC_BBBB_auto_comps_df[logFC_BBBB_auto_comps_df$treatment == "Mouse DIP",]$logFC < 0), sep = "")) +
  
  annotate("text", size = 2, x = 2, y = summary(logFC_BBBB_auto_comps_df[grep("SP", logFC_BBBB_auto_comps_df$treatment),]$logFC)[6]+2.1, label= paste("u = ", sum(logFC_BBBB_auto_comps_df[logFC_BBBB_auto_comps_df$treatment == "Hamster SP",]$logFC > 0), sep = "")) +
  annotate("text", size = 2, x = 2, y = summary(logFC_BBBB_auto_comps_df[grep("SP", logFC_BBBB_auto_comps_df$treatment),]$logFC)[6]+1.3, label= paste("d  = ", sum(logFC_BBBB_auto_comps_df[logFC_BBBB_auto_comps_df$treatment == "Hamster SP",]$logFC < 0), sep = "")) +
  annotate("text", size = 2, x = 4, y = summary(logFC_BBBB_auto_comps_df[grep("LZ", logFC_BBBB_auto_comps_df$treatment),]$logFC)[6]+2.1, label= paste("u = ", sum(logFC_BBBB_auto_comps_df[logFC_BBBB_auto_comps_df$treatment == "Hamster LZ",]$logFC > 0), sep = "")) +
  annotate("text", size = 2, x = 4, y = summary(logFC_BBBB_auto_comps_df[grep("LZ", logFC_BBBB_auto_comps_df$treatment),]$logFC)[6]+1.3, label= paste("d  = ", sum(logFC_BBBB_auto_comps_df[logFC_BBBB_auto_comps_df$treatment == "Hamster LZ",]$logFC < 0), sep = "")) +
  annotate("text", size = 2, x = 6, y = summary(logFC_BBBB_auto_comps_df[grep("DIP", logFC_BBBB_auto_comps_df$treatment),]$logFC)[6]+2.1, label= paste("u = ", sum(logFC_BBBB_auto_comps_df[logFC_BBBB_auto_comps_df$treatment == "Hamster DIP",]$logFC > 0), sep = "")) +
  annotate("text", size = 2, x = 6, y = summary(logFC_BBBB_auto_comps_df[grep("DIP", logFC_BBBB_auto_comps_df$treatment),]$logFC)[6]+1.3, label= paste("d  = ", sum(logFC_BBBB_auto_comps_df[logFC_BBBB_auto_comps_df$treatment == "Hamster DIP",]$logFC < 0), sep = ""))


if(exportFigs == TRUE) {
  ggsave(filename ="exported_figures/hybrid_logFC_BBBB_both_comps_mouse_v_hamster.pdf", 
         plot_grid(logFC_BBBB_auto_comps_plot, logFC_BBBB_X_comps_plot, nrow = 1), width=full_plot_width, height=3)
} else {
  plot_grid(logFC_BBBB_auto_comps_plot, logFC_BBBB_X_comps_plot, nrow = 1)
}


#print average logFC for hybrid mouse and ham DE genes across spermatogenesis
logFC_BBBB_X_comps_df %>% group_by(treatment) %>%
  summarize_at('logFC',mean)

logFC_BBBB_auto_comps_df %>% group_by(treatment) %>%
  summarize_at('logFC',mean)



chi_square_table_df <- data.frame(rbind(calculate_chisq(logFC_BBBB_X_comps_df, "Mouse SP", "X"),
                                        calculate_chisq(logFC_BBBB_X_comps_df, "Mouse LZ", "X"),
                                        calculate_chisq(logFC_BBBB_X_comps_df, "Mouse DIP", "X"),
                                        calculate_chisq(logFC_BBBB_auto_comps_df, "Mouse SP", "auto"),
                                        calculate_chisq(logFC_BBBB_auto_comps_df, "Mouse LZ", "auto"),
                                        calculate_chisq(logFC_BBBB_auto_comps_df, "Mouse DIP", "auto"),
                                        calculate_chisq(logFC_BBBB_X_comps_df, "Hamster SP", "X"),
                                        calculate_chisq(logFC_BBBB_X_comps_df, "Hamster LZ", "X"),
                                        calculate_chisq(logFC_BBBB_X_comps_df, "Hamster DIP", "X"),
                                        calculate_chisq(logFC_BBBB_auto_comps_df, "Hamster SP", "auto"),
                                        calculate_chisq(logFC_BBBB_auto_comps_df, "Hamster LZ", "auto"),
                                        calculate_chisq(logFC_BBBB_auto_comps_df, "Hamster DIP", "auto")))

colnames(chi_square_table_df) <- c("Comparison", "df", "N", "X^2", "Unadj_Pval")
chi_square_table_df$Adj_Pval <- p.adjust(chi_square_table_df$Unadj_Pval)

chi_square_table_df
```

<h3>Compare the logFC of Differentially Expressed Genes between Hybrids and Parents with Different X Chromosome for Both Mouse and Hamster across Spermatogenesis</h3>
```{r hybrid-DE-logFC-graphs-SSSS}
#this section produces a supplemental companion to figure 3 in the manuscript and demonstrates how patterns of up and downregulation across stages in hybrids differs between systems (comparison to parent with different X chromosome)
Hyb.Mouse.SP.de <- read.csv("exported_files/hybrid_mouse_DomP_SP_DE_Genes.csv")
Hyb.Mouse.LZ.de <- read.csv("exported_files/hybrid_mouse_DomP_LZ_DE_Genes.csv")
Hyb.Mouse.DIP.de <-read.csv("exported_files/hybrid_mouse_DomP_DIP_DE_Genes.csv")

#X-linked
logFC_mouse_SP_X_df <- data.frame(logFC =  Hyb.Mouse.SP.de[Hyb.Mouse.SP.de$chr == "X",]$logFC, treatment = "Mouse SP")
logFC_mouse_LZ_X_df <- data.frame(logFC =  Hyb.Mouse.LZ.de[Hyb.Mouse.LZ.de$chr == "X",]$logFC, treatment = "Mouse LZ")
logFC_mouse_DIP_X_df <- data.frame(logFC = Hyb.Mouse.DIP.de[Hyb.Mouse.DIP.de$chr == "X",]$logFC, treatment = "Mouse DIP")

logFC_hamster_SP_SSSS_X_df <-  data.frame(logFC = Hyb.SS.SP.Dev.de[grep("^X", Hyb.SS.SP.Dev.de$chr),]$logFC, treatment  = "Hamster SP")
logFC_hamster_LZ_SSSS_X_df <-  data.frame(logFC = Hyb.SS.LZ.Dev.de[grep("^X", Hyb.SS.LZ.Dev.de$chr),]$logFC, treatment  = "Hamster LZ")
logFC_hamster_DIP_SSSS_X_df <- data.frame(logFC = Hyb.SS.DIP.Dev.de[grep("^X", Hyb.SS.DIP.Dev.de$chr),]$logFC, treatment = "Hamster DIP")

#auto
logFC_mouse_SP_auto_df <- data.frame(logFC =  Hyb.Mouse.SP.de[Hyb.Mouse.SP.de$chr != "X",]$logFC, treatment = "Mouse SP")
logFC_mouse_LZ_auto_df <- data.frame(logFC =  Hyb.Mouse.LZ.de[Hyb.Mouse.LZ.de$chr != "X",]$logFC, treatment = "Mouse LZ")
logFC_mouse_DIP_auto_df <- data.frame(logFC = Hyb.Mouse.DIP.de[Hyb.Mouse.DIP.de$chr != "X",]$logFC, treatment = "Mouse DIP")

logFC_hamster_SP_SSSS_auto_df <-  data.frame(logFC = Hyb.SS.SP.Dev.de[grep("^X", Hyb.SS.SP.Dev.de$chr, invert = TRUE),]$logFC, treatment  = "Hamster SP")
logFC_hamster_LZ_SSSS_auto_df <-  data.frame(logFC = Hyb.SS.LZ.Dev.de[grep("^X", Hyb.SS.LZ.Dev.de$chr, invert = TRUE),]$logFC, treatment  = "Hamster LZ")
logFC_hamster_DIP_SSSS_auto_df <- data.frame(logFC = Hyb.SS.DIP.Dev.de[grep("^X", Hyb.SS.DIP.Dev.de$chr, invert = TRUE),]$logFC, treatment = "Hamster DIP")

# logFC_hamster_SP_SSSS_df <-  data.frame(logFC = Hyb.SS.SP.Dev.de[grep("^X", Hyb.SS.SP.Dev.de$chr),]$logFC, treatment  = "SSSS_SP")
# logFC_hamster_LZ_SSSS_df <-  data.frame(logFC = Hyb.SS.LZ.Dev.de[grep("^X", Hyb.SS.LZ.Dev.de$chr),]$logFC, treatment  = "SSSS_LZ")
# logFC_hamster_DIP_SSSS_df <- data.frame(logFC = Hyb.SS.DIP.Dev.de[grep("^X", Hyb.SS.DIP.Dev.de$chr),]$logFC, treatment = "SSSS_DIP")


#X-linked analysis
logFC_SSSS_X_comps_df <- rbind(logFC_mouse_SP_X_df, logFC_hamster_SP_SSSS_X_df,
                               logFC_mouse_LZ_X_df, logFC_hamster_LZ_SSSS_X_df,
                               logFC_mouse_DIP_X_df, logFC_hamster_DIP_SSSS_X_df)

logFC_SSSS_X_comps_df$treatment <- factor(logFC_SSSS_X_comps_df$treatment, levels = c("Mouse SP", "Hamster SP", 
                                                                                      "Mouse LZ", "Hamster LZ", 
                                                                                      "Mouse DIP", "Hamster DIP"))


logFC_SSSS_auto_comps_df <- rbind(logFC_mouse_SP_auto_df, logFC_hamster_SP_SSSS_auto_df,
                                  logFC_mouse_LZ_auto_df, logFC_hamster_LZ_SSSS_auto_df,
                                  logFC_mouse_DIP_auto_df, logFC_hamster_DIP_SSSS_auto_df)

logFC_SSSS_auto_comps_df$treatment <- factor(logFC_SSSS_auto_comps_df$treatment, levels = c("Mouse SP", "Hamster SP", 
                                                                                            "Mouse LZ", "Hamster LZ", 
                                                                                            "Mouse DIP", "Hamster DIP"))

#paired wilcox tests to test for difference within stage between mouse and hamster that are then FDR corrected
logFC_SSSS_X_pvalues <- data.frame(names = c("SP", "LZ", "DIP"),
                                   pValues = c(wilcox.test(logFC_SSSS_X_comps_df[logFC_SSSS_X_comps_df$treatment == "Mouse SP",]$logFC, 
                                                           logFC_SSSS_X_comps_df[logFC_SSSS_X_comps_df$treatment == "Hamster SP",]$logFC)$p.value,
                                               
                                               wilcox.test(logFC_SSSS_X_comps_df[logFC_SSSS_X_comps_df$treatment == "Mouse LZ",]$logFC, 
                                                           logFC_SSSS_X_comps_df[logFC_SSSS_X_comps_df$treatment == "Hamster LZ",]$logFC)$p.value,
                                               
                                               wilcox.test(logFC_SSSS_X_comps_df[logFC_SSSS_X_comps_df$treatment == "Mouse DIP",]$logFC, 
                                                           logFC_SSSS_X_comps_df[logFC_SSSS_X_comps_df$treatment == "Hamster DIP",]$logFC)$p.value))

logFC_SSSS_X_pvalues$pValues[1:3] <- signif(p.adjust(logFC_SSSS_X_pvalues[1:3,]$pValues, method = "fdr"), 3)
logFC_SSSS_X_pvalues$pSymbols <- sig_conversion(logFC_SSSS_X_pvalues$pValues)


logFC_SSSS_X_comps_plot <- ggplot(logFC_SSSS_X_comps_df, aes(x = treatment, y = logFC)) + 
  geom_hline(yintercept = 0, linewidth = 0.5) +
  geom_quasirandom(aes(col = treatment, fill = treatment), alpha = 0.3, size = point_size, width = 0.4) + 
  geom_violin(aes(col = treatment), width = 1, fill = NA) + 
  geom_boxplot(width=boxplot_width, outlier.shape = NA, alpha = 0.8) + 
  theme_minimal() + 
  scale_color_manual(values=c(sp_color, sp_color_dark,
                              lz_color, lz_color_dark,
                              dip_color, dip_color_dark)) +
  scale_fill_manual(values=c(sp_color, sp_color_dark,
                             lz_color, lz_color_dark,
                             dip_color, dip_color_dark)) +
  ylab("Differential expression (logFC)") + 
  ggtitle("X Chromosome") + 
  scale_x_discrete(labels= rep(c("Mouse", "Hamster"), 3)) +
  ylim(-13, 16) +
  geom_signif(
    y_position = c(summary(logFC_SSSS_X_comps_df[grep("SP", logFC_SSSS_X_comps_df$treatment),]$logFC)[6]+3,
                   summary(logFC_SSSS_X_comps_df[grep("LZ", logFC_SSSS_X_comps_df$treatment),]$logFC)[6]+3,
                   summary(logFC_SSSS_X_comps_df[grep("DIP", logFC_SSSS_X_comps_df$treatment),]$logFC)[6]+3), 
    xmin = c(0.8,2.8,4.8), 
    xmax = c(2.2,4.2,6.2), 
    annotation = c(logFC_SSSS_X_pvalues[logFC_SSSS_X_pvalues$names == "SP",]$pSymbols,
                   logFC_SSSS_X_pvalues[logFC_SSSS_X_pvalues$names == "LZ",]$pSymbols,
                   logFC_SSSS_X_pvalues[logFC_SSSS_X_pvalues$names == "DIP",]$pSymbols), 
    tip_length = 0.01, textsize = text_size*0.5, size = 0.5) +
  theme(legend.position = "none", 
        text = element_text(size = text_size*1.5),
        panel.grid.minor = element_blank(),
        panel.grid.major.x = element_blank(),
        axis.title.x = element_blank(),
        axis.text.x = element_text(angle = 45),
        plot.title = element_text(hjust = 0.5)) +
  annotate("text", size = 2, x = 1, y = summary(logFC_SSSS_X_comps_df[grep("SP", logFC_SSSS_X_comps_df$treatment),]$logFC)[6]+2.1, label= paste("u = ", sum(logFC_SSSS_X_comps_df[logFC_SSSS_X_comps_df$treatment == "Mouse SP",]$logFC > 0), sep = "")) +
  annotate("text", size = 2, x = 1, y = summary(logFC_SSSS_X_comps_df[grep("SP", logFC_SSSS_X_comps_df$treatment),]$logFC)[6]+1.3, label= paste("d  = ", sum(logFC_SSSS_X_comps_df[logFC_SSSS_X_comps_df$treatment == "Mouse SP",]$logFC < 0), sep = "")) +
  annotate("text", size = 2, x = 3, y = summary(logFC_SSSS_X_comps_df[grep("LZ", logFC_SSSS_X_comps_df$treatment),]$logFC)[6]+2.1, label= paste("u = ", sum(logFC_SSSS_X_comps_df[logFC_SSSS_X_comps_df$treatment == "Mouse LZ",]$logFC > 0), sep = "")) +
  annotate("text", size = 2, x = 3, y = summary(logFC_SSSS_X_comps_df[grep("LZ", logFC_SSSS_X_comps_df$treatment),]$logFC)[6]+1.3, label= paste("d  = ", sum(logFC_SSSS_X_comps_df[logFC_SSSS_X_comps_df$treatment == "Mouse LZ",]$logFC < 0), sep = "")) +
  annotate("text", size = 2, x = 5, y = summary(logFC_SSSS_X_comps_df[grep("DIP", logFC_SSSS_X_comps_df$treatment),]$logFC)[6]+2.1, label= paste("u = ", sum(logFC_SSSS_X_comps_df[logFC_SSSS_X_comps_df$treatment == "Mouse DIP",]$logFC > 0), sep = "")) +
  annotate("text", size = 2, x = 5, y = summary(logFC_SSSS_X_comps_df[grep("DIP", logFC_SSSS_X_comps_df$treatment),]$logFC)[6]+1.3, label= paste("d  = ", sum(logFC_SSSS_X_comps_df[logFC_SSSS_X_comps_df$treatment == "Mouse DIP",]$logFC < 0), sep = "")) +
  
  annotate("text", size = 2, x = 2, y = summary(logFC_SSSS_X_comps_df[grep("SP", logFC_SSSS_X_comps_df$treatment),]$logFC)[6]+2.1, label= paste("u = ", sum(logFC_SSSS_X_comps_df[logFC_SSSS_X_comps_df$treatment == "Hamster SP",]$logFC > 0), sep = "")) +
  annotate("text", size = 2, x = 2, y = summary(logFC_SSSS_X_comps_df[grep("SP", logFC_SSSS_X_comps_df$treatment),]$logFC)[6]+1.3, label= paste("d  = ", sum(logFC_SSSS_X_comps_df[logFC_SSSS_X_comps_df$treatment == "Hamster SP",]$logFC < 0), sep = "")) +
  annotate("text", size = 2, x = 4, y = summary(logFC_SSSS_X_comps_df[grep("LZ", logFC_SSSS_X_comps_df$treatment),]$logFC)[6]+2.1, label= paste("u = ", sum(logFC_SSSS_X_comps_df[logFC_SSSS_X_comps_df$treatment == "Hamster LZ",]$logFC > 0), sep = "")) +
  annotate("text", size = 2, x = 4, y = summary(logFC_SSSS_X_comps_df[grep("LZ", logFC_SSSS_X_comps_df$treatment),]$logFC)[6]+1.3, label= paste("d  = ", sum(logFC_SSSS_X_comps_df[logFC_SSSS_X_comps_df$treatment == "Hamster LZ",]$logFC < 0), sep = "")) +
  annotate("text", size = 2, x = 6, y = summary(logFC_SSSS_X_comps_df[grep("DIP", logFC_SSSS_X_comps_df$treatment),]$logFC)[6]+2.1, label= paste("u = ", sum(logFC_SSSS_X_comps_df[logFC_SSSS_X_comps_df$treatment == "Hamster DIP",]$logFC > 0), sep = "")) +
  annotate("text", size = 2, x = 6, y = summary(logFC_SSSS_X_comps_df[grep("DIP", logFC_SSSS_X_comps_df$treatment),]$logFC)[6]+1.3, label= paste("d  = ", sum(logFC_SSSS_X_comps_df[logFC_SSSS_X_comps_df$treatment == "Hamster DIP",]$logFC < 0), sep = ""))

#Autosomal analysis
#paired wilcox tests to test for difference within stage between mouse and hamster that are then FDR corrected
logFC_SSSS_auto_pvalues <- data.frame(names = c("SP", "LZ", "DIP"),
                                      pValues = c(wilcox.test(logFC_SSSS_auto_comps_df[logFC_SSSS_auto_comps_df$treatment == "Mouse SP",]$logFC, 
                                                              logFC_SSSS_auto_comps_df[logFC_SSSS_auto_comps_df$treatment == "Hamster SP",]$logFC)$p.value,
                                                  
                                                  wilcox.test(logFC_SSSS_auto_comps_df[logFC_SSSS_auto_comps_df$treatment == "Mouse LZ",]$logFC, 
                                                              logFC_SSSS_auto_comps_df[logFC_SSSS_auto_comps_df$treatment == "Hamster LZ",]$logFC)$p.value,
                                                  
                                                  wilcox.test(logFC_SSSS_auto_comps_df[logFC_SSSS_auto_comps_df$treatment == "Mouse DIP",]$logFC, 
                                                              logFC_SSSS_auto_comps_df[logFC_SSSS_auto_comps_df$treatment == "Hamster DIP",]$logFC)$p.value))

logFC_SSSS_auto_pvalues$pValues[1:3] <- signif(p.adjust(logFC_SSSS_auto_pvalues[1:3,]$pValues, method = "fdr"), 3)
logFC_SSSS_auto_pvalues$pSymbols <- sig_conversion(logFC_SSSS_auto_pvalues$pValues)


logFC_SSSS_auto_comps_plot <- ggplot(logFC_SSSS_auto_comps_df, aes(x = treatment, y = logFC)) + 
  geom_hline(yintercept = 0, linewidth = 0.5) +
  geom_quasirandom(aes(col = treatment, fill = treatment), alpha = 0.3, size = point_size, width = 0.4) + 
  geom_violin(aes(col = treatment), width = 1, fill = NA) + 
  geom_boxplot(width=boxplot_width, outlier.shape = NA, alpha = 0.8) + 
  theme_minimal() + 
  scale_color_manual(values=c(sp_color, sp_color_dark,
                              lz_color, lz_color_dark,
                              dip_color, dip_color_dark)) +
  scale_fill_manual(values=c(sp_color, sp_color_dark,
                             lz_color, lz_color_dark,
                             dip_color, dip_color_dark)) +
  ylab("Differential expression (logFC)") + 
  ggtitle("Autosomes") + 
  scale_x_discrete(labels= rep(c("Mouse", "Hamster"), 3)) +
  ylim(-13, 16) +
  geom_signif(
    y_position = c(summary(logFC_SSSS_auto_comps_df[grep("SP", logFC_SSSS_auto_comps_df$treatment),]$logFC)[6]+3,
                   summary(logFC_SSSS_auto_comps_df[grep("LZ", logFC_SSSS_auto_comps_df$treatment),]$logFC)[6]+3,
                   summary(logFC_SSSS_auto_comps_df[grep("DIP", logFC_SSSS_auto_comps_df$treatment),]$logFC)[6]+3), 
    xmin = c(0.8,2.8,4.8), 
    xmax = c(2.2,4.2,6.2), 
    annotation = c(logFC_SSSS_auto_pvalues[logFC_SSSS_auto_pvalues$names == "SP",]$pSymbols,
                   logFC_SSSS_auto_pvalues[logFC_SSSS_auto_pvalues$names == "LZ",]$pSymbols,
                   logFC_SSSS_auto_pvalues[logFC_SSSS_auto_pvalues$names == "DIP",]$pSymbols), 
    tip_length = 0.01, textsize = text_size*0.5, size = 0.5) +
  theme(legend.position = "none", 
        text = element_text(size = text_size*1.5),
        panel.grid.minor = element_blank(),
        panel.grid.major.x = element_blank(),
        axis.title.x = element_blank(),
        axis.text.x = element_text(angle = 45),
        plot.title = element_text(hjust = 0.5)) +
  annotate("text", size = 2, x = 1, y = summary(logFC_SSSS_auto_comps_df[grep("SP", logFC_SSSS_auto_comps_df$treatment),]$logFC)[6]+2.1, label= paste("u = ", sum(logFC_SSSS_auto_comps_df[logFC_SSSS_auto_comps_df$treatment == "Mouse SP",]$logFC > 0), sep = "")) +
  annotate("text", size = 2, x = 1, y = summary(logFC_SSSS_auto_comps_df[grep("SP", logFC_SSSS_auto_comps_df$treatment),]$logFC)[6]+1.3, label= paste("d  = ", sum(logFC_SSSS_auto_comps_df[logFC_SSSS_auto_comps_df$treatment == "Mouse SP",]$logFC < 0), sep = "")) +
  annotate("text", size = 2, x = 3, y = summary(logFC_SSSS_auto_comps_df[grep("LZ", logFC_SSSS_auto_comps_df$treatment),]$logFC)[6]+2.1, label= paste("u = ", sum(logFC_SSSS_auto_comps_df[logFC_SSSS_auto_comps_df$treatment == "Mouse LZ",]$logFC > 0), sep = "")) +
  annotate("text", size = 2, x = 3, y = summary(logFC_SSSS_auto_comps_df[grep("LZ", logFC_SSSS_auto_comps_df$treatment),]$logFC)[6]+1.3, label= paste("d  = ", sum(logFC_SSSS_auto_comps_df[logFC_SSSS_auto_comps_df$treatment == "Mouse LZ",]$logFC < 0), sep = "")) +
  annotate("text", size = 2, x = 5, y = summary(logFC_SSSS_auto_comps_df[grep("DIP", logFC_SSSS_auto_comps_df$treatment),]$logFC)[6]+2.1, label= paste("u = ", sum(logFC_SSSS_auto_comps_df[logFC_SSSS_auto_comps_df$treatment == "Mouse DIP",]$logFC > 0), sep = "")) +
  annotate("text", size = 2, x = 5, y = summary(logFC_SSSS_auto_comps_df[grep("DIP", logFC_SSSS_auto_comps_df$treatment),]$logFC)[6]+1.3, label= paste("d  = ", sum(logFC_SSSS_auto_comps_df[logFC_SSSS_auto_comps_df$treatment == "Mouse DIP",]$logFC < 0), sep = "")) +
  
  annotate("text", size = 2, x = 2, y = summary(logFC_SSSS_auto_comps_df[grep("SP", logFC_SSSS_auto_comps_df$treatment),]$logFC)[6]+2.1, label= paste("u = ", sum(logFC_SSSS_auto_comps_df[logFC_SSSS_auto_comps_df$treatment == "Hamster SP",]$logFC > 0), sep = "")) +
  annotate("text", size = 2, x = 2, y = summary(logFC_SSSS_auto_comps_df[grep("SP", logFC_SSSS_auto_comps_df$treatment),]$logFC)[6]+1.3, label= paste("d  = ", sum(logFC_SSSS_auto_comps_df[logFC_SSSS_auto_comps_df$treatment == "Hamster SP",]$logFC < 0), sep = "")) +
  annotate("text", size = 2, x = 4, y = summary(logFC_SSSS_auto_comps_df[grep("LZ", logFC_SSSS_auto_comps_df$treatment),]$logFC)[6]+2.1, label= paste("u = ", sum(logFC_SSSS_auto_comps_df[logFC_SSSS_auto_comps_df$treatment == "Hamster LZ",]$logFC > 0), sep = "")) +
  annotate("text", size = 2, x = 4, y = summary(logFC_SSSS_auto_comps_df[grep("LZ", logFC_SSSS_auto_comps_df$treatment),]$logFC)[6]+1.3, label= paste("d  = ", sum(logFC_SSSS_auto_comps_df[logFC_SSSS_auto_comps_df$treatment == "Hamster LZ",]$logFC < 0), sep = "")) +
  annotate("text", size = 2, x = 6, y = summary(logFC_SSSS_auto_comps_df[grep("DIP", logFC_SSSS_auto_comps_df$treatment),]$logFC)[6]+2.1, label= paste("u = ", sum(logFC_SSSS_auto_comps_df[logFC_SSSS_auto_comps_df$treatment == "Hamster DIP",]$logFC > 0), sep = "")) +
  annotate("text", size = 2, x = 6, y = summary(logFC_SSSS_auto_comps_df[grep("DIP", logFC_SSSS_auto_comps_df$treatment),]$logFC)[6]+1.3, label= paste("d  = ", sum(logFC_SSSS_auto_comps_df[logFC_SSSS_auto_comps_df$treatment == "Hamster DIP",]$logFC < 0), sep = ""))



if(exportFigs == TRUE) {
  ggsave(filename ="exported_figures/hybrid_logFC_SSSS_both_comps_mouse_v_hamster.pdf", 
         plot_grid(logFC_SSSS_auto_comps_plot, logFC_SSSS_X_comps_plot, nrow = 1), width=full_plot_width, height=3)
} else {
  plot_grid(logFC_SSSS_auto_comps_plot, logFC_SSSS_X_comps_plot, nrow = 1)
}

```

<h3>Hybrid hypergeometric tests to test if different chromosomes are enriched for DEs</h3>
```{r hybrid-DE-hypergeometric-tests}
#testing the Hyb x BBBB contrast
contrasts2sum <- c("Hyb.BB") 
cell.exp.list<- Hyb.BB.cell.exp.list

# uncomment if you want to test the Hyb x SSSS contrast
# contrasts2sum <- c("Hyb.SS")
# cell.exp.list<- Hyb.SS.cell.exp.list

# uncomment if you want to test the Hyb x Comb (SSSS and BBBB) contrast
# contrasts2sum <- c("Hyb.Comb")
# cell.exp.list<- Hyb.Comb.cell.exp.list

#set up an empty data frame to add results to
results_printing <- c()

#set up a list of chromosomes to iterate through (only including major scaffolds)
chr.list <- chrom_sizes_large$chromosome

#set up colors for plotting
under_colors <- c(sp_color, lz_color, dip_color)
names(under_colors) <- c("SP", "LZ", "DIP")
over_colors <- c(sp_color_dark, lz_color_dark, dip_color_dark)
names(over_colors) <- c("SP", "LZ", "DIP")

#create graph title objects for easier plotting
enrichment_title_list <- c("Obs vs Exp Hybrid DE Genes in SP","Obs vs Exp Hybrid DE Genes in LZ","Obs vs Exp Hybrid DE Genes in DIP")

#set FDR
enrichment_FDR <- 0.05
enrichment_title_list <- paste(enrichment_title_list, " with FDR = ", enrichment_FDR, sep = "")

#retrieve list of expressed genes and DE genes for hypergeometric tests
focal.cont <- c(paste(contrasts2sum, ".SP", sep = ""), 
                paste(contrasts2sum, ".LZ", sep = ""), 
                paste(contrasts2sum, ".DIP", sep = ""))
de.list <- c(paste(contrasts2sum, ".SP", ".Dev.de" , sep = ""), 
             paste(contrasts2sum, ".LZ", ".Dev.de" , sep = ""), 
             paste(contrasts2sum, ".DIP", ".Dev.de" , sep = ""))

#set up empty dataframes to take under- and over-enrichment data
under.enrich <- data.frame(row.names = chr.list)
under.enrich.fdr <- data.frame(row.names = chr.list)

over.enrich <- data.frame(row.names = chr.list)
over.enrich.fdr <- data.frame(row.names = chr.list)

#loop through each stage then each chromosome to determine if chromosome is over- or under-enriched for DE genes relative to expectatations
for (contrast in focal.cont){
  under <- c()
  over <- c()
  exp <- c()
  obs <- c()
  
  for (chr in chr.list){
    #hypergeometric tests should take the form of:
    #phyper(Overlap, group2, Total-group2, group1,  lower.tail= TRUE) for under-enrichment
    #phyper(Overlap - 1, group2, Total-group2, group1,  lower.tail= FALSE) for over-enrichment
    #where:
    #Total = expressed genes on all chromosomes for appropriate contrast: exp_genes_on_chr
    #Group1 = expressed genes on given chromosome for appropriate contrast: exp_genes_on_chr
    #Group2 = all DE genes across chromosomes for appropriate contrast: all_DE_genes
    #Overlap = DE genes on given chromosome for appropriate contrast: DE_on_chr
    DE_on_chr <- dim(get(grep(contrast, de.list, value = TRUE))[get(grep(contrast, de.list, value = TRUE))$chr == chr,])[1]
    all_DE_genes <- dim(get(grep(contrast, de.list, value = TRUE)))[1]
    all_expressed_genes <- sum(get(grep(contrast, cell.exp.list, value = TRUE)) %in% rownames(annotations))
    non_DE_genes <- all_expressed_genes - all_DE_genes
    exp_genes_on_chr <- sum(get(grep(contrast, cell.exp.list, value = TRUE)) %in% rownames(annotations[grep(paste("^",chr,"$", sep = ""), annotations$chr),]))
    
    #run phyper tests
    under.p <- phyper(DE_on_chr, all_DE_genes, non_DE_genes, exp_genes_on_chr, lower.tail = TRUE)
    over.p <- phyper(DE_on_chr - 1, all_DE_genes, non_DE_genes, exp_genes_on_chr, lower.tail = FALSE)
    #save enrichment p-values for later FDR correction
    under <- c(under, round(under.p, digits = 5))
    over <- c(over, round(over.p, digits = 5))
    
    #save number of observed and expected genes for plotting comparisons
    exp <- c(exp, round((all_DE_genes/all_expressed_genes)*exp_genes_on_chr,digits = 0))
    obs <- c(obs, DE_on_chr)
  }			
  
  #FDR correct p_values
  under.enrich.fdr[,which(focal.cont %in% contrast)] <- round(p.adjust(under, method = "fdr"), digits = 3)
  under.enrich[,which(focal.cont %in% contrast)] <- under
  over.enrich.fdr[,which(focal.cont %in% contrast)] <- round(p.adjust(over, method = "fdr"), digits = 3)
  over.enrich[,which(focal.cont %in% contrast)] <- over 
  names(exp) <- paste(chr.list)
  names(obs) <- paste(chr.list)
}

#writing out chromosomes with significant enrichment for DEs to processing files
for(stage in 1:3) {
  sig_df <- data.frame(chr = rownames(under.enrich.fdr), 
                       underP = under.enrich.fdr[,stage], 
                       overP = over.enrich.fdr[,stage])
  sig_df$enriched <- ""
  sig_df$direction <- ""
  sig_df$sig <- ""
  sig_df$ringText <- ""
  
  #processing under-enriched chromosomes
  if(sum(sig_df$underP < 0.05) != 0) {
    sig_df[sig_df$underP < 0.05 ,]$ringText <- "-"
    sig_df[sig_df$underP < 0.05 ,]$sig <- "*"
    sig_df[sig_df$underP < 0.05,]$direction <- "(-)"   
    sig_df[sig_df$underP < 0.05,]$enriched <- "yes"
  } else {
    sig_df[!(sig_df$underP < 0.05),]$enriched <- "no"
  }
  if(sum(sig_df$underP < 0.01) != 0) {
    sig_df[sig_df$underP < 0.01 ,]$ringText <- "- -"
    sig_df[sig_df$underP < 0.01 ,]$sig <- "**"
  }
  if(sum(sig_df$underP < 0.001) != 0) {
    sig_df[sig_df$underP < 0.001,]$ringText <- "- - -"
    sig_df[sig_df$underP < 0.001,]$sig <- "***"
  }
  
  #processing over-enriched chromosomes
  if(sum(sig_df$overP < 0.05) != 0) {
    sig_df[sig_df$overP < 0.05 ,]$ringText <- "+"
    sig_df[sig_df$overP < 0.05,]$sig <- "*"
    sig_df[sig_df$overP < 0.05,]$direction <- "(+)" 
    sig_df[sig_df$overP < 0.05,]$enriched <- "yes"
  } else {
    sig_df[!(sig_df$overP < 0.05),]$enriched <- "no"
  }
  
  if(sum(sig_df$overP < 0.01) != 0) {
    sig_df[sig_df$overP < 0.01 ,]$ringText <- "++"
    sig_df[sig_df$overP < 0.01,]$sig <- "**"
  }
  if(sum(sig_df$overP < 0.001) != 0) {
    sig_df[sig_df$overP < 0.001,]$ringText <- "+++"
    sig_df[sig_df$overP < 0.001,]$sig <- "***"
  }
  
  write.csv(sig_df, file = paste("circos_processing_scripts/hamster_circos_hybrids/data/", focal.cont[stage], "_sig.csv", sep = ""), row.names = FALSE)
}
```

<h3>DE Gene List Export</h3>
```{r export_DE_genes_with_orthologs_for_GO_analysis}
#this chunk creates lists of DE genes and background gene lists for GO enrichment analysis in David (visualization of GO terms is next chunk)
#write file of significant DE genes for SP but only for DE genes with orthologs
SP_DE_list <- Hyb.BB.SP.Dev.tt$table[,"PValue"]
names(SP_DE_list) <- recode(rownames(Hyb.BB.SP.Dev.tt), 
                            !!!setNames(as.character(orthology_df$mGene), orthology_df$hGene))
SP_DE_list <- SP_DE_list[grepl("ENS", names(SP_DE_list))]

#DE_lists
write.table(names(SP_DE_list[SP_DE_list < 0.05]), file = "go_enrichment/Hyb.BB.SP.Dev.de_list.txt", 
            sep = "\t", quote = FALSE, row.names = FALSE, col.names = FALSE)
#background is all genes expressed in contrast
write.table(names(SP_DE_list), file = "go_enrichment/Hyb.BB.SP.Dev.bg_list.txt", 
            sep = "\t", quote = FALSE, row.names = FALSE, col.names = FALSE)

write.table(grep("ENS", recode(Hyb.Comb.SP.Dev.de$gene, 
                               !!!setNames(as.character(orthology_df$mGene), orthology_df$hGene)), value = TRUE), 
            file = "go_enrichment/Hyb.Comb.SP.Dev.de_list.txt", 
            sep = "\t", quote = FALSE, row.names = FALSE, col.names = FALSE)


#write file of significant DE genes for LZ but only for DE genes with orthologs
LZ_DE_list <- Hyb.BB.LZ.Dev.tt$table[,"PValue"]
names(LZ_DE_list) <- recode(rownames(Hyb.BB.LZ.Dev.tt), 
                            !!!setNames(as.character(orthology_df$mGene), orthology_df$hGene))
LZ_DE_list <- LZ_DE_list[grepl("ENS", names(LZ_DE_list))]

write.table(names(LZ_DE_list[LZ_DE_list < 0.05]), file = "go_enrichment/Hyb.BB.LZ.Dev.de_list.txt", 
            sep = "\t", quote = FALSE, row.names = FALSE, col.names = FALSE)
write.table(names(LZ_DE_list), file = "go_enrichment/Hyb.BB.LZ.Dev.bg_list.txt", 
            sep = "\t", quote = FALSE, row.names = FALSE, col.names = FALSE)

write.table(grep("ENS", recode(Hyb.Comb.LZ.Dev.de$gene, 
                               !!!setNames(as.character(orthology_df$mGene), orthology_df$hGene)), value = TRUE), 
            file = "go_enrichment/Hyb.Comb.LZ.Dev.de_list.txt", 
            sep = "\t", quote = FALSE, row.names = FALSE, col.names = FALSE)


#write file of significant DE genes for DIP but only for DE genes with orthologs
DIP_DE_list <- Hyb.BB.DIP.Dev.tt$table[,"PValue"]
names(DIP_DE_list) <- recode(rownames(Hyb.BB.DIP.Dev.tt), 
                             !!!setNames(as.character(orthology_df$mGene), orthology_df$hGene))
DIP_DE_list <- DIP_DE_list[grepl("ENS", names(DIP_DE_list))]

write.table(names(DIP_DE_list[DIP_DE_list < 0.05]), file = "go_enrichment/Hyb.BB.DIP.Dev.de_list.txt", 
            sep = "\t", quote = FALSE, row.names = FALSE, col.names = FALSE)
write.table(names(DIP_DE_list), file = "go_enrichment/Hyb.BB.DIP.Dev.bg_list.txt", 
            sep = "\t", quote = FALSE, row.names = FALSE, col.names = FALSE)

write.table(grep("ENS", recode(Hyb.Comb.DIP.Dev.de$gene, 
                               !!!setNames(as.character(orthology_df$mGene), orthology_df$hGene)), value = TRUE), 
            file = "go_enrichment/Hyb.Comb.DIP.Dev.de_list.txt", 
            sep = "\t", quote = FALSE, row.names = FALSE, col.names = FALSE)
```

<h3>Plot GO term enrichment from DE genes</h3>
```{r go-enrichment-bp-visualization}
#produces supplement figure S10 and main figure 5D
#format DAVID results for plotting for genes that were DE in SP between hybrids and both parents
Hyb.Comb.SP.Dev.go_terms <- read.table("go_enrichment/Hyb.Comb.SP.Dev.go_terms.txt", 
                                       header = TRUE, row.names=NULL, sep = "\t")
Hyb.Comb.SP.Dev.sig_go_terms <- Hyb.Comb.SP.Dev.go_terms[Hyb.Comb.SP.Dev.go_terms$FDR < 0.05,]
Hyb.Comb.SP.Dev.bp <- Hyb.Comb.SP.Dev.sig_go_terms[Hyb.Comb.SP.Dev.sig_go_terms$Category == "GOTERM_BP_DIRECT",]
Hyb.Comb.SP.Dev.bp$Term <- gsub("^GO:[0-9]+~", "", Hyb.Comb.SP.Dev.bp$Term)
setorder(Hyb.Comb.SP.Dev.go_terms, FDR)
Hyb.Comb.SP.Dev.top10_bp <- head(Hyb.Comb.SP.Dev.go_terms[Hyb.Comb.SP.Dev.go_terms$Category == "GOTERM_BP_DIRECT",], 10)
Hyb.Comb.SP.Dev.top10_bp$sig <- ""
Hyb.Comb.SP.Dev.top10_bp[Hyb.Comb.SP.Dev.top10_bp$FDR < 0.05,]$sig <- "yes"
Hyb.Comb.SP.Dev.top10_bp$Term <- gsub("^GO:[0-9]+~", "", Hyb.Comb.SP.Dev.top10_bp$Term)

#plot top ten most significant GO terms for SP
sp_go_plot_top10 <- ggplot(data = Hyb.Comb.SP.Dev.top10_bp, aes(x = Fold.Enrichment, y = reorder(Term, Fold.Enrichment), size = Count, col = sig)) +
  geom_point() +
  theme_minimal() + 
  scale_color_manual(values = c(sp_color), name = "Significance") +
  labs(x = "Fold enrichment", y = "") + 
  scale_size_continuous(name = "# Genes")


#format David results for plotting for genes that were DE in LZ between hybrids and both parents
Hyb.Comb.LZ.Dev.go_terms <- read.table("go_enrichment/Hyb.Comb.LZ.Dev.go_terms.txt", 
                                       header = TRUE, row.names=NULL, sep = "\t")
Hyb.Comb.LZ.Dev.sig_go_terms <- Hyb.Comb.LZ.Dev.go_terms[Hyb.Comb.LZ.Dev.go_terms$FDR < 0.05,]
Hyb.Comb.LZ.Dev.bp <- Hyb.Comb.LZ.Dev.sig_go_terms[Hyb.Comb.LZ.Dev.sig_go_terms$Category == "GOTERM_BP_DIRECT",]
Hyb.Comb.LZ.Dev.bp$Term <- gsub("^GO:[0-9]+~", "", Hyb.Comb.LZ.Dev.bp$Term)
setorder(Hyb.Comb.SP.Dev.go_terms, FDR)
Hyb.Comb.LZ.Dev.top10_bp <- head(Hyb.Comb.LZ.Dev.go_terms[Hyb.Comb.LZ.Dev.go_terms$Category == "GOTERM_BP_DIRECT",], 10)
Hyb.Comb.LZ.Dev.top10_bp$sig <- ""
Hyb.Comb.LZ.Dev.top10_bp[Hyb.Comb.LZ.Dev.top10_bp$FDR < 0.05,]$sig <- "yes"
Hyb.Comb.LZ.Dev.top10_bp[Hyb.Comb.LZ.Dev.top10_bp$FDR > 0.05,]$sig <- "no"
Hyb.Comb.LZ.Dev.top10_bp$Term <- gsub("^GO:[0-9]+~", "", Hyb.Comb.LZ.Dev.top10_bp$Term)

#plot top ten most significant GO terms for SP
lz_go_plot_top10 <- ggplot(data = Hyb.Comb.LZ.Dev.top10_bp, aes(x = Fold.Enrichment, y = reorder(Term, Fold.Enrichment), size = Count, col = sig)) +
  geom_point() +
  theme_minimal() + 
  scale_color_manual(values = c(gray_3, lz_color), name = "Significance") +
  labs(x = "Fold enrichment", y = "") + 
  scale_size_continuous(name = "# Genes")

#format David results for plotting for genes that were DE in DIP between hybrids and both parents
Hyb.Comb.DIP.Dev.go_terms <- suppressWarnings(read.table("go_enrichment/Hyb.Comb.DIP.Dev.go_terms.txt", 
                                        header = TRUE, row.names=NULL, sep = "\t"))

Hyb.Comb.DIP.Dev.go_terms <- Hyb.Comb.DIP.Dev.go_terms[!is.na(Hyb.Comb.DIP.Dev.go_terms$FDR),]
Hyb.Comb.DIP.Dev.sig_go_terms <- Hyb.Comb.DIP.Dev.go_terms[Hyb.Comb.DIP.Dev.go_terms$FDR < 0.05,]
Hyb.Comb.DIP.Dev.bp <- Hyb.Comb.DIP.Dev.sig_go_terms[Hyb.Comb.DIP.Dev.sig_go_terms$Category == "GOTERM_BP_DIRECT",]
Hyb.Comb.DIP.Dev.bp$Term <- gsub("^GO:[0-9]+~", "", Hyb.Comb.DIP.Dev.bp$Term)
setorder(Hyb.Comb.DIP.Dev.go_terms, FDR)
Hyb.Comb.DIP.Dev.top10_bp <- head(Hyb.Comb.DIP.Dev.go_terms[Hyb.Comb.DIP.Dev.go_terms$Category == "GOTERM_BP_DIRECT",], 10)
Hyb.Comb.DIP.Dev.top10_bp$sig <- ""
Hyb.Comb.DIP.Dev.top10_bp[Hyb.Comb.DIP.Dev.top10_bp$FDR < 0.05,]$sig <- "yes"
Hyb.Comb.DIP.Dev.top10_bp[Hyb.Comb.DIP.Dev.top10_bp$FDR > 0.05,]$sig <- "no"
Hyb.Comb.DIP.Dev.top10_bp$Term <- gsub("^GO:[0-9]+~", "", Hyb.Comb.DIP.Dev.top10_bp$Term)

#plot top ten most significant GO terms for DIP
dip_go_plot_top10 <- ggplot(data = Hyb.Comb.DIP.Dev.top10_bp, aes(x = Fold.Enrichment, y = reorder(Term, Fold.Enrichment), size = Count, col = sig)) +
  geom_point() +
  theme_minimal() + 
  scale_color_manual(values = c(gray_3, dip_color), name = "Significance") +
  labs(x = "Fold enrichment", y = "") + 
  scale_size_continuous(name = "# Genes")

go_grid_plot <- plot_grid(sp_go_plot_top10, lz_go_plot_top10, dip_go_plot_top10, ncol=1, labels = c("A", "B", "C"), align = "v")

if(exportFigs == TRUE) {
  ggsave("exported_figures/hybrid_go_enrichment.pdf", go_grid_plot, width = 1.25*full_plot_width, height = full_plot_height*1.5)
}else if (exportFigs == FALSE) {
  go_grid_plot
}

#producing the figure 5 multipanel with the dip go results
hybDIP_corr_multiPanel <-  
  plot_grid(
    plot_grid(NULL, plot_grid(mouse_correlation_plot, hamster_correlation_plot, labels = c('A', 'B'), nrow = 2), both_species_hybrid_escapees_plot, rel_widths = c(0.5, 5, 2), labels = c('', '', 'C'), nrow = 1),
    dip_go_plot_top10, labels = c('', 'D'), rel_heights = c(2,1), nrow = 2)

if(exportFigs == TRUE) {
  ggsave(filename = "exported_figures/hybrid_DIP_correlation_multiPanel_plot.pdf", hybDIP_corr_multiPanel, width = 10, height = 8)
} else if(exportFigs == FALSE) {
  print(hybDIP_corr_multiPanel)
}
```

<h3>PAR Behavior in Hybrid versus Parental Dwarf Hamsters</h3>
```{r par-behavior}
#The starting position of the dwarf hamster PAR as reported by Moore et al. 2022
PAR_coord <- 115350000
#create a data frame of the genes located in the PAR
par_genes <- annotations[annotations$chr == "X" & annotations$start > PAR_coord,]

#a function to help make the output table more readible for inclusion in the supplement
convert_DE_labels <- function(true_false_column) {
  if(sum(true_false_column != "TRUE") != length(true_false_column)){
    true_false_column[true_false_column != "TRUE"] <- ""
    true_false_column[true_false_column == "TRUE"] <- "X"
  } else {
    true_false_column <- ""
  }
  return(true_false_column)
}

#set up an empty data frame to contain par gene expression values/ DE status
par_gene_behavior <- data.frame()

#add DE status for both species for each ortholog for each stage
par_gene_behavior <- cbind(par_genes,       SP_BB_DE =  rownames(par_genes) %in% Hyb.BB.SP.Dev.de$gene)
par_gene_behavior <- cbind(par_gene_behavior, SP_SS_DE =  rownames(par_genes)  %in% Hyb.SS.SP.Dev.de$gene)
par_gene_behavior <- cbind(par_gene_behavior, LZ_BB_DE =  rownames(par_genes)  %in% Hyb.BB.LZ.Dev.de$gene)
par_gene_behavior <- cbind(par_gene_behavior, LZ_SS_DE =  rownames(par_genes)  %in% Hyb.SS.LZ.Dev.de$gene)
par_gene_behavior <- cbind(par_gene_behavior, DIP_BB_DE = rownames(par_genes) %in% Hyb.BB.DIP.Dev.de$gene)
par_gene_behavior <- cbind(par_gene_behavior, DIP_SS_DE = rownames(par_genes) %in% Hyb.SS.DIP.Dev.de$gene)

par_gene_behavior$SP_BB_DE <- convert_DE_labels(par_gene_behavior$SP_BB_DE)
par_gene_behavior$SP_SS_DE <- convert_DE_labels(par_gene_behavior$SP_SS_DE)
par_gene_behavior$LZ_BB_DE <- convert_DE_labels(par_gene_behavior$LZ_BB_DE)
par_gene_behavior$LZ_SS_DE <- convert_DE_labels(par_gene_behavior$LZ_SS_DE)
par_gene_behavior$DIP_BB_DE <- convert_DE_labels(par_gene_behavior$DIP_BB_DE)
par_gene_behavior$DIP_SS_DE <- convert_DE_labels(par_gene_behavior$DIP_SS_DE)

#print table to html file and save to exported files
kable(par_gene_behavior, format = "simple")
write.csv(x = par_gene_behavior, file = "exported_files/Hamster_PAR_genes.csv")

#details to be included in text
print(paste("There are ", sum(table(par_gene_behavior[par_gene_behavior$Length != "NA",]$mouse_gene) == 1), " MSCI genes present as single-copy", sep = ""))
print(paste("There are ", sum(table(par_gene_behavior[par_gene_behavior$Length != "NA",]$mouse_gene) != 1), " MSCI genes present as multi-copy", sep = ""))
print(paste("There are ", sum(table(par_gene_behavior[par_gene_behavior$Length == "NA",]$mouse_gene) == 1), " MSCI genes not present", sep = ""))

#summarizing expression data for plotting; generates Figure 6 in main text
par_genes_rpkm_Dev.norm <- rpkm_Dev.norm[rownames(rpkm_Dev.norm) %in% rownames(par_genes),]
par_genes_rpkm_Dev.norm$Psun_ID <- rownames(par_genes_rpkm_Dev.norm)
par_genes_rpkm_Dev.norm <- pivot_longer(par_genes_rpkm_Dev.norm, cols = 1:(dim(par_genes_rpkm_Dev.norm)[2]-1), names_to = "ID", values_to = "RPKM")
par_genes_rpkm_Dev.norm$CrossXStage <- gsub("[0-9].*M_", "", par_genes_rpkm_Dev.norm$ID)

par_genes_rpkm_Dev.norm$CrossXStage <- factor(par_genes_rpkm_Dev.norm$CrossXStage, levels =
                                                c("BBBB_SP", "BBSS_SP", "SSSS_SP", 
                                                  "BBBB_LZ", "BBSS_LZ", "SSSS_LZ", 
                                                  "BBBB_DIP", "BBSS_DIP", "SSSS_DIP", 
                                                  "BBBB_RS", "SSSS_RS"))
par_genes_rpkm_Dev.norm$cross <- gsub("_.*", "", par_genes_rpkm_Dev.norm$CrossXStage)
par_genes_rpkm_Dev.norm$cross <- factor(par_genes_rpkm_Dev.norm$cross, levels =
                                          c("BBBB", "BBSS", "SSSS"))
par_genes_rpkm_Dev.norm$stage <- gsub(".*_", "", par_genes_rpkm_Dev.norm$CrossXStage)
par_genes_rpkm_Dev.norm$stage <- factor(par_genes_rpkm_Dev.norm$stage, levels =
                                          c("SP", "LZ", "DIP", "RS"))

par_genes_rpkm_Dev.norm$Mouse_ID <- par_genes_rpkm_Dev.norm$Psun_ID

for (hamsterGene in unique(par_genes_rpkm_Dev.norm$Psun_ID)) {
  if(!is.na(par_genes[grep(hamsterGene, par_genes$gene),]$mouse_gene)) {
    par_genes_rpkm_Dev.norm$Mouse_ID <- gsub(hamsterGene, 
                                             par_genes[grep(hamsterGene, par_genes$gene),]$mouse_gene,
                                             par_genes_rpkm_Dev.norm$Mouse_ID)
  }
}

par_genes_rpkm_Dev.norm$Psun_ID <- factor(par_genes_rpkm_Dev.norm$Psun_ID)
par_genes_rpkm_Dev.norm$Mouse_ID <- factor(par_genes_rpkm_Dev.norm$Mouse_ID)


PAR_average_exp_df <- data.frame()

for(gene in unique(par_genes_rpkm_Dev.norm$Mouse_ID)){
  gene_tmp_df <- as.data.frame(subset(par_genes_rpkm_Dev.norm,par_genes_rpkm_Dev.norm$Mouse_ID == gene))
  PAR_sem_exp <- c()
  for(condition in unique(par_genes_rpkm_Dev.norm$CrossXStage)) {
    crossXstage_tmp <- subset(gene_tmp_df, gene_tmp_df$CrossXStage == condition)
    PAR_sem_exp <- c(PAR_sem_exp, calculate_sem(crossXstage_tmp$RPKM))
  }
  PAR_average_exp_df <- rbind(PAR_average_exp_df, cbind(aggregate(gene_tmp_df$RPKM, list(gene_tmp_df$CrossXStage), FUN=mean), PAR_sem_exp, rep(gene, length(unique(gene_tmp_df$CrossXStage)))))
}
PAR_average_exp_df <- separate(PAR_average_exp_df, Group.1, into = c("cross", "stage"))
colnames(PAR_average_exp_df) <- c("cross", "stage", "mean_RPKM", "sem", "gene")
PAR_average_exp_df$stage <- factor(PAR_average_exp_df$stage, levels = c("SP", "LZ", "DIP", "RS"))

#first plots expression without the gene trends collapsed on top of one another
PAR_trends_plot_byGene <- ggplot(data = subset(PAR_average_exp_df, PAR_average_exp_df$stage != "RS"), aes(x = stage, y = mean_RPKM, group = cross, col = cross, shape = cross, fill = cross)) +
  geom_hline(yintercept = 0) +
  geom_line(linewidth = 1) +
  geom_errorbar(aes(ymin=mean_RPKM-sem, ymax=mean_RPKM+sem), width=.1, color = "black") +
  geom_point(color = "black") +
  theme_minimal() +
  theme(panel.grid.minor.y = element_blank()) +
  scale_color_manual(values=c(gray_5, gray_4, gray_2)) + 
  scale_fill_manual(values=c(gray_5, gray_4, gray_2)) + 
  scale_shape_manual(values=c(24,4,25)) + 
  facet_wrap(vars(gene, cross), nrow = length(unique(PAR_average_exp_df$gene)))

cross.labs <- c("P. campbelli", "Hybrid", "P. sungorus")
names(cross.labs) <- c("BBBB", "BBSS", "SSSS")

colorBrownLightest  <- "#D4CCC4";
colorBrownLight     <- "#BAADA1";
colorBrownMedium    <- "#A08F7E";
colorBrownMediumest <- "#81705F";
colorBrownDark      <- "#584D41";
colorBrownDarkest   <- "#3A332C";
colorBrownBlack     <- "#231F1A";

#then plots expression with the gene trends collapsed on top of one another (version in manuscript)
PAR_trends_plot_collapsed <- ggplot(data = subset(PAR_average_exp_df, PAR_average_exp_df$stage != "RS"), aes(x = stage, y = mean_RPKM, group = gene, col = gene)) +
  geom_hline(yintercept = 0) +
  geom_line(linewidth = 0.75, aes(linetype = gene, color = gene), position=position_jitter(w=0.05)) +
  theme_minimal() +
  theme(panel.grid.minor.y = element_blank()) +
  scale_color_manual(values = c(colorBrownLightest, colorBrownLight, colorBrownMedium, colorBrownMediumest, colorBrownDark, colorBrownDarkest, colorBrownBlack)) + 
  scale_linetype_manual(values = c("solid", "dashed", "solid", "dashed", "solid", "dashed", "solid")) +
  labs(x = "Stage of spermatogenesis", y = "Average expression of PAR genes (normalized RPKM)") +
  facet_wrap(vars(cross), nrow = 1, labeller = labeller(cross = cross.labs))

if(exportFigs == TRUE) {
  ggsave(filename = "exported_figures/hamster_PAR_trends_byGene_plot.pdf", PAR_trends_plot_byGene, width = 1.25*full_plot_width, height = 1.5*full_plot_height)
} else {
  print(PAR_trends_plot_byGene)
}

if(exportFigs == TRUE) {
  ggsave(filename = "exported_figures/hamster_PAR_trends_collapsed_plot.pdf", PAR_trends_plot_collapsed, height = 5, width = full_plot_width*1.5)
} else {
  print(PAR_trends_plot_collapsed)
}
```

<h3>Calculate BCV for each Species/Hybrid</h3>
```{r calculate-BCV-for-each-hamster-cross}
#the purpose of this analysis is to see how variation within hybrids and within parents compares
#variation across hybrids for SP/LZ/DIP
dgeBBSSDev <- DGEList(just_counts[,grep("BBSS_.*_[D,S,L]",colnames(just_counts))], genes=annotations, group=grep("BBSS-[D,S,L]",metadata$condition_name, value = TRUE))
dgeBBSSDev <- normalize_dge(dgeBBSSDev)

#variation across hybrids for SP/LZ/DIP
design_BBSSDev <- design[grep("BBSS\\.[D,L,S]", design$condition_name),]
design_BBSSDev$condition_name <- factor(design_BBSSDev$condition_name)

design_BBSSDev_Matrix  <-  model.matrix(~0+design_BBSSDev$condition_name)
colnames(design_BBSSDev_Matrix)  <-  levels(design_BBSSDev$condition_name)
rownames(design_BBSSDev_Matrix)  <-  design_BBSSDev$sample_ID

dgeBBSSDev <-  estimateGLMCommonDisp(dgeBBSSDev, design_BBSSDev_Matrix, verbose = TRUE)
dgeBBSSDev <- estimateGLMTrendedDisp(dgeBBSSDev, design_BBSSDev_Matrix) 
dgeBBSSDev <- estimateGLMTagwiseDisp(dgeBBSSDev, design_BBSSDev_Matrix)  # estimate genewise dispersion


#change exportFigs to TRUE to export as PDF if desired
if(exportFigs == TRUE) {
  pdf(file="exported_figures/Dispersion_plots_justBBSS_dataset.pdf", width=10, height=8, pointsize=12)
  plotBCV(dgeBBSSDev, main = paste("Dispersion estimates for BBSS dataset; BCV = ",round(sqrt(dgeBBSSDev$common.dispersion), 3)), ylim = c(0, 12), col.common = "#C2A5A2", col.trend = "#745756", xlab = "Average log(CPM)", xlim = c(-4,13), col = alpha("#6E6E6E", 0.15), cex = 1.25)
  dev.off()
}
plotBCV(dgeBBSSDev, main = paste("Dispersion estimates for BBSS dataset; BCV = ",round(sqrt(dgeBBSSDev$common.dispersion), 3)), ylim = c(0, 12), col.common = "#C2A5A2", col.trend = "#745756", xlab = "Average log(CPM)", xlim = c(-4,13), col = alpha("#6E6E6E", 0.15), cex = 1.25)


#variation across P. cam for SP/LZ/DIP
dgeBBBBDev <- DGEList(just_counts[,grep("BBBB_.*_[D,S,L]",colnames(just_counts))], genes=annotations, group=grep("BBBB-[D,S,L]",metadata$condition_name, value = TRUE))
dgeBBBBDev <- normalize_dge(dgeBBBBDev)

#variation across P. cam for SP/LZ/DIP
design_BBBBDev <- design[grep("BBBB\\.[D,L,S]", design$condition_name),]
design_BBBBDev$condition_name <- factor(design_BBBBDev$condition_name)

design_BBBBDev_Matrix  <-  model.matrix(~0+design_BBBBDev$condition_name)
colnames(design_BBBBDev_Matrix)  <-  levels(design_BBBBDev$condition_name)
rownames(design_BBBBDev_Matrix)  <-  design_BBBBDev$sample_ID

dgeBBBBDev <-  estimateGLMCommonDisp(dgeBBBBDev, design_BBBBDev_Matrix, verbose = TRUE)
dgeBBBBDev <- estimateGLMTrendedDisp(dgeBBBBDev, design_BBBBDev_Matrix) 
dgeBBBBDev <- estimateGLMTagwiseDisp(dgeBBBBDev, design_BBBBDev_Matrix)  # estimate genewise dispersion


#change exportFigs to TRUE to export as PDF if desired
if(exportFigs == TRUE) {
  pdf(file="exported_figures/Dispersion_plots_justBBBB_dataset.pdf", width=10, height=8, pointsize=12)
  plotBCV(dgeBBBBDev, main = paste("Dispersion estimates for BBBB dataset; BCV = ",round(sqrt(dgeBBBBDev$common.dispersion), 3)), ylim = c(0, 12), col.common = "#C2A5A2", col.trend = "#745756", xlab = "Average log(CPM)", xlim = c(-4,13), col = alpha("#6E6E6E", 0.15), cex = 1.25)
  dev.off()
}
plotBCV(dgeBBBBDev, main = paste("Dispersion estimates for BBBB dataset; BCV = ",round(sqrt(dgeBBBBDev$common.dispersion), 3)), ylim = c(0, 12), col.common = "#C2A5A2", col.trend = "#745756", xlab = "Average log(CPM)", xlim = c(-4,13), col = alpha("#6E6E6E", 0.15), cex = 1.25)


#variation across P. sun for SP/LZ/DIP
dgeSSSSDev <- DGEList(just_counts[,grep("SSSS_.*_[D,S,L]",colnames(just_counts))], genes=annotations, group=grep("SSSS-[D,S,L]",metadata$condition_name, value = TRUE))
dgeSSSSDev <- normalize_dge(dgeSSSSDev)

#variation across P. sun for SP/LZ/DIP
design_SSSSDev <- design[grep("SSSS\\.[D,L,S]", design$condition_name),]
design_SSSSDev$condition_name <- factor(design_SSSSDev$condition_name)

design_SSSSDev_Matrix  <-  model.matrix(~0+design_SSSSDev$condition_name)
colnames(design_SSSSDev_Matrix)  <-  levels(design_SSSSDev$condition_name)
rownames(design_SSSSDev_Matrix)  <-  design_SSSSDev$sample_ID

dgeSSSSDev <-  estimateGLMCommonDisp(dgeSSSSDev, design_SSSSDev_Matrix, verbose = TRUE)
dgeSSSSDev <- estimateGLMTrendedDisp(dgeSSSSDev, design_SSSSDev_Matrix) 
dgeSSSSDev <- estimateGLMTagwiseDisp(dgeSSSSDev, design_SSSSDev_Matrix)  # estimate genewise dispersion


#change exportFigs to TRUE to export as PDF if desired
if(exportFigs == TRUE) {
  pdf(file="exported_figures/Dispersion_plots_justSSSS_dataset.pdf", width=10, height=8, pointsize=12)
  plotBCV(dgeSSSSDev, main = paste("Dispersion estimates for SSSS dataset; BCV = ",round(sqrt(dgeSSSSDev$common.dispersion), 3)), ylim = c(0, 12), col.common = "#C2A5A2", col.trend = "#745756", xlab = "Average log(CPM)", xlim = c(-4,13), col = alpha("#6E6E6E", 0.15), cex = 1.25)
  dev.off()
}
plotBCV(dgeSSSSDev, main = paste("Dispersion estimates for SSSS dataset; BCV = ",round(sqrt(dgeSSSSDev$common.dispersion), 3)), ylim = c(0, 12), col.common = "#C2A5A2", col.trend = "#745756", xlab = "Average log(CPM)", xlim = c(-4,13), col = alpha("#6E6E6E", 0.15), cex = 1.25)
```